// ng2-activiti-form/src/components/widgets/tabs/tabs.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var TabsWidget = (function () {
    function TabsWidget() {
        this.tabs = [];
        this.formTabChanged = new core_1.EventEmitter();
        this.visibleTabs = [];
    }
    TabsWidget.prototype.hasTabs = function () {
        return this.tabs && this.tabs.length > 0;
    };
    TabsWidget.prototype.ngAfterContentChecked = function () {
        this.filterVisibleTabs();
    };
    TabsWidget.prototype.ngAfterViewInit = function () {
        this.setupMaterialComponents();
    };
    TabsWidget.prototype.filterVisibleTabs = function () {
        this.visibleTabs = this.tabs.filter(function (tab) {
            return tab.isVisible;
        });
    };
    TabsWidget.prototype.setupMaterialComponents = function () {
        if (componentHandler) {
            componentHandler.upgradeAllRegistered();
            return true;
        }
        return false;
    };
    TabsWidget.prototype.tabChanged = function (field) {
        this.formTabChanged.emit(field);
    };
    return TabsWidget;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", Array)
], TabsWidget.prototype, "tabs", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], TabsWidget.prototype, "formTabChanged", void 0);
TabsWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'tabs-widget',
        template: "<div *ngIf=\"hasTabs()\" class=\"alfresco-tabs-widget\">     <div alfresco-mdl-tabs>         <div class=\"mdl-tabs__tab-bar\">             <a *ngFor=\"let tab of visibleTabs; let isFirst = first\"                id=\"title-{{tab.id}}\"                [href]=\"'#' + tab.id\"                class=\"mdl-tabs__tab\" [class.is-active]=\"isFirst\">                 {{tab.title}}             </a>         </div>         <div *ngFor=\"let tab of visibleTabs; let isFirst = first\"              class=\"mdl-tabs__panel\"              [class.is-active]=\"isFirst\"              [attr.id]=\"tab.id\">              <div *ngFor=\"let field of tab.fields\">                 <form-field [field]=\"field.field\"></form-field>              </div>         </div>     </div> </div>"
    })
], TabsWidget);
exports.TabsWidget = TabsWidget;
