// ng2-activiti-processlist/src/components/activiti-process-instance-header.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var process_instance_model_1 = require("../models/process-instance.model");
var common_1 = require("@angular/common");
var ActivitiProcessInstanceHeader = (function () {
    function ActivitiProcessInstanceHeader(translate, logService) {
        this.translate = translate;
        this.logService = logService;
        this.onError = new core_1.EventEmitter();
        if (translate) {
            translate.addTranslationFolder('ng2-activiti-processlist', 'node_modules/ng2-activiti-processlist/src');
        }
    }
    ActivitiProcessInstanceHeader.prototype.getStartedByFullName = function () {
        if (this.processInstance && this.processInstance.startedBy) {
            return (this.processInstance.startedBy.firstName && this.processInstance.startedBy.firstName !== 'null'
                ? this.processInstance.startedBy.firstName + ' ' : '') +
                this.processInstance.startedBy.lastName;
        }
        return '';
    };
    ActivitiProcessInstanceHeader.prototype.getFormatDate = function (value, format) {
        var datePipe = new common_1.DatePipe('en-US');
        try {
            return datePipe.transform(value, format);
        }
        catch (err) {
            this.logService.error("ProcessListInstanceHeader: error parsing date " + value + " to format " + format);
        }
    };
    ActivitiProcessInstanceHeader.prototype.isRunning = function () {
        return this.processInstance && !this.processInstance.ended;
    };
    return ActivitiProcessInstanceHeader;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", process_instance_model_1.ProcessInstance)
], ActivitiProcessInstanceHeader.prototype, "processInstance", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], ActivitiProcessInstanceHeader.prototype, "onError", void 0);
ActivitiProcessInstanceHeader = __decorate([
    core_1.Component({
        selector: 'activiti-process-instance-header',
        moduleId: module.id,
        template: "<div *ngIf=\"processInstance\">     <div class=\"mdl-grid\">         <div class=\"mdl-cell mdl-cell--4-col\" data-automation-id=\"header-started-by\">             <span class=\"activiti-label\">{{ 'DETAILS.LABELS.STARTED_BY' | translate }}</span>:             <span class=\"activiti-process-header__value\">{{getStartedByFullName()}}</span>         </div>         <div class=\"mdl-cell mdl-cell--4-col\" data-automation-id=\"header-started\">             <span class=\"activiti-label\">{{ 'DETAILS.LABELS.STARTED' | translate }}</span>:             <span class=\"activiti-process-header__value\">{{getFormatDate(processInstance.started, 'medium')}}</span>         </div>         <div class=\"mdl-cell mdl-cell--4-col\">             <activiti-process-instance-comments [processInstanceId]=\"processInstance?.id\"></activiti-process-instance-comments>         </div>         <div class=\"mdl-cell mdl-cell--4-col\" data-automation-id=\"header-status\" *ngIf=\"!isRunning()\">             <span class=\"activiti-label\">{{ 'DETAILS.LABELS.ENDED' | translate }}</span>:             <span class=\"activiti-process-header__value\">{{getFormatDate(processInstance.ended, 'medium')}}</span>         </div>     </div> </div>",
        styles: [":host {     width: 100%; }  .activiti-label {     font-weight: bolder; }  .activiti-process-header__value {     color: rgb(68, 138, 255); }"]
    }),
    __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoTranslationService,
        ng2_alfresco_core_1.LogService])
], ActivitiProcessInstanceHeader);
exports.ActivitiProcessInstanceHeader = ActivitiProcessInstanceHeader;
