// ng2-alfresco-webscript/bundles/ng2-alfresco-webscript.js
!function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in v||(v[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==g.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=v[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(g.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=v[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}n=!n}}function u(e){return y[e]||(y[e]={name:e,dependencies:[],exports:{},importers:[]})}function d(r){if(!r.module){var t=r.module=u(r.name),n=r.module.exports,o=r.declare.call(e,function(e,r){if(t.locked=!0,"object"==typeof e)for(var o in e)n[o]=e[o];else n[e]=r;for(var a=0,u=t.importers.length;u>a;a++){var d=t.importers[a];if(!d.locked)for(var i=0;i<d.dependencies.length;++i)d.dependencies[i]===t&&d.setters[i](n)}return t.locked=!1,r},{id:r.name});t.setters=o.setters,t.execute=o.execute;for(var a=0,i=r.normalizedDeps.length;i>a;a++){var l,s=r.normalizedDeps[a],c=v[s],f=y[s];f?l=f.exports:c&&!c.declarative?l=c.esModule:c?(d(c),f=c.module,l=f.exports):l=p(s),f&&f.importers?(f.importers.push(t),t.dependencies.push(f)):t.dependencies.push(null),t.setters[a]&&t.setters[a](l)}}}function i(e){var r,t=v[e];if(t)t.declarative?f(e,[]):t.evaluated||l(t),r=t.module.exports;else if(r=p(e),!r)throw new Error("Unable to load dependency "+e+".");return(!t||t.declarative)&&r&&r.__useDefault?r["default"]:r}function l(r){if(!r.module){var t={},n=r.module={exports:t,id:r.name};if(!r.executingRequire)for(var o=0,a=r.normalizedDeps.length;a>o;o++){var u=r.normalizedDeps[o],d=v[u];d&&l(d)}r.evaluated=!0;var c=r.execute.call(e,function(e){for(var t=0,n=r.deps.length;n>t;t++)if(r.deps[t]==e)return i(r.normalizedDeps[t]);throw new TypeError("Module "+e+" not declared as a dependency.")},t,n);void 0!==c&&(n.exports=c),t=n.exports,t&&t.__esModule?r.esModule=t:r.esModule=s(t)}}function s(r){var t={};if(("object"==typeof r||"function"==typeof r)&&r!==e)if(m)for(var n in r)"default"!==n&&c(t,r,n);else{var o=r&&r.hasOwnProperty;for(var n in r)"default"===n||o&&!r.hasOwnProperty(n)||(t[n]=r[n])}return t["default"]=r,x(t,"__useDefault",{value:!0}),t}function c(e,r,t){try{var n;(n=Object.getOwnPropertyDescriptor(r,t))&&x(e,t,n)}catch(o){return e[t]=r[t],!1}}function f(r,t){var n=v[r];if(n&&!n.evaluated&&n.declarative){t.push(r);for(var o=0,a=n.normalizedDeps.length;a>o;o++){var u=n.normalizedDeps[o];-1==g.call(t,u)&&(v[u]?f(u,t):p(u))}n.evaluated||(n.evaluated=!0,n.module.execute.call(e))}}function p(e){if(I[e])return I[e];if("@node/"==e.substr(0,6))return I[e]=s(D(e.substr(6)));var r=v[e];if(!r)throw"Module "+e+" not present.";return a(e),f(e,[]),v[e]=void 0,r.declarative&&x(r.module.exports,"__esModule",{value:!0}),I[e]=r.declarative?r.module.exports:r.esModule}var v={},g=Array.prototype.indexOf||function(e){for(var r=0,t=this.length;t>r;r++)if(this[r]===e)return r;return-1},m=!0;try{Object.getOwnPropertyDescriptor({a:0},"a")}catch(h){m=!1}var x;!function(){try{Object.defineProperty({},"a",{})&&(x=Object.defineProperty)}catch(e){x=function(e,r,t){try{e[r]=t.value||t.get.call(e)}catch(n){}}}}();var y={},D="undefined"!=typeof System&&System._nodeRequire||"undefined"!=typeof require&&"undefined"!=typeof require.resolve&&"undefined"!=typeof process&&process.platform&&require,I={"@empty":{}};return function(e,n,o,a){return function(u){u(function(u){for(var d={_nodeRequire:D,register:r,registerDynamic:t,get:p,set:function(e,r){I[e]=r},newModule:function(e){return e}},i=0;i<n.length;i++)(function(e,r){r&&r.__esModule?I[e]=r:I[e]=s(r)})(n[i],arguments[i]);a(d);var l=p(e[0]);if(e.length>1)for(var i=1;i<e.length;i++)p(e[i]);return o?l["default"]:l})}}}("undefined"!=typeof self?self:global)

(["1"], ["3","4","5"], true, function($__System) {
var require = this.require, exports = this.exports, module = this.module;
$__System.registerDynamic("2", ["3", "4", "5"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("4");
    var ng2_alfresco_datatable_1 = $__require("5");
    var WebscriptComponent = function () {
        function WebscriptComponent(apiService, logService) {
            this.apiService = apiService;
            this.logService = logService;
            this.showData = true;
            this.contextRoot = 'alfresco';
            this.servicePath = 'service';
            this.contentType = 'TEXT';
            this.onSuccess = new core_1.EventEmitter();
            this.data = undefined;
            this.showError = false;
        }
        WebscriptComponent.prototype.ngOnChanges = function (changes) {
            var _this = this;
            if (this.showData) {
                this.clean();
            }
            return new Promise(function (resolve, reject) {
                _this.apiService.getInstance().webScript.executeWebScript('GET', _this.scriptPath, _this.scriptArgs, _this.contextRoot, _this.servicePath).then(function (webScriptdata) {
                    _this.data = webScriptdata;
                    if (_this.showData) {
                        if (_this.contentType === 'DATATABLE') {
                            _this.data = _this.showDataAsDataTable(webScriptdata);
                        }
                    }
                    _this.onSuccess.emit(_this.data);
                    resolve();
                }, function (error) {
                    _this.logService.log('Error' + error);
                    reject();
                });
            });
        };
        WebscriptComponent.prototype.showDataAsDataTable = function (data) {
            var datatableData = null;
            try {
                if (!data.schema) {
                    data.schema = ng2_alfresco_datatable_1.ObjectDataTableAdapter.generateSchema(data.data);
                }
                if (data.schema && data.schema.length > 0) {
                    this.data = new ng2_alfresco_datatable_1.ObjectDataTableAdapter(data.data, data.schema);
                }
            } catch (e) {
                this.logService.error('error during the cast as datatable');
            }
            return datatableData;
        };
        WebscriptComponent.prototype.clean = function () {
            this.data = undefined;
        };
        WebscriptComponent.prototype.isDataTableContent = function () {
            return this.contentType === 'DATATABLE';
        };
        return WebscriptComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String)], WebscriptComponent.prototype, "scriptPath", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Object)], WebscriptComponent.prototype, "scriptArgs", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], WebscriptComponent.prototype, "showData", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], WebscriptComponent.prototype, "contextRoot", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], WebscriptComponent.prototype, "servicePath", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], WebscriptComponent.prototype, "contentType", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], WebscriptComponent.prototype, "onSuccess", void 0);
    WebscriptComponent = __decorate([core_1.Component({
        selector: 'alfresco-webscript-get',
        template: "\n        <div *ngIf=\"showData\">\n            <div *ngIf=\"contentType === 'JSON'\" id=\"webscript-data-JSON\">{{data | json}}</div>\n            <div *ngIf=\"contentType === 'HTML'\" id=\"webscript-data-HTML\" [innerHTML]=\"data\"></div>\n            <div *ngIf=\"contentType === 'TEXT'\" id=\"webscript-data-TEXT\" >{{data}}</div>\n            <div *ngIf=\"isDataTableContent()\"><alfresco-datatable id=\"webscript-datatable-wrapper\" [data]=\"data\"></alfresco-datatable><div>\n            <div *ngIf=\"showError\" id=\"error\">Error during the deserialization of {{data}} as {{contentType}}</div>\n        </div>\n    "
    }), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoApiService, ng2_alfresco_core_1.LogService])], WebscriptComponent);
    exports.WebscriptComponent = WebscriptComponent;
});
$__System.registerDynamic("1", ["3", "4", "5", "2"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("4");
    var ng2_alfresco_datatable_1 = $__require("5");
    var webscript_component_1 = $__require("2");
    __export($__require("2"));
    exports.WEBSCRIPT_DIRECTIVES = [webscript_component_1.WebscriptComponent];
    var WebScriptModule = WebScriptModule_1 = function () {
        function WebScriptModule() {}
        WebScriptModule.forRoot = function () {
            return {
                ngModule: WebScriptModule_1
            };
        };
        return WebScriptModule;
    }();
    WebScriptModule = WebScriptModule_1 = __decorate([core_1.NgModule({
        imports: [ng2_alfresco_core_1.CoreModule, ng2_alfresco_datatable_1.DataTableModule],
        declarations: exports.WEBSCRIPT_DIRECTIVES.slice(),
        exports: exports.WEBSCRIPT_DIRECTIVES.slice()
    })], WebScriptModule);
    exports.WebScriptModule = WebScriptModule;
    var WebScriptModule_1;
    
});
})
(function(factory) {
  if (typeof define == 'function' && define.amd)
    define(["@angular/core","ng2-alfresco-core","ng2-alfresco-datatable"], factory);
  else if (typeof module == 'object' && module.exports && typeof require == 'function')
    module.exports = factory(require("@angular/core"), require("ng2-alfresco-core"), require("ng2-alfresco-datatable"));
  else
    throw new Error("Module must be loaded as AMD or CommonJS");
});
