// ng2-alfresco-core/bundles/ng2-alfresco-core.js
!function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in v||(v[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==g.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=v[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(g.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=v[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}n=!n}}function u(e){return y[e]||(y[e]={name:e,dependencies:[],exports:{},importers:[]})}function d(r){if(!r.module){var t=r.module=u(r.name),n=r.module.exports,o=r.declare.call(e,function(e,r){if(t.locked=!0,"object"==typeof e)for(var o in e)n[o]=e[o];else n[e]=r;for(var a=0,u=t.importers.length;u>a;a++){var d=t.importers[a];if(!d.locked)for(var i=0;i<d.dependencies.length;++i)d.dependencies[i]===t&&d.setters[i](n)}return t.locked=!1,r},{id:r.name});t.setters=o.setters,t.execute=o.execute;for(var a=0,i=r.normalizedDeps.length;i>a;a++){var l,s=r.normalizedDeps[a],c=v[s],f=y[s];f?l=f.exports:c&&!c.declarative?l=c.esModule:c?(d(c),f=c.module,l=f.exports):l=p(s),f&&f.importers?(f.importers.push(t),t.dependencies.push(f)):t.dependencies.push(null),t.setters[a]&&t.setters[a](l)}}}function i(e){var r,t=v[e];if(t)t.declarative?f(e,[]):t.evaluated||l(t),r=t.module.exports;else if(r=p(e),!r)throw new Error("Unable to load dependency "+e+".");return(!t||t.declarative)&&r&&r.__useDefault?r["default"]:r}function l(r){if(!r.module){var t={},n=r.module={exports:t,id:r.name};if(!r.executingRequire)for(var o=0,a=r.normalizedDeps.length;a>o;o++){var u=r.normalizedDeps[o],d=v[u];d&&l(d)}r.evaluated=!0;var c=r.execute.call(e,function(e){for(var t=0,n=r.deps.length;n>t;t++)if(r.deps[t]==e)return i(r.normalizedDeps[t]);throw new TypeError("Module "+e+" not declared as a dependency.")},t,n);void 0!==c&&(n.exports=c),t=n.exports,t&&t.__esModule?r.esModule=t:r.esModule=s(t)}}function s(r){var t={};if(("object"==typeof r||"function"==typeof r)&&r!==e)if(m)for(var n in r)"default"!==n&&c(t,r,n);else{var o=r&&r.hasOwnProperty;for(var n in r)"default"===n||o&&!r.hasOwnProperty(n)||(t[n]=r[n])}return t["default"]=r,x(t,"__useDefault",{value:!0}),t}function c(e,r,t){try{var n;(n=Object.getOwnPropertyDescriptor(r,t))&&x(e,t,n)}catch(o){return e[t]=r[t],!1}}function f(r,t){var n=v[r];if(n&&!n.evaluated&&n.declarative){t.push(r);for(var o=0,a=n.normalizedDeps.length;a>o;o++){var u=n.normalizedDeps[o];-1==g.call(t,u)&&(v[u]?f(u,t):p(u))}n.evaluated||(n.evaluated=!0,n.module.execute.call(e))}}function p(e){if(I[e])return I[e];if("@node/"==e.substr(0,6))return I[e]=s(D(e.substr(6)));var r=v[e];if(!r)throw"Module "+e+" not present.";return a(e),f(e,[]),v[e]=void 0,r.declarative&&x(r.module.exports,"__esModule",{value:!0}),I[e]=r.declarative?r.module.exports:r.esModule}var v={},g=Array.prototype.indexOf||function(e){for(var r=0,t=this.length;t>r;r++)if(this[r]===e)return r;return-1},m=!0;try{Object.getOwnPropertyDescriptor({a:0},"a")}catch(h){m=!1}var x;!function(){try{Object.defineProperty({},"a",{})&&(x=Object.defineProperty)}catch(e){x=function(e,r,t){try{e[r]=t.value||t.get.call(e)}catch(n){}}}}();var y={},D="undefined"!=typeof System&&System._nodeRequire||"undefined"!=typeof require&&"undefined"!=typeof require.resolve&&"undefined"!=typeof process&&process.platform&&require,I={"@empty":{}};return function(e,n,o,a){return function(u){u(function(u){for(var d={_nodeRequire:D,register:r,registerDynamic:t,get:p,set:function(e,r){I[e]=r},newModule:function(e){return e}},i=0;i<n.length;i++)(function(e,r){r&&r.__esModule?I[e]=r:I[e]=s(r)})(n[i],arguments[i]);a(d);var l=p(e[0]);if(e.length>1)for(var i=1;i<e.length;i++)p(e[i]);return o?l["default"]:l})}}}("undefined"!=typeof self?self:global)

(["1"], ["26","3","27","17","e","a","12","14","7","10"], true, function($__System) {
var require = this.require, exports = this.exports, module = this.module;
$__System.registerDynamic("2", ["3", "4", "5"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var alfresco_authentication_service_1 = $__require("4");
    var alfresco_api_service_1 = $__require("5");
    var AlfrescoContentService = function () {
        function AlfrescoContentService(authService, apiService) {
            this.authService = authService;
            this.apiService = apiService;
        }
        AlfrescoContentService.prototype.getDocumentThumbnailUrl = function (document) {
            return this.apiService.getInstance().content.getDocumentThumbnailUrl(document.entry.id);
        };
        AlfrescoContentService.prototype.getContentUrl = function (document) {
            return this.apiService.getInstance().content.getContentUrl(document.entry.id);
        };
        return AlfrescoContentService;
    }();
    AlfrescoContentService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [alfresco_authentication_service_1.AlfrescoAuthenticationService, alfresco_api_service_1.AlfrescoApiService])], AlfrescoContentService);
    exports.AlfrescoContentService = AlfrescoContentService;
});
$__System.registerDynamic("6", ["3", "7", "5", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("7");
    var alfresco_api_service_1 = $__require("5");
    var log_service_1 = $__require("8");
    var RenditionsService = function () {
        function RenditionsService(apiService, logService) {
            this.apiService = apiService;
            this.logService = logService;
        }
        RenditionsService.prototype.isRenditionAvailable = function (nodeId, encoding) {
            var _this = this;
            return Rx_1.Observable.create(function (observer) {
                _this.getRendition(nodeId, encoding).subscribe(function (res) {
                    var isAvailable = true;
                    if (res.entry.status === 'NOT_CREATED') {
                        isAvailable = false;
                    }
                    observer.next(isAvailable);
                    observer.complete();
                }, function () {
                    observer.next(false);
                    observer.complete();
                });
            });
        };
        RenditionsService.prototype.isConversionPossible = function (nodeId, encoding) {
            var _this = this;
            return Rx_1.Observable.create(function (observer) {
                _this.getRendition(nodeId, encoding).subscribe(function () {
                    observer.next(true);
                    observer.complete();
                }, function () {
                    observer.next(false);
                    observer.complete();
                });
            });
        };
        RenditionsService.prototype.getRendition = function (nodeId, encoding) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.renditionsApi.getRendition(nodeId, encoding)).catch(function (err) {
                return _this.handleError(err);
            });
        };
        RenditionsService.prototype.getRenditionsListByNodeId = function (nodeId) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.renditionsApi.getRenditions(nodeId)).catch(function (err) {
                return _this.handleError(err);
            });
        };
        RenditionsService.prototype.createRendition = function (nodeId, encoding) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.renditionsApi.createRendition(nodeId, { id: encoding })).catch(function (err) {
                return _this.handleError(err);
            });
        };
        RenditionsService.prototype.handleError = function (error) {
            this.logService.error(error);
            return Rx_1.Observable.throw(error || 'Server error');
        };
        return RenditionsService;
    }();
    RenditionsService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [alfresco_api_service_1.AlfrescoApiService, log_service_1.LogService])], RenditionsService);
    exports.RenditionsService = RenditionsService;
});
$__System.registerDynamic("9", ["3", "a", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var router_1 = $__require("a");
    var alfresco_authentication_service_1 = $__require("4");
    var AuthGuard = function () {
        function AuthGuard(authService, router) {
            this.authService = authService;
            this.router = router;
        }
        AuthGuard.prototype.canActivate = function (route, state) {
            return this.checkLogin();
        };
        AuthGuard.prototype.canActivateChild = function (route, state) {
            return this.canActivate(route, state);
        };
        AuthGuard.prototype.checkLogin = function () {
            if (this.authService.isLoggedIn()) {
                return true;
            }
            this.router.navigate(['/login']);
            return false;
        };
        return AuthGuard;
    }();
    AuthGuard = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [alfresco_authentication_service_1.AlfrescoAuthenticationService, router_1.Router])], AuthGuard);
    exports.AuthGuard = AuthGuard;
});
$__System.registerDynamic("b", ["3", "a", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var router_1 = $__require("a");
    var alfresco_authentication_service_1 = $__require("4");
    var AuthGuardEcm = function () {
        function AuthGuardEcm(authService, router) {
            this.authService = authService;
            this.router = router;
        }
        AuthGuardEcm.prototype.canActivate = function (route, state) {
            return this.checkLogin();
        };
        AuthGuardEcm.prototype.canActivateChild = function (route, state) {
            return this.canActivate(route, state);
        };
        AuthGuardEcm.prototype.checkLogin = function () {
            if (this.authService.isEcmLoggedIn()) {
                return true;
            }
            this.router.navigate(['/login']);
            return false;
        };
        return AuthGuardEcm;
    }();
    AuthGuardEcm = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [alfresco_authentication_service_1.AlfrescoAuthenticationService, router_1.Router])], AuthGuardEcm);
    exports.AuthGuardEcm = AuthGuardEcm;
});
$__System.registerDynamic("c", ["3", "a", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var router_1 = $__require("a");
    var alfresco_authentication_service_1 = $__require("4");
    var AuthGuardBpm = function () {
        function AuthGuardBpm(authService, router) {
            this.authService = authService;
            this.router = router;
        }
        AuthGuardBpm.prototype.canActivate = function (route, state) {
            return this.checkLogin();
        };
        AuthGuardBpm.prototype.canActivateChild = function (route, state) {
            return this.canActivate(route, state);
        };
        AuthGuardBpm.prototype.checkLogin = function () {
            if (this.authService.isBpmLoggedIn()) {
                return true;
            }
            this.router.navigate(['/login']);
            return false;
        };
        return AuthGuardBpm;
    }();
    AuthGuardBpm = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [alfresco_authentication_service_1.AlfrescoAuthenticationService, router_1.Router])], AuthGuardBpm);
    exports.AuthGuardBpm = AuthGuardBpm;
});
$__System.registerDynamic("d", ["3", "e"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var material_1 = $__require("e");
    var NotificationService = NotificationService_1 = function () {
        function NotificationService(snackbar) {
            this.snackbar = snackbar;
        }
        NotificationService.prototype.openSnackMessage = function (message, millisecondsDuration) {
            return this.snackbar.open(message, null, {
                duration: millisecondsDuration || NotificationService_1.DEFAULT_DURATION_MESSAGE
            });
        };
        NotificationService.prototype.openSnackMessageAction = function (message, action, millisecondsDuration) {
            return this.snackbar.open(message, action, {
                duration: millisecondsDuration || NotificationService_1.DEFAULT_DURATION_MESSAGE
            });
        };
        return NotificationService;
    }();
    NotificationService.DEFAULT_DURATION_MESSAGE = 5000;
    NotificationService = NotificationService_1 = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [material_1.MdSnackBar])], NotificationService);
    exports.NotificationService = NotificationService;
    var NotificationService_1;
});
$__System.registerDynamic("f", ["3", "10"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var core_1 = $__require("3");
    var Subject_1 = $__require("10");
    var AlfrescoSettingsService = AlfrescoSettingsService_1 = function () {
        function AlfrescoSettingsService() {
            this._ecmHost = AlfrescoSettingsService_1.DEFAULT_ECM_ADDRESS;
            this._bpmHost = AlfrescoSettingsService_1.DEFAULT_BPM_ADDRESS;
            this._csrfDisabled = AlfrescoSettingsService_1.DEFAULT_CSRF_CONFIG;
            this._bpmContextPath = AlfrescoSettingsService_1.DEFAULT_BPM_CONTEXT_PATH;
            this.providers = 'ALL';
            this.bpmHostSubject = new Subject_1.Subject();
            this.ecmHostSubject = new Subject_1.Subject();
            this.csrfSubject = new Subject_1.Subject();
            this.providerSubject = new Subject_1.Subject();
        }
        Object.defineProperty(AlfrescoSettingsService.prototype, "ecmHost", {
            get: function () {
                return this._ecmHost;
            },
            set: function (ecmHostUrl) {
                this.ecmHostSubject.next(ecmHostUrl);
                this._ecmHost = ecmHostUrl;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(AlfrescoSettingsService.prototype, "csrfDisabled", {
            set: function (csrfDisabled) {
                this.csrfSubject.next(csrfDisabled);
                this._csrfDisabled = csrfDisabled;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(AlfrescoSettingsService.prototype, "bpmHost", {
            get: function () {
                return this._bpmHost;
            },
            set: function (bpmHostUrl) {
                this.bpmHostSubject.next(bpmHostUrl);
                this._bpmHost = bpmHostUrl;
            },
            enumerable: true,
            configurable: true
        });
        AlfrescoSettingsService.prototype.getBPMApiBaseUrl = function () {
            return this._bpmHost + this._bpmContextPath;
        };
        AlfrescoSettingsService.prototype.getProviders = function () {
            return this.providers;
        };
        AlfrescoSettingsService.prototype.setProviders = function (providers) {
            this.providerSubject.next(providers);
            this.providers = providers;
        };
        return AlfrescoSettingsService;
    }();
    AlfrescoSettingsService.DEFAULT_ECM_ADDRESS = 'http://' + window.location.hostname + ':8080';
    AlfrescoSettingsService.DEFAULT_BPM_ADDRESS = 'http://' + window.location.hostname + ':9999';
    AlfrescoSettingsService.DEFAULT_CSRF_CONFIG = false;
    AlfrescoSettingsService.DEFAULT_BPM_CONTEXT_PATH = '/activiti-app';
    AlfrescoSettingsService = AlfrescoSettingsService_1 = __decorate([core_1.Injectable()], AlfrescoSettingsService);
    exports.AlfrescoSettingsService = AlfrescoSettingsService;
    var AlfrescoSettingsService_1;
});
$__System.registerDynamic("11", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var StorageService = function () {
        function StorageService() {
            this.memoryStore = {};
            this.useLocalStorage = false;
            this.useLocalStorage = this.storageAvailable('localStorage');
        }
        StorageService.prototype.getItem = function (key) {
            if (this.useLocalStorage) {
                return localStorage.getItem(key);
            } else {
                return this.memoryStore.hasOwnProperty(key) ? this.memoryStore[key] : null;
            }
        };
        StorageService.prototype.setItem = function (key, data) {
            if (this.useLocalStorage) {
                localStorage.setItem(key, data);
            } else {
                this.memoryStore[key] = data.toString();
            }
        };
        StorageService.prototype.clear = function () {
            if (this.useLocalStorage) {
                localStorage.clear();
            } else {
                this.memoryStore = {};
            }
        };
        StorageService.prototype.removeItem = function (key) {
            if (this.useLocalStorage) {
                localStorage.removeItem(key);
            } else {
                delete this.memoryStore[key];
            }
        };
        StorageService.prototype.hasItem = function (key) {
            if (this.useLocalStorage) {
                return localStorage.getItem(key) ? true : false;
            } else {
                return this.memoryStore.hasOwnProperty(key);
            }
        };
        StorageService.prototype.storageAvailable = function (type) {
            try {
                var storage = window[type];
                var key = '__storage_test__';
                storage.setItem(key, key);
                storage.removeItem(key, key);
                return true;
            } catch (e) {
                return false;
            }
        };
        return StorageService;
    }();
    StorageService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [])], StorageService);
    exports.StorageService = StorageService;
});
$__System.registerDynamic("5", ["3", "12", "f", "11"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var alfrescoApi = $__require("12");
    var alfresco_settings_service_1 = $__require("f");
    var storage_service_1 = $__require("11");
    var AlfrescoApiService = function () {
        function AlfrescoApiService(settingsService, storage) {
            var _this = this;
            this.settingsService = settingsService;
            this.storage = storage;
            this.provider = this.settingsService.getProviders();
            this.ticketEcm = this.getTicketEcm();
            this.ticketBpm = this.getTicketBpm();
            this.hostEcm = this.settingsService.ecmHost;
            this.hostBpm = this.settingsService.bpmHost;
            this.contextRoot = 'alfresco';
            this.disableCsrf = false;
            this.init();
            settingsService.bpmHostSubject.subscribe(function (hostBpm) {
                _this.hostBpm = hostBpm;
                _this.init();
            });
            settingsService.ecmHostSubject.subscribe(function (hostEcm) {
                _this.hostEcm = hostEcm;
                _this.init();
            });
            settingsService.csrfSubject.subscribe(function (disableCsrf) {
                _this.disableCsrf = disableCsrf;
                _this.init();
            });
            settingsService.providerSubject.subscribe(function (provider) {
                _this.provider = provider;
                _this.init();
            });
        }
        AlfrescoApiService.prototype.getInstance = function () {
            return this.alfrescoApi;
        };
        AlfrescoApiService.prototype.init = function () {
            this.alfrescoApi = new alfrescoApi({
                provider: this.provider,
                ticketEcm: this.ticketEcm,
                ticketBpm: this.ticketBpm,
                hostEcm: this.hostEcm,
                hostBpm: this.hostBpm,
                contextRoot: this.contextRoot,
                disableCsrf: this.disableCsrf
            });
        };
        AlfrescoApiService.prototype.getTicketEcm = function () {
            return this.storage.getItem('ticket-ECM');
        };
        AlfrescoApiService.prototype.getTicketBpm = function () {
            return this.storage.getItem('ticket-BPM');
        };
        return AlfrescoApiService;
    }();
    AlfrescoApiService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [alfresco_settings_service_1.AlfrescoSettingsService, storage_service_1.StorageService])], AlfrescoApiService);
    exports.AlfrescoApiService = AlfrescoApiService;
});
$__System.registerDynamic("4", ["3", "7", "f", "11", "8", "5"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("7");
    var alfresco_settings_service_1 = $__require("f");
    var storage_service_1 = $__require("11");
    var log_service_1 = $__require("8");
    var alfresco_api_service_1 = $__require("5");
    var AlfrescoAuthenticationService = function () {
        function AlfrescoAuthenticationService(settingsService, alfrescoApi, storage, logService) {
            this.settingsService = settingsService;
            this.alfrescoApi = alfrescoApi;
            this.storage = storage;
            this.logService = logService;
            this.loginSubject = new Rx_1.Subject();
            this.logoutSubject = new Rx_1.Subject();
        }
        AlfrescoAuthenticationService.prototype.isLoggedIn = function () {
            return !!this.alfrescoApi.getInstance().isLoggedIn();
        };
        AlfrescoAuthenticationService.prototype.login = function (username, password) {
            var _this = this;
            this.removeTicket();
            return Rx_1.Observable.fromPromise(this.callApiLogin(username, password)).map(function (response) {
                _this.saveTickets();
                _this.loginSubject.next(response);
                return { type: _this.settingsService.getProviders(), ticket: response };
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AlfrescoAuthenticationService.prototype.callApiLogin = function (username, password) {
            return this.alfrescoApi.getInstance().login(username, password);
        };
        AlfrescoAuthenticationService.prototype.logout = function () {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.callApiLogout()).map(function (res) {
                return res;
            }).do(function (response) {
                _this.removeTicket();
                _this.logoutSubject.next(response);
                return response;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AlfrescoAuthenticationService.prototype.callApiLogout = function () {
            if (this.alfrescoApi.getInstance()) {
                return this.alfrescoApi.getInstance().logout();
            }
        };
        AlfrescoAuthenticationService.prototype.removeTicket = function () {
            this.storage.removeItem('ticket-ECM');
            this.storage.removeItem('ticket-BPM');
            this.alfrescoApi.getInstance().setTicket(undefined, undefined);
        };
        AlfrescoAuthenticationService.prototype.getTicketEcm = function () {
            return this.storage.getItem('ticket-ECM');
        };
        AlfrescoAuthenticationService.prototype.getTicketBpm = function () {
            return this.storage.getItem('ticket-BPM');
        };
        AlfrescoAuthenticationService.prototype.getTicketEcmBase64 = function () {
            var ticket = this.storage.getItem('ticket-ECM');
            if (ticket) {
                return 'Basic ' + btoa(ticket);
            }
            return null;
        };
        AlfrescoAuthenticationService.prototype.saveTickets = function () {
            this.saveTicketEcm();
            this.saveTicketBpm();
        };
        AlfrescoAuthenticationService.prototype.saveTicketEcm = function () {
            if (this.alfrescoApi.getInstance() && this.alfrescoApi.getInstance().getTicketEcm()) {
                this.storage.setItem('ticket-ECM', this.alfrescoApi.getInstance().getTicketEcm());
            }
        };
        AlfrescoAuthenticationService.prototype.saveTicketBpm = function () {
            if (this.alfrescoApi.getInstance() && this.alfrescoApi.getInstance().getTicketBpm()) {
                this.storage.setItem('ticket-BPM', this.alfrescoApi.getInstance().getTicketBpm());
            }
        };
        AlfrescoAuthenticationService.prototype.isEcmLoggedIn = function () {
            return this.alfrescoApi.getInstance().ecmAuth && !!this.alfrescoApi.getInstance().ecmAuth.isLoggedIn();
        };
        AlfrescoAuthenticationService.prototype.isBpmLoggedIn = function () {
            return this.alfrescoApi.getInstance().bpmAuth && !!this.alfrescoApi.getInstance().bpmAuth.isLoggedIn();
        };
        AlfrescoAuthenticationService.prototype.handleError = function (error) {
            this.logService.error('Error when logging in', error);
            return Rx_1.Observable.throw(error || 'Server error');
        };
        return AlfrescoAuthenticationService;
    }();
    AlfrescoAuthenticationService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [alfresco_settings_service_1.AlfrescoSettingsService, alfresco_api_service_1.AlfrescoApiService, storage_service_1.StorageService, log_service_1.LogService])], AlfrescoAuthenticationService);
    exports.AlfrescoAuthenticationService = AlfrescoAuthenticationService;
});
$__System.registerDynamic("13", ["3", "14"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var ng2_translate_1 = $__require("14");
    var AlfrescoTranslationService = function () {
        function AlfrescoTranslationService(translate) {
            this.translate = translate;
            this.userLang = 'en';
            this.userLang = translate.getBrowserLang() || 'en';
            translate.setDefaultLang(this.userLang);
            this.customLoader = this.translate.currentLoader;
            this.customLoader.init(this.userLang);
        }
        AlfrescoTranslationService.prototype.addTranslationFolder = function (name, path) {
            var _this = this;
            if (name === void 0) {
                name = '';
            }
            if (path === void 0) {
                path = '';
            }
            if (!this.customLoader.existComponent(name)) {
                this.customLoader.addComponentList(name, path);
                this.translate.getTranslation(this.userLang).subscribe(function () {
                    _this.translate.use(_this.userLang);
                });
            }
        };
        AlfrescoTranslationService.prototype.use = function (lang) {
            this.customLoader.init(lang);
            return this.translate.use(lang);
        };
        AlfrescoTranslationService.prototype.get = function (key, interpolateParams) {
            return this.translate.get(key, interpolateParams);
        };
        return AlfrescoTranslationService;
    }();
    AlfrescoTranslationService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [ng2_translate_1.TranslateService])], AlfrescoTranslationService);
    exports.AlfrescoTranslationService = AlfrescoTranslationService;
});
$__System.registerDynamic("15", [], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var ComponentTranslationModel = function () {
        function ComponentTranslationModel(obj) {
            this.name = obj && obj.name;
            this.path = obj && obj.path;
            this.json = obj && obj.json || [];
        }
        return ComponentTranslationModel;
    }();
    exports.ComponentTranslationModel = ComponentTranslationModel;
});
$__System.registerDynamic("8", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var core_1 = $__require("3");
    var LogService = function () {
        function LogService() {}
        Object.defineProperty(LogService.prototype, "assert", {
            get: function () {
                return console.assert.bind(console);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LogService.prototype, "error", {
            get: function () {
                return console.error.bind(console);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LogService.prototype, "group", {
            get: function () {
                return console.group.bind(console);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LogService.prototype, "groupEnd", {
            get: function () {
                return console.groupEnd.bind(console);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LogService.prototype, "info", {
            get: function () {
                return console.info.bind(console);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LogService.prototype, "log", {
            get: function () {
                return console.log.bind(console);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(LogService.prototype, "warn", {
            get: function () {
                return console.warn.bind(console);
            },
            enumerable: true,
            configurable: true
        });
        return LogService;
    }();
    LogService = __decorate([core_1.Injectable()], LogService);
    exports.LogService = LogService;
    var LogServiceMock = function () {
        function LogServiceMock() {}
        LogServiceMock.prototype.assert = function (message) {
            var optionalParams = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                optionalParams[_i - 1] = arguments[_i];
            }
        };
        LogServiceMock.prototype.error = function (message) {
            var optionalParams = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                optionalParams[_i - 1] = arguments[_i];
            }
        };
        LogServiceMock.prototype.group = function (message) {
            var optionalParams = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                optionalParams[_i - 1] = arguments[_i];
            }
        };
        LogServiceMock.prototype.groupEnd = function (message) {
            var optionalParams = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                optionalParams[_i - 1] = arguments[_i];
            }
        };
        LogServiceMock.prototype.info = function (message) {
            var optionalParams = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                optionalParams[_i - 1] = arguments[_i];
            }
        };
        LogServiceMock.prototype.log = function (message) {
            var optionalParams = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                optionalParams[_i - 1] = arguments[_i];
            }
        };
        LogServiceMock.prototype.warn = function (message) {
            var optionalParams = [];
            for (var _i = 1; _i < arguments.length; _i++) {
                optionalParams[_i - 1] = arguments[_i];
            }
        };
        return LogServiceMock;
    }();
    exports.LogServiceMock = LogServiceMock;
});
$__System.registerDynamic("16", ["3", "17", "7", "15", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var http_1 = $__require("17");
    var Rx_1 = $__require("7");
    var component_model_1 = $__require("15");
    var log_service_1 = $__require("8");
    var AlfrescoTranslateLoader = function () {
        function AlfrescoTranslateLoader(http, logService) {
            this.http = http;
            this.logService = logService;
            this.prefix = 'i18n';
            this.suffix = '.json';
            this._componentList = [];
            this.queue = [];
        }
        AlfrescoTranslateLoader.prototype.addComponentList = function (nameInput, pathInput) {
            this._componentList.push(new component_model_1.ComponentTranslationModel({ name: nameInput, path: pathInput }));
        };
        AlfrescoTranslateLoader.prototype.existComponent = function (name) {
            return this._componentList.find(function (x) {
                return x.name === name;
            }) ? true : false;
        };
        AlfrescoTranslateLoader.prototype.getComponentToFetch = function (lang) {
            var _this = this;
            var observableBatch = [];
            this._componentList.forEach(function (component) {
                if (!_this.isComponentInQueue(lang, component.name)) {
                    _this.queue[lang].push(component.name);
                    observableBatch.push(_this.http.get(component.path + "/" + _this.prefix + "/" + lang + _this.suffix).map(function (res) {
                        component.json[lang] = res.json();
                    }).catch(function () {
                        return Rx_1.Observable.of('');
                    }));
                }
            });
            return observableBatch;
        };
        AlfrescoTranslateLoader.prototype.init = function (lang) {
            if (this.queue[lang] === undefined) {
                this.queue[lang] = [];
            }
        };
        AlfrescoTranslateLoader.prototype.isComponentInQueue = function (lang, name) {
            return this.queue[lang].find(function (x) {
                return x === name;
            }) ? true : false;
        };
        AlfrescoTranslateLoader.prototype.getFullTranslationJSON = function (lang) {
            var fullTranslation = '';
            var cloneList = this._componentList.slice(0);
            cloneList.reverse().forEach(function (component) {
                if (component.json && component.json[lang]) {
                    fullTranslation += JSON.stringify(component.json[lang]);
                }
            });
            if (fullTranslation !== '') {
                return JSON.parse(fullTranslation.replace(/}{/g, ','));
            }
        };
        AlfrescoTranslateLoader.prototype.getTranslation = function (lang) {
            var _this = this;
            var observableBatch = this.getComponentToFetch(lang);
            return Rx_1.Observable.create(function (observer) {
                if (observableBatch.length > 0) {
                    Rx_1.Observable.forkJoin(observableBatch).subscribe(function () {
                        var fullTranslation = _this.getFullTranslationJSON(lang);
                        if (fullTranslation) {
                            observer.next(fullTranslation);
                        }
                        observer.complete();
                    }, function (err) {
                        _this.logService.error(err);
                    });
                } else {
                    var fullTranslation = _this.getFullTranslationJSON(lang);
                    if (fullTranslation) {
                        observer.next(fullTranslation);
                    }
                }
            });
        };
        return AlfrescoTranslateLoader;
    }();
    AlfrescoTranslateLoader = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [http_1.Http, log_service_1.LogService])], AlfrescoTranslateLoader);
    exports.AlfrescoTranslateLoader = AlfrescoTranslateLoader;
});
$__System.registerDynamic("18", ["11", "5", "f", "2", "6", "9", "b", "c", "d", "8", "4", "13", "16"], true, function ($__require, exports, module) {
  /*!
   * @license
   * Copyright 2016 Alfresco Software, Ltd.
   *
   * Licensed under the Apache License, Version 2.0 (the "License");
   * you may not use this file except in compliance with the License.
   * You may obtain a copy of the License at
   *
   *     http://www.apache.org/licenses/LICENSE-2.0
   *
   * Unless required by applicable law or agreed to in writing, software
   * distributed under the License is distributed on an "AS IS" BASIS,
   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   * See the License for the specific language governing permissions and
   * limitations under the License.
   */
  "use strict";

  var global = this || self,
      GLOBAL = global;
  function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
  }
  __export($__require("11"));
  __export($__require("5"));
  __export($__require("f"));
  __export($__require("2"));
  __export($__require("6"));
  __export($__require("9"));
  __export($__require("b"));
  __export($__require("c"));
  __export($__require("d"));
  __export($__require("8"));
  __export($__require("4"));
  __export($__require("13"));
  __export($__require("16"));
});
$__System.registerDynamic("19", ["3", "1a"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var context_menu_service_1 = $__require("1a");
    var ContextMenuHolderComponent = function () {
        function ContextMenuHolderComponent(_contextMenuService) {
            var _this = this;
            this._contextMenuService = _contextMenuService;
            this.links = [];
            this.isShown = false;
            this.mouseLocation = { left: 0, top: 0 };
            _contextMenuService.show.subscribe(function (e) {
                return _this.showMenu(e.event, e.obj);
            });
        }
        Object.defineProperty(ContextMenuHolderComponent.prototype, "locationCss", {
            get: function () {
                return {
                    position: 'fixed',
                    display: this.isShown ? 'block' : 'none',
                    left: this.mouseLocation.left + 'px',
                    top: this.mouseLocation.top + 'px'
                };
            },
            enumerable: true,
            configurable: true
        });
        ContextMenuHolderComponent.prototype.clickedOutside = function () {
            this.isShown = false;
        };
        ContextMenuHolderComponent.prototype.showMenu = function (e, links) {
            this.isShown = true;
            this.links = links;
            if (e) {
                this.mouseLocation = {
                    left: e.clientX,
                    top: e.clientY
                };
            }
        };
        ContextMenuHolderComponent.prototype.onShowContextMenu = function (event) {
            if (event) {
                event.preventDefault();
            }
        };
        return ContextMenuHolderComponent;
    }();
    __decorate([core_1.HostListener('document:click'), __metadata("design:type", Function), __metadata("design:paramtypes", []), __metadata("design:returntype", void 0)], ContextMenuHolderComponent.prototype, "clickedOutside", null);
    __decorate([core_1.HostListener('contextmenu', ['$event']), __metadata("design:type", Function), __metadata("design:paramtypes", [MouseEvent]), __metadata("design:returntype", void 0)], ContextMenuHolderComponent.prototype, "onShowContextMenu", null);
    ContextMenuHolderComponent = __decorate([core_1.Component({
        selector: 'context-menu-holder',
        styles: ["\n        .menu-container {\n            background: #fff;\n            display: block;\n            margin: 0;\n            padding: 0;\n            border: none;\n            overflow: visible;\n            z-index: 9999;\n        }\n\n        .context-menu {\n            list-style-type: none;\n            position: static;\n            height: auto;\n            width: auto;\n            min-width: 124px;\n            padding: 8px 0;\n            margin: 0;\n            box-shadow: 0 2px 2px 0 rgba(0,0,0,.14),0 3px 1px -2px rgba(0,0,0,.2),0 1px 5px 0 rgba(0,0,0,.12);\n            border-radius: 2px;\n        }\n\n        .context-menu .link {\n            opacity: 1;\n        }\n        "],
        template: "\n        <div [ngStyle]=\"locationCss\" class=\"menu-container\">\n            <ul class=\"context-menu\">\n                <li (click)=\"link.subject.next(link)\" class=\"mdl-menu__item link\" *ngFor=\"let link of links\">\n                    {{link.title || link.model?.title}}\n                </li>\n            </ul>\n        </div>\n    "
    }), __metadata("design:paramtypes", [context_menu_service_1.ContextMenuService])], ContextMenuHolderComponent);
    exports.ContextMenuHolderComponent = ContextMenuHolderComponent;
});
$__System.registerDynamic("1a", ["3", "7"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("7");
    var ContextMenuService = function () {
        function ContextMenuService() {
            this.show = new Rx_1.Subject();
        }
        return ContextMenuService;
    }();
    ContextMenuService = __decorate([core_1.Injectable()], ContextMenuService);
    exports.ContextMenuService = ContextMenuService;
});
$__System.registerDynamic("1b", ["3", "1a"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var context_menu_service_1 = $__require("1a");
    var ContextMenuDirective = function () {
        function ContextMenuDirective(_contextMenuService) {
            this._contextMenuService = _contextMenuService;
        }
        ContextMenuDirective.prototype.onShowContextMenu = function (event) {
            if (event) {
                event.preventDefault();
            }
            if (this.links && this.links.length > 0) {
                if (this._contextMenuService) {
                    this._contextMenuService.show.next({ event: event, obj: this.links });
                }
            }
        };
        return ContextMenuDirective;
    }();
    __decorate([core_1.Input('context-menu'), __metadata("design:type", Array)], ContextMenuDirective.prototype, "links", void 0);
    __decorate([core_1.HostListener('contextmenu', ['$event']), __metadata("design:type", Function), __metadata("design:paramtypes", [MouseEvent]), __metadata("design:returntype", void 0)], ContextMenuDirective.prototype, "onShowContextMenu", null);
    ContextMenuDirective = __decorate([core_1.Directive({
        selector: '[context-menu]'
    }), __metadata("design:paramtypes", [context_menu_service_1.ContextMenuService])], ContextMenuDirective);
    exports.ContextMenuDirective = ContextMenuDirective;
});
$__System.registerDynamic("1c", ["1a", "19", "1b"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var context_menu_service_1 = $__require("1a");
    var context_menu_holder_component_1 = $__require("19");
    var context_menu_directive_1 = $__require("1b");
    __export($__require("1a"));
    __export($__require("19"));
    __export($__require("1b"));
    exports.CONTEXT_MENU_PROVIDERS = [context_menu_service_1.ContextMenuService];
    exports.CONTEXT_MENU_DIRECTIVES = [context_menu_holder_component_1.ContextMenuHolderComponent, context_menu_directive_1.ContextMenuDirective];
});
$__System.registerDynamic("1d", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var core_1 = $__require("3");
    var MDL = function () {
        function MDL() {}
        MDL.prototype.ngAfterViewInit = function () {
            componentHandler.upgradeAllRegistered();
        };
        return MDL;
    }();
    MDL = __decorate([core_1.Directive({
        selector: '[mdl]'
    })], MDL);
    exports.MDL = MDL;
});
$__System.registerDynamic("1e", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var AlfrescoMdlButtonDirective = function () {
        function AlfrescoMdlButtonDirective(element) {
            this.element = element;
        }
        AlfrescoMdlButtonDirective.prototype.ngAfterViewInit = function () {
            if (componentHandler) {
                var el = this.element.nativeElement;
                el.classList.add('mdl-button');
                el.classList.add('mdl-js-button');
                el.classList.add('mdl-js-ripple-effect');
                componentHandler.upgradeElement(el, 'MaterialButton');
            }
        };
        return AlfrescoMdlButtonDirective;
    }();
    AlfrescoMdlButtonDirective = __decorate([core_1.Directive({
        selector: '[alfresco-mdl-button]'
    }), __metadata("design:paramtypes", [core_1.ElementRef])], AlfrescoMdlButtonDirective);
    exports.AlfrescoMdlButtonDirective = AlfrescoMdlButtonDirective;
});
$__System.registerDynamic("1f", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var AlfrescoMdlMenuDirective = function () {
        function AlfrescoMdlMenuDirective(element) {
            this.element = element;
        }
        AlfrescoMdlMenuDirective.prototype.ngAfterViewInit = function () {
            if (componentHandler) {
                var el = this.element.nativeElement;
                el.classList.add('mdl-menu');
                el.classList.add('mdl-js-menu');
                el.classList.add('mdl-js-ripple-effect');
                componentHandler.upgradeElement(el, 'MaterialMenu');
            }
        };
        return AlfrescoMdlMenuDirective;
    }();
    AlfrescoMdlMenuDirective = __decorate([core_1.Directive({
        selector: '[alfresco-mdl-menu]'
    }), __metadata("design:paramtypes", [core_1.ElementRef])], AlfrescoMdlMenuDirective);
    exports.AlfrescoMdlMenuDirective = AlfrescoMdlMenuDirective;
});
$__System.registerDynamic("20", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var AlfrescoMdlTabsDirective = function () {
        function AlfrescoMdlTabsDirective(element) {
            this.element = element;
        }
        AlfrescoMdlTabsDirective.prototype.ngAfterViewInit = function () {
            if (componentHandler) {
                var el_1 = this.element.nativeElement;
                el_1.classList.add('mdl-tabs');
                el_1.classList.add('mdl-js-tabs');
                el_1.classList.add('mdl-js-ripple-effect');
                componentHandler.upgradeElement(el_1, 'MaterialTabs');
                var tabBar = el_1.querySelector('.mdl-tabs__tab-bar');
                if (tabBar) {
                    this.observer = new MutationObserver(function (mutations) {
                        var upgrade = false;
                        mutations.forEach(function (mutation) {
                            if (mutation.addedNodes && mutation.addedNodes.length > 0) {
                                upgrade = true;
                            }
                            if (mutation.removedNodes && mutation.removedNodes.length > 0) {
                                upgrade = true;
                            }
                        });
                        if (upgrade) {
                            componentHandler.downgradeElements([el_1]);
                            componentHandler.upgradeElement(el_1);
                        }
                    });
                    this.observer.observe(tabBar, {
                        childList: true,
                        subtree: false
                    });
                }
            }
        };
        AlfrescoMdlTabsDirective.prototype.ngOnDestroy = function () {
            if (this.observer) {
                this.observer.disconnect();
            }
        };
        return AlfrescoMdlTabsDirective;
    }();
    AlfrescoMdlTabsDirective = __decorate([core_1.Directive({
        selector: '[alfresco-mdl-tabs]'
    }), __metadata("design:paramtypes", [core_1.ElementRef])], AlfrescoMdlTabsDirective);
    exports.AlfrescoMdlTabsDirective = AlfrescoMdlTabsDirective;
});
$__System.registerDynamic("21", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var AlfrescoMdlTextFieldDirective = function () {
        function AlfrescoMdlTextFieldDirective(element) {
            this.element = element;
        }
        AlfrescoMdlTextFieldDirective.prototype.ngAfterViewInit = function () {
            if (componentHandler) {
                var el = this.element.nativeElement;
                el.classList.add('mdl-textfield');
                el.classList.add('mdl-js-textfield');
                el.classList.add('mdl-textfield--floating-label');
                componentHandler.upgradeElement(el, 'MaterialTextfield');
            }
        };
        return AlfrescoMdlTextFieldDirective;
    }();
    AlfrescoMdlTextFieldDirective = __decorate([core_1.Directive({
        selector: '[alfresco-mdl-textfield]'
    }), __metadata("design:paramtypes", [core_1.ElementRef])], AlfrescoMdlTextFieldDirective);
    exports.AlfrescoMdlTextFieldDirective = AlfrescoMdlTextFieldDirective;
});
$__System.registerDynamic("22", ["1d", "1e", "1f", "20", "21"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var mdl_upgrade_element_directive_1 = $__require("1d");
    var mdl_button_directive_1 = $__require("1e");
    var mdl_menu_directive_1 = $__require("1f");
    var mdl_tabs_directive_1 = $__require("20");
    var mdl_textfield_directive_1 = $__require("21");
    __export($__require("1d"));
    __export($__require("1e"));
    __export($__require("1f"));
    __export($__require("20"));
    __export($__require("21"));
    exports.MATERIAL_DESIGN_DIRECTIVES = [mdl_upgrade_element_directive_1.MDL, mdl_button_directive_1.AlfrescoMdlButtonDirective, mdl_menu_directive_1.AlfrescoMdlMenuDirective, mdl_tabs_directive_1.AlfrescoMdlTabsDirective, mdl_textfield_directive_1.AlfrescoMdlTextFieldDirective];
});
$__System.registerDynamic("23", ["1c", "22"], true, function ($__require, exports, module) {
  /*!
   * @license
   * Copyright 2016 Alfresco Software, Ltd.
   *
   * Licensed under the Apache License, Version 2.0 (the "License");
   * you may not use this file except in compliance with the License.
   * You may obtain a copy of the License at
   *
   *     http://www.apache.org/licenses/LICENSE-2.0
   *
   * Unless required by applicable law or agreed to in writing, software
   * distributed under the License is distributed on an "AS IS" BASIS,
   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   * See the License for the specific language governing permissions and
   * limitations under the License.
   */
  "use strict";

  var global = this || self,
      GLOBAL = global;
  function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
  }
  __export($__require("1c"));
  __export($__require("22"));
});
$__System.registerDynamic('24', [], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var ObjectUtils = function () {
        function ObjectUtils() {}
        ObjectUtils.getValue = function (target, key) {
            if (!target) {
                return undefined;
            }
            var keys = key.split('.');
            key = '';
            do {
                key += keys.shift();
                var value = target[key];
                if (value !== undefined && (typeof value === 'object' || !keys.length)) {
                    target = value;
                    key = '';
                } else if (!keys.length) {
                    target = undefined;
                } else {
                    key += '.';
                }
            } while (keys.length);
            return target;
        };
        return ObjectUtils;
    }();
    exports.ObjectUtils = ObjectUtils;
});
$__System.registerDynamic("25", ["24"], true, function ($__require, exports, module) {
  /*!
   * @license
   * Copyright 2016 Alfresco Software, Ltd.
   *
   * Licensed under the Apache License, Version 2.0 (the "License");
   * you may not use this file except in compliance with the License.
   * You may obtain a copy of the License at
   *
   *     http://www.apache.org/licenses/LICENSE-2.0
   *
   * Unless required by applicable law or agreed to in writing, software
   * distributed under the License is distributed on an "AS IS" BASIS,
   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   * See the License for the specific language governing permissions and
   * limitations under the License.
   */
  "use strict";

  var global = this || self,
      GLOBAL = global;
  function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
  }
  __export($__require("24"));
});
$__System.registerDynamic("1", ["3", "17", "26", "27", "14", "e", "18", "22", "1c", "23", "25"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var core_1 = $__require("3");
    var http_1 = $__require("17");
    var common_1 = $__require("26");
    var forms_1 = $__require("27");
    var ng2_translate_1 = $__require("14");
    var material_1 = $__require("e");
    var index_1 = $__require("18");
    var index_2 = $__require("22");
    var index_3 = $__require("1c");
    __export($__require("18"));
    __export($__require("23"));
    __export($__require("25"));
    exports.ALFRESCO_CORE_PROVIDERS = [index_1.NotificationService, index_1.LogService, index_1.LogServiceMock, index_1.AlfrescoAuthenticationService, index_1.AlfrescoContentService, index_1.AlfrescoSettingsService, index_1.StorageService, index_1.AlfrescoApiService, index_1.AlfrescoTranslateLoader, index_1.AlfrescoTranslationService, index_1.RenditionsService, index_1.AuthGuard, index_1.AuthGuardEcm, index_1.AuthGuardBpm].concat(index_3.CONTEXT_MENU_PROVIDERS);
    function createTranslateLoader(http, logService) {
        return new index_1.AlfrescoTranslateLoader(http, logService);
    }
    exports.createTranslateLoader = createTranslateLoader;
    var CoreModule = CoreModule_1 = function () {
        function CoreModule() {}
        CoreModule.forRoot = function () {
            return {
                ngModule: CoreModule_1,
                providers: exports.ALFRESCO_CORE_PROVIDERS.slice()
            };
        };
        return CoreModule;
    }();
    CoreModule = CoreModule_1 = __decorate([core_1.NgModule({
        imports: [common_1.CommonModule, forms_1.FormsModule, forms_1.ReactiveFormsModule, http_1.HttpModule, material_1.MaterialModule.forRoot(), ng2_translate_1.TranslateModule.forRoot({
            provide: ng2_translate_1.TranslateLoader,
            useFactory: createTranslateLoader,
            deps: [http_1.Http, index_1.LogService]
        })],
        declarations: index_2.MATERIAL_DESIGN_DIRECTIVES.concat(index_3.CONTEXT_MENU_DIRECTIVES),
        providers: exports.ALFRESCO_CORE_PROVIDERS.slice(),
        exports: [common_1.CommonModule, forms_1.FormsModule, material_1.MaterialModule, forms_1.ReactiveFormsModule, http_1.HttpModule, ng2_translate_1.TranslateModule].concat(index_2.MATERIAL_DESIGN_DIRECTIVES, index_3.CONTEXT_MENU_DIRECTIVES)
    })], CoreModule);
    exports.CoreModule = CoreModule;
    var CoreModule_1;
    
});
})
(function(factory) {
  if (typeof define == 'function' && define.amd)
    define(["@angular/common","@angular/core","@angular/forms","@angular/http","@angular/material","@angular/router","alfresco-js-api","ng2-translate/ng2-translate","rxjs/Rx","rxjs/Subject"], factory);
  else if (typeof module == 'object' && module.exports && typeof require == 'function')
    module.exports = factory(require("@angular/common"), require("@angular/core"), require("@angular/forms"), require("@angular/http"), require("@angular/material"), require("@angular/router"), require("alfresco-js-api"), require("ng2-translate/ng2-translate"), require("rxjs/Rx"), require("rxjs/Subject"));
  else
    throw new Error("Module must be loaded as AMD or CommonJS");
});
