// ng2-activiti-analytics/bundles/ng2-activiti-analytics.js
!function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in v||(v[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==g.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=v[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(g.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=v[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}n=!n}}function u(e){return y[e]||(y[e]={name:e,dependencies:[],exports:{},importers:[]})}function d(r){if(!r.module){var t=r.module=u(r.name),n=r.module.exports,o=r.declare.call(e,function(e,r){if(t.locked=!0,"object"==typeof e)for(var o in e)n[o]=e[o];else n[e]=r;for(var a=0,u=t.importers.length;u>a;a++){var d=t.importers[a];if(!d.locked)for(var i=0;i<d.dependencies.length;++i)d.dependencies[i]===t&&d.setters[i](n)}return t.locked=!1,r},{id:r.name});t.setters=o.setters,t.execute=o.execute;for(var a=0,i=r.normalizedDeps.length;i>a;a++){var l,s=r.normalizedDeps[a],c=v[s],f=y[s];f?l=f.exports:c&&!c.declarative?l=c.esModule:c?(d(c),f=c.module,l=f.exports):l=p(s),f&&f.importers?(f.importers.push(t),t.dependencies.push(f)):t.dependencies.push(null),t.setters[a]&&t.setters[a](l)}}}function i(e){var r,t=v[e];if(t)t.declarative?f(e,[]):t.evaluated||l(t),r=t.module.exports;else if(r=p(e),!r)throw new Error("Unable to load dependency "+e+".");return(!t||t.declarative)&&r&&r.__useDefault?r["default"]:r}function l(r){if(!r.module){var t={},n=r.module={exports:t,id:r.name};if(!r.executingRequire)for(var o=0,a=r.normalizedDeps.length;a>o;o++){var u=r.normalizedDeps[o],d=v[u];d&&l(d)}r.evaluated=!0;var c=r.execute.call(e,function(e){for(var t=0,n=r.deps.length;n>t;t++)if(r.deps[t]==e)return i(r.normalizedDeps[t]);throw new TypeError("Module "+e+" not declared as a dependency.")},t,n);void 0!==c&&(n.exports=c),t=n.exports,t&&t.__esModule?r.esModule=t:r.esModule=s(t)}}function s(r){var t={};if(("object"==typeof r||"function"==typeof r)&&r!==e)if(m)for(var n in r)"default"!==n&&c(t,r,n);else{var o=r&&r.hasOwnProperty;for(var n in r)"default"===n||o&&!r.hasOwnProperty(n)||(t[n]=r[n])}return t["default"]=r,x(t,"__useDefault",{value:!0}),t}function c(e,r,t){try{var n;(n=Object.getOwnPropertyDescriptor(r,t))&&x(e,t,n)}catch(o){return e[t]=r[t],!1}}function f(r,t){var n=v[r];if(n&&!n.evaluated&&n.declarative){t.push(r);for(var o=0,a=n.normalizedDeps.length;a>o;o++){var u=n.normalizedDeps[o];-1==g.call(t,u)&&(v[u]?f(u,t):p(u))}n.evaluated||(n.evaluated=!0,n.module.execute.call(e))}}function p(e){if(I[e])return I[e];if("@node/"==e.substr(0,6))return I[e]=s(D(e.substr(6)));var r=v[e];if(!r)throw"Module "+e+" not present.";return a(e),f(e,[]),v[e]=void 0,r.declarative&&x(r.module.exports,"__esModule",{value:!0}),I[e]=r.declarative?r.module.exports:r.esModule}var v={},g=Array.prototype.indexOf||function(e){for(var r=0,t=this.length;t>r;r++)if(this[r]===e)return r;return-1},m=!0;try{Object.getOwnPropertyDescriptor({a:0},"a")}catch(h){m=!1}var x;!function(){try{Object.defineProperty({},"a",{})&&(x=Object.defineProperty)}catch(e){x=function(e,r,t){try{e[r]=t.value||t.get.call(e)}catch(n){}}}}();var y={},D="undefined"!=typeof System&&System._nodeRequire||"undefined"!=typeof require&&"undefined"!=typeof require.resolve&&"undefined"!=typeof process&&process.platform&&require,I={"@empty":{}};return function(e,n,o,a){return function(u){u(function(u){for(var d={_nodeRequire:D,register:r,registerDynamic:t,get:p,set:function(e,r){I[e]=r},newModule:function(e){return e}},i=0;i<n.length;i++)(function(e,r){r&&r.__esModule?I[e]=r:I[e]=s(r)})(n[i],arguments[i]);a(d);var l=p(e[0]);if(e.length>1)for(var i=1;i<e.length;i++)p(e[i]);return o?l["default"]:l})}}}("undefined"!=typeof self?self:global)

(["1"], ["3","8","16","5","17","4"], true, function($__System) {
var require = this.require, exports = this.exports, module = this.module;
!function(e){function n(e,n){e=e.replace(l,"");var r=e.match(u),t=(r[1].split(",")[n]||"require").replace(s,""),i=p[t]||(p[t]=new RegExp(a+t+f,"g"));i.lastIndex=0;for(var o,c=[];o=i.exec(e);)c.push(o[2]||o[3]);return c}function r(e,n,t,o){if("object"==typeof e&&!(e instanceof Array))return r.apply(null,Array.prototype.splice.call(arguments,1,arguments.length-1));if("string"==typeof e&&"function"==typeof n&&(e=[e]),!(e instanceof Array)){if("string"==typeof e){var l=i.get(e);return l.__useDefault?l["default"]:l}throw new TypeError("Invalid require")}for(var a=[],f=0;f<e.length;f++)a.push(i["import"](e[f],o));Promise.all(a).then(function(e){n&&n.apply(null,e)},t)}function t(t,l,a){"string"!=typeof t&&(a=l,l=t,t=null),l instanceof Array||(a=l,l=["require","exports","module"].splice(0,a.length)),"function"!=typeof a&&(a=function(e){return function(){return e}}(a)),void 0===l[l.length-1]&&l.pop();var f,u,s;-1!=(f=o.call(l,"require"))&&(l.splice(f,1),t||(l=l.concat(n(a.toString(),f)))),-1!=(u=o.call(l,"exports"))&&l.splice(u,1),-1!=(s=o.call(l,"module"))&&l.splice(s,1);var p={name:t,deps:l,execute:function(n,t,o){for(var p=[],c=0;c<l.length;c++)p.push(n(l[c]));o.uri=o.id,o.config=function(){},-1!=s&&p.splice(s,0,o),-1!=u&&p.splice(u,0,t),-1!=f&&p.splice(f,0,function(e,t,l){return"string"==typeof e&&"function"!=typeof t?n(e):r.call(i,e,t,l,o.id)});var d=a.apply(-1==u?e:t,p);return"undefined"==typeof d&&o&&(d=o.exports),"undefined"!=typeof d?d:void 0}};if(t)c.anonDefine||c.isBundle?c.anonDefine&&c.anonDefine.name&&(c.anonDefine=null):c.anonDefine=p,c.isBundle=!0,i.registerDynamic(p.name,p.deps,!1,p.execute);else{if(c.anonDefine&&!c.anonDefine.name)throw new Error("Multiple anonymous defines in module "+t);c.anonDefine=p}}var i=$__System,o=Array.prototype.indexOf||function(e){for(var n=0,r=this.length;r>n;n++)if(this[n]===e)return n;return-1},l=/(\/\*([\s\S]*?)\*\/|([^:]|^)\/\/(.*)$)/gm,a="(?:^|[^$_a-zA-Z\\xA0-\\uFFFF.])",f="\\s*\\(\\s*(\"([^\"]+)\"|'([^']+)')\\s*\\)",u=/\(([^\)]*)\)/,s=/^\s+|\s+$/g,p={};t.amd={};var c={isBundle:!1,anonDefine:null};i.amdDefine=t,i.amdRequire=r}("undefined"!=typeof self?self:global);
$__System.registerDynamic("2", ["3", "4", "5", "6"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("4");
    var ng2_alfresco_core_1 = $__require("5");
    var analytics_service_1 = $__require("6");
    var AnalyticsReportListComponent = function () {
        function AnalyticsReportListComponent(analyticsService, logService) {
            var _this = this;
            this.analyticsService = analyticsService;
            this.logService = logService;
            this.reportClick = new core_1.EventEmitter();
            this.onSuccess = new core_1.EventEmitter();
            this.onError = new core_1.EventEmitter();
            this.reports = [];
            this.report$ = new Rx_1.Observable(function (observer) {
                return _this.reportObserver = observer;
            }).share();
        }
        AnalyticsReportListComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.report$.subscribe(function (report) {
                _this.reports.push(report);
            });
            this.getReportList();
        };
        AnalyticsReportListComponent.prototype.reload = function () {
            this.reset();
            this.getReportList();
        };
        AnalyticsReportListComponent.prototype.getReportList = function () {
            var _this = this;
            this.analyticsService.getReportList().subscribe(function (res) {
                if (res && res.length === 0) {
                    _this.createDefaultReports();
                } else {
                    res.forEach(function (report) {
                        _this.reportObserver.next(report);
                    });
                    _this.onSuccess.emit(res);
                }
            }, function (err) {
                _this.onError.emit(err);
                _this.logService.error(err);
            });
        };
        AnalyticsReportListComponent.prototype.createDefaultReports = function () {
            var _this = this;
            this.analyticsService.createDefaultReports().subscribe(function () {
                _this.analyticsService.getReportList().subscribe(function (response) {
                    response.forEach(function (report) {
                        _this.reportObserver.next(report);
                    });
                    _this.onSuccess.emit(response);
                });
            });
        };
        AnalyticsReportListComponent.prototype.isReportsEmpty = function () {
            return this.reports === undefined || this.reports && this.reports.length === 0;
        };
        AnalyticsReportListComponent.prototype.reset = function () {
            if (!this.isReportsEmpty()) {
                this.reports = [];
            }
        };
        AnalyticsReportListComponent.prototype.selectReport = function (report) {
            this.currentReport = report;
            this.reportClick.emit(report);
        };
        return AnalyticsReportListComponent;
    }();
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], AnalyticsReportListComponent.prototype, "reportClick", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportListComponent.prototype, "onSuccess", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportListComponent.prototype, "onError", void 0);
    AnalyticsReportListComponent = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'analytics-report-list',
        template: "<div class=\"menu-container\">     <ul class='mdl-list'>         <li class=\"mdl-list__item activiti-filters__entry\" (click)=\"selectReport(report)\" *ngFor=\"let report of reports; let idx = index\"             [class.active]=\"currentReport === report\">             <span [attr.id]=\"'report-list-' + idx\" class=\"mdl-list__item-primary-content\">             <i class=\"material-icons mdl-list__item-icon activiti-filters__entry-icon\" [attr.data-automation-id]=\"report.name + '_filter'\">assignment</i>             <span class=\"text\">{{report.name}}</span>             </span>         </li>     </ul> </div>",
        styles: [".mdl-list__item {     cursor: pointer; }  .activiti-filters__entry {     cursor: pointer; }  .activiti-filters__entry-icon {     margin-right: 12px !important; }  .activiti-filters__entry.active {     color: rgb(68,138,255); }  .activiti-filters__entry.active .activiti-filters__entry-icon {     color: rgb(68,138,255); }"]
    }), __metadata("design:paramtypes", [analytics_service_1.AnalyticsService, ng2_alfresco_core_1.LogService])], AnalyticsReportListComponent);
    exports.AnalyticsReportListComponent = AnalyticsReportListComponent;
});
$__System.registerDynamic("7", ["3", "8", "9", "5", "6", "a"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var forms_1 = $__require("8");
    var moment = $__require("9");
    var ng2_alfresco_core_1 = $__require("5");
    var analytics_service_1 = $__require("6");
    var report_model_1 = $__require("a");
    var AnalyticsReportParametersComponent = AnalyticsReportParametersComponent_1 = function () {
        function AnalyticsReportParametersComponent(translateService, analyticsService, formBuilder, logService) {
            this.translateService = translateService;
            this.analyticsService = analyticsService;
            this.formBuilder = formBuilder;
            this.logService = logService;
            this.debug = false;
            this.onSuccess = new core_1.EventEmitter();
            this.onError = new core_1.EventEmitter();
            this.onEdit = new core_1.EventEmitter();
            this.onFormValueChanged = new core_1.EventEmitter();
            this.onDropdownChanged = new core_1.EventEmitter();
            this.onSuccessReportParams = new core_1.EventEmitter();
            this.onSuccessParamOpt = new core_1.EventEmitter();
            this.isEditable = false;
            this.hideParameters = true;
            if (translateService) {
                translateService.addTranslationFolder('ng2-activiti-analytics', 'node_modules/ng2-activiti-analytics/src');
            }
        }
        AnalyticsReportParametersComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.initForm();
            this.dropDownSub = this.onDropdownChanged.subscribe(function (field) {
                var paramDependOn = _this.reportParameters.definition.parameters.find(function (p) {
                    return p.dependsOn === field.id;
                });
                if (paramDependOn) {
                    _this.retrieveParameterOptions(_this.reportParameters.definition.parameters, _this.appId, _this.reportId, field.value);
                }
            });
            this.paramOpts = this.onSuccessReportParams.subscribe(function (report) {
                if (report.hasParameters()) {
                    _this.retrieveParameterOptions(report.definition.parameters, _this.appId);
                }
            });
            this.reportForm.valueChanges.subscribe(function (data) {
                return _this.onValueChanged(data);
            });
        };
        AnalyticsReportParametersComponent.prototype.ngOnChanges = function (changes) {
            this.isEditable = false;
            var reportId = changes['reportId'];
            if (reportId && reportId.currentValue) {
                this.getReportParams(reportId.currentValue);
            }
            var appId = changes['appId'];
            if (appId && (appId.currentValue || appId.currentValue === null)) {
                this.getReportParams(this.reportId);
            }
        };
        AnalyticsReportParametersComponent.prototype.initForm = function () {
            this.reportForm = this.formBuilder.group({
                dateRange: new forms_1.FormGroup({}),
                statusGroup: new forms_1.FormGroup({
                    status: new forms_1.FormControl()
                }),
                processInstanceGroup: new forms_1.FormGroup({
                    slowProcessInstanceInteger: new forms_1.FormControl()
                }),
                taskGroup: new forms_1.FormGroup({
                    taskName: new forms_1.FormControl()
                }),
                typeFilteringGroup: new forms_1.FormGroup({
                    typeFiltering: new forms_1.FormControl()
                }),
                dateIntervalGroup: new forms_1.FormGroup({
                    dateRangeInterval: new forms_1.FormControl()
                }),
                durationGroup: new forms_1.FormGroup({
                    duration: new forms_1.FormControl()
                }),
                processDefGroup: new forms_1.FormGroup({
                    processDefinitionId: new forms_1.FormControl()
                })
            });
        };
        AnalyticsReportParametersComponent.prototype.getReportParams = function (reportId) {
            var _this = this;
            this.reportParamsSub = this.analyticsService.getReportParams(reportId).subscribe(function (res) {
                _this.reportParameters = res;
                if (_this.reportParameters.hasParameters()) {
                    _this.onSuccessReportParams.emit(res);
                } else {
                    _this.onSuccess.emit();
                }
            }, function (err) {
                _this.logService.error(err);
                _this.onError.emit(err);
            });
        };
        AnalyticsReportParametersComponent.prototype.retrieveParameterOptions = function (parameters, appId, reportId, processDefinitionId) {
            var _this = this;
            parameters.forEach(function (param) {
                _this.analyticsService.getParamValuesByType(param.type, appId, reportId, processDefinitionId).subscribe(function (opts) {
                    param.options = opts;
                    _this.onSuccessParamOpt.emit(opts);
                }, function (err) {
                    _this.logService.error(err);
                    _this.onError.emit(err);
                });
            });
        };
        AnalyticsReportParametersComponent.prototype.onProcessDefinitionChanges = function (field) {
            if (field.value) {
                this.onDropdownChanged.emit(field);
            }
        };
        AnalyticsReportParametersComponent.prototype.submit = function (values) {
            var reportParamQuery = this.convertFormValuesToReportParamQuery(values);
            this.onSuccess.emit(reportParamQuery);
        };
        AnalyticsReportParametersComponent.prototype.onValueChanged = function (values) {
            this.onFormValueChanged.emit(values);
            if (this.reportForm && this.reportForm.valid) {
                this.submit(values);
            }
        };
        AnalyticsReportParametersComponent.prototype.convertMomentDate = function (date) {
            return moment(date, AnalyticsReportParametersComponent_1.FORMAT_DATE_ACTIVITI, true).format(AnalyticsReportParametersComponent_1.FORMAT_DATE_ACTIVITI) + 'T00:00:00.000Z';
        };
        AnalyticsReportParametersComponent.prototype.convertNumber = function (value) {
            return value != null ? parseInt(value, 10) : 0;
        };
        AnalyticsReportParametersComponent.prototype.convertFormValuesToReportParamQuery = function (values) {
            var reportParamQuery = new report_model_1.ReportQuery();
            reportParamQuery.dateRange.startDate = this.convertMomentDate(values.dateRange.startDate);
            reportParamQuery.dateRange.endDate = this.convertMomentDate(values.dateRange.endDate);
            reportParamQuery.status = values.statusGroup.status;
            reportParamQuery.processDefinitionId = values.processDefGroup.processDefinitionId;
            reportParamQuery.taskName = values.taskGroup.taskName;
            reportParamQuery.duration = values.durationGroup.duration;
            reportParamQuery.dateRangeInterval = values.dateIntervalGroup.dateRangeInterval;
            reportParamQuery.slowProcessInstanceInteger = this.convertNumber(values.processInstanceGroup.slowProcessInstanceInteger);
            reportParamQuery.typeFiltering = values.typeFilteringGroup.typeFiltering;
            return reportParamQuery;
        };
        AnalyticsReportParametersComponent.prototype.ngOnDestroy = function () {
            this.dropDownSub.unsubscribe();
            this.paramOpts.unsubscribe();
            if (this.reportParamsSub) {
                this.reportParamsSub.unsubscribe();
            }
        };
        AnalyticsReportParametersComponent.prototype.editEnable = function () {
            this.isEditable = true;
        };
        AnalyticsReportParametersComponent.prototype.editDisable = function () {
            this.isEditable = false;
        };
        AnalyticsReportParametersComponent.prototype.editTitle = function () {
            var _this = this;
            this.reportParamsSub = this.analyticsService.updateReport(this.reportParameters.id, this.reportParameters.name).subscribe(function (res) {
                _this.editDisable();
                _this.onEdit.emit(_this.reportParameters.name);
            }, function (err) {
                _this.logService.error(err);
                _this.onError.emit(err);
            });
        };
        AnalyticsReportParametersComponent.prototype.ngAfterViewChecked = function () {
            if (componentHandler) {
                componentHandler.upgradeAllRegistered();
            }
        };
        AnalyticsReportParametersComponent.prototype.toggleParameters = function () {
            this.hideParameters = !this.hideParameters;
        };
        AnalyticsReportParametersComponent.prototype.isParametersHide = function () {
            return this.hideParameters;
        };
        return AnalyticsReportParametersComponent;
    }();
    AnalyticsReportParametersComponent.FORMAT_DATE_ACTIVITI = 'YYYY-MM-DD';
    __decorate([core_1.Input(), __metadata("design:type", String)], AnalyticsReportParametersComponent.prototype, "appId", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], AnalyticsReportParametersComponent.prototype, "reportId", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], AnalyticsReportParametersComponent.prototype, "debug", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportParametersComponent.prototype, "onSuccess", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportParametersComponent.prototype, "onError", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportParametersComponent.prototype, "onEdit", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportParametersComponent.prototype, "onFormValueChanged", void 0);
    AnalyticsReportParametersComponent = AnalyticsReportParametersComponent_1 = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'analytics-report-parameters',
        template: "<div class=\"report-container\">     <a class=\"mdl-navigation__link setting-button\" data-automation-id=\"settings\">         <button (click)=\"toggleParameters()\" class=\"mdl-button mdl-js-button mdl-button--fab mdl-button--colored\">             <i class=\"material-icons\">settings</i>         </button>         <span class=\"report-container-setting\">{{'ANALYTICS.MESSAGES.SETTING-TITLE' | translate}}</span>     </a>     <div class=\"col-md-6\" [class.is-hide]=\"isParametersHide()\" >         <div *ngIf=\"reportParameters\">             <form [formGroup]=\"reportForm\" novalidate>             <div *ngIf=\"isEditable\">                 <input                         type=\"text\"                         class=\"mdl-textfield__input large\"                         id=\"reportName\"                         autofocus                         data-automation-id=\"reportName\"                         [value]=\"reportParameters.name\"                         (input)=\"reportParameters.name=$event.target.value\"                         (blur)=\"editTitle($event)\"                         />             </div>             <div *ngIf=\"!isEditable\">                 <span class=\"icon-small\">                 <i class=\"material-icons\">mode_edit</i>                 <h4 (click)=\"editEnable()\">{{reportParameters.name}}</h4>                 </span>             </div><hr>             <div *ngFor=\"let field of reportParameters.definition.parameters\">                 <div [ngSwitch]=\"field.type\">                     <div *ngSwitchCase=\"'integer'\">                         <br>                         <number-widget [field]=\"field\" [group]=\"reportForm.controls.processInstanceGroup\" [controllerName]=\"'slowProcessInstanceInteger'\"                                        [required]=\"true\"></number-widget>                     </div>                     <div *ngSwitchCase=\"'duration'\">                         <br>                         <duration-widget [field]=\"field\" [group]=\"reportForm.controls.durationGroup\"                                          [controllerName]=\"'duration'\"></duration-widget>                     </div>                     <div *ngSwitchCase=\"'boolean'\">                         <br>                         <checkbox-widget [field]=\"field\" [group]=\"reportForm.controls.typeFilteringGroup\"                                          [controllerName]=\"'typeFiltering'\"></checkbox-widget>                     </div>                     <div *ngSwitchCase=\"'status'\">                         <br>                         <dropdown-widget [field]=\"field\" [group]=\"reportForm.controls.statusGroup\" [controllerName]=\"'status'\"                                          [required]=\"true\"></dropdown-widget>                     </div>                     <div *ngSwitchCase=\"'processDefinition'\">                         <br>                         <dropdown-widget [field]=\"field\" [group]=\"reportForm.controls.processDefGroup\" [controllerName]=\"'processDefinitionId'\"                                          [required]=\"true\" (fieldChanged)=\"onProcessDefinitionChanges(field)\"></dropdown-widget>                     </div>                     <div *ngSwitchCase=\"'task'\">                         <br>                         <dropdown-widget [field]=\"field\" [group]=\"reportForm.controls.taskGroup\" [controllerName]=\"'taskName'\"                                          [required]=\"true\"></dropdown-widget>                     </div>                     <div *ngSwitchCase=\"'dateRange'\">                         <br>                         <date-range-widget [field]=\"field\" [group]=\"reportForm.controls.dateRange\"></date-range-widget>                     </div>                     <div *ngSwitchCase=\"'dateInterval'\">                         <br>                         <dropdown-widget [field]=\"field\" [group]=\"reportForm.controls.dateIntervalGroup\" [controllerName]=\"'dateRangeInterval'\"                                          [required]=\"true\" [showDefaultOption]=\"false\"></dropdown-widget>                     </div>                     <div *ngSwitchDefault>                         <span>{{'ANALYTICS.MESSAGES.UNKNOWN-WIDGET-TYPE' | translate}}: {{field.type}}</span>                     </div>                 </div>             </div>             <div *ngIf=\"debug\">                 <p>ReportForm valid : {{ reportForm.valid }}</p>                 <p>ReportForm status : {{ reportForm.errors | json }}</p>                 <p>ReportForm FormGroup valid : {{reportForm && reportForm.controls.dateRange.valid | json }}</p>             </div>             </form>         </div>     </div> </div>",
        styles: [".dropdown-widget {     width: 100%; }  .dropdown-widget__select {     width: 100%; }  .dropdown-widget__invalid .dropdown-widget__select {     border-color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label {     color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label:after {     background-color: #d50000; }  .dropdown-widget__invalid .mdl-textfield__error {     visibility: visible !important; }  .large {     font-size: x-large;     margin-top: 24px;     margin-left: 12px; }  .icon-small i {     float: left;     margin-right: 10px;     display: none;     position: absolute; }  .icon-small h4 {     clear: left;     margin-left: 26px; }  .icon-small:hover {     color: rgb(68,138,255);     cursor: pointer; }  .icon-small:hover .material-icons {     display: block; }  .is-hide {     height: 0px;     overflow: hidden;     transition: height 0.5s; }  .report-container {     border: solid 1px rgb(212, 212, 212);     padding: 10px 10px 10px 10px; }  .report-container-setting {     padding-left: 10px; }"]
    }), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoTranslationService, analytics_service_1.AnalyticsService, forms_1.FormBuilder, ng2_alfresco_core_1.LogService])], AnalyticsReportParametersComponent);
    exports.AnalyticsReportParametersComponent = AnalyticsReportParametersComponent;
    var AnalyticsReportParametersComponent_1;
});
$__System.registerDynamic("b", ["3", "5", "6", "c"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("5");
    var analytics_service_1 = $__require("6");
    var analytics_generator_component_1 = $__require("c");
    var AnalyticsComponent = function () {
        function AnalyticsComponent(translateService, analyticsService, logService) {
            this.translateService = translateService;
            this.analyticsService = analyticsService;
            this.logService = logService;
            this.debug = false;
            this.editReport = new core_1.EventEmitter();
            logService.info('AnalyticsComponent');
            if (translateService) {
                translateService.addTranslationFolder('ng2-activiti-analytics', 'node_modules/ng2-activiti-analytics/src');
            }
        }
        AnalyticsComponent.prototype.ngOnChanges = function (changes) {
            this.analyticsgenerator.reset();
        };
        AnalyticsComponent.prototype.showReport = function ($event) {
            this.analyticsgenerator.generateReport(this.reportId, $event);
        };
        AnalyticsComponent.prototype.reset = function () {
            this.analyticsgenerator.reset();
        };
        AnalyticsComponent.prototype.onEditReport = function (name) {
            this.editReport.emit(name);
        };
        return AnalyticsComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", Number)], AnalyticsComponent.prototype, "appId", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Number)], AnalyticsComponent.prototype, "reportId", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], AnalyticsComponent.prototype, "debug", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsComponent.prototype, "editReport", void 0);
    __decorate([core_1.ViewChild('analyticsgenerator'), __metadata("design:type", analytics_generator_component_1.AnalyticsGeneratorComponent)], AnalyticsComponent.prototype, "analyticsgenerator", void 0);
    AnalyticsComponent = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'activiti-analytics',
        template: "<div class=\"col-md-6\">     <analytics-report-parameters [appId]=\"appId\" [reportId]=\"reportId\"                                  (onFormValueChanged)=\"reset()\"                                  (onSuccess)=\"showReport($event)\"                                  (onEdit)=\"onEditReport($event)\">     </analytics-report-parameters>      <activiti-analytics-generator [reportId]=\"reportId\"                                   [reportParamQuery]=\"reportParamQuery\"                                   #analyticsgenerator>     </activiti-analytics-generator> </div>",
        styles: [".chart {display: block; width: 100%;}"]
    }), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoTranslationService, analytics_service_1.AnalyticsService, ng2_alfresco_core_1.LogService])], AnalyticsComponent);
    exports.AnalyticsComponent = AnalyticsComponent;
});
$__System.registerDynamic("c", ["3", "5", "6", "a"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("5");
    var analytics_service_1 = $__require("6");
    var report_model_1 = $__require("a");
    var AnalyticsGeneratorComponent = function () {
        function AnalyticsGeneratorComponent(translateService, analyticsService, logService) {
            this.translateService = translateService;
            this.analyticsService = analyticsService;
            this.logService = logService;
            this.reportParamQuery = undefined;
            this.onSuccess = new core_1.EventEmitter();
            this.onError = new core_1.EventEmitter();
            this.showDetails = false;
            this.barChartOptions = {
                responsive: true,
                scales: {
                    yAxes: [{
                        ticks: {
                            beginAtZero: true,
                            stepSize: 1
                        }
                    }],
                    xAxes: [{
                        ticks: {},
                        stacked: true
                    }]
                }
            };
            logService.info('AnalyticsGeneratorComponent');
            if (translateService) {
                translateService.addTranslationFolder('ng2-activiti-analytics', 'node_modules/ng2-activiti-analytics/src');
            }
        }
        AnalyticsGeneratorComponent.prototype.ngOnChanges = function (changes) {
            if (this.reportId && this.reportParamQuery) {
                this.generateReport(this.reportId, this.reportParamQuery);
            } else {
                this.reset();
            }
        };
        AnalyticsGeneratorComponent.prototype.generateReport = function (reportId, reportParamQuery) {
            var _this = this;
            this.analyticsService.getReportsByParams(reportId, reportParamQuery).subscribe(function (res) {
                _this.reports = res;
                _this.onSuccess.emit(res);
            }, function (err) {
                _this.onError.emit(err);
                _this.logService.error(err);
            });
        };
        AnalyticsGeneratorComponent.prototype.reset = function () {
            if (this.reports) {
                this.reports = undefined;
            }
        };
        AnalyticsGeneratorComponent.prototype.refresh = function (report) {
            var clone = JSON.parse(JSON.stringify(report));
            report.datasets = clone.datasets;
        };
        AnalyticsGeneratorComponent.prototype.toggleDetailsTable = function () {
            this.showDetails = !this.showDetails;
        };
        AnalyticsGeneratorComponent.prototype.isShowDetails = function () {
            return this.showDetails;
        };
        return AnalyticsGeneratorComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", Number)], AnalyticsGeneratorComponent.prototype, "reportId", void 0);
    __decorate([core_1.Input(), __metadata("design:type", report_model_1.ReportQuery)], AnalyticsGeneratorComponent.prototype, "reportParamQuery", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsGeneratorComponent.prototype, "onSuccess", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsGeneratorComponent.prototype, "onError", void 0);
    AnalyticsGeneratorComponent = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'activiti-analytics-generator',
        template: "<div *ngIf=\"reports\">     <div *ngFor=\"let report of reports\">         <h4>{{report.title}}</h4>         <div [ngSwitch]=\"report.type\">             <div *ngSwitchCase=\"'pie'\">                 <div class=\"col-md-6\">                     <div *ngIf=\"!report.hasData()\">{{'ANALYTICS.MESSAGES.NO-DATA-FOUND' | translate}}</div>                     <div *ngIf=\"report.hasData()\">                         <div *ngIf=\"report.hasZeroValues()\">{{'ANALYTICS.MESSAGES.ZERO-DATA-FOUND' | translate}}</div>                         <base-chart *ngIf=\"!report.hasZeroValues()\" class=\"chart\"                                     [data]=\"report.data\"                                     [labels]=\"report.labels\"                                     [chartType]=\"report.type\"></base-chart>                     </div>                 </div>             </div>             <div *ngSwitchCase=\"'table'\">                 <div *ngIf=\"!report.hasDatasets()\">{{'ANALYTICS.MESSAGES.NO-DATA-FOUND' | translate}}</div>                 <div [attr.id]=\"'chart-table-' + report.id\" *ngIf=\"report.hasDatasets()\">                     <table class=\"table table-responsive table-condensed\" style=\"width: 100%\">                         <tr>                             <th *ngFor=\"let label of report.labels\">{{label | translate}}</th>                         </tr>                         <tr *ngFor=\"let rows of report.datasets\" style=\"text-align: center;\">                             <td *ngFor=\"let row of rows\">{{row | translate }}</td>                         </tr>                     </table>                 </div>             </div>             <div *ngSwitchCase=\"'masterDetailTable'\">                 <div *ngIf=\"!report.hasDatasets()\">{{'ANALYTICS.MESSAGES.NO-DATA-FOUND' | translate}}</div>                 <div [attr.id]=\"'chart-master-detail-table-' + report.id\" *ngIf=\"report.hasDatasets()\">                     <table class=\"table table-responsive table-condensed\" style=\"width: 100%\">                         <tr>                             <th *ngFor=\"let label of report.labels\">{{label | translate}}</th>                         </tr>                         <tr *ngFor=\"let rows of report.datasets\" class=\"analytics-row__entry\" style=\"text-align: center;\">                             <td *ngFor=\"let row of rows\" (click)=\"toggleDetailsTable()\">{{row | translate }}</td>                         </tr>                     </table>                 </div>                 <div [attr.id]=\"'chart-master-detail-' + report.id\" *ngIf=\"isShowDetails()\">                     <table class=\"table table-responsive table-condensed\" style=\"width: 100%\">                         <tr>                             <th *ngFor=\"let label of report.detailsTable.labels\">{{label | translate}}</th>                         </tr>                         <tr *ngFor=\"let rows of report.detailsTable.datasets\" style=\"text-align: center;\">                             <td *ngFor=\"let row of rows\">{{row | translate }}</td>                         </tr>                     </table>                 </div>             </div>             <div *ngSwitchCase=\"'bar'\">                 <div class=\"col-md-6\">                     <div *ngIf=\"!report.hasDatasets()\">{{'ANALYTICS.MESSAGES.NO-DATA-FOUND' | translate}}</div>                     <base-chart *ngIf=\"report.hasDatasets()\" class=\"chart\"                                 [datasets]=\"report.datasets\"                                 [labels]=\"report.labels\"                                 [options]=\"report.options\"                                 [chartType]=\"report.type\"></base-chart>                 </div>             </div>             <div *ngSwitchCase=\"'multiBar'\">                 <div class=\"col-md-6\">                     <div *ngIf=\"!report.hasDatasets()\">{{'ANALYTICS.MESSAGES.NO-DATA-FOUND' | translate}}</div>                     <div *ngIf=\"report.hasDatasets()\">                         <label class=\"mdl-checkbox mdl-js-checkbox mdl-js-ripple-effect\" [attr.for]=\"'stacked-id'\">                             <input type=\"checkbox\" [attr.id]=\"'stacked-id'\" class=\"mdl-checkbox__input\"                                    [checked]=\"report.options.scales.xAxes[0].stacked\"                                    [(ngModel)]=\"report.options.scales.xAxes[0].stacked\"                                    (change)=\"refresh(report)\">                             <span class=\"mdl-checkbox__label\">Stacked</span>                         </label>                         <base-chart class=\"chart\"                                     [datasets]=\"report.datasets\"                                     [labels]=\"report.labels\"                                     [options]=\"report.options\"                                     [chartType]=\"'bar'\"></base-chart>                     </div>                 </div>             </div>             <div *ngSwitchCase=\"'HeatMap'\">                 <analytics-report-heat-map [report]=\"report\"></analytics-report-heat-map>             </div>             <div *ngSwitchDefault>                 <span>{{'ANALYTICS.MESSAGES.UNKNOWN-WIDGET-TYPE' | translate}}: {{report.type}}</span>             </div>         </div>     </div> </div> <br><br><br> <div *ngIf=\"!reports\">{{'ANALYTICS.MESSAGES.FILL-PARAMETER' | translate}}</div>",
        styles: [".chart {display: block; width: 100%;}  .analytics-row__entry {     cursor: pointer; }"]
    }), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoTranslationService, analytics_service_1.AnalyticsService, ng2_alfresco_core_1.LogService])], AnalyticsGeneratorComponent);
    exports.AnalyticsGeneratorComponent = AnalyticsGeneratorComponent;
});
$__System.registerDynamic("d", ["3", "5", "6", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("5");
    var analytics_service_1 = $__require("6");
    var forms_1 = $__require("8");
    var AnalyticsReportHeatMapComponent = function () {
        function AnalyticsReportHeatMapComponent(translateService, analyticsService, formBuilder) {
            this.translateService = translateService;
            this.analyticsService = analyticsService;
            this.formBuilder = formBuilder;
            this.onSuccess = new core_1.EventEmitter();
            this.onError = new core_1.EventEmitter();
            this.field = {};
            if (translateService) {
                translateService.addTranslationFolder('ng2-activiti-analytics', 'node_modules/ng2-activiti-analytics/src');
            }
        }
        AnalyticsReportHeatMapComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.initForm();
            this.field.id = 'metrics';
            this.field.value = 'totalCount';
            this.analyticsService.getMetricValues().subscribe(function (opts) {
                _this.field.options = opts;
                _this.onSuccess.emit(opts);
            });
        };
        AnalyticsReportHeatMapComponent.prototype.onMetricChanges = function (field) {
            if (field.value === 'totalCount') {
                this.currentMetric = this.report.totalCountValues;
                this.currentMetricColors = this.report.totalCountsPercentages;
                this.metricType = 'times';
            } else if (field.value === 'totalTime') {
                this.currentMetric = this.report.totalTimeValues;
                this.currentMetricColors = this.report.totalTimePercentages;
                this.metricType = 'hours';
            } else if (field.value === 'avgTime') {
                this.currentMetric = this.report.avgTimeValues;
                this.currentMetricColors = this.report.avgTimePercentages;
                this.metricType = 'hours';
            }
        };
        AnalyticsReportHeatMapComponent.prototype.initForm = function () {
            this.metricForm = this.formBuilder.group({
                metricGroup: new forms_1.FormGroup({
                    metric: new forms_1.FormControl()
                })
            });
        };
        AnalyticsReportHeatMapComponent.prototype.hasMetric = function () {
            return this.report.totalCountsPercentages || this.report.totalTimePercentages || this.report.avgTimePercentages ? true : false;
        };
        return AnalyticsReportHeatMapComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", Object)], AnalyticsReportHeatMapComponent.prototype, "report", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportHeatMapComponent.prototype, "onSuccess", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], AnalyticsReportHeatMapComponent.prototype, "onError", void 0);
    AnalyticsReportHeatMapComponent = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'analytics-report-heat-map',
        template: "<h4>Process Heat map</h4> <div *ngIf=\"hasMetric()\">     <form [formGroup]=\"metricForm\" novalidate>         <dropdown-widget [field]=\"field\" [group]=\"metricForm.controls.metricGroup\" [controllerName]=\"'metric'\"                          (fieldChanged)=\"onMetricChanges(field)\" [showDefaultOption]=\"false\"></dropdown-widget>     </form>     <activiti-diagram *ngIf=\"currentMetric\" [processDefinitionId]=\"report.processDefinitionId\" [metricPercentages]=\"currentMetric\" [metricColor]=\"currentMetricColors\" [metricType]=\"metricType\"></activiti-diagram> </div> <div *ngIf=\"!hasMetric()\">No metric found</div>"
    }), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoTranslationService, analytics_service_1.AnalyticsService, forms_1.FormBuilder])], AnalyticsReportHeatMapComponent);
    exports.AnalyticsReportHeatMapComponent = AnalyticsReportHeatMapComponent;
});
$__System.registerDynamic("e", ["9"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __extends = this && this.__extends || function (d, b) {
        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
    var moment = $__require("9");
    var Chart = function () {
        function Chart(obj) {
            this.id = obj && obj.id || null;
            if (obj && obj.type) {
                this.type = this.convertType(obj.type);
            }
        }
        Chart.prototype.convertType = function (type) {
            var chartType = '';
            switch (type) {
                case 'pieChart':
                    chartType = 'pie';
                    break;
                case 'table':
                    chartType = 'table';
                    break;
                case 'line':
                    chartType = 'line';
                    break;
                case 'barChart':
                    chartType = 'bar';
                    break;
                case 'multiBarChart':
                    chartType = 'multiBar';
                    break;
                case 'processDefinitionHeatMap':
                    chartType = 'HeatMap';
                    break;
                case 'masterDetailTable':
                    chartType = 'masterDetailTable';
                    break;
                default:
                    chartType = 'table';
                    break;
            }
            return chartType;
        };
        return Chart;
    }();
    exports.Chart = Chart;
    var LineChart = function (_super) {
        __extends(LineChart, _super);
        function LineChart(obj) {
            var _this = _super.call(this, obj) || this;
            _this.labels = [];
            _this.datasets = [];
            _this.title = obj && obj.title || null;
            _this.titleKey = obj && obj.titleKey || null;
            _this.labels = obj && obj.columnNames.slice(1, obj.columnNames.length);
            obj.rows.forEach(function (value) {
                _this.datasets.push({ data: value.slice(1, value.length), label: value[0] });
            });
            return _this;
        }
        return LineChart;
    }(Chart);
    exports.LineChart = LineChart;
    var BarChart = function (_super) {
        __extends(BarChart, _super);
        function BarChart(obj) {
            var _this = _super.call(this, obj) || this;
            _this.labels = [];
            _this.datasets = [];
            _this.data = [];
            _this.options = {
                responsive: true,
                scales: {
                    yAxes: [{
                        ticks: {
                            beginAtZero: true,
                            stepSize: 1
                        }
                    }],
                    xAxes: [{
                        ticks: {},
                        stacked: false
                    }]
                }
            };
            _this.xAxisTickFormatFunction = function (xAxisType) {
                return function (value) {
                    if (xAxisType !== null && xAxisType !== undefined) {
                        if ('date_day' === xAxisType) {
                            return moment(new Date(value)).format('DD');
                        } else if ('date_month' === xAxisType) {
                            return moment(new Date(value)).format('MMMM');
                        } else if ('date_year' === xAxisType) {
                            return moment(new Date(value)).format('YYYY');
                        }
                    }
                    return value;
                };
            };
            _this.yAxisTickFormatFunction = function (yAxisType) {
                return function (value) {
                    if (yAxisType !== null && yAxisType !== undefined) {
                        if ('count' === yAxisType) {
                            var label = '' + value;
                            if (label.indexOf('.') !== -1) {
                                return '';
                            }
                        }
                    }
                    return value;
                };
            };
            _this.title = obj && obj.title || null;
            _this.titleKey = obj && obj.titleKey || null;
            _this.xAxisType = obj && obj.xAxisType || null;
            _this.yAxisType = obj && obj.yAxisType || null;
            _this.options.scales.xAxes[0].ticks.callback = _this.xAxisTickFormatFunction(_this.xAxisType);
            _this.options.scales.yAxes[0].ticks.callback = _this.yAxisTickFormatFunction(_this.yAxisType);
            if (obj.values) {
                obj.values.forEach(function (params) {
                    var dataValue = [];
                    params.values.forEach(function (info) {
                        info.forEach(function (value, index) {
                            if (index % 2 === 0) {
                                if (!_this.labels.includes(value)) {
                                    _this.labels.push(value);
                                }
                            } else {
                                dataValue.push(value);
                            }
                        });
                    });
                    if (dataValue && dataValue.length > 0) {
                        _this.datasets.push({ data: dataValue, label: params.key });
                    }
                });
            }
            return _this;
        }
        BarChart.prototype.hasDatasets = function () {
            return this.datasets && this.datasets.length > 0 ? true : false;
        };
        return BarChart;
    }(Chart);
    exports.BarChart = BarChart;
    var MultiBarChart = function (_super) {
        __extends(MultiBarChart, _super);
        function MultiBarChart(obj) {
            return _super.call(this, obj) || this;
        }
        return MultiBarChart;
    }(BarChart);
    exports.MultiBarChart = MultiBarChart;
    var TableChart = function (_super) {
        __extends(TableChart, _super);
        function TableChart(obj) {
            var _this = _super.call(this, obj) || this;
            _this.labels = [];
            _this.datasets = [];
            _this.title = obj && obj.title || null;
            _this.titleKey = obj && obj.titleKey || null;
            _this.labels = obj && obj.columnNames;
            if (obj.rows) {
                _this.datasets = obj && obj.rows;
            }
            return _this;
        }
        TableChart.prototype.hasDatasets = function () {
            return this.datasets && this.datasets.length > 0 ? true : false;
        };
        return TableChart;
    }(Chart);
    exports.TableChart = TableChart;
    var DetailsTableChart = function (_super) {
        __extends(DetailsTableChart, _super);
        function DetailsTableChart(obj) {
            var _this = _super.call(this, obj) || this;
            _this.showDetails = false;
            if (obj.detailTables) {
                _this.detailsTable = new TableChart(obj.detailTables[0]);
            }
            return _this;
        }
        DetailsTableChart.prototype.hasDetailsTable = function () {
            return this.detailsTable ? true : false;
        };
        return DetailsTableChart;
    }(TableChart);
    exports.DetailsTableChart = DetailsTableChart;
    var HeatMapChart = function (_super) {
        __extends(HeatMapChart, _super);
        function HeatMapChart(obj) {
            var _this = _super.call(this, obj) || this;
            _this.avgTimePercentages = obj && obj.avgTimePercentages || null;
            _this.avgTimeValues = obj && obj.avgTimeValues || null;
            _this.processDefinitionId = obj && obj.processDefinitionId || null;
            _this.totalCountValues = obj && obj.totalCountValues || null;
            _this.titleKey = obj && obj.titleKey || null;
            _this.totalCountsPercentages = obj && obj.totalCountsPercentages || null;
            _this.totalTimePercentages = obj && obj.totalTimePercentages || null;
            _this.totalTimeValues = obj && obj.totalTimeValues || null;
            return _this;
        }
        return HeatMapChart;
    }(Chart);
    exports.HeatMapChart = HeatMapChart;
    var PieChart = function (_super) {
        __extends(PieChart, _super);
        function PieChart(obj) {
            var _this = _super.call(this, obj) || this;
            _this.labels = [];
            _this.data = [];
            _this.title = obj && obj.title || null;
            _this.titleKey = obj && obj.titleKey || null;
            if (obj.values) {
                obj.values.forEach(function (value) {
                    _this.add(value.key, value.y);
                });
            }
            return _this;
        }
        PieChart.prototype.add = function (label, data) {
            this.labels.push(label);
            this.data.push(data);
        };
        PieChart.prototype.hasData = function () {
            return this.data && this.data.length > 0 ? true : false;
        };
        PieChart.prototype.hasZeroValues = function () {
            var isZeroValues = false;
            if (this.hasData()) {
                isZeroValues = true;
                this.data.forEach(function (value) {
                    if (value.toString() !== '0') {
                        isZeroValues = false;
                    }
                });
            }
            return isZeroValues;
        };
        return PieChart;
    }(Chart);
    exports.PieChart = PieChart;
});
$__System.registerDynamic("6", ["3", "4", "5", "a", "e"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("4");
    var ng2_alfresco_core_1 = $__require("5");
    var report_model_1 = $__require("a");
    var chart_model_1 = $__require("e");
    var AnalyticsService = function () {
        function AnalyticsService(apiService, logService) {
            this.apiService = apiService;
            this.logService = logService;
        }
        AnalyticsService.prototype.getReportList = function () {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getReportList()).map(function (res) {
                var reports = [];
                res.forEach(function (report) {
                    var reportModel = new report_model_1.ReportParametersModel(report);
                    reports.push(reportModel);
                });
                return reports;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.getReportParams = function (reportId) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getReportParams(reportId)).map(function (res) {
                return new report_model_1.ReportParametersModel(res);
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.getParamValuesByType = function (type, appId, reportId, processDefinitionId) {
            if (type === 'status') {
                return this.getProcessStatusValues();
            } else if (type === 'processDefinition') {
                if (appId === null || appId === undefined) {
                    return this.getProcessDefinitionsValuesNoApp();
                } else {
                    return this.getProcessDefinitionsValues(appId);
                }
            } else if (type === 'dateInterval') {
                return this.getDateIntervalValues();
            } else if (type === 'task' && reportId && processDefinitionId) {
                return this.getTasksByProcessDefinitionId(reportId, processDefinitionId);
            } else {
                return Rx_1.Observable.create(function (observer) {
                    observer.next(null);
                    observer.complete();
                });
            }
        };
        AnalyticsService.prototype.getProcessStatusValues = function () {
            var paramOptions = [];
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'All', name: 'All' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'Active', name: 'Active' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'Complete', name: 'Complete' }));
            return Rx_1.Observable.create(function (observer) {
                observer.next(paramOptions);
                observer.complete();
            });
        };
        AnalyticsService.prototype.getDateIntervalValues = function () {
            var paramOptions = [];
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byHour', name: 'By hour' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byDay', name: 'By day' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byWeek', name: 'By week' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byMonth', name: 'By month' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byYear', name: 'By year' }));
            return Rx_1.Observable.create(function (observer) {
                observer.next(paramOptions);
                observer.complete();
            });
        };
        AnalyticsService.prototype.getMetricValues = function () {
            var paramOptions = [];
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'totalCount', name: 'Number of times a step is executed' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'totalTime', name: 'Total time spent in a process step' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: 'avgTime', name: 'Average time spent in a process step' }));
            return Rx_1.Observable.create(function (observer) {
                observer.next(paramOptions);
                observer.complete();
            });
        };
        AnalyticsService.prototype.getProcessDefinitionsValuesNoApp = function () {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getProcessDefinitions()).map(function (res) {
                var paramOptions = [];
                res.forEach(function (opt) {
                    paramOptions.push(new report_model_1.ParameterValueModel(opt));
                });
                return paramOptions;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.getProcessDefinitionsValues = function (appId) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.processDefinitionsApi.getProcessDefinitions(appId)).map(function (res) {
                var paramOptions = [];
                res.data.forEach(function (opt) {
                    paramOptions.push(new report_model_1.ParameterValueModel(opt));
                });
                return paramOptions;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.getTasksByProcessDefinitionId = function (reportId, processDefinitionId) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getTasksByProcessDefinitionId(reportId, processDefinitionId)).map(function (res) {
                var paramOptions = [];
                res.forEach(function (opt) {
                    paramOptions.push(new report_model_1.ParameterValueModel({ id: opt, name: opt }));
                });
                return paramOptions;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.getReportsByParams = function (reportId, paramsQuery) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getReportsByParams(reportId, paramsQuery)).map(function (res) {
                var elements = [];
                res.elements.forEach(function (chartData) {
                    if (chartData.type === 'pieChart') {
                        elements.push(new chart_model_1.PieChart(chartData));
                    } else if (chartData.type === 'table') {
                        elements.push(new chart_model_1.TableChart(chartData));
                    } else if (chartData.type === 'processDefinitionHeatMap') {
                        elements.push(new chart_model_1.HeatMapChart(chartData));
                    } else if (chartData.type === 'masterDetailTable') {
                        elements.push(new chart_model_1.DetailsTableChart(chartData));
                    } else if (chartData.type === 'barChart') {
                        elements.push(new chart_model_1.BarChart(chartData));
                    } else if (chartData.type === 'multiBarChart') {
                        elements.push(new chart_model_1.MultiBarChart(chartData));
                    }
                });
                return elements;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.createDefaultReports = function () {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.createDefaultReports()).map(this.toJson).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.updateReport = function (reportId, name) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.updateReport(reportId, name)).map(function (res) {
                _this.logService.info('upload');
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        AnalyticsService.prototype.handleError = function (error) {
            this.logService.error(error);
            return Rx_1.Observable.throw(error.json().error || 'Server error');
        };
        AnalyticsService.prototype.toJson = function (res) {
            return res || {};
        };
        return AnalyticsService;
    }();
    AnalyticsService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoApiService, ng2_alfresco_core_1.LogService])], AnalyticsService);
    exports.AnalyticsService = AnalyticsService;
});
$__System.registerDynamic("f", ["3", "8", "10"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __extends = this && this.__extends || function (d, b) {
        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var forms_1 = $__require("8");
    var widget_component_1 = $__require("10");
    var DropdownWidget = function (_super) {
        __extends(DropdownWidget, _super);
        function DropdownWidget() {
            var _this = _super.call(this) || this;
            _this.fieldChanged = new core_1.EventEmitter();
            _this.showDefaultOption = true;
            _this.required = false;
            _this.defaultOptionText = 'Choose One';
            return _this;
        }
        DropdownWidget.prototype.ngOnInit = function () {
            if (this.required) {
                this.formGroup.get(this.controllerName).setValidators(forms_1.Validators.required);
            }
        };
        return DropdownWidget;
    }(widget_component_1.WidgetComponent);
    __decorate([core_1.Input(), __metadata("design:type", Object)], DropdownWidget.prototype, "field", void 0);
    __decorate([core_1.Input('group'), __metadata("design:type", forms_1.FormGroup)], DropdownWidget.prototype, "formGroup", void 0);
    __decorate([core_1.Input('controllerName'), __metadata("design:type", String)], DropdownWidget.prototype, "controllerName", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DropdownWidget.prototype, "fieldChanged", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DropdownWidget.prototype, "showDefaultOption", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DropdownWidget.prototype, "required", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], DropdownWidget.prototype, "defaultOptionText", void 0);
    DropdownWidget = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'dropdown-widget',
        template: "<div class=\"dropdown-widget\" [formGroup]=\"formGroup\">     <label class=\"dropdown-widget__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label>     <select [formControlName]=\"controllerName\" [attr.id]=\"'select-' + field.id\" class=\"dropdown-widget__select\"             [(ngModel)]=\"field.value\" (ngModelChange)=\"changeValue($event)\">         <option *ngIf=\"showDefaultOption\" value=\"null\">{{defaultOptionText}}</option>         <option *ngFor=\"let opt of field.options\" [value]=\"opt.id\">{{opt.label}}</option>     </select> </div>",
        styles: [".dropdown-widget {     width: 100%; }  .dropdown-widget__select {     width: 100%; }  .dropdown-widget__invalid .dropdown-widget__select {     border-color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label {     color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label:after {     background-color: #d50000; }  .dropdown-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }), __metadata("design:paramtypes", [])], DropdownWidget);
    exports.DropdownWidget = DropdownWidget;
});
$__System.registerDynamic("11", ["3", "10", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __extends = this && this.__extends || function (d, b) {
        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var widget_component_1 = $__require("10");
    var forms_1 = $__require("8");
    var NumberWidget = function (_super) {
        __extends(NumberWidget, _super);
        function NumberWidget(elementRef) {
            var _this = _super.call(this) || this;
            _this.elementRef = elementRef;
            _this.required = false;
            return _this;
        }
        NumberWidget.prototype.ngOnInit = function () {
            if (this.required) {
                this.formGroup.get(this.controllerName).setValidators(forms_1.Validators.required);
            }
        };
        NumberWidget.prototype.setupMaterialComponents = function (handler) {
            if (handler) {
                handler.upgradeAllRegistered();
                if (this.elementRef && this.hasValue()) {
                    var container = this.elementRef.nativeElement.querySelector('.mdl-textfield');
                    if (container && container.MaterialTextfield) {
                        container.MaterialTextfield.change(this.field.value.toString());
                    }
                }
                return true;
            }
            return false;
        };
        return NumberWidget;
    }(widget_component_1.WidgetComponent);
    __decorate([core_1.Input(), __metadata("design:type", Object)], NumberWidget.prototype, "field", void 0);
    __decorate([core_1.Input('group'), __metadata("design:type", forms_1.FormGroup)], NumberWidget.prototype, "formGroup", void 0);
    __decorate([core_1.Input('controllerName'), __metadata("design:type", String)], NumberWidget.prototype, "controllerName", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], NumberWidget.prototype, "required", void 0);
    NumberWidget = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'number-widget',
        template: "<div class=\"mdl-textfield mdl-js-textfield mdl-textfield--floating-label number-widget\" [formGroup]=\"formGroup\">     <input formControlName=\"{{controllerName}}\" class=\"mdl-textfield__input\"            type=\"text\"            pattern=\"-?[0-9]*(\\.[0-9]+)?\"            [attr.id]=\"field.id\"            [value]=\"field.value\"            [(ngModel)]=\"field.value\"            (ngModelChange)=\"changeValue(field)\">     <label class=\"mdl-textfield__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label> </div>",
        styles: [".number-widget {     width: 100%; }  .number-widget__invalid .mdl-textfield__input {     border-color: #d50000; }  .number-widget__invalid .mdl-textfield__label {     color: #d50000; }  .number-widget__invalid .mdl-textfield__label:after {     background-color: #d50000; }  .number-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }), __metadata("design:paramtypes", [core_1.ElementRef])], NumberWidget);
    exports.NumberWidget = NumberWidget;
});
$__System.registerDynamic("a", [], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var ReportParametersModel = function () {
        function ReportParametersModel(obj) {
            this.id = obj && obj.id;
            this.name = obj && obj.name || null;
            if (obj && obj.definition) {
                this.definition = new ReportDefinitionModel(JSON.parse(obj.definition));
            }
            this.created = obj && obj.created || null;
        }
        ReportParametersModel.prototype.hasParameters = function () {
            return this.definition && this.definition.parameters && this.definition.parameters.length > 0 ? true : false;
        };
        return ReportParametersModel;
    }();
    exports.ReportParametersModel = ReportParametersModel;
    var ReportDefinitionModel = function () {
        function ReportDefinitionModel(obj) {
            var _this = this;
            this.parameters = [];
            obj.parameters.forEach(function (params) {
                var reportParamsModel = new ReportParameterDetailsModel(params);
                _this.parameters.push(reportParamsModel);
            });
        }
        ReportDefinitionModel.prototype.findParam = function (name) {
            this.parameters.forEach(function (param) {
                return param.type === name ? param : null;
            });
            return null;
        };
        return ReportDefinitionModel;
    }();
    exports.ReportDefinitionModel = ReportDefinitionModel;
    var ReportParameterDetailsModel = function () {
        function ReportParameterDetailsModel(obj) {
            this.id = obj && obj.id;
            this.name = obj && obj.name || null;
            this.nameKey = obj && obj.nameKey || null;
            this.type = obj && obj.type || null;
            this.value = obj && obj.value || null;
            this.options = obj && obj.options || null;
            this.dependsOn = obj && obj.dependsOn || null;
        }
        return ReportParameterDetailsModel;
    }();
    exports.ReportParameterDetailsModel = ReportParameterDetailsModel;
    var ParameterValueModel = function () {
        function ParameterValueModel(obj) {
            this.id = obj && obj.id;
            this.name = obj && obj.name || null;
            this.value = obj && obj.value || null;
            this.version = obj && obj.version || null;
        }
        Object.defineProperty(ParameterValueModel.prototype, "label", {
            get: function () {
                return this.version ? this.name + " (v " + this.version + ") " : this.name;
            },
            enumerable: true,
            configurable: true
        });
        return ParameterValueModel;
    }();
    exports.ParameterValueModel = ParameterValueModel;
    var ReportQuery = function () {
        function ReportQuery(obj) {
            this.processDefinitionId = obj && obj.processDefinitionId || null;
            this.status = obj && obj.status || null;
            this.taskName = obj && obj.taskName || null;
            this.dateRangeInterval = obj && obj.dateRangeInterval || null;
            this.typeFiltering = obj && obj.typeFiltering || true;
            this.slowProcessInstanceInteger = obj && obj.slowProcessInstanceInteger || 0;
            this.duration = obj && obj.duration || 0;
            this.dateRange = new ReportDateRange(obj);
        }
        return ReportQuery;
    }();
    exports.ReportQuery = ReportQuery;
    var ReportDateRange = function () {
        function ReportDateRange(obj) {
            this.startDate = obj && obj.startDate || null;
            this.endDate = obj && obj.endDate || null;
        }
        return ReportDateRange;
    }();
    exports.ReportDateRange = ReportDateRange;
});
$__System.registerDynamic("12", ["3", "11", "a", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __extends = this && this.__extends || function (d, b) {
        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var number_widget_1 = $__require("11");
    var report_model_1 = $__require("a");
    var forms_1 = $__require("8");
    var DurationWidget = function (_super) {
        __extends(DurationWidget, _super);
        function DurationWidget(elementRef) {
            var _this = _super.call(this, elementRef) || this;
            _this.elementRef = elementRef;
            _this.required = false;
            return _this;
        }
        DurationWidget.prototype.ngOnInit = function () {
            var timeType = new forms_1.FormControl();
            this.formGroup.addControl('timeType', timeType);
            if (this.required) {
                this.formGroup.get(this.controllerName).setValidators(forms_1.Validators.required);
            }
            if (this.field.value === null) {
                this.field.value = 0;
            }
            var paramOptions = [];
            paramOptions.push(new report_model_1.ParameterValueModel({ id: '1', name: 'Seconds' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: '60', name: 'Minutes' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: '3600', name: 'Hours' }));
            paramOptions.push(new report_model_1.ParameterValueModel({ id: '86400', name: 'Days', selected: true }));
            this.duration = new report_model_1.ReportParameterDetailsModel({ id: 'duration', name: 'duration', options: paramOptions });
            this.duration.value = paramOptions[0].id;
        };
        DurationWidget.prototype.calculateDuration = function () {
            if (this.field && this.duration.value) {
                this.currentValue = parseInt(this.field.value, 10) * parseInt(this.duration.value, 10);
                this.formGroup.get(this.controllerName).setValue(this.currentValue);
                this.fieldChanged.emit({ value: this.currentValue });
            }
        };
        return DurationWidget;
    }(number_widget_1.NumberWidget);
    __decorate([core_1.Input(), __metadata("design:type", Object)], DurationWidget.prototype, "field", void 0);
    __decorate([core_1.Input('group'), __metadata("design:type", forms_1.FormGroup)], DurationWidget.prototype, "formGroup", void 0);
    __decorate([core_1.Input('controllerName'), __metadata("design:type", String)], DurationWidget.prototype, "controllerName", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DurationWidget.prototype, "required", void 0);
    DurationWidget = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'duration-widget',
        template: "<div class=\"mdl-grid\">     <div class=\"mdl-cell mdl-cell--6-col\">         <div class=\"mdl-textfield mdl-js-textfield mdl-textfield--floating-label number-widget\">             <input class=\"mdl-textfield__input\"                    type=\"text\"                    pattern=\"-?[0-9]*(\\.[0-9]+)?\"                    [attr.id]=\"field.id\"                    [value]=\"field.value\"                    [(ngModel)]=\"field.value\"                    (ngModelChange)=\"calculateDuration()\">             <label class=\"mdl-textfield__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label>         </div>     </div>     <div class=\"mdl-cell mdl-cell--6-col\">         <div style=\"margin-top: 30px\">             <dropdown-widget [field]=\"duration\" [group]=\"formGroup\" [controllerName]=\"'timeType'\"                              [showDefaultOption]=\"false\"                              (fieldChanged)=\"calculateDuration()\"></dropdown-widget>         </div>     </div> </div>",
        styles: [".number-widget {     width: 100%; }  .number-widget__invalid .mdl-textfield__input {     border-color: #d50000; }  .number-widget__invalid .mdl-textfield__label {     color: #d50000; }  .number-widget__invalid .mdl-textfield__label:after {     background-color: #d50000; }  .number-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }), __metadata("design:paramtypes", [core_1.ElementRef])], DurationWidget);
    exports.DurationWidget = DurationWidget;
});
$__System.registerDynamic("13", ["3", "10", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __extends = this && this.__extends || function (d, b) {
        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var widget_component_1 = $__require("10");
    var forms_1 = $__require("8");
    var CheckboxWidget = function (_super) {
        __extends(CheckboxWidget, _super);
        function CheckboxWidget(elementRef) {
            var _this = _super.call(this) || this;
            _this.elementRef = elementRef;
            return _this;
        }
        return CheckboxWidget;
    }(widget_component_1.WidgetComponent);
    __decorate([core_1.Input(), __metadata("design:type", Object)], CheckboxWidget.prototype, "field", void 0);
    __decorate([core_1.Input('group'), __metadata("design:type", forms_1.FormGroup)], CheckboxWidget.prototype, "formGroup", void 0);
    __decorate([core_1.Input('controllerName'), __metadata("design:type", String)], CheckboxWidget.prototype, "controllerName", void 0);
    CheckboxWidget = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'checkbox-widget',
        template: "<label class=\"mdl-checkbox mdl-js-checkbox mdl-js-ripple-effect\" [attr.for]=\"field.id\" [formGroup]=\"formGroup\">     <input formControlName=\"{{controllerName}}\" type=\"checkbox\"            [attr.id]=\"field.id\"            class=\"mdl-checkbox__input\"            [checked]=\"field.value\"            [(ngModel)]=\"field.value\"            (ngModelChange)=\"changeValue(field)\">     <span class=\"mdl-checkbox__label\">{{field.nameKey | translate}}</span> </label>"
    }), __metadata("design:paramtypes", [core_1.ElementRef])], CheckboxWidget);
    exports.CheckboxWidget = CheckboxWidget;
});
$__System.registerDynamic("10", ["3"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var componentHandler;
    var WidgetComponent = function () {
        function WidgetComponent() {
            this.fieldChanged = new core_1.EventEmitter();
        }
        WidgetComponent.prototype.ngOnChanges = function (changes) {
            var field = changes['field'];
            if (field && field.currentValue) {
                this.fieldChanged.emit(field.currentValue.value);
                return;
            }
        };
        WidgetComponent.prototype.hasField = function () {
            return this.field ? true : false;
        };
        WidgetComponent.prototype.hasValue = function () {
            return this.field && this.field.value !== null && this.field.value !== undefined;
        };
        WidgetComponent.prototype.changeValue = function (field) {
            this.fieldChanged.emit(field);
        };
        WidgetComponent.prototype.ngAfterViewInit = function () {
            this.setupMaterialComponents(componentHandler);
        };
        WidgetComponent.prototype.setupMaterialComponents = function (handler) {
            if (handler) {
                handler.upgradeAllRegistered();
                return true;
            }
            return false;
        };
        return WidgetComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", Object)], WidgetComponent.prototype, "field", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], WidgetComponent.prototype, "fieldChanged", void 0);
    exports.WidgetComponent = WidgetComponent;
});
(function() {
var define = $__System.amdDefine;
!function(a, b) {
  "object" == typeof exports && "undefined" != typeof module ? module.exports = b() : "function" == typeof define && define.amd ? define("9", [], b) : a.moment = b();
}(this, function() {
  "use strict";
  function a() {
    return md.apply(null, arguments);
  }
  function b(a) {
    md = a;
  }
  function c(a) {
    return a instanceof Array || "[object Array]" === Object.prototype.toString.call(a);
  }
  function d(a) {
    return null != a && "[object Object]" === Object.prototype.toString.call(a);
  }
  function e(a) {
    var b;
    for (b in a)
      return !1;
    return !0;
  }
  function f(a) {
    return a instanceof Date || "[object Date]" === Object.prototype.toString.call(a);
  }
  function g(a, b) {
    var c,
        d = [];
    for (c = 0; c < a.length; ++c)
      d.push(b(a[c], c));
    return d;
  }
  function h(a, b) {
    return Object.prototype.hasOwnProperty.call(a, b);
  }
  function i(a, b) {
    for (var c in b)
      h(b, c) && (a[c] = b[c]);
    return h(b, "toString") && (a.toString = b.toString), h(b, "valueOf") && (a.valueOf = b.valueOf), a;
  }
  function j(a, b, c, d) {
    return qb(a, b, c, d, !0).utc();
  }
  function k() {
    return {
      empty: !1,
      unusedTokens: [],
      unusedInput: [],
      overflow: -2,
      charsLeftOver: 0,
      nullInput: !1,
      invalidMonth: null,
      invalidFormat: !1,
      userInvalidated: !1,
      iso: !1,
      parsedDateParts: [],
      meridiem: null
    };
  }
  function l(a) {
    return null == a._pf && (a._pf = k()), a._pf;
  }
  function m(a) {
    if (null == a._isValid) {
      var b = l(a),
          c = nd.call(b.parsedDateParts, function(a) {
            return null != a;
          }),
          d = !isNaN(a._d.getTime()) && b.overflow < 0 && !b.empty && !b.invalidMonth && !b.invalidWeekday && !b.nullInput && !b.invalidFormat && !b.userInvalidated && (!b.meridiem || b.meridiem && c);
      if (a._strict && (d = d && 0 === b.charsLeftOver && 0 === b.unusedTokens.length && void 0 === b.bigHour), null != Object.isFrozen && Object.isFrozen(a))
        return d;
      a._isValid = d;
    }
    return a._isValid;
  }
  function n(a) {
    var b = j(NaN);
    return null != a ? i(l(b), a) : l(b).userInvalidated = !0, b;
  }
  function o(a) {
    return void 0 === a;
  }
  function p(a, b) {
    var c,
        d,
        e;
    if (o(b._isAMomentObject) || (a._isAMomentObject = b._isAMomentObject), o(b._i) || (a._i = b._i), o(b._f) || (a._f = b._f), o(b._l) || (a._l = b._l), o(b._strict) || (a._strict = b._strict), o(b._tzm) || (a._tzm = b._tzm), o(b._isUTC) || (a._isUTC = b._isUTC), o(b._offset) || (a._offset = b._offset), o(b._pf) || (a._pf = l(b)), o(b._locale) || (a._locale = b._locale), od.length > 0)
      for (c in od)
        d = od[c], e = b[d], o(e) || (a[d] = e);
    return a;
  }
  function q(b) {
    p(this, b), this._d = new Date(null != b._d ? b._d.getTime() : NaN), pd === !1 && (pd = !0, a.updateOffset(this), pd = !1);
  }
  function r(a) {
    return a instanceof q || null != a && null != a._isAMomentObject;
  }
  function s(a) {
    return 0 > a ? Math.ceil(a) || 0 : Math.floor(a);
  }
  function t(a) {
    var b = +a,
        c = 0;
    return 0 !== b && isFinite(b) && (c = s(b)), c;
  }
  function u(a, b, c) {
    var d,
        e = Math.min(a.length, b.length),
        f = Math.abs(a.length - b.length),
        g = 0;
    for (d = 0; e > d; d++)
      (c && a[d] !== b[d] || !c && t(a[d]) !== t(b[d])) && g++;
    return g + f;
  }
  function v(b) {
    a.suppressDeprecationWarnings === !1 && "undefined" != typeof console && console.warn && console.warn("Deprecation warning: " + b);
  }
  function w(b, c) {
    var d = !0;
    return i(function() {
      if (null != a.deprecationHandler && a.deprecationHandler(null, b), d) {
        for (var e,
            f = [],
            g = 0; g < arguments.length; g++) {
          if (e = "", "object" == typeof arguments[g]) {
            e += "\n[" + g + "] ";
            for (var h in arguments[0])
              e += h + ": " + arguments[0][h] + ", ";
            e = e.slice(0, -2);
          } else
            e = arguments[g];
          f.push(e);
        }
        v(b + "\nArguments: " + Array.prototype.slice.call(f).join("") + "\n" + (new Error).stack), d = !1;
      }
      return c.apply(this, arguments);
    }, c);
  }
  function x(b, c) {
    null != a.deprecationHandler && a.deprecationHandler(b, c), qd[b] || (v(c), qd[b] = !0);
  }
  function y(a) {
    return a instanceof Function || "[object Function]" === Object.prototype.toString.call(a);
  }
  function z(a) {
    var b,
        c;
    for (c in a)
      b = a[c], y(b) ? this[c] = b : this["_" + c] = b;
    this._config = a, this._ordinalParseLenient = new RegExp(this._ordinalParse.source + "|" + /\d{1,2}/.source);
  }
  function A(a, b) {
    var c,
        e = i({}, a);
    for (c in b)
      h(b, c) && (d(a[c]) && d(b[c]) ? (e[c] = {}, i(e[c], a[c]), i(e[c], b[c])) : null != b[c] ? e[c] = b[c] : delete e[c]);
    for (c in a)
      h(a, c) && !h(b, c) && d(a[c]) && (e[c] = i({}, e[c]));
    return e;
  }
  function B(a) {
    null != a && this.set(a);
  }
  function C(a, b, c) {
    var d = this._calendar[a] || this._calendar.sameElse;
    return y(d) ? d.call(b, c) : d;
  }
  function D(a) {
    var b = this._longDateFormat[a],
        c = this._longDateFormat[a.toUpperCase()];
    return b || !c ? b : (this._longDateFormat[a] = c.replace(/MMMM|MM|DD|dddd/g, function(a) {
      return a.slice(1);
    }), this._longDateFormat[a]);
  }
  function E() {
    return this._invalidDate;
  }
  function F(a) {
    return this._ordinal.replace("%d", a);
  }
  function G(a, b, c, d) {
    var e = this._relativeTime[c];
    return y(e) ? e(a, b, c, d) : e.replace(/%d/i, a);
  }
  function H(a, b) {
    var c = this._relativeTime[a > 0 ? "future" : "past"];
    return y(c) ? c(b) : c.replace(/%s/i, b);
  }
  function I(a, b) {
    var c = a.toLowerCase();
    zd[c] = zd[c + "s"] = zd[b] = a;
  }
  function J(a) {
    return "string" == typeof a ? zd[a] || zd[a.toLowerCase()] : void 0;
  }
  function K(a) {
    var b,
        c,
        d = {};
    for (c in a)
      h(a, c) && (b = J(c), b && (d[b] = a[c]));
    return d;
  }
  function L(a, b) {
    Ad[a] = b;
  }
  function M(a) {
    var b = [];
    for (var c in a)
      b.push({
        unit: c,
        priority: Ad[c]
      });
    return b.sort(function(a, b) {
      return a.priority - b.priority;
    }), b;
  }
  function N(b, c) {
    return function(d) {
      return null != d ? (P(this, b, d), a.updateOffset(this, c), this) : O(this, b);
    };
  }
  function O(a, b) {
    return a.isValid() ? a._d["get" + (a._isUTC ? "UTC" : "") + b]() : NaN;
  }
  function P(a, b, c) {
    a.isValid() && a._d["set" + (a._isUTC ? "UTC" : "") + b](c);
  }
  function Q(a) {
    return a = J(a), y(this[a]) ? this[a]() : this;
  }
  function R(a, b) {
    if ("object" == typeof a) {
      a = K(a);
      for (var c = M(a),
          d = 0; d < c.length; d++)
        this[c[d].unit](a[c[d].unit]);
    } else if (a = J(a), y(this[a]))
      return this[a](b);
    return this;
  }
  function S(a, b, c) {
    var d = "" + Math.abs(a),
        e = b - d.length,
        f = a >= 0;
    return (f ? c ? "+" : "" : "-") + Math.pow(10, Math.max(0, e)).toString().substr(1) + d;
  }
  function T(a, b, c, d) {
    var e = d;
    "string" == typeof d && (e = function() {
      return this[d]();
    }), a && (Ed[a] = e), b && (Ed[b[0]] = function() {
      return S(e.apply(this, arguments), b[1], b[2]);
    }), c && (Ed[c] = function() {
      return this.localeData().ordinal(e.apply(this, arguments), a);
    });
  }
  function U(a) {
    return a.match(/\[[\s\S]/) ? a.replace(/^\[|\]$/g, "") : a.replace(/\\/g, "");
  }
  function V(a) {
    var b,
        c,
        d = a.match(Bd);
    for (b = 0, c = d.length; c > b; b++)
      Ed[d[b]] ? d[b] = Ed[d[b]] : d[b] = U(d[b]);
    return function(b) {
      var e,
          f = "";
      for (e = 0; c > e; e++)
        f += d[e] instanceof Function ? d[e].call(b, a) : d[e];
      return f;
    };
  }
  function W(a, b) {
    return a.isValid() ? (b = X(b, a.localeData()), Dd[b] = Dd[b] || V(b), Dd[b](a)) : a.localeData().invalidDate();
  }
  function X(a, b) {
    function c(a) {
      return b.longDateFormat(a) || a;
    }
    var d = 5;
    for (Cd.lastIndex = 0; d >= 0 && Cd.test(a); )
      a = a.replace(Cd, c), Cd.lastIndex = 0, d -= 1;
    return a;
  }
  function Y(a, b, c) {
    Wd[a] = y(b) ? b : function(a, d) {
      return a && c ? c : b;
    };
  }
  function Z(a, b) {
    return h(Wd, a) ? Wd[a](b._strict, b._locale) : new RegExp($(a));
  }
  function $(a) {
    return _(a.replace("\\", "").replace(/\\(\[)|\\(\])|\[([^\]\[]*)\]|\\(.)/g, function(a, b, c, d, e) {
      return b || c || d || e;
    }));
  }
  function _(a) {
    return a.replace(/[-\/\\^$*+?.()|[\]{}]/g, "\\$&");
  }
  function aa(a, b) {
    var c,
        d = b;
    for ("string" == typeof a && (a = [a]), "number" == typeof b && (d = function(a, c) {
      c[b] = t(a);
    }), c = 0; c < a.length; c++)
      Xd[a[c]] = d;
  }
  function ba(a, b) {
    aa(a, function(a, c, d, e) {
      d._w = d._w || {}, b(a, d._w, d, e);
    });
  }
  function ca(a, b, c) {
    null != b && h(Xd, a) && Xd[a](b, c._a, c, a);
  }
  function da(a, b) {
    return new Date(Date.UTC(a, b + 1, 0)).getUTCDate();
  }
  function ea(a, b) {
    return a ? c(this._months) ? this._months[a.month()] : this._months[(this._months.isFormat || fe).test(b) ? "format" : "standalone"][a.month()] : this._months;
  }
  function fa(a, b) {
    return a ? c(this._monthsShort) ? this._monthsShort[a.month()] : this._monthsShort[fe.test(b) ? "format" : "standalone"][a.month()] : this._monthsShort;
  }
  function ga(a, b, c) {
    var d,
        e,
        f,
        g = a.toLocaleLowerCase();
    if (!this._monthsParse)
      for (this._monthsParse = [], this._longMonthsParse = [], this._shortMonthsParse = [], d = 0; 12 > d; ++d)
        f = j([2e3, d]), this._shortMonthsParse[d] = this.monthsShort(f, "").toLocaleLowerCase(), this._longMonthsParse[d] = this.months(f, "").toLocaleLowerCase();
    return c ? "MMM" === b ? (e = sd.call(this._shortMonthsParse, g), -1 !== e ? e : null) : (e = sd.call(this._longMonthsParse, g), -1 !== e ? e : null) : "MMM" === b ? (e = sd.call(this._shortMonthsParse, g), -1 !== e ? e : (e = sd.call(this._longMonthsParse, g), -1 !== e ? e : null)) : (e = sd.call(this._longMonthsParse, g), -1 !== e ? e : (e = sd.call(this._shortMonthsParse, g), -1 !== e ? e : null));
  }
  function ha(a, b, c) {
    var d,
        e,
        f;
    if (this._monthsParseExact)
      return ga.call(this, a, b, c);
    for (this._monthsParse || (this._monthsParse = [], this._longMonthsParse = [], this._shortMonthsParse = []), d = 0; 12 > d; d++) {
      if (e = j([2e3, d]), c && !this._longMonthsParse[d] && (this._longMonthsParse[d] = new RegExp("^" + this.months(e, "").replace(".", "") + "$", "i"), this._shortMonthsParse[d] = new RegExp("^" + this.monthsShort(e, "").replace(".", "") + "$", "i")), c || this._monthsParse[d] || (f = "^" + this.months(e, "") + "|^" + this.monthsShort(e, ""), this._monthsParse[d] = new RegExp(f.replace(".", ""), "i")), c && "MMMM" === b && this._longMonthsParse[d].test(a))
        return d;
      if (c && "MMM" === b && this._shortMonthsParse[d].test(a))
        return d;
      if (!c && this._monthsParse[d].test(a))
        return d;
    }
  }
  function ia(a, b) {
    var c;
    if (!a.isValid())
      return a;
    if ("string" == typeof b)
      if (/^\d+$/.test(b))
        b = t(b);
      else if (b = a.localeData().monthsParse(b), "number" != typeof b)
        return a;
    return c = Math.min(a.date(), da(a.year(), b)), a._d["set" + (a._isUTC ? "UTC" : "") + "Month"](b, c), a;
  }
  function ja(b) {
    return null != b ? (ia(this, b), a.updateOffset(this, !0), this) : O(this, "Month");
  }
  function ka() {
    return da(this.year(), this.month());
  }
  function la(a) {
    return this._monthsParseExact ? (h(this, "_monthsRegex") || na.call(this), a ? this._monthsShortStrictRegex : this._monthsShortRegex) : (h(this, "_monthsShortRegex") || (this._monthsShortRegex = ie), this._monthsShortStrictRegex && a ? this._monthsShortStrictRegex : this._monthsShortRegex);
  }
  function ma(a) {
    return this._monthsParseExact ? (h(this, "_monthsRegex") || na.call(this), a ? this._monthsStrictRegex : this._monthsRegex) : (h(this, "_monthsRegex") || (this._monthsRegex = je), this._monthsStrictRegex && a ? this._monthsStrictRegex : this._monthsRegex);
  }
  function na() {
    function a(a, b) {
      return b.length - a.length;
    }
    var b,
        c,
        d = [],
        e = [],
        f = [];
    for (b = 0; 12 > b; b++)
      c = j([2e3, b]), d.push(this.monthsShort(c, "")), e.push(this.months(c, "")), f.push(this.months(c, "")), f.push(this.monthsShort(c, ""));
    for (d.sort(a), e.sort(a), f.sort(a), b = 0; 12 > b; b++)
      d[b] = _(d[b]), e[b] = _(e[b]);
    for (b = 0; 24 > b; b++)
      f[b] = _(f[b]);
    this._monthsRegex = new RegExp("^(" + f.join("|") + ")", "i"), this._monthsShortRegex = this._monthsRegex, this._monthsStrictRegex = new RegExp("^(" + e.join("|") + ")", "i"), this._monthsShortStrictRegex = new RegExp("^(" + d.join("|") + ")", "i");
  }
  function oa(a) {
    return pa(a) ? 366 : 365;
  }
  function pa(a) {
    return a % 4 === 0 && a % 100 !== 0 || a % 400 === 0;
  }
  function qa() {
    return pa(this.year());
  }
  function ra(a, b, c, d, e, f, g) {
    var h = new Date(a, b, c, d, e, f, g);
    return 100 > a && a >= 0 && isFinite(h.getFullYear()) && h.setFullYear(a), h;
  }
  function sa(a) {
    var b = new Date(Date.UTC.apply(null, arguments));
    return 100 > a && a >= 0 && isFinite(b.getUTCFullYear()) && b.setUTCFullYear(a), b;
  }
  function ta(a, b, c) {
    var d = 7 + b - c,
        e = (7 + sa(a, 0, d).getUTCDay() - b) % 7;
    return -e + d - 1;
  }
  function ua(a, b, c, d, e) {
    var f,
        g,
        h = (7 + c - d) % 7,
        i = ta(a, d, e),
        j = 1 + 7 * (b - 1) + h + i;
    return 0 >= j ? (f = a - 1, g = oa(f) + j) : j > oa(a) ? (f = a + 1, g = j - oa(a)) : (f = a, g = j), {
      year: f,
      dayOfYear: g
    };
  }
  function va(a, b, c) {
    var d,
        e,
        f = ta(a.year(), b, c),
        g = Math.floor((a.dayOfYear() - f - 1) / 7) + 1;
    return 1 > g ? (e = a.year() - 1, d = g + wa(e, b, c)) : g > wa(a.year(), b, c) ? (d = g - wa(a.year(), b, c), e = a.year() + 1) : (e = a.year(), d = g), {
      week: d,
      year: e
    };
  }
  function wa(a, b, c) {
    var d = ta(a, b, c),
        e = ta(a + 1, b, c);
    return (oa(a) - d + e) / 7;
  }
  function xa(a) {
    return va(a, this._week.dow, this._week.doy).week;
  }
  function ya() {
    return this._week.dow;
  }
  function za() {
    return this._week.doy;
  }
  function Aa(a) {
    var b = this.localeData().week(this);
    return null == a ? b : this.add(7 * (a - b), "d");
  }
  function Ba(a) {
    var b = va(this, 1, 4).week;
    return null == a ? b : this.add(7 * (a - b), "d");
  }
  function Ca(a, b) {
    return "string" != typeof a ? a : isNaN(a) ? (a = b.weekdaysParse(a), "number" == typeof a ? a : null) : parseInt(a, 10);
  }
  function Da(a, b) {
    return "string" == typeof a ? b.weekdaysParse(a) % 7 || 7 : isNaN(a) ? null : a;
  }
  function Ea(a, b) {
    return a ? c(this._weekdays) ? this._weekdays[a.day()] : this._weekdays[this._weekdays.isFormat.test(b) ? "format" : "standalone"][a.day()] : this._weekdays;
  }
  function Fa(a) {
    return a ? this._weekdaysShort[a.day()] : this._weekdaysShort;
  }
  function Ga(a) {
    return a ? this._weekdaysMin[a.day()] : this._weekdaysMin;
  }
  function Ha(a, b, c) {
    var d,
        e,
        f,
        g = a.toLocaleLowerCase();
    if (!this._weekdaysParse)
      for (this._weekdaysParse = [], this._shortWeekdaysParse = [], this._minWeekdaysParse = [], d = 0; 7 > d; ++d)
        f = j([2e3, 1]).day(d), this._minWeekdaysParse[d] = this.weekdaysMin(f, "").toLocaleLowerCase(), this._shortWeekdaysParse[d] = this.weekdaysShort(f, "").toLocaleLowerCase(), this._weekdaysParse[d] = this.weekdays(f, "").toLocaleLowerCase();
    return c ? "dddd" === b ? (e = sd.call(this._weekdaysParse, g), -1 !== e ? e : null) : "ddd" === b ? (e = sd.call(this._shortWeekdaysParse, g), -1 !== e ? e : null) : (e = sd.call(this._minWeekdaysParse, g), -1 !== e ? e : null) : "dddd" === b ? (e = sd.call(this._weekdaysParse, g), -1 !== e ? e : (e = sd.call(this._shortWeekdaysParse, g), -1 !== e ? e : (e = sd.call(this._minWeekdaysParse, g), -1 !== e ? e : null))) : "ddd" === b ? (e = sd.call(this._shortWeekdaysParse, g), -1 !== e ? e : (e = sd.call(this._weekdaysParse, g), -1 !== e ? e : (e = sd.call(this._minWeekdaysParse, g), -1 !== e ? e : null))) : (e = sd.call(this._minWeekdaysParse, g), -1 !== e ? e : (e = sd.call(this._weekdaysParse, g), -1 !== e ? e : (e = sd.call(this._shortWeekdaysParse, g), -1 !== e ? e : null)));
  }
  function Ia(a, b, c) {
    var d,
        e,
        f;
    if (this._weekdaysParseExact)
      return Ha.call(this, a, b, c);
    for (this._weekdaysParse || (this._weekdaysParse = [], this._minWeekdaysParse = [], this._shortWeekdaysParse = [], this._fullWeekdaysParse = []), d = 0; 7 > d; d++) {
      if (e = j([2e3, 1]).day(d), c && !this._fullWeekdaysParse[d] && (this._fullWeekdaysParse[d] = new RegExp("^" + this.weekdays(e, "").replace(".", ".?") + "$", "i"), this._shortWeekdaysParse[d] = new RegExp("^" + this.weekdaysShort(e, "").replace(".", ".?") + "$", "i"), this._minWeekdaysParse[d] = new RegExp("^" + this.weekdaysMin(e, "").replace(".", ".?") + "$", "i")), this._weekdaysParse[d] || (f = "^" + this.weekdays(e, "") + "|^" + this.weekdaysShort(e, "") + "|^" + this.weekdaysMin(e, ""), this._weekdaysParse[d] = new RegExp(f.replace(".", ""), "i")), c && "dddd" === b && this._fullWeekdaysParse[d].test(a))
        return d;
      if (c && "ddd" === b && this._shortWeekdaysParse[d].test(a))
        return d;
      if (c && "dd" === b && this._minWeekdaysParse[d].test(a))
        return d;
      if (!c && this._weekdaysParse[d].test(a))
        return d;
    }
  }
  function Ja(a) {
    if (!this.isValid())
      return null != a ? this : NaN;
    var b = this._isUTC ? this._d.getUTCDay() : this._d.getDay();
    return null != a ? (a = Ca(a, this.localeData()), this.add(a - b, "d")) : b;
  }
  function Ka(a) {
    if (!this.isValid())
      return null != a ? this : NaN;
    var b = (this.day() + 7 - this.localeData()._week.dow) % 7;
    return null == a ? b : this.add(a - b, "d");
  }
  function La(a) {
    if (!this.isValid())
      return null != a ? this : NaN;
    if (null != a) {
      var b = Da(a, this.localeData());
      return this.day(this.day() % 7 ? b : b - 7);
    }
    return this.day() || 7;
  }
  function Ma(a) {
    return this._weekdaysParseExact ? (h(this, "_weekdaysRegex") || Pa.call(this), a ? this._weekdaysStrictRegex : this._weekdaysRegex) : (h(this, "_weekdaysRegex") || (this._weekdaysRegex = pe), this._weekdaysStrictRegex && a ? this._weekdaysStrictRegex : this._weekdaysRegex);
  }
  function Na(a) {
    return this._weekdaysParseExact ? (h(this, "_weekdaysRegex") || Pa.call(this), a ? this._weekdaysShortStrictRegex : this._weekdaysShortRegex) : (h(this, "_weekdaysShortRegex") || (this._weekdaysShortRegex = qe), this._weekdaysShortStrictRegex && a ? this._weekdaysShortStrictRegex : this._weekdaysShortRegex);
  }
  function Oa(a) {
    return this._weekdaysParseExact ? (h(this, "_weekdaysRegex") || Pa.call(this), a ? this._weekdaysMinStrictRegex : this._weekdaysMinRegex) : (h(this, "_weekdaysMinRegex") || (this._weekdaysMinRegex = re), this._weekdaysMinStrictRegex && a ? this._weekdaysMinStrictRegex : this._weekdaysMinRegex);
  }
  function Pa() {
    function a(a, b) {
      return b.length - a.length;
    }
    var b,
        c,
        d,
        e,
        f,
        g = [],
        h = [],
        i = [],
        k = [];
    for (b = 0; 7 > b; b++)
      c = j([2e3, 1]).day(b), d = this.weekdaysMin(c, ""), e = this.weekdaysShort(c, ""), f = this.weekdays(c, ""), g.push(d), h.push(e), i.push(f), k.push(d), k.push(e), k.push(f);
    for (g.sort(a), h.sort(a), i.sort(a), k.sort(a), b = 0; 7 > b; b++)
      h[b] = _(h[b]), i[b] = _(i[b]), k[b] = _(k[b]);
    this._weekdaysRegex = new RegExp("^(" + k.join("|") + ")", "i"), this._weekdaysShortRegex = this._weekdaysRegex, this._weekdaysMinRegex = this._weekdaysRegex, this._weekdaysStrictRegex = new RegExp("^(" + i.join("|") + ")", "i"), this._weekdaysShortStrictRegex = new RegExp("^(" + h.join("|") + ")", "i"), this._weekdaysMinStrictRegex = new RegExp("^(" + g.join("|") + ")", "i");
  }
  function Qa() {
    return this.hours() % 12 || 12;
  }
  function Ra() {
    return this.hours() || 24;
  }
  function Sa(a, b) {
    T(a, 0, 0, function() {
      return this.localeData().meridiem(this.hours(), this.minutes(), b);
    });
  }
  function Ta(a, b) {
    return b._meridiemParse;
  }
  function Ua(a) {
    return "p" === (a + "").toLowerCase().charAt(0);
  }
  function Va(a, b, c) {
    return a > 11 ? c ? "pm" : "PM" : c ? "am" : "AM";
  }
  function Wa(a) {
    return a ? a.toLowerCase().replace("_", "-") : a;
  }
  function Xa(a) {
    for (var b,
        c,
        d,
        e,
        f = 0; f < a.length; ) {
      for (e = Wa(a[f]).split("-"), b = e.length, c = Wa(a[f + 1]), c = c ? c.split("-") : null; b > 0; ) {
        if (d = Ya(e.slice(0, b).join("-")))
          return d;
        if (c && c.length >= b && u(e, c, !0) >= b - 1)
          break;
        b--;
      }
      f++;
    }
    return null;
  }
  function Ya(a) {
    var b = null;
    if (!we[a] && "undefined" != typeof module && module && module.exports)
      try {
        b = se._abbr, require("./locale/" + a), Za(b);
      } catch (c) {}
    return we[a];
  }
  function Za(a, b) {
    var c;
    return a && (c = o(b) ? ab(a) : $a(a, b), c && (se = c)), se._abbr;
  }
  function $a(a, b) {
    if (null !== b) {
      var c = ve;
      return b.abbr = a, null != we[a] ? (x("defineLocaleOverride", "use moment.updateLocale(localeName, config) to change an existing locale. moment.defineLocale(localeName, config) should only be used for creating a new locale See http://momentjs.com/guides/#/warnings/define-locale/ for more info."), c = we[a]._config) : null != b.parentLocale && (null != we[b.parentLocale] ? c = we[b.parentLocale]._config : x("parentLocaleUndefined", "specified parentLocale is not defined yet. See http://momentjs.com/guides/#/warnings/parent-locale/")), we[a] = new B(A(c, b)), Za(a), we[a];
    }
    return delete we[a], null;
  }
  function _a(a, b) {
    if (null != b) {
      var c,
          d = ve;
      null != we[a] && (d = we[a]._config), b = A(d, b), c = new B(b), c.parentLocale = we[a], we[a] = c, Za(a);
    } else
      null != we[a] && (null != we[a].parentLocale ? we[a] = we[a].parentLocale : null != we[a] && delete we[a]);
    return we[a];
  }
  function ab(a) {
    var b;
    if (a && a._locale && a._locale._abbr && (a = a._locale._abbr), !a)
      return se;
    if (!c(a)) {
      if (b = Ya(a))
        return b;
      a = [a];
    }
    return Xa(a);
  }
  function bb() {
    return rd(we);
  }
  function cb(a) {
    var b,
        c = a._a;
    return c && -2 === l(a).overflow && (b = c[Zd] < 0 || c[Zd] > 11 ? Zd : c[$d] < 1 || c[$d] > da(c[Yd], c[Zd]) ? $d : c[_d] < 0 || c[_d] > 24 || 24 === c[_d] && (0 !== c[ae] || 0 !== c[be] || 0 !== c[ce]) ? _d : c[ae] < 0 || c[ae] > 59 ? ae : c[be] < 0 || c[be] > 59 ? be : c[ce] < 0 || c[ce] > 999 ? ce : -1, l(a)._overflowDayOfYear && (Yd > b || b > $d) && (b = $d), l(a)._overflowWeeks && -1 === b && (b = de), l(a)._overflowWeekday && -1 === b && (b = ee), l(a).overflow = b), a;
  }
  function db(a) {
    var b,
        c,
        d,
        e,
        f,
        g,
        h = a._i,
        i = xe.exec(h) || ye.exec(h);
    if (i) {
      for (l(a).iso = !0, b = 0, c = Ae.length; c > b; b++)
        if (Ae[b][1].exec(i[1])) {
          e = Ae[b][0], d = Ae[b][2] !== !1;
          break;
        }
      if (null == e)
        return void(a._isValid = !1);
      if (i[3]) {
        for (b = 0, c = Be.length; c > b; b++)
          if (Be[b][1].exec(i[3])) {
            f = (i[2] || " ") + Be[b][0];
            break;
          }
        if (null == f)
          return void(a._isValid = !1);
      }
      if (!d && null != f)
        return void(a._isValid = !1);
      if (i[4]) {
        if (!ze.exec(i[4]))
          return void(a._isValid = !1);
        g = "Z";
      }
      a._f = e + (f || "") + (g || ""), jb(a);
    } else
      a._isValid = !1;
  }
  function eb(b) {
    var c = Ce.exec(b._i);
    return null !== c ? void(b._d = new Date(+c[1])) : (db(b), void(b._isValid === !1 && (delete b._isValid, a.createFromInputFallback(b))));
  }
  function fb(a, b, c) {
    return null != a ? a : null != b ? b : c;
  }
  function gb(b) {
    var c = new Date(a.now());
    return b._useUTC ? [c.getUTCFullYear(), c.getUTCMonth(), c.getUTCDate()] : [c.getFullYear(), c.getMonth(), c.getDate()];
  }
  function hb(a) {
    var b,
        c,
        d,
        e,
        f = [];
    if (!a._d) {
      for (d = gb(a), a._w && null == a._a[$d] && null == a._a[Zd] && ib(a), a._dayOfYear && (e = fb(a._a[Yd], d[Yd]), a._dayOfYear > oa(e) && (l(a)._overflowDayOfYear = !0), c = sa(e, 0, a._dayOfYear), a._a[Zd] = c.getUTCMonth(), a._a[$d] = c.getUTCDate()), b = 0; 3 > b && null == a._a[b]; ++b)
        a._a[b] = f[b] = d[b];
      for (; 7 > b; b++)
        a._a[b] = f[b] = null == a._a[b] ? 2 === b ? 1 : 0 : a._a[b];
      24 === a._a[_d] && 0 === a._a[ae] && 0 === a._a[be] && 0 === a._a[ce] && (a._nextDay = !0, a._a[_d] = 0), a._d = (a._useUTC ? sa : ra).apply(null, f), null != a._tzm && a._d.setUTCMinutes(a._d.getUTCMinutes() - a._tzm), a._nextDay && (a._a[_d] = 24);
    }
  }
  function ib(a) {
    var b,
        c,
        d,
        e,
        f,
        g,
        h,
        i;
    b = a._w, null != b.GG || null != b.W || null != b.E ? (f = 1, g = 4, c = fb(b.GG, a._a[Yd], va(rb(), 1, 4).year), d = fb(b.W, 1), e = fb(b.E, 1), (1 > e || e > 7) && (i = !0)) : (f = a._locale._week.dow, g = a._locale._week.doy, c = fb(b.gg, a._a[Yd], va(rb(), f, g).year), d = fb(b.w, 1), null != b.d ? (e = b.d, (0 > e || e > 6) && (i = !0)) : null != b.e ? (e = b.e + f, (b.e < 0 || b.e > 6) && (i = !0)) : e = f), 1 > d || d > wa(c, f, g) ? l(a)._overflowWeeks = !0 : null != i ? l(a)._overflowWeekday = !0 : (h = ua(c, d, e, f, g), a._a[Yd] = h.year, a._dayOfYear = h.dayOfYear);
  }
  function jb(b) {
    if (b._f === a.ISO_8601)
      return void db(b);
    b._a = [], l(b).empty = !0;
    var c,
        d,
        e,
        f,
        g,
        h = "" + b._i,
        i = h.length,
        j = 0;
    for (e = X(b._f, b._locale).match(Bd) || [], c = 0; c < e.length; c++)
      f = e[c], d = (h.match(Z(f, b)) || [])[0], d && (g = h.substr(0, h.indexOf(d)), g.length > 0 && l(b).unusedInput.push(g), h = h.slice(h.indexOf(d) + d.length), j += d.length), Ed[f] ? (d ? l(b).empty = !1 : l(b).unusedTokens.push(f), ca(f, d, b)) : b._strict && !d && l(b).unusedTokens.push(f);
    l(b).charsLeftOver = i - j, h.length > 0 && l(b).unusedInput.push(h), b._a[_d] <= 12 && l(b).bigHour === !0 && b._a[_d] > 0 && (l(b).bigHour = void 0), l(b).parsedDateParts = b._a.slice(0), l(b).meridiem = b._meridiem, b._a[_d] = kb(b._locale, b._a[_d], b._meridiem), hb(b), cb(b);
  }
  function kb(a, b, c) {
    var d;
    return null == c ? b : null != a.meridiemHour ? a.meridiemHour(b, c) : null != a.isPM ? (d = a.isPM(c), d && 12 > b && (b += 12), d || 12 !== b || (b = 0), b) : b;
  }
  function lb(a) {
    var b,
        c,
        d,
        e,
        f;
    if (0 === a._f.length)
      return l(a).invalidFormat = !0, void(a._d = new Date(NaN));
    for (e = 0; e < a._f.length; e++)
      f = 0, b = p({}, a), null != a._useUTC && (b._useUTC = a._useUTC), b._f = a._f[e], jb(b), m(b) && (f += l(b).charsLeftOver, f += 10 * l(b).unusedTokens.length, l(b).score = f, (null == d || d > f) && (d = f, c = b));
    i(a, c || b);
  }
  function mb(a) {
    if (!a._d) {
      var b = K(a._i);
      a._a = g([b.year, b.month, b.day || b.date, b.hour, b.minute, b.second, b.millisecond], function(a) {
        return a && parseInt(a, 10);
      }), hb(a);
    }
  }
  function nb(a) {
    var b = new q(cb(ob(a)));
    return b._nextDay && (b.add(1, "d"), b._nextDay = void 0), b;
  }
  function ob(a) {
    var b = a._i,
        d = a._f;
    return a._locale = a._locale || ab(a._l), null === b || void 0 === d && "" === b ? n({nullInput: !0}) : ("string" == typeof b && (a._i = b = a._locale.preparse(b)), r(b) ? new q(cb(b)) : (c(d) ? lb(a) : f(b) ? a._d = b : d ? jb(a) : pb(a), m(a) || (a._d = null), a));
  }
  function pb(b) {
    var d = b._i;
    void 0 === d ? b._d = new Date(a.now()) : f(d) ? b._d = new Date(d.valueOf()) : "string" == typeof d ? eb(b) : c(d) ? (b._a = g(d.slice(0), function(a) {
      return parseInt(a, 10);
    }), hb(b)) : "object" == typeof d ? mb(b) : "number" == typeof d ? b._d = new Date(d) : a.createFromInputFallback(b);
  }
  function qb(a, b, f, g, h) {
    var i = {};
    return "boolean" == typeof f && (g = f, f = void 0), (d(a) && e(a) || c(a) && 0 === a.length) && (a = void 0), i._isAMomentObject = !0, i._useUTC = i._isUTC = h, i._l = f, i._i = a, i._f = b, i._strict = g, nb(i);
  }
  function rb(a, b, c, d) {
    return qb(a, b, c, d, !1);
  }
  function sb(a, b) {
    var d,
        e;
    if (1 === b.length && c(b[0]) && (b = b[0]), !b.length)
      return rb();
    for (d = b[0], e = 1; e < b.length; ++e)
      b[e].isValid() && !b[e][a](d) || (d = b[e]);
    return d;
  }
  function tb() {
    var a = [].slice.call(arguments, 0);
    return sb("isBefore", a);
  }
  function ub() {
    var a = [].slice.call(arguments, 0);
    return sb("isAfter", a);
  }
  function vb(a) {
    var b = K(a),
        c = b.year || 0,
        d = b.quarter || 0,
        e = b.month || 0,
        f = b.week || 0,
        g = b.day || 0,
        h = b.hour || 0,
        i = b.minute || 0,
        j = b.second || 0,
        k = b.millisecond || 0;
    this._milliseconds = +k + 1e3 * j + 6e4 * i + 1e3 * h * 60 * 60, this._days = +g + 7 * f, this._months = +e + 3 * d + 12 * c, this._data = {}, this._locale = ab(), this._bubble();
  }
  function wb(a) {
    return a instanceof vb;
  }
  function xb(a) {
    return 0 > a ? -1 * Math.round(-1 * a) : Math.round(a);
  }
  function yb(a, b) {
    T(a, 0, 0, function() {
      var a = this.utcOffset(),
          c = "+";
      return 0 > a && (a = -a, c = "-"), c + S(~~(a / 60), 2) + b + S(~~a % 60, 2);
    });
  }
  function zb(a, b) {
    var c = (b || "").match(a) || [],
        d = c[c.length - 1] || [],
        e = (d + "").match(Ge) || ["-", 0, 0],
        f = +(60 * e[1]) + t(e[2]);
    return "+" === e[0] ? f : -f;
  }
  function Ab(b, c) {
    var d,
        e;
    return c._isUTC ? (d = c.clone(), e = (r(b) || f(b) ? b.valueOf() : rb(b).valueOf()) - d.valueOf(), d._d.setTime(d._d.valueOf() + e), a.updateOffset(d, !1), d) : rb(b).local();
  }
  function Bb(a) {
    return 15 * -Math.round(a._d.getTimezoneOffset() / 15);
  }
  function Cb(b, c) {
    var d,
        e = this._offset || 0;
    return this.isValid() ? null != b ? ("string" == typeof b ? b = zb(Td, b) : Math.abs(b) < 16 && (b = 60 * b), !this._isUTC && c && (d = Bb(this)), this._offset = b, this._isUTC = !0, null != d && this.add(d, "m"), e !== b && (!c || this._changeInProgress ? Sb(this, Nb(b - e, "m"), 1, !1) : this._changeInProgress || (this._changeInProgress = !0, a.updateOffset(this, !0), this._changeInProgress = null)), this) : this._isUTC ? e : Bb(this) : null != b ? this : NaN;
  }
  function Db(a, b) {
    return null != a ? ("string" != typeof a && (a = -a), this.utcOffset(a, b), this) : -this.utcOffset();
  }
  function Eb(a) {
    return this.utcOffset(0, a);
  }
  function Fb(a) {
    return this._isUTC && (this.utcOffset(0, a), this._isUTC = !1, a && this.subtract(Bb(this), "m")), this;
  }
  function Gb() {
    if (this._tzm)
      this.utcOffset(this._tzm);
    else if ("string" == typeof this._i) {
      var a = zb(Sd, this._i);
      0 === a ? this.utcOffset(0, !0) : this.utcOffset(zb(Sd, this._i));
    }
    return this;
  }
  function Hb(a) {
    return this.isValid() ? (a = a ? rb(a).utcOffset() : 0, (this.utcOffset() - a) % 60 === 0) : !1;
  }
  function Ib() {
    return this.utcOffset() > this.clone().month(0).utcOffset() || this.utcOffset() > this.clone().month(5).utcOffset();
  }
  function Jb() {
    if (!o(this._isDSTShifted))
      return this._isDSTShifted;
    var a = {};
    if (p(a, this), a = ob(a), a._a) {
      var b = a._isUTC ? j(a._a) : rb(a._a);
      this._isDSTShifted = this.isValid() && u(a._a, b.toArray()) > 0;
    } else
      this._isDSTShifted = !1;
    return this._isDSTShifted;
  }
  function Kb() {
    return this.isValid() ? !this._isUTC : !1;
  }
  function Lb() {
    return this.isValid() ? this._isUTC : !1;
  }
  function Mb() {
    return this.isValid() ? this._isUTC && 0 === this._offset : !1;
  }
  function Nb(a, b) {
    var c,
        d,
        e,
        f = a,
        g = null;
    return wb(a) ? f = {
      ms: a._milliseconds,
      d: a._days,
      M: a._months
    } : "number" == typeof a ? (f = {}, b ? f[b] = a : f.milliseconds = a) : (g = He.exec(a)) ? (c = "-" === g[1] ? -1 : 1, f = {
      y: 0,
      d: t(g[$d]) * c,
      h: t(g[_d]) * c,
      m: t(g[ae]) * c,
      s: t(g[be]) * c,
      ms: t(xb(1e3 * g[ce])) * c
    }) : (g = Ie.exec(a)) ? (c = "-" === g[1] ? -1 : 1, f = {
      y: Ob(g[2], c),
      M: Ob(g[3], c),
      w: Ob(g[4], c),
      d: Ob(g[5], c),
      h: Ob(g[6], c),
      m: Ob(g[7], c),
      s: Ob(g[8], c)
    }) : null == f ? f = {} : "object" == typeof f && ("from" in f || "to" in f) && (e = Qb(rb(f.from), rb(f.to)), f = {}, f.ms = e.milliseconds, f.M = e.months), d = new vb(f), wb(a) && h(a, "_locale") && (d._locale = a._locale), d;
  }
  function Ob(a, b) {
    var c = a && parseFloat(a.replace(",", "."));
    return (isNaN(c) ? 0 : c) * b;
  }
  function Pb(a, b) {
    var c = {
      milliseconds: 0,
      months: 0
    };
    return c.months = b.month() - a.month() + 12 * (b.year() - a.year()), a.clone().add(c.months, "M").isAfter(b) && --c.months, c.milliseconds = +b - +a.clone().add(c.months, "M"), c;
  }
  function Qb(a, b) {
    var c;
    return a.isValid() && b.isValid() ? (b = Ab(b, a), a.isBefore(b) ? c = Pb(a, b) : (c = Pb(b, a), c.milliseconds = -c.milliseconds, c.months = -c.months), c) : {
      milliseconds: 0,
      months: 0
    };
  }
  function Rb(a, b) {
    return function(c, d) {
      var e,
          f;
      return null === d || isNaN(+d) || (x(b, "moment()." + b + "(period, number) is deprecated. Please use moment()." + b + "(number, period). See http://momentjs.com/guides/#/warnings/add-inverted-param/ for more info."), f = c, c = d, d = f), c = "string" == typeof c ? +c : c, e = Nb(c, d), Sb(this, e, a), this;
    };
  }
  function Sb(b, c, d, e) {
    var f = c._milliseconds,
        g = xb(c._days),
        h = xb(c._months);
    b.isValid() && (e = null == e ? !0 : e, f && b._d.setTime(b._d.valueOf() + f * d), g && P(b, "Date", O(b, "Date") + g * d), h && ia(b, O(b, "Month") + h * d), e && a.updateOffset(b, g || h));
  }
  function Tb(a, b) {
    var c = a.diff(b, "days", !0);
    return -6 > c ? "sameElse" : -1 > c ? "lastWeek" : 0 > c ? "lastDay" : 1 > c ? "sameDay" : 2 > c ? "nextDay" : 7 > c ? "nextWeek" : "sameElse";
  }
  function Ub(b, c) {
    var d = b || rb(),
        e = Ab(d, this).startOf("day"),
        f = a.calendarFormat(this, e) || "sameElse",
        g = c && (y(c[f]) ? c[f].call(this, d) : c[f]);
    return this.format(g || this.localeData().calendar(f, this, rb(d)));
  }
  function Vb() {
    return new q(this);
  }
  function Wb(a, b) {
    var c = r(a) ? a : rb(a);
    return this.isValid() && c.isValid() ? (b = J(o(b) ? "millisecond" : b), "millisecond" === b ? this.valueOf() > c.valueOf() : c.valueOf() < this.clone().startOf(b).valueOf()) : !1;
  }
  function Xb(a, b) {
    var c = r(a) ? a : rb(a);
    return this.isValid() && c.isValid() ? (b = J(o(b) ? "millisecond" : b), "millisecond" === b ? this.valueOf() < c.valueOf() : this.clone().endOf(b).valueOf() < c.valueOf()) : !1;
  }
  function Yb(a, b, c, d) {
    return d = d || "()", ("(" === d[0] ? this.isAfter(a, c) : !this.isBefore(a, c)) && (")" === d[1] ? this.isBefore(b, c) : !this.isAfter(b, c));
  }
  function Zb(a, b) {
    var c,
        d = r(a) ? a : rb(a);
    return this.isValid() && d.isValid() ? (b = J(b || "millisecond"), "millisecond" === b ? this.valueOf() === d.valueOf() : (c = d.valueOf(), this.clone().startOf(b).valueOf() <= c && c <= this.clone().endOf(b).valueOf())) : !1;
  }
  function $b(a, b) {
    return this.isSame(a, b) || this.isAfter(a, b);
  }
  function _b(a, b) {
    return this.isSame(a, b) || this.isBefore(a, b);
  }
  function ac(a, b, c) {
    var d,
        e,
        f,
        g;
    return this.isValid() ? (d = Ab(a, this), d.isValid() ? (e = 6e4 * (d.utcOffset() - this.utcOffset()), b = J(b), "year" === b || "month" === b || "quarter" === b ? (g = bc(this, d), "quarter" === b ? g /= 3 : "year" === b && (g /= 12)) : (f = this - d, g = "second" === b ? f / 1e3 : "minute" === b ? f / 6e4 : "hour" === b ? f / 36e5 : "day" === b ? (f - e) / 864e5 : "week" === b ? (f - e) / 6048e5 : f), c ? g : s(g)) : NaN) : NaN;
  }
  function bc(a, b) {
    var c,
        d,
        e = 12 * (b.year() - a.year()) + (b.month() - a.month()),
        f = a.clone().add(e, "months");
    return 0 > b - f ? (c = a.clone().add(e - 1, "months"), d = (b - f) / (f - c)) : (c = a.clone().add(e + 1, "months"), d = (b - f) / (c - f)), -(e + d) || 0;
  }
  function cc() {
    return this.clone().locale("en").format("ddd MMM DD YYYY HH:mm:ss [GMT]ZZ");
  }
  function dc() {
    var a = this.clone().utc();
    return 0 < a.year() && a.year() <= 9999 ? y(Date.prototype.toISOString) ? this.toDate().toISOString() : W(a, "YYYY-MM-DD[T]HH:mm:ss.SSS[Z]") : W(a, "YYYYYY-MM-DD[T]HH:mm:ss.SSS[Z]");
  }
  function ec(b) {
    b || (b = this.isUtc() ? a.defaultFormatUtc : a.defaultFormat);
    var c = W(this, b);
    return this.localeData().postformat(c);
  }
  function fc(a, b) {
    return this.isValid() && (r(a) && a.isValid() || rb(a).isValid()) ? Nb({
      to: this,
      from: a
    }).locale(this.locale()).humanize(!b) : this.localeData().invalidDate();
  }
  function gc(a) {
    return this.from(rb(), a);
  }
  function hc(a, b) {
    return this.isValid() && (r(a) && a.isValid() || rb(a).isValid()) ? Nb({
      from: this,
      to: a
    }).locale(this.locale()).humanize(!b) : this.localeData().invalidDate();
  }
  function ic(a) {
    return this.to(rb(), a);
  }
  function jc(a) {
    var b;
    return void 0 === a ? this._locale._abbr : (b = ab(a), null != b && (this._locale = b), this);
  }
  function kc() {
    return this._locale;
  }
  function lc(a) {
    switch (a = J(a)) {
      case "year":
        this.month(0);
      case "quarter":
      case "month":
        this.date(1);
      case "week":
      case "isoWeek":
      case "day":
      case "date":
        this.hours(0);
      case "hour":
        this.minutes(0);
      case "minute":
        this.seconds(0);
      case "second":
        this.milliseconds(0);
    }
    return "week" === a && this.weekday(0), "isoWeek" === a && this.isoWeekday(1), "quarter" === a && this.month(3 * Math.floor(this.month() / 3)), this;
  }
  function mc(a) {
    return a = J(a), void 0 === a || "millisecond" === a ? this : ("date" === a && (a = "day"), this.startOf(a).add(1, "isoWeek" === a ? "week" : a).subtract(1, "ms"));
  }
  function nc() {
    return this._d.valueOf() - 6e4 * (this._offset || 0);
  }
  function oc() {
    return Math.floor(this.valueOf() / 1e3);
  }
  function pc() {
    return new Date(this.valueOf());
  }
  function qc() {
    var a = this;
    return [a.year(), a.month(), a.date(), a.hour(), a.minute(), a.second(), a.millisecond()];
  }
  function rc() {
    var a = this;
    return {
      years: a.year(),
      months: a.month(),
      date: a.date(),
      hours: a.hours(),
      minutes: a.minutes(),
      seconds: a.seconds(),
      milliseconds: a.milliseconds()
    };
  }
  function sc() {
    return this.isValid() ? this.toISOString() : null;
  }
  function tc() {
    return m(this);
  }
  function uc() {
    return i({}, l(this));
  }
  function vc() {
    return l(this).overflow;
  }
  function wc() {
    return {
      input: this._i,
      format: this._f,
      locale: this._locale,
      isUTC: this._isUTC,
      strict: this._strict
    };
  }
  function xc(a, b) {
    T(0, [a, a.length], 0, b);
  }
  function yc(a) {
    return Cc.call(this, a, this.week(), this.weekday(), this.localeData()._week.dow, this.localeData()._week.doy);
  }
  function zc(a) {
    return Cc.call(this, a, this.isoWeek(), this.isoWeekday(), 1, 4);
  }
  function Ac() {
    return wa(this.year(), 1, 4);
  }
  function Bc() {
    var a = this.localeData()._week;
    return wa(this.year(), a.dow, a.doy);
  }
  function Cc(a, b, c, d, e) {
    var f;
    return null == a ? va(this, d, e).year : (f = wa(a, d, e), b > f && (b = f), Dc.call(this, a, b, c, d, e));
  }
  function Dc(a, b, c, d, e) {
    var f = ua(a, b, c, d, e),
        g = sa(f.year, 0, f.dayOfYear);
    return this.year(g.getUTCFullYear()), this.month(g.getUTCMonth()), this.date(g.getUTCDate()), this;
  }
  function Ec(a) {
    return null == a ? Math.ceil((this.month() + 1) / 3) : this.month(3 * (a - 1) + this.month() % 3);
  }
  function Fc(a) {
    var b = Math.round((this.clone().startOf("day") - this.clone().startOf("year")) / 864e5) + 1;
    return null == a ? b : this.add(a - b, "d");
  }
  function Gc(a, b) {
    b[ce] = t(1e3 * ("0." + a));
  }
  function Hc() {
    return this._isUTC ? "UTC" : "";
  }
  function Ic() {
    return this._isUTC ? "Coordinated Universal Time" : "";
  }
  function Jc(a) {
    return rb(1e3 * a);
  }
  function Kc() {
    return rb.apply(null, arguments).parseZone();
  }
  function Lc(a) {
    return a;
  }
  function Mc(a, b, c, d) {
    var e = ab(),
        f = j().set(d, b);
    return e[c](f, a);
  }
  function Nc(a, b, c) {
    if ("number" == typeof a && (b = a, a = void 0), a = a || "", null != b)
      return Mc(a, b, c, "month");
    var d,
        e = [];
    for (d = 0; 12 > d; d++)
      e[d] = Mc(a, d, c, "month");
    return e;
  }
  function Oc(a, b, c, d) {
    "boolean" == typeof a ? ("number" == typeof b && (c = b, b = void 0), b = b || "") : (b = a, c = b, a = !1, "number" == typeof b && (c = b, b = void 0), b = b || "");
    var e = ab(),
        f = a ? e._week.dow : 0;
    if (null != c)
      return Mc(b, (c + f) % 7, d, "day");
    var g,
        h = [];
    for (g = 0; 7 > g; g++)
      h[g] = Mc(b, (g + f) % 7, d, "day");
    return h;
  }
  function Pc(a, b) {
    return Nc(a, b, "months");
  }
  function Qc(a, b) {
    return Nc(a, b, "monthsShort");
  }
  function Rc(a, b, c) {
    return Oc(a, b, c, "weekdays");
  }
  function Sc(a, b, c) {
    return Oc(a, b, c, "weekdaysShort");
  }
  function Tc(a, b, c) {
    return Oc(a, b, c, "weekdaysMin");
  }
  function Uc() {
    var a = this._data;
    return this._milliseconds = Ue(this._milliseconds), this._days = Ue(this._days), this._months = Ue(this._months), a.milliseconds = Ue(a.milliseconds), a.seconds = Ue(a.seconds), a.minutes = Ue(a.minutes), a.hours = Ue(a.hours), a.months = Ue(a.months), a.years = Ue(a.years), this;
  }
  function Vc(a, b, c, d) {
    var e = Nb(b, c);
    return a._milliseconds += d * e._milliseconds, a._days += d * e._days, a._months += d * e._months, a._bubble();
  }
  function Wc(a, b) {
    return Vc(this, a, b, 1);
  }
  function Xc(a, b) {
    return Vc(this, a, b, -1);
  }
  function Yc(a) {
    return 0 > a ? Math.floor(a) : Math.ceil(a);
  }
  function Zc() {
    var a,
        b,
        c,
        d,
        e,
        f = this._milliseconds,
        g = this._days,
        h = this._months,
        i = this._data;
    return f >= 0 && g >= 0 && h >= 0 || 0 >= f && 0 >= g && 0 >= h || (f += 864e5 * Yc(_c(h) + g), g = 0, h = 0), i.milliseconds = f % 1e3, a = s(f / 1e3), i.seconds = a % 60, b = s(a / 60), i.minutes = b % 60, c = s(b / 60), i.hours = c % 24, g += s(c / 24), e = s($c(g)), h += e, g -= Yc(_c(e)), d = s(h / 12), h %= 12, i.days = g, i.months = h, i.years = d, this;
  }
  function $c(a) {
    return 4800 * a / 146097;
  }
  function _c(a) {
    return 146097 * a / 4800;
  }
  function ad(a) {
    var b,
        c,
        d = this._milliseconds;
    if (a = J(a), "month" === a || "year" === a)
      return b = this._days + d / 864e5, c = this._months + $c(b), "month" === a ? c : c / 12;
    switch (b = this._days + Math.round(_c(this._months)), a) {
      case "week":
        return b / 7 + d / 6048e5;
      case "day":
        return b + d / 864e5;
      case "hour":
        return 24 * b + d / 36e5;
      case "minute":
        return 1440 * b + d / 6e4;
      case "second":
        return 86400 * b + d / 1e3;
      case "millisecond":
        return Math.floor(864e5 * b) + d;
      default:
        throw new Error("Unknown unit " + a);
    }
  }
  function bd() {
    return this._milliseconds + 864e5 * this._days + this._months % 12 * 2592e6 + 31536e6 * t(this._months / 12);
  }
  function cd(a) {
    return function() {
      return this.as(a);
    };
  }
  function dd(a) {
    return a = J(a), this[a + "s"]();
  }
  function ed(a) {
    return function() {
      return this._data[a];
    };
  }
  function fd() {
    return s(this.days() / 7);
  }
  function gd(a, b, c, d, e) {
    return e.relativeTime(b || 1, !!c, a, d);
  }
  function hd(a, b, c) {
    var d = Nb(a).abs(),
        e = jf(d.as("s")),
        f = jf(d.as("m")),
        g = jf(d.as("h")),
        h = jf(d.as("d")),
        i = jf(d.as("M")),
        j = jf(d.as("y")),
        k = e < kf.s && ["s", e] || 1 >= f && ["m"] || f < kf.m && ["mm", f] || 1 >= g && ["h"] || g < kf.h && ["hh", g] || 1 >= h && ["d"] || h < kf.d && ["dd", h] || 1 >= i && ["M"] || i < kf.M && ["MM", i] || 1 >= j && ["y"] || ["yy", j];
    return k[2] = b, k[3] = +a > 0, k[4] = c, gd.apply(null, k);
  }
  function id(a) {
    return void 0 === a ? jf : "function" == typeof a ? (jf = a, !0) : !1;
  }
  function jd(a, b) {
    return void 0 === kf[a] ? !1 : void 0 === b ? kf[a] : (kf[a] = b, !0);
  }
  function kd(a) {
    var b = this.localeData(),
        c = hd(this, !a, b);
    return a && (c = b.pastFuture(+this, c)), b.postformat(c);
  }
  function ld() {
    var a,
        b,
        c,
        d = lf(this._milliseconds) / 1e3,
        e = lf(this._days),
        f = lf(this._months);
    a = s(d / 60), b = s(a / 60), d %= 60, a %= 60, c = s(f / 12), f %= 12;
    var g = c,
        h = f,
        i = e,
        j = b,
        k = a,
        l = d,
        m = this.asSeconds();
    return m ? (0 > m ? "-" : "") + "P" + (g ? g + "Y" : "") + (h ? h + "M" : "") + (i ? i + "D" : "") + (j || k || l ? "T" : "") + (j ? j + "H" : "") + (k ? k + "M" : "") + (l ? l + "S" : "") : "P0D";
  }
  var md,
      nd;
  nd = Array.prototype.some ? Array.prototype.some : function(a) {
    for (var b = Object(this),
        c = b.length >>> 0,
        d = 0; c > d; d++)
      if (d in b && a.call(this, b[d], d, b))
        return !0;
    return !1;
  };
  var od = a.momentProperties = [],
      pd = !1,
      qd = {};
  a.suppressDeprecationWarnings = !1, a.deprecationHandler = null;
  var rd;
  rd = Object.keys ? Object.keys : function(a) {
    var b,
        c = [];
    for (b in a)
      h(a, b) && c.push(b);
    return c;
  };
  var sd,
      td = {
        sameDay: "[Today at] LT",
        nextDay: "[Tomorrow at] LT",
        nextWeek: "dddd [at] LT",
        lastDay: "[Yesterday at] LT",
        lastWeek: "[Last] dddd [at] LT",
        sameElse: "L"
      },
      ud = {
        LTS: "h:mm:ss A",
        LT: "h:mm A",
        L: "MM/DD/YYYY",
        LL: "MMMM D, YYYY",
        LLL: "MMMM D, YYYY h:mm A",
        LLLL: "dddd, MMMM D, YYYY h:mm A"
      },
      vd = "Invalid date",
      wd = "%d",
      xd = /\d{1,2}/,
      yd = {
        future: "in %s",
        past: "%s ago",
        s: "a few seconds",
        m: "a minute",
        mm: "%d minutes",
        h: "an hour",
        hh: "%d hours",
        d: "a day",
        dd: "%d days",
        M: "a month",
        MM: "%d months",
        y: "a year",
        yy: "%d years"
      },
      zd = {},
      Ad = {},
      Bd = /(\[[^\[]*\])|(\\)?([Hh]mm(ss)?|Mo|MM?M?M?|Do|DDDo|DD?D?D?|ddd?d?|do?|w[o|w]?|W[o|W]?|Qo?|YYYYYY|YYYYY|YYYY|YY|gg(ggg?)?|GG(GGG?)?|e|E|a|A|hh?|HH?|kk?|mm?|ss?|S{1,9}|x|X|zz?|ZZ?|.)/g,
      Cd = /(\[[^\[]*\])|(\\)?(LTS|LT|LL?L?L?|l{1,4})/g,
      Dd = {},
      Ed = {},
      Fd = /\d/,
      Gd = /\d\d/,
      Hd = /\d{3}/,
      Id = /\d{4}/,
      Jd = /[+-]?\d{6}/,
      Kd = /\d\d?/,
      Ld = /\d\d\d\d?/,
      Md = /\d\d\d\d\d\d?/,
      Nd = /\d{1,3}/,
      Od = /\d{1,4}/,
      Pd = /[+-]?\d{1,6}/,
      Qd = /\d+/,
      Rd = /[+-]?\d+/,
      Sd = /Z|[+-]\d\d:?\d\d/gi,
      Td = /Z|[+-]\d\d(?::?\d\d)?/gi,
      Ud = /[+-]?\d+(\.\d{1,3})?/,
      Vd = /[0-9]*['a-z\u00A0-\u05FF\u0700-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+|[\u0600-\u06FF\/]+(\s*?[\u0600-\u06FF]+){1,2}/i,
      Wd = {},
      Xd = {},
      Yd = 0,
      Zd = 1,
      $d = 2,
      _d = 3,
      ae = 4,
      be = 5,
      ce = 6,
      de = 7,
      ee = 8;
  sd = Array.prototype.indexOf ? Array.prototype.indexOf : function(a) {
    var b;
    for (b = 0; b < this.length; ++b)
      if (this[b] === a)
        return b;
    return -1;
  }, T("M", ["MM", 2], "Mo", function() {
    return this.month() + 1;
  }), T("MMM", 0, 0, function(a) {
    return this.localeData().monthsShort(this, a);
  }), T("MMMM", 0, 0, function(a) {
    return this.localeData().months(this, a);
  }), I("month", "M"), L("month", 8), Y("M", Kd), Y("MM", Kd, Gd), Y("MMM", function(a, b) {
    return b.monthsShortRegex(a);
  }), Y("MMMM", function(a, b) {
    return b.monthsRegex(a);
  }), aa(["M", "MM"], function(a, b) {
    b[Zd] = t(a) - 1;
  }), aa(["MMM", "MMMM"], function(a, b, c, d) {
    var e = c._locale.monthsParse(a, d, c._strict);
    null != e ? b[Zd] = e : l(c).invalidMonth = a;
  });
  var fe = /D[oD]?(\[[^\[\]]*\]|\s+)+MMMM?/,
      ge = "January_February_March_April_May_June_July_August_September_October_November_December".split("_"),
      he = "Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec".split("_"),
      ie = Vd,
      je = Vd;
  T("Y", 0, 0, function() {
    var a = this.year();
    return 9999 >= a ? "" + a : "+" + a;
  }), T(0, ["YY", 2], 0, function() {
    return this.year() % 100;
  }), T(0, ["YYYY", 4], 0, "year"), T(0, ["YYYYY", 5], 0, "year"), T(0, ["YYYYYY", 6, !0], 0, "year"), I("year", "y"), L("year", 1), Y("Y", Rd), Y("YY", Kd, Gd), Y("YYYY", Od, Id), Y("YYYYY", Pd, Jd), Y("YYYYYY", Pd, Jd), aa(["YYYYY", "YYYYYY"], Yd), aa("YYYY", function(b, c) {
    c[Yd] = 2 === b.length ? a.parseTwoDigitYear(b) : t(b);
  }), aa("YY", function(b, c) {
    c[Yd] = a.parseTwoDigitYear(b);
  }), aa("Y", function(a, b) {
    b[Yd] = parseInt(a, 10);
  }), a.parseTwoDigitYear = function(a) {
    return t(a) + (t(a) > 68 ? 1900 : 2e3);
  };
  var ke = N("FullYear", !0);
  T("w", ["ww", 2], "wo", "week"), T("W", ["WW", 2], "Wo", "isoWeek"), I("week", "w"), I("isoWeek", "W"), L("week", 5), L("isoWeek", 5), Y("w", Kd), Y("ww", Kd, Gd), Y("W", Kd), Y("WW", Kd, Gd), ba(["w", "ww", "W", "WW"], function(a, b, c, d) {
    b[d.substr(0, 1)] = t(a);
  });
  var le = {
    dow: 0,
    doy: 6
  };
  T("d", 0, "do", "day"), T("dd", 0, 0, function(a) {
    return this.localeData().weekdaysMin(this, a);
  }), T("ddd", 0, 0, function(a) {
    return this.localeData().weekdaysShort(this, a);
  }), T("dddd", 0, 0, function(a) {
    return this.localeData().weekdays(this, a);
  }), T("e", 0, 0, "weekday"), T("E", 0, 0, "isoWeekday"), I("day", "d"), I("weekday", "e"), I("isoWeekday", "E"), L("day", 11), L("weekday", 11), L("isoWeekday", 11), Y("d", Kd), Y("e", Kd), Y("E", Kd), Y("dd", function(a, b) {
    return b.weekdaysMinRegex(a);
  }), Y("ddd", function(a, b) {
    return b.weekdaysShortRegex(a);
  }), Y("dddd", function(a, b) {
    return b.weekdaysRegex(a);
  }), ba(["dd", "ddd", "dddd"], function(a, b, c, d) {
    var e = c._locale.weekdaysParse(a, d, c._strict);
    null != e ? b.d = e : l(c).invalidWeekday = a;
  }), ba(["d", "e", "E"], function(a, b, c, d) {
    b[d] = t(a);
  });
  var me = "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"),
      ne = "Sun_Mon_Tue_Wed_Thu_Fri_Sat".split("_"),
      oe = "Su_Mo_Tu_We_Th_Fr_Sa".split("_"),
      pe = Vd,
      qe = Vd,
      re = Vd;
  T("H", ["HH", 2], 0, "hour"), T("h", ["hh", 2], 0, Qa), T("k", ["kk", 2], 0, Ra), T("hmm", 0, 0, function() {
    return "" + Qa.apply(this) + S(this.minutes(), 2);
  }), T("hmmss", 0, 0, function() {
    return "" + Qa.apply(this) + S(this.minutes(), 2) + S(this.seconds(), 2);
  }), T("Hmm", 0, 0, function() {
    return "" + this.hours() + S(this.minutes(), 2);
  }), T("Hmmss", 0, 0, function() {
    return "" + this.hours() + S(this.minutes(), 2) + S(this.seconds(), 2);
  }), Sa("a", !0), Sa("A", !1), I("hour", "h"), L("hour", 13), Y("a", Ta), Y("A", Ta), Y("H", Kd), Y("h", Kd), Y("HH", Kd, Gd), Y("hh", Kd, Gd), Y("hmm", Ld), Y("hmmss", Md), Y("Hmm", Ld), Y("Hmmss", Md), aa(["H", "HH"], _d), aa(["a", "A"], function(a, b, c) {
    c._isPm = c._locale.isPM(a), c._meridiem = a;
  }), aa(["h", "hh"], function(a, b, c) {
    b[_d] = t(a), l(c).bigHour = !0;
  }), aa("hmm", function(a, b, c) {
    var d = a.length - 2;
    b[_d] = t(a.substr(0, d)), b[ae] = t(a.substr(d)), l(c).bigHour = !0;
  }), aa("hmmss", function(a, b, c) {
    var d = a.length - 4,
        e = a.length - 2;
    b[_d] = t(a.substr(0, d)), b[ae] = t(a.substr(d, 2)), b[be] = t(a.substr(e)), l(c).bigHour = !0;
  }), aa("Hmm", function(a, b, c) {
    var d = a.length - 2;
    b[_d] = t(a.substr(0, d)), b[ae] = t(a.substr(d));
  }), aa("Hmmss", function(a, b, c) {
    var d = a.length - 4,
        e = a.length - 2;
    b[_d] = t(a.substr(0, d)), b[ae] = t(a.substr(d, 2)), b[be] = t(a.substr(e));
  });
  var se,
      te = /[ap]\.?m?\.?/i,
      ue = N("Hours", !0),
      ve = {
        calendar: td,
        longDateFormat: ud,
        invalidDate: vd,
        ordinal: wd,
        ordinalParse: xd,
        relativeTime: yd,
        months: ge,
        monthsShort: he,
        week: le,
        weekdays: me,
        weekdaysMin: oe,
        weekdaysShort: ne,
        meridiemParse: te
      },
      we = {},
      xe = /^\s*((?:[+-]\d{6}|\d{4})-(?:\d\d-\d\d|W\d\d-\d|W\d\d|\d\d\d|\d\d))(?:(T| )(\d\d(?::\d\d(?::\d\d(?:[.,]\d+)?)?)?)([\+\-]\d\d(?::?\d\d)?|\s*Z)?)?/,
      ye = /^\s*((?:[+-]\d{6}|\d{4})(?:\d\d\d\d|W\d\d\d|W\d\d|\d\d\d|\d\d))(?:(T| )(\d\d(?:\d\d(?:\d\d(?:[.,]\d+)?)?)?)([\+\-]\d\d(?::?\d\d)?|\s*Z)?)?/,
      ze = /Z|[+-]\d\d(?::?\d\d)?/,
      Ae = [["YYYYYY-MM-DD", /[+-]\d{6}-\d\d-\d\d/], ["YYYY-MM-DD", /\d{4}-\d\d-\d\d/], ["GGGG-[W]WW-E", /\d{4}-W\d\d-\d/], ["GGGG-[W]WW", /\d{4}-W\d\d/, !1], ["YYYY-DDD", /\d{4}-\d{3}/], ["YYYY-MM", /\d{4}-\d\d/, !1], ["YYYYYYMMDD", /[+-]\d{10}/], ["YYYYMMDD", /\d{8}/], ["GGGG[W]WWE", /\d{4}W\d{3}/], ["GGGG[W]WW", /\d{4}W\d{2}/, !1], ["YYYYDDD", /\d{7}/]],
      Be = [["HH:mm:ss.SSSS", /\d\d:\d\d:\d\d\.\d+/], ["HH:mm:ss,SSSS", /\d\d:\d\d:\d\d,\d+/], ["HH:mm:ss", /\d\d:\d\d:\d\d/], ["HH:mm", /\d\d:\d\d/], ["HHmmss.SSSS", /\d\d\d\d\d\d\.\d+/], ["HHmmss,SSSS", /\d\d\d\d\d\d,\d+/], ["HHmmss", /\d\d\d\d\d\d/], ["HHmm", /\d\d\d\d/], ["HH", /\d\d/]],
      Ce = /^\/?Date\((\-?\d+)/i;
  a.createFromInputFallback = w("value provided is not in a recognized ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.", function(a) {
    a._d = new Date(a._i + (a._useUTC ? " UTC" : ""));
  }), a.ISO_8601 = function() {};
  var De = w("moment().min is deprecated, use moment.max instead. http://momentjs.com/guides/#/warnings/min-max/", function() {
    var a = rb.apply(null, arguments);
    return this.isValid() && a.isValid() ? this > a ? this : a : n();
  }),
      Ee = w("moment().max is deprecated, use moment.min instead. http://momentjs.com/guides/#/warnings/min-max/", function() {
        var a = rb.apply(null, arguments);
        return this.isValid() && a.isValid() ? a > this ? this : a : n();
      }),
      Fe = function() {
        return Date.now ? Date.now() : +new Date;
      };
  yb("Z", ":"), yb("ZZ", ""), Y("Z", Td), Y("ZZ", Td), aa(["Z", "ZZ"], function(a, b, c) {
    c._useUTC = !0, c._tzm = zb(Td, a);
  });
  var Ge = /([\+\-]|\d\d)/gi;
  a.updateOffset = function() {};
  var He = /^(\-)?(?:(\d*)[. ])?(\d+)\:(\d+)(?:\:(\d+)(\.\d*)?)?$/,
      Ie = /^(-)?P(?:(-?[0-9,.]*)Y)?(?:(-?[0-9,.]*)M)?(?:(-?[0-9,.]*)W)?(?:(-?[0-9,.]*)D)?(?:T(?:(-?[0-9,.]*)H)?(?:(-?[0-9,.]*)M)?(?:(-?[0-9,.]*)S)?)?$/;
  Nb.fn = vb.prototype;
  var Je = Rb(1, "add"),
      Ke = Rb(-1, "subtract");
  a.defaultFormat = "YYYY-MM-DDTHH:mm:ssZ", a.defaultFormatUtc = "YYYY-MM-DDTHH:mm:ss[Z]";
  var Le = w("moment().lang() is deprecated. Instead, use moment().localeData() to get the language configuration. Use moment().locale() to change languages.", function(a) {
    return void 0 === a ? this.localeData() : this.locale(a);
  });
  T(0, ["gg", 2], 0, function() {
    return this.weekYear() % 100;
  }), T(0, ["GG", 2], 0, function() {
    return this.isoWeekYear() % 100;
  }), xc("gggg", "weekYear"), xc("ggggg", "weekYear"), xc("GGGG", "isoWeekYear"), xc("GGGGG", "isoWeekYear"), I("weekYear", "gg"), I("isoWeekYear", "GG"), L("weekYear", 1), L("isoWeekYear", 1), Y("G", Rd), Y("g", Rd), Y("GG", Kd, Gd), Y("gg", Kd, Gd), Y("GGGG", Od, Id), Y("gggg", Od, Id), Y("GGGGG", Pd, Jd), Y("ggggg", Pd, Jd), ba(["gggg", "ggggg", "GGGG", "GGGGG"], function(a, b, c, d) {
    b[d.substr(0, 2)] = t(a);
  }), ba(["gg", "GG"], function(b, c, d, e) {
    c[e] = a.parseTwoDigitYear(b);
  }), T("Q", 0, "Qo", "quarter"), I("quarter", "Q"), L("quarter", 7), Y("Q", Fd), aa("Q", function(a, b) {
    b[Zd] = 3 * (t(a) - 1);
  }), T("D", ["DD", 2], "Do", "date"), I("date", "D"), L("date", 9), Y("D", Kd), Y("DD", Kd, Gd), Y("Do", function(a, b) {
    return a ? b._ordinalParse : b._ordinalParseLenient;
  }), aa(["D", "DD"], $d), aa("Do", function(a, b) {
    b[$d] = t(a.match(Kd)[0], 10);
  });
  var Me = N("Date", !0);
  T("DDD", ["DDDD", 3], "DDDo", "dayOfYear"), I("dayOfYear", "DDD"), L("dayOfYear", 4), Y("DDD", Nd), Y("DDDD", Hd), aa(["DDD", "DDDD"], function(a, b, c) {
    c._dayOfYear = t(a);
  }), T("m", ["mm", 2], 0, "minute"), I("minute", "m"), L("minute", 14), Y("m", Kd), Y("mm", Kd, Gd), aa(["m", "mm"], ae);
  var Ne = N("Minutes", !1);
  T("s", ["ss", 2], 0, "second"), I("second", "s"), L("second", 15), Y("s", Kd), Y("ss", Kd, Gd), aa(["s", "ss"], be);
  var Oe = N("Seconds", !1);
  T("S", 0, 0, function() {
    return ~~(this.millisecond() / 100);
  }), T(0, ["SS", 2], 0, function() {
    return ~~(this.millisecond() / 10);
  }), T(0, ["SSS", 3], 0, "millisecond"), T(0, ["SSSS", 4], 0, function() {
    return 10 * this.millisecond();
  }), T(0, ["SSSSS", 5], 0, function() {
    return 100 * this.millisecond();
  }), T(0, ["SSSSSS", 6], 0, function() {
    return 1e3 * this.millisecond();
  }), T(0, ["SSSSSSS", 7], 0, function() {
    return 1e4 * this.millisecond();
  }), T(0, ["SSSSSSSS", 8], 0, function() {
    return 1e5 * this.millisecond();
  }), T(0, ["SSSSSSSSS", 9], 0, function() {
    return 1e6 * this.millisecond();
  }), I("millisecond", "ms"), L("millisecond", 16), Y("S", Nd, Fd), Y("SS", Nd, Gd), Y("SSS", Nd, Hd);
  var Pe;
  for (Pe = "SSSS"; Pe.length <= 9; Pe += "S")
    Y(Pe, Qd);
  for (Pe = "S"; Pe.length <= 9; Pe += "S")
    aa(Pe, Gc);
  var Qe = N("Milliseconds", !1);
  T("z", 0, 0, "zoneAbbr"), T("zz", 0, 0, "zoneName");
  var Re = q.prototype;
  Re.add = Je, Re.calendar = Ub, Re.clone = Vb, Re.diff = ac, Re.endOf = mc, Re.format = ec, Re.from = fc, Re.fromNow = gc, Re.to = hc, Re.toNow = ic, Re.get = Q, Re.invalidAt = vc, Re.isAfter = Wb, Re.isBefore = Xb, Re.isBetween = Yb, Re.isSame = Zb, Re.isSameOrAfter = $b, Re.isSameOrBefore = _b, Re.isValid = tc, Re.lang = Le, Re.locale = jc, Re.localeData = kc, Re.max = Ee, Re.min = De, Re.parsingFlags = uc, Re.set = R, Re.startOf = lc, Re.subtract = Ke, Re.toArray = qc, Re.toObject = rc, Re.toDate = pc, Re.toISOString = dc, Re.toJSON = sc, Re.toString = cc, Re.unix = oc, Re.valueOf = nc, Re.creationData = wc, Re.year = ke, Re.isLeapYear = qa, Re.weekYear = yc, Re.isoWeekYear = zc, Re.quarter = Re.quarters = Ec, Re.month = ja, Re.daysInMonth = ka, Re.week = Re.weeks = Aa, Re.isoWeek = Re.isoWeeks = Ba, Re.weeksInYear = Bc, Re.isoWeeksInYear = Ac, Re.date = Me, Re.day = Re.days = Ja, Re.weekday = Ka, Re.isoWeekday = La, Re.dayOfYear = Fc, Re.hour = Re.hours = ue, Re.minute = Re.minutes = Ne, Re.second = Re.seconds = Oe, Re.millisecond = Re.milliseconds = Qe, Re.utcOffset = Cb, Re.utc = Eb, Re.local = Fb, Re.parseZone = Gb, Re.hasAlignedHourOffset = Hb, Re.isDST = Ib, Re.isLocal = Kb, Re.isUtcOffset = Lb, Re.isUtc = Mb, Re.isUTC = Mb, Re.zoneAbbr = Hc, Re.zoneName = Ic, Re.dates = w("dates accessor is deprecated. Use date instead.", Me), Re.months = w("months accessor is deprecated. Use month instead", ja), Re.years = w("years accessor is deprecated. Use year instead", ke), Re.zone = w("moment().zone is deprecated, use moment().utcOffset instead. http://momentjs.com/guides/#/warnings/zone/", Db), Re.isDSTShifted = w("isDSTShifted is deprecated. See http://momentjs.com/guides/#/warnings/dst-shifted/ for more information", Jb);
  var Se = Re,
      Te = B.prototype;
  Te.calendar = C, Te.longDateFormat = D, Te.invalidDate = E, Te.ordinal = F, Te.preparse = Lc, Te.postformat = Lc, Te.relativeTime = G, Te.pastFuture = H, Te.set = z, Te.months = ea, Te.monthsShort = fa, Te.monthsParse = ha, Te.monthsRegex = ma, Te.monthsShortRegex = la, Te.week = xa, Te.firstDayOfYear = za, Te.firstDayOfWeek = ya, Te.weekdays = Ea, Te.weekdaysMin = Ga, Te.weekdaysShort = Fa, Te.weekdaysParse = Ia, Te.weekdaysRegex = Ma, Te.weekdaysShortRegex = Na, Te.weekdaysMinRegex = Oa, Te.isPM = Ua, Te.meridiem = Va, Za("en", {
    ordinalParse: /\d{1,2}(th|st|nd|rd)/,
    ordinal: function(a) {
      var b = a % 10,
          c = 1 === t(a % 100 / 10) ? "th" : 1 === b ? "st" : 2 === b ? "nd" : 3 === b ? "rd" : "th";
      return a + c;
    }
  }), a.lang = w("moment.lang is deprecated. Use moment.locale instead.", Za), a.langData = w("moment.langData is deprecated. Use moment.localeData instead.", ab);
  var Ue = Math.abs,
      Ve = cd("ms"),
      We = cd("s"),
      Xe = cd("m"),
      Ye = cd("h"),
      Ze = cd("d"),
      $e = cd("w"),
      _e = cd("M"),
      af = cd("y"),
      bf = ed("milliseconds"),
      cf = ed("seconds"),
      df = ed("minutes"),
      ef = ed("hours"),
      ff = ed("days"),
      gf = ed("months"),
      hf = ed("years"),
      jf = Math.round,
      kf = {
        s: 45,
        m: 45,
        h: 22,
        d: 26,
        M: 11
      },
      lf = Math.abs,
      mf = vb.prototype;
  mf.abs = Uc, mf.add = Wc, mf.subtract = Xc, mf.as = ad, mf.asMilliseconds = Ve, mf.asSeconds = We, mf.asMinutes = Xe, mf.asHours = Ye, mf.asDays = Ze, mf.asWeeks = $e, mf.asMonths = _e, mf.asYears = af, mf.valueOf = bd, mf._bubble = Zc, mf.get = dd, mf.milliseconds = bf, mf.seconds = cf, mf.minutes = df, mf.hours = ef, mf.days = ff, mf.weeks = fd, mf.months = gf, mf.years = hf, mf.humanize = kd, mf.toISOString = ld, mf.toString = ld, mf.toJSON = ld, mf.locale = jc, mf.localeData = kc, mf.toIsoString = w("toIsoString() is deprecated. Please use toISOString() instead (notice the capitals)", ld), mf.lang = Le, T("X", 0, 0, "unix"), T("x", 0, 0, "valueOf"), Y("x", Rd), Y("X", Ud), aa("X", function(a, b, c) {
    c._d = new Date(1e3 * parseFloat(a, 10));
  }), aa("x", function(a, b, c) {
    c._d = new Date(t(a));
  }), a.version = "2.15.1", b(rb), a.fn = Se, a.min = tb, a.max = ub, a.now = Fe, a.utc = j, a.unix = Jc, a.months = Pc, a.isDate = f, a.locale = Za, a.invalid = n, a.duration = Nb, a.isMoment = r, a.weekdays = Rc, a.parseZone = Kc, a.localeData = ab, a.isDuration = wb, a.monthsShort = Qc, a.weekdaysMin = Tc, a.defineLocale = $a, a.updateLocale = _a, a.locales = bb, a.weekdaysShort = Sc, a.normalizeUnits = J, a.relativeTimeRounding = id, a.relativeTimeThreshold = jd, a.calendarFormat = Tb, a.prototype = Se;
  var nf = a;
  return nf;
});

})();
$__System.registerDynamic("14", ["3", "8", "10", "9"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __extends = this && this.__extends || function (d, b) {
        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var forms_1 = $__require("8");
    var widget_component_1 = $__require("10");
    var moment = $__require("9");
    function dateCheck(c) {
        var startDate = moment(c.get('startDate').value);
        var endDate = moment(c.get('endDate').value);
        var result = startDate.isAfter(endDate);
        return result ? { 'greaterThan': true } : null;
    }
    var DateRangeWidget = DateRangeWidget_1 = function (_super) {
        __extends(DateRangeWidget, _super);
        function DateRangeWidget(elementRef, formBuilder) {
            var _this = _super.call(this) || this;
            _this.elementRef = elementRef;
            _this.formBuilder = formBuilder;
            _this.dateRangeChanged = new core_1.EventEmitter();
            _this.debug = false;
            return _this;
        }
        DateRangeWidget.prototype.ngOnInit = function () {
            this.initForm();
            this.addAccessibilityLabelToDatePicker();
        };
        DateRangeWidget.prototype.initForm = function () {
            var _this = this;
            var startDateForm = this.field.value ? this.field.value.startDate : '';
            var startDate = this.convertToMomentDate(startDateForm);
            var endDateForm = this.field.value ? this.field.value.endDate : '';
            var endDate = this.convertToMomentDate(endDateForm);
            var startDateControl = new forms_1.FormControl(startDate);
            startDateControl.setValidators(forms_1.Validators.required);
            this.dateRange.addControl('startDate', startDateControl);
            var endDateControl = new forms_1.FormControl(endDate);
            endDateControl.setValidators(forms_1.Validators.required);
            this.dateRange.addControl('endDate', endDateControl);
            this.dateRange.setValidators(dateCheck);
            this.dateRange.valueChanges.subscribe(function (data) {
                return _this.onGroupValueChanged(data);
            });
            this.initSartDateDialog(startDate);
            this.initEndDateDialog(endDate);
        };
        DateRangeWidget.prototype.initSartDateDialog = function (date) {
            var _this = this;
            var settings = {
                type: 'date',
                past: moment().subtract(100, 'years'),
                future: moment().add(100, 'years')
            };
            settings.init = moment(date, DateRangeWidget_1.FORMAT_DATE_ACTIVITI);
            this.dialogStart = new mdDateTimePicker.default(settings);
            this.dialogStart.trigger = this.startElement.nativeElement;
            var startDateButton = document.getElementById('startDateButton');
            startDateButton.addEventListener('click', function () {
                _this.dialogStart.toggle();
            });
        };
        DateRangeWidget.prototype.addAccessibilityLabelToDatePicker = function () {
            var left = document.querySelector('#mddtp-date__left');
            if (left) {
                left.appendChild(this.createCustomElement('date left'));
            }
            var right = document.querySelector('#mddtp-date__right');
            if (right) {
                right.appendChild(this.createCustomElement('date right'));
            }
            var cancel = document.querySelector('#mddtp-date__cancel');
            if (cancel) {
                cancel.appendChild(this.createCustomElement('date cancel'));
            }
            var ok = document.querySelector('#mddtp-date__ok');
            if (ok) {
                ok.appendChild(this.createCustomElement('date ok'));
            }
        };
        DateRangeWidget.prototype.createCustomElement = function (text) {
            var span = document.createElement('span');
            span.style.visibility = 'hidden';
            var rightSpanText = document.createTextNode(text);
            span.appendChild(rightSpanText);
            return span;
        };
        DateRangeWidget.prototype.initEndDateDialog = function (date) {
            var _this = this;
            var settings = {
                type: 'date',
                past: moment().subtract(100, 'years'),
                future: moment().add(100, 'years')
            };
            settings.init = moment(date, DateRangeWidget_1.FORMAT_DATE_ACTIVITI);
            this.dialogEnd = new mdDateTimePicker.default(settings);
            this.dialogEnd.trigger = this.endElement.nativeElement;
            var endDateButton = document.getElementById('endDateButton');
            endDateButton.addEventListener('click', function () {
                _this.dialogEnd.toggle();
            });
        };
        DateRangeWidget.prototype.onOkStart = function (inputEl) {
            var date = this.dialogStart.time.format(DateRangeWidget_1.FORMAT_DATE_ACTIVITI);
            this.dateRange.patchValue({
                startDate: date
            });
            var materialElemen = inputEl.parentElement;
            if (materialElemen) {
                materialElemen.MaterialTextfield.change(date);
            }
        };
        DateRangeWidget.prototype.onOkEnd = function (inputEl) {
            var date = this.dialogEnd.time.format(DateRangeWidget_1.FORMAT_DATE_ACTIVITI);
            this.dateRange.patchValue({
                endDate: date
            });
            var materialElemen = inputEl.parentElement;
            if (materialElemen) {
                materialElemen.MaterialTextfield.change(date);
            }
        };
        DateRangeWidget.prototype.onGroupValueChanged = function (data) {
            if (this.dateRange.valid) {
                var dateStart = this.convertToMomentDateWithTime(this.dateRange.controls['startDate'].value);
                var endStart = this.convertToMomentDateWithTime(this.dateRange.controls['endDate'].value);
                this.dateRangeChanged.emit({ startDate: dateStart, endDate: endStart });
            }
        };
        DateRangeWidget.prototype.convertToMomentDateWithTime = function (date) {
            return moment(date, DateRangeWidget_1.FORMAT_DATE_ACTIVITI, true).format(DateRangeWidget_1.FORMAT_DATE_ACTIVITI) + 'T00:00:00.000Z';
        };
        DateRangeWidget.prototype.convertToMomentDate = function (date) {
            if (date) {
                return moment(date).format(DateRangeWidget_1.FORMAT_DATE_ACTIVITI);
            } else {
                return moment().format(DateRangeWidget_1.FORMAT_DATE_ACTIVITI);
            }
        };
        DateRangeWidget.prototype.ngOnDestroy = function () {};
        return DateRangeWidget;
    }(widget_component_1.WidgetComponent);
    DateRangeWidget.FORMAT_DATE_ACTIVITI = 'YYYY-MM-DD';
    __decorate([core_1.ViewChild('startElement'), __metadata("design:type", Object)], DateRangeWidget.prototype, "startElement", void 0);
    __decorate([core_1.ViewChild('endElement'), __metadata("design:type", Object)], DateRangeWidget.prototype, "endElement", void 0);
    __decorate([core_1.Input('group'), __metadata("design:type", forms_1.FormGroup)], DateRangeWidget.prototype, "dateRange", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Object)], DateRangeWidget.prototype, "field", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DateRangeWidget.prototype, "dateRangeChanged", void 0);
    DateRangeWidget = DateRangeWidget_1 = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'date-range-widget',
        template: "<label>{{field.nameKey | translate}}</label><br> <div [formGroup]=\"dateRange\">     <small *ngIf=\"dateRange && dateRange.errors && dateRange.errors.greaterThan\" [hidden]=\"!dateRange.errors\" class=\"text-danger\">         Start date must be less than End date     </small>     <div class=\"mdl-grid\">         <div class=\"mdl-cell mdl-cell--4-col\">             <div class=\"mdl-textfield mdl-js-textfield mdl-textfield--floating-label\">                 <input class=\"mdl-textfield__input\"                        formControlName=\"startDate\"                        type=\"text\"                        (onOk)=\"onOkStart(startElement)\"                        id=\"startDateInput\" #startElement>                 <label class=\"mdl-textfield__label\" for=\"startDateInput\">Start Date</label>                 <small [hidden]=\"dateRange && dateRange.controls.startDate && dateRange.controls.startDate.valid\" class=\"text-danger\">                     Start is required                 </small>             </div>         </div>         <div class=\"mdl-cell mdl-cell--2-col\">             <button id=\"startDateButton\" class=\"mdl-button mdl-js-button mdl-button--fab date-picker-mdl\">                 <i class=\"material-icons\">date_range</i>             </button>         </div>         <div class=\"mdl-cell mdl-cell--4-col\">             <div class=\"mdl-textfield mdl-js-textfield mdl-textfield--floating-label\">                 <input class=\"mdl-textfield__input\"                        formControlName=\"endDate\"                        type=\"text\"                        (onOk)=\"onOkEnd(endElement)\"                        id=\"endDateInput\" #endElement>                 <label class=\"mdl-textfield__label\" for=\"endDateInput\">End Date</label>             </div>         </div>         <div class=\"mdl-cell mdl-cell--2-col\">             <button id=\"endDateButton\" class=\"mdl-button mdl-js-button mdl-button--fab date-picker-mdl\">                 <i class=\"material-icons\">date_range</i>             </button>         </div>     </div> </div> <div *ngIf=\"debug\">     <p>FormGroup : {{dateRange && dateRange.value | json }}</p>     <p>FormGroup valid : {{dateRange && dateRange.valid }}</p>     <p>FormGroup status : {{dateRange && dateRange.errors | json }}</p>     <p>FormGroup start status : {{dateRange && dateRange.controls.startDate && dateRange.controls.startDate.errors | json }}</p>     <p>FormGroup end status: {{dateRange && dateRange.controls.endDate.errors | json }}</p> </div>",
        styles: [".date-picker-mdl {     margin-left: 20px; }"]
    }), __metadata("design:paramtypes", [core_1.ElementRef, forms_1.FormBuilder])], DateRangeWidget);
    exports.DateRangeWidget = DateRangeWidget;
    var DateRangeWidget_1;
});
$__System.registerDynamic("15", ["f", "11", "12", "13", "14"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var dropdown_widget_1 = $__require("f");
    var number_widget_1 = $__require("11");
    var duration_widget_1 = $__require("12");
    var checkbox_widget_1 = $__require("13");
    var date_range_widget_1 = $__require("14");
    __export($__require("f"));
    __export($__require("11"));
    __export($__require("12"));
    __export($__require("13"));
    __export($__require("14"));
    exports.WIDGET_DIRECTIVES = [dropdown_widget_1.DropdownWidget, number_widget_1.NumberWidget, duration_widget_1.DurationWidget, checkbox_widget_1.CheckboxWidget, date_range_widget_1.DateRangeWidget];
});
$__System.registerDynamic("1", ["3", "5", "16", "2", "7", "b", "c", "d", "6", "17", "15"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("5");
    var ng2_activiti_diagrams_1 = $__require("16");
    var analytics_report_list_component_1 = $__require("2");
    var analytics_report_parameters_component_1 = $__require("7");
    var analytics_component_1 = $__require("b");
    var analytics_generator_component_1 = $__require("c");
    var analytics_report_heat_map_component_1 = $__require("d");
    var analytics_service_1 = $__require("6");
    var ng2_charts_1 = $__require("17");
    var index_1 = $__require("15");
    __export($__require("b"));
    __export($__require("c"));
    __export($__require("2"));
    __export($__require("7"));
    __export($__require("6"));
    __export($__require("15"));
    exports.ANALYTICS_DIRECTIVES = [analytics_component_1.AnalyticsComponent, analytics_report_list_component_1.AnalyticsReportListComponent, analytics_report_parameters_component_1.AnalyticsReportParametersComponent, analytics_generator_component_1.AnalyticsGeneratorComponent, analytics_report_heat_map_component_1.AnalyticsReportHeatMapComponent, index_1.WIDGET_DIRECTIVES];
    exports.ANALYTICS_PROVIDERS = [analytics_service_1.AnalyticsService];
    var AnalyticsModule = AnalyticsModule_1 = function () {
        function AnalyticsModule() {}
        AnalyticsModule.forRoot = function () {
            return {
                ngModule: AnalyticsModule_1,
                providers: exports.ANALYTICS_PROVIDERS.slice()
            };
        };
        return AnalyticsModule;
    }();
    AnalyticsModule = AnalyticsModule_1 = __decorate([core_1.NgModule({
        imports: [ng2_alfresco_core_1.CoreModule, ng2_activiti_diagrams_1.DiagramsModule],
        declarations: exports.ANALYTICS_DIRECTIVES.concat(ng2_charts_1.CHART_DIRECTIVES),
        providers: exports.ANALYTICS_PROVIDERS.slice(),
        exports: exports.ANALYTICS_DIRECTIVES.slice()
    })], AnalyticsModule);
    exports.AnalyticsModule = AnalyticsModule;
    var AnalyticsModule_1;
    
});
})
(function(factory) {
  if (typeof define == 'function' && define.amd)
    define(["@angular/core","@angular/forms","ng2-activiti-diagrams","ng2-alfresco-core","ng2-charts/ng2-charts","rxjs/Rx"], factory);
  else if (typeof module == 'object' && module.exports && typeof require == 'function')
    module.exports = factory(require("@angular/core"), require("@angular/forms"), require("ng2-activiti-diagrams"), require("ng2-alfresco-core"), require("ng2-charts/ng2-charts"), require("rxjs/Rx"));
  else
    throw new Error("Module must be loaded as AMD or CommonJS");
});
