// ng2-activiti-tasklist/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var ng2_alfresco_datatable_1 = require("ng2-alfresco-datatable");
var ng2_activiti_form_1 = require("ng2-activiti-form");
var activiti_people_service_1 = require("./src/services/activiti-people.service");
var activiti_tasklist_service_1 = require("./src/services/activiti-tasklist.service");
var index_1 = require("./src/components/index");
__export(require("./src/components/index"));
__export(require("./src/services/activiti-tasklist.service"));
__export(require("./src/models/index"));
exports.ACTIVITI_TASKLIST_DIRECTIVES = [
    index_1.NoTaskDetailsTemplateComponent,
    index_1.ActivitiApps,
    index_1.ActivitiFilters,
    index_1.ActivitiTaskList,
    index_1.ActivitiTaskDetails,
    index_1.ActivitiChecklist,
    index_1.ActivitiComments,
    index_1.ActivitiPeople,
    index_1.ActivitiTaskHeader,
    index_1.ActivitiStartTaskButton,
    index_1.ActivitiPeopleSearch
];
exports.ACTIVITI_TASKLIST_PROVIDERS = [
    activiti_tasklist_service_1.ActivitiTaskListService,
    activiti_people_service_1.ActivitiPeopleService
];
var ActivitiTaskListModule = ActivitiTaskListModule_1 = (function () {
    function ActivitiTaskListModule() {
    }
    ActivitiTaskListModule.forRoot = function () {
        return {
            ngModule: ActivitiTaskListModule_1,
            providers: exports.ACTIVITI_TASKLIST_PROVIDERS.slice()
        };
    };
    return ActivitiTaskListModule;
}());
ActivitiTaskListModule = ActivitiTaskListModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng2_alfresco_core_1.CoreModule,
            ng2_alfresco_datatable_1.DataTableModule,
            ng2_activiti_form_1.ActivitiFormModule
        ],
        declarations: exports.ACTIVITI_TASKLIST_DIRECTIVES.slice(),
        providers: exports.ACTIVITI_TASKLIST_PROVIDERS.slice(),
        exports: exports.ACTIVITI_TASKLIST_DIRECTIVES.slice()
    })
], ActivitiTaskListModule);
exports.ActivitiTaskListModule = ActivitiTaskListModule;
var ActivitiTaskListModule_1;
