// ng2-activiti-analytics/src/services/analytics.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var Rx_1 = require("rxjs/Rx");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var report_model_1 = require("../models/report.model");
var chart_model_1 = require("../models/chart.model");
var AnalyticsService = (function () {
    function AnalyticsService(apiService, logService) {
        this.apiService = apiService;
        this.logService = logService;
    }
    AnalyticsService.prototype.getReportList = function () {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getReportList())
            .map(function (res) {
            var reports = [];
            res.forEach(function (report) {
                var reportModel = new report_model_1.ReportParametersModel(report);
                reports.push(reportModel);
            });
            return reports;
        }).catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.getReportParams = function (reportId) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getReportParams(reportId))
            .map(function (res) {
            return new report_model_1.ReportParametersModel(res);
        }).catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.getParamValuesByType = function (type, appId, reportId, processDefinitionId) {
        if (type === 'status') {
            return this.getProcessStatusValues();
        }
        else if (type === 'processDefinition') {
            if (appId === null || appId === undefined) {
                return this.getProcessDefinitionsValuesNoApp();
            }
            else {
                return this.getProcessDefinitionsValues(appId);
            }
        }
        else if (type === 'dateInterval') {
            return this.getDateIntervalValues();
        }
        else if (type === 'task' && reportId && processDefinitionId) {
            return this.getTasksByProcessDefinitionId(reportId, processDefinitionId);
        }
        else {
            return Rx_1.Observable.create(function (observer) {
                observer.next(null);
                observer.complete();
            });
        }
    };
    AnalyticsService.prototype.getProcessStatusValues = function () {
        var paramOptions = [];
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'All', name: 'All' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'Active', name: 'Active' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'Complete', name: 'Complete' }));
        return Rx_1.Observable.create(function (observer) {
            observer.next(paramOptions);
            observer.complete();
        });
    };
    AnalyticsService.prototype.getDateIntervalValues = function () {
        var paramOptions = [];
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byHour', name: 'By hour' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byDay', name: 'By day' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byWeek', name: 'By week' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byMonth', name: 'By month' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'byYear', name: 'By year' }));
        return Rx_1.Observable.create(function (observer) {
            observer.next(paramOptions);
            observer.complete();
        });
    };
    AnalyticsService.prototype.getMetricValues = function () {
        var paramOptions = [];
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'totalCount', name: 'Number of times a step is executed' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'totalTime', name: 'Total time spent in a process step' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: 'avgTime', name: 'Average time spent in a process step' }));
        return Rx_1.Observable.create(function (observer) {
            observer.next(paramOptions);
            observer.complete();
        });
    };
    AnalyticsService.prototype.getProcessDefinitionsValuesNoApp = function () {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getProcessDefinitions())
            .map(function (res) {
            var paramOptions = [];
            res.forEach(function (opt) {
                paramOptions.push(new report_model_1.ParameterValueModel(opt));
            });
            return paramOptions;
        }).catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.getProcessDefinitionsValues = function (appId) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.processDefinitionsApi.getProcessDefinitions(appId))
            .map(function (res) {
            var paramOptions = [];
            res.data.forEach(function (opt) {
                paramOptions.push(new report_model_1.ParameterValueModel(opt));
            });
            return paramOptions;
        }).catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.getTasksByProcessDefinitionId = function (reportId, processDefinitionId) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getTasksByProcessDefinitionId(reportId, processDefinitionId))
            .map(function (res) {
            var paramOptions = [];
            res.forEach(function (opt) {
                paramOptions.push(new report_model_1.ParameterValueModel({ id: opt, name: opt }));
            });
            return paramOptions;
        }).catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.getReportsByParams = function (reportId, paramsQuery) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getReportsByParams(reportId, paramsQuery))
            .map(function (res) {
            var elements = [];
            res.elements.forEach(function (chartData) {
                if (chartData.type === 'pieChart') {
                    elements.push(new chart_model_1.PieChart(chartData));
                }
                else if (chartData.type === 'table') {
                    elements.push(new chart_model_1.TableChart(chartData));
                }
                else if (chartData.type === 'processDefinitionHeatMap') {
                    elements.push(new chart_model_1.HeatMapChart(chartData));
                }
                else if (chartData.type === 'masterDetailTable') {
                    elements.push(new chart_model_1.DetailsTableChart(chartData));
                }
                else if (chartData.type === 'barChart') {
                    elements.push(new chart_model_1.BarChart(chartData));
                }
                else if (chartData.type === 'multiBarChart') {
                    elements.push(new chart_model_1.MultiBarChart(chartData));
                }
            });
            return elements;
        }).catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.createDefaultReports = function () {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.createDefaultReports())
            .map(this.toJson)
            .catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.updateReport = function (reportId, name) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.updateReport(reportId, name))
            .map(function (res) {
            _this.logService.info('upload');
        }).catch(function (err) { return _this.handleError(err); });
    };
    AnalyticsService.prototype.handleError = function (error) {
        this.logService.error(error);
        return Rx_1.Observable.throw(error.json().error || 'Server error');
    };
    AnalyticsService.prototype.toJson = function (res) {
        return res || {};
    };
    return AnalyticsService;
}());
AnalyticsService = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoApiService,
        ng2_alfresco_core_1.LogService])
], AnalyticsService);
exports.AnalyticsService = AnalyticsService;
