// ng2-activiti-processlist/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var ng2_alfresco_datatable_1 = require("ng2-alfresco-datatable");
var ng2_activiti_form_1 = require("ng2-activiti-form");
var ng2_activiti_tasklist_1 = require("ng2-activiti-tasklist");
var index_1 = require("./src/components/index");
var activiti_process_service_1 = require("./src/services/activiti-process.service");
__export(require("./src/components/activiti-processlist.component"));
__export(require("./src/components/activiti-filters.component"));
__export(require("./src/components/activiti-process-instance-details.component"));
__export(require("./src/components/activiti-start-process.component"));
__export(require("./src/models/index"));
__export(require("./src/services/activiti-process.service"));
exports.ACTIVITI_PROCESSLIST_DIRECTIVES = [
    index_1.ActivitiProcessInstanceListComponent,
    index_1.ActivitiProcessFilters,
    index_1.ActivitiProcessInstanceDetails,
    index_1.ActivitiProcessInstanceHeader,
    index_1.ActivitiProcessInstanceTasks,
    index_1.ActivitiProcessInstanceVariables,
    index_1.ActivitiProcessComments,
    index_1.ActivitiStartProcessInstance
];
exports.ACTIVITI_PROCESSLIST_PROVIDERS = [
    activiti_process_service_1.ActivitiProcessService
];
var ActivitiProcessListModule = ActivitiProcessListModule_1 = (function () {
    function ActivitiProcessListModule() {
    }
    ActivitiProcessListModule.forRoot = function () {
        return {
            ngModule: ActivitiProcessListModule_1,
            providers: exports.ACTIVITI_PROCESSLIST_PROVIDERS.slice()
        };
    };
    return ActivitiProcessListModule;
}());
ActivitiProcessListModule = ActivitiProcessListModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng2_alfresco_core_1.CoreModule,
            ng2_alfresco_datatable_1.DataTableModule,
            ng2_activiti_form_1.ActivitiFormModule,
            ng2_activiti_tasklist_1.ActivitiTaskListModule
        ],
        declarations: exports.ACTIVITI_PROCESSLIST_DIRECTIVES.slice(),
        providers: exports.ACTIVITI_PROCESSLIST_PROVIDERS.slice(),
        exports: exports.ACTIVITI_PROCESSLIST_DIRECTIVES.slice()
    })
], ActivitiProcessListModule);
exports.ActivitiProcessListModule = ActivitiProcessListModule;
var ActivitiProcessListModule_1;
