// ng2-activiti-tasklist/src/components/activiti-task-header.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var task_details_model_1 = require("../models/task-details.model");
var activiti_tasklist_service_1 = require("./../services/activiti-tasklist.service");
var ActivitiTaskHeader = (function () {
    function ActivitiTaskHeader(translateService, activitiTaskService, logService) {
        this.translateService = translateService;
        this.activitiTaskService = activitiTaskService;
        this.logService = logService;
        this.formName = null;
        this.claim = new core_1.EventEmitter();
        if (translateService) {
            translateService.addTranslationFolder('ng2-activiti-tasklist', 'node_modules/ng2-activiti-tasklist/src');
        }
    }
    ActivitiTaskHeader.prototype.hasAssignee = function () {
        return (this.taskDetails && this.taskDetails.assignee) ? true : false;
    };
    ActivitiTaskHeader.prototype.isAssignedToMe = function () {
        return this.taskDetails.assignee ? true : false;
    };
    ActivitiTaskHeader.prototype.claimTask = function (taskId) {
        var _this = this;
        this.activitiTaskService.claimTask(taskId).subscribe(function (res) {
            _this.logService.info('Task claimed');
            _this.claim.emit(taskId);
        });
    };
    return ActivitiTaskHeader;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ActivitiTaskHeader.prototype, "formName", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", task_details_model_1.TaskDetailsModel)
], ActivitiTaskHeader.prototype, "taskDetails", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], ActivitiTaskHeader.prototype, "claim", void 0);
ActivitiTaskHeader = __decorate([
    core_1.Component({
        selector: 'activiti-task-header',
        moduleId: module.id,
        template: "<div *ngIf=\"taskDetails\">     <div class=\"mdl-grid\">         <div class=\"mdl-cell mdl-cell--4-col\" data-automation-id=\"header-assignee\">             <span class=\"activiti-task-header__label\">{{ 'TASK_DETAILS.LABELS.ASSIGNEE' | translate }}: </span>             <span class=\"activiti-task-header__value\" *ngIf=\"hasAssignee()\">{{ taskDetails.assignee.firstName }} {{ taskDetails.assignee.lastName }}</span>             <span class=\"activiti-task-header__value\" *ngIf=\"!hasAssignee()\">{{ 'TASK_DETAILS.ASSIGNEE.NONE' | translate }}</span>         </div>         <div class=\"mdl-cell mdl-cell--4-col\" data-automation-id=\"header-due-date\">             <span class=\"activiti-task-header__label\">{{ 'TASK_DETAILS.LABELS.DUE' | translate }}: </span>             <span class=\"activiti-task-header__value\" *ngIf=\"taskDetails?.dueDate\">{{ taskDetails.dueDate }}</span>             <span class=\"activiti-task-header__value\" *ngIf=\"!taskDetails?.dueDate\">{{ 'TASK_DETAILS.DUE.NONE' |translate }}</span>         </div>         <div *ngIf=\"formName\" class=\"mdl-cell mdl-cell--4-col\" data-automation-id=\"header-form-name\">             <span class=\"activiti-task-header__label\">{{ 'TASK_DETAILS.LABELS.FORM' | translate }}: </span>             <span class=\"activiti-task-header__value\" *ngIf=\"formName\">{{ formName }}</span>             <span class=\"activiti-task-header__value\" *ngIf=\"!formName\">{{ 'TASK_DETAILS.FORM.NONE' | translate }}</span>         </div>          <button *ngIf=\"!isAssignedToMe()\" data-automation-id=\"header-claim-button\" type=\"button\" id=\"claim-task\"                 (click)=\"claimTask(taskDetails.id)\" class=\"mdl-button\">{{ 'TASK_DETAILS.BUTTON.CLAIM' | translate }}         </button>      </div> </div>",
        styles: [":host {     width: 100%; }  .activiti-task-header__label {     font-weight: bold; }  .activiti-task-header__value {     color: rgb(68, 138, 255); }  .material-icons {     cursor: pointer; }"]
    }),
    __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoTranslationService,
        activiti_tasklist_service_1.ActivitiTaskListService,
        ng2_alfresco_core_1.LogService])
], ActivitiTaskHeader);
exports.ActivitiTaskHeader = ActivitiTaskHeader;
