// ng2-alfresco-documentlist/src/services/document-list.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var Rx_1 = require("rxjs/Rx");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var DocumentListService = DocumentListService_1 = (function () {
    function DocumentListService(authService, contentService, apiService, logService) {
        this.authService = authService;
        this.contentService = contentService;
        this.apiService = apiService;
        this.logService = logService;
        this.mimeTypeIcons = {
            'image/png': 'ft_ic_raster_image.svg',
            'image/jpeg': 'ft_ic_raster_image.svg',
            'image/gif': 'ft_ic_raster_image.svg',
            'application/pdf': 'ft_ic_pdf.svg',
            'application/vnd.ms-excel': 'ft_ic_ms_excel.svg',
            'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet': 'ft_ic_ms_excel.svg',
            'application/vnd.openxmlformats-officedocument.spreadsheetml.template': 'ft_ic_ms_excel.svg',
            'application/msword': 'ft_ic_ms_word.svg',
            'application/vnd.openxmlformats-officedocument.wordprocessingml.document': 'ft_ic_ms_word.svg',
            'application/vnd.openxmlformats-officedocument.wordprocessingml.template': 'ft_ic_ms_word.svg',
            'application/vnd.ms-powerpoint': 'ft_ic_ms_powerpoint.svg',
            'application/vnd.openxmlformats-officedocument.presentationml.presentation': 'ft_ic_ms_powerpoint.svg',
            'application/vnd.openxmlformats-officedocument.presentationml.template': 'ft_ic_ms_powerpoint.svg',
            'application/vnd.openxmlformats-officedocument.presentationml.slideshow': 'ft_ic_ms_powerpoint.svg',
            'video/mp4': 'ft_ic_video.svg',
            'text/plain': 'ft_ic_document.svg',
            'application/x-javascript': 'ft_ic_document.svg',
            'application/json': 'ft_ic_document.svg',
            'image/svg+xml': 'ft_ic_vector_image.svg',
            'text/html': 'ft_ic_website.svg',
            'application/x-compressed': 'ft_ic_archive.svg',
            'application/x-zip-compressed': 'ft_ic_archive.svg',
            'application/zip': 'ft_ic_archive.svg',
            'application/vnd.apple.keynote': 'ft_ic_presentation.svg',
            'application/vnd.apple.pages': 'ft_ic_document.svg',
            'application/vnd.apple.numbers': 'ft_ic_spreadsheet.svg'
        };
    }
    DocumentListService.prototype.getNodesPromise = function (folder, opts) {
        var rootNodeId = DocumentListService_1.ROOT_ID;
        if (opts && opts.rootFolderId) {
            rootNodeId = opts.rootFolderId;
        }
        var params = {
            includeSource: true,
            include: ['path', 'properties']
        };
        if (folder) {
            params.relativePath = folder;
        }
        if (opts) {
            if (opts.maxItems) {
                params.maxItems = opts.maxItems;
            }
            if (opts.skipCount) {
                params.skipCount = opts.skipCount;
            }
        }
        return this.apiService.getInstance().nodes.getNodeChildren(rootNodeId, params);
    };
    DocumentListService.prototype.deleteNode = function (nodeId) {
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().nodes.deleteNode(nodeId));
    };
    DocumentListService.prototype.createFolder = function (name, parentId) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().nodes.createFolder(name, '/', parentId))
            .catch(function (err) { return _this.handleError(err); });
    };
    DocumentListService.prototype.getFolder = function (folder, opts) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.getNodesPromise(folder, opts))
            .map(function (res) { return res; })
            .catch(function (err) { return _this.handleError(err); });
    };
    DocumentListService.prototype.getFolderNode = function (nodeId) {
        var opts = {
            includeSource: true,
            include: ['path', 'properties']
        };
        var nodes = this.apiService.getInstance().nodes;
        return nodes.getNodeInfo(nodeId, opts);
    };
    DocumentListService.prototype.getDocumentThumbnailUrl = function (node) {
        if (node && this.contentService) {
            return this.contentService.getDocumentThumbnailUrl(node);
        }
        return null;
    };
    DocumentListService.prototype.getMimeTypeIcon = function (mimeType) {
        var icon = this.mimeTypeIcons[mimeType];
        return icon || DocumentListService_1.DEFAULT_MIME_TYPE_ICON;
    };
    DocumentListService.prototype.handleError = function (error) {
        this.logService.error(error);
        return Rx_1.Observable.throw(error || 'Server error');
    };
    return DocumentListService;
}());
DocumentListService.DEFAULT_MIME_TYPE_ICON = 'ft_ic_miscellaneous.svg';
DocumentListService.ROOT_ID = '-root-';
DocumentListService = DocumentListService_1 = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoAuthenticationService,
        ng2_alfresco_core_1.AlfrescoContentService,
        ng2_alfresco_core_1.AlfrescoApiService,
        ng2_alfresco_core_1.LogService])
], DocumentListService);
exports.DocumentListService = DocumentListService;
var DocumentListService_1;
