// ng2-alfresco-documentlist/src/components/breadcrumb/breadcrumb.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var document_list_component_1 = require("../document-list.component");
var DocumentListBreadcrumbComponent = (function () {
    function DocumentListBreadcrumbComponent() {
        this.route = [];
        this.navigate = new core_1.EventEmitter();
    }
    DocumentListBreadcrumbComponent.prototype.ngOnChanges = function (changes) {
        if (changes['folderNode']) {
            var node = changes['folderNode'].currentValue;
            if (node) {
                var route = (node.path.elements || []);
                route.push({
                    id: node.id,
                    name: node.name
                });
                this.route = route;
            }
        }
    };
    DocumentListBreadcrumbComponent.prototype.onRoutePathClick = function (route, e) {
        if (e) {
            e.preventDefault();
        }
        if (route) {
            this.navigate.emit({
                value: route
            });
            if (this.target) {
                this.target.loadFolderByNodeId(route.id);
            }
        }
    };
    return DocumentListBreadcrumbComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], DocumentListBreadcrumbComponent.prototype, "folderNode", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", document_list_component_1.DocumentListComponent)
], DocumentListBreadcrumbComponent.prototype, "target", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], DocumentListBreadcrumbComponent.prototype, "navigate", void 0);
DocumentListBreadcrumbComponent = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-document-list-breadcrumb',
        template: "<div>     <ol *ngIf=\"folderNode\" data-automation-id=\"breadcrumb\" class=\"breadcrumb\">         <li *ngFor=\"let r of route; let last = last\" [class.active]=\"last\" [ngSwitch]=\"last\">             <span *ngSwitchCase=\"true\">{{r.name}}</span>             <a *ngSwitchDefault href=\"#\" [attr.data-automation-id]=\"'breadcrumb_' + r.name\" (click)=\"onRoutePathClick(r, $event)\">                 {{r.name}}             </a>         </li>     </ol> </div>",
        styles: ["/* breadcrumb */  :host .breadcrumb {     text-align: left;     padding: 8px 15px;     list-style: none;     background-color: #fafafa;     margin: 0 0 4px; }  :host .breadcrumb > li {     display: inline-block;     box-sizing: border-box; }  :host .breadcrumb > li+li:before {     content: \">\\00a0\";     padding: 0 0 0 5px;     opacity: 0.54;     color: #000000; }  :host .breadcrumb > li > a {     text-decoration: none;     opacity: 0.54;     font-family: 'Muli', \"Helvetica\", \"Arial\", sans-serif;     font-size: 14px;     font-weight: 600;     line-height: 1.43;     letter-spacing: -0.2px;     color: #000000; }  :host .breadcrumb > li:hover > a, :host .breadcrumb > .active {     opacity: 0.87;     font-size: 14px;     font-weight: 600;     line-height: 1.43;     letter-spacing: -0.2px;     color: #000000; }"]
    })
], DocumentListBreadcrumbComponent);
exports.DocumentListBreadcrumbComponent = DocumentListBreadcrumbComponent;
