// ng2-alfresco-documentlist/src/components/content-column/content-column.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var content_column_list_component_1 = require("./content-column-list.component");
var ContentColumnComponent = (function () {
    function ContentColumnComponent(list) {
        this.list = list;
        this.type = 'text';
        this.sortable = false;
        this.title = '';
    }
    ContentColumnComponent.prototype.ngOnInit = function () {
        if (!this.srTitle && this.key === '$thumbnail') {
            this.srTitle = 'Thumbnail';
        }
    };
    ContentColumnComponent.prototype.ngAfterContentInit = function () {
        this.register();
    };
    ContentColumnComponent.prototype.register = function () {
        if (this.list) {
            return this.list.registerColumn(this);
        }
        return false;
    };
    return ContentColumnComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentColumnComponent.prototype, "key", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentColumnComponent.prototype, "type", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentColumnComponent.prototype, "format", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], ContentColumnComponent.prototype, "sortable", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentColumnComponent.prototype, "title", void 0);
__decorate([
    core_1.ContentChild(core_1.TemplateRef),
    __metadata("design:type", Object)
], ContentColumnComponent.prototype, "template", void 0);
__decorate([
    core_1.Input('sr-title'),
    __metadata("design:type", String)
], ContentColumnComponent.prototype, "srTitle", void 0);
__decorate([
    core_1.Input('class'),
    __metadata("design:type", String)
], ContentColumnComponent.prototype, "cssClass", void 0);
ContentColumnComponent = __decorate([
    core_1.Component({
        selector: 'content-column',
        template: ''
    }),
    __metadata("design:paramtypes", [content_column_list_component_1.ContentColumnListComponent])
], ContentColumnComponent);
exports.ContentColumnComponent = ContentColumnComponent;
