// ng2-alfresco-core/src/components/context-menu/context-menu-holder.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var context_menu_service_1 = require("./context-menu.service");
var ContextMenuHolderComponent = (function () {
    function ContextMenuHolderComponent(_contextMenuService) {
        var _this = this;
        this._contextMenuService = _contextMenuService;
        this.links = [];
        this.isShown = false;
        this.mouseLocation = { left: 0, top: 0 };
        _contextMenuService.show.subscribe(function (e) { return _this.showMenu(e.event, e.obj); });
    }
    Object.defineProperty(ContextMenuHolderComponent.prototype, "locationCss", {
        get: function () {
            return {
                position: 'fixed',
                display: this.isShown ? 'block' : 'none',
                left: this.mouseLocation.left + 'px',
                top: this.mouseLocation.top + 'px'
            };
        },
        enumerable: true,
        configurable: true
    });
    ContextMenuHolderComponent.prototype.clickedOutside = function () {
        this.isShown = false;
    };
    ContextMenuHolderComponent.prototype.showMenu = function (e, links) {
        this.isShown = true;
        this.links = links;
        if (e) {
            this.mouseLocation = {
                left: e.clientX,
                top: e.clientY
            };
        }
    };
    ContextMenuHolderComponent.prototype.onShowContextMenu = function (event) {
        if (event) {
            event.preventDefault();
        }
    };
    return ContextMenuHolderComponent;
}());
__decorate([
    core_1.HostListener('document:click'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], ContextMenuHolderComponent.prototype, "clickedOutside", null);
__decorate([
    core_1.HostListener('contextmenu', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [MouseEvent]),
    __metadata("design:returntype", void 0)
], ContextMenuHolderComponent.prototype, "onShowContextMenu", null);
ContextMenuHolderComponent = __decorate([
    core_1.Component({
        selector: 'context-menu-holder',
        styles: [
            "\n        .menu-container {\n            background: #fff;\n            display: block;\n            margin: 0;\n            padding: 0;\n            border: none;\n            overflow: visible;\n            z-index: 9999;\n        }\n\n        .context-menu {\n            list-style-type: none;\n            position: static;\n            height: auto;\n            width: auto;\n            min-width: 124px;\n            padding: 8px 0;\n            margin: 0;\n            box-shadow: 0 2px 2px 0 rgba(0,0,0,.14),0 3px 1px -2px rgba(0,0,0,.2),0 1px 5px 0 rgba(0,0,0,.12);\n            border-radius: 2px;\n        }\n\n        .context-menu .link {\n            opacity: 1;\n        }\n        "
        ],
        template: "\n        <div [ngStyle]=\"locationCss\" class=\"menu-container\">\n            <ul class=\"context-menu\">\n                <li (click)=\"link.subject.next(link)\" class=\"mdl-menu__item link\" *ngFor=\"let link of links\">\n                    {{link.title || link.model?.title}}\n                </li>\n            </ul>\n        </div>\n    "
    }),
    __metadata("design:paramtypes", [context_menu_service_1.ContextMenuService])
], ContextMenuHolderComponent);
exports.ContextMenuHolderComponent = ContextMenuHolderComponent;
