// ng2-activiti-diagrams/src/components/events/diagram-event.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var diagram_color_service_1 = require("../../services/diagram-color.service");
var DiagramEventComponent = (function () {
    function DiagramEventComponent(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: '', radius: '' };
        this.onError = new core_1.EventEmitter();
        this.center = {};
    }
    DiagramEventComponent.prototype.ngOnInit = function () {
        this.center.x = this.data.x + (this.data.width / 2);
        this.center.y = this.data.y + (this.data.height / 2);
    };
    return DiagramEventComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], DiagramEventComponent.prototype, "data", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], DiagramEventComponent.prototype, "options", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], DiagramEventComponent.prototype, "iconFillColor", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], DiagramEventComponent.prototype, "onError", void 0);
DiagramEventComponent = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'diagram-event',
        template: "<raphael-circle [elementId]=\"data.id\" [center]=\"center\" [radius]=\"options.radius\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"                 [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle> <diagram-container-icon-event [data]=\"data\" [type]=\"data.eventDefinition && data.eventDefinition.type\"                               [fillColor]=\"iconFillColor\"></diagram-container-icon-event> <diagram-tooltip [data]=\"data\"></diagram-tooltip>"
    }),
    __metadata("design:paramtypes", [core_1.ElementRef,
        diagram_color_service_1.DiagramColorService])
], DiagramEventComponent);
exports.DiagramEventComponent = DiagramEventComponent;
