// ng2-alfresco-datatable/src/models/pagination.data.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var PaginationData = (function () {
    function PaginationData(count, totalItems, skipCount, maxItems, hasMoreItems) {
        this.count = count;
        this.hasMoreItems = hasMoreItems;
        this.totalItems = totalItems;
        this.skipCount = skipCount;
        this.maxItems = maxItems;
    }
    return PaginationData;
}());
exports.PaginationData = PaginationData;
