// ng2-alfresco-core/src/components/material/mdl-tabs.directive.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var AlfrescoMdlTabsDirective = (function () {
    function AlfrescoMdlTabsDirective(element) {
        this.element = element;
    }
    AlfrescoMdlTabsDirective.prototype.ngAfterViewInit = function () {
        if (componentHandler) {
            var el_1 = this.element.nativeElement;
            el_1.classList.add('mdl-tabs');
            el_1.classList.add('mdl-js-tabs');
            el_1.classList.add('mdl-js-ripple-effect');
            componentHandler.upgradeElement(el_1, 'MaterialTabs');
            var tabBar = el_1.querySelector('.mdl-tabs__tab-bar');
            if (tabBar) {
                this.observer = new MutationObserver(function (mutations) {
                    var upgrade = false;
                    mutations.forEach(function (mutation) {
                        if (mutation.addedNodes && mutation.addedNodes.length > 0) {
                            upgrade = true;
                        }
                        if (mutation.removedNodes && mutation.removedNodes.length > 0) {
                            upgrade = true;
                        }
                    });
                    if (upgrade) {
                        componentHandler.downgradeElements([el_1]);
                        componentHandler.upgradeElement(el_1);
                    }
                });
                this.observer.observe(tabBar, {
                    childList: true,
                    subtree: false
                });
            }
        }
    };
    AlfrescoMdlTabsDirective.prototype.ngOnDestroy = function () {
        if (this.observer) {
            this.observer.disconnect();
        }
    };
    return AlfrescoMdlTabsDirective;
}());
AlfrescoMdlTabsDirective = __decorate([
    core_1.Directive({
        selector: '[alfresco-mdl-tabs]'
    }),
    __metadata("design:paramtypes", [core_1.ElementRef])
], AlfrescoMdlTabsDirective);
exports.AlfrescoMdlTabsDirective = AlfrescoMdlTabsDirective;
