// ng2-alfresco-userinfo/src/assets/fake-ecm-user.service.mock.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var ecm_user_model_1 = require("../models/ecm-user.model");
exports.EcmUserModel = ecm_user_model_1.EcmUserModel;
var ecm_user_service_1 = require("../services/ecm-user.service");
exports.EcmUserService = ecm_user_service_1.EcmUserService;
var Rx_1 = require("rxjs/Rx");
exports.fakeEcmCompany = {
    organization: 'company-fake-name',
    address1: 'fake-address-1',
    address2: 'fake-address-2',
    address3: 'fake-address-3',
    postcode: 'fAk1',
    telephone: '00000000',
    fax: '11111111',
    email: 'fakeCompany@fake.com'
};
exports.fakeEcmUserNoImage = {
    id: 'fake-id',
    firstName: 'fake-first-name',
    lastName: 'fake-last-name',
    description: 'i am a fake user for test',
    avatarId: undefined,
    email: 'fakeEcm@ecmUser.com',
    skypeId: 'fake-skype-id',
    googleId: 'fake-googleId-id',
    instantMessageId: 'fake-instantMessageId-id',
    company: exports.fakeEcmCompany,
    jobTitle: 'test job',
    location: 'fake location',
    mobile: '000000000',
    telephone: '11111111',
    statusUpdatedAt: 'fake-date',
    userStatus: 'active',
    enabled: true,
    emailNotificationsEnabled: true
};
exports.fakeEcmUser = {
    id: 'fake-id',
    firstName: 'fake-first-name',
    lastName: 'fake-last-name',
    description: 'i am a fake user for test',
    avatarId: 'fake-avatar-id',
    email: 'fakeEcm@ecmUser.com',
    skypeId: 'fake-skype-id',
    googleId: 'fake-googleId-id',
    instantMessageId: 'fake-instantMessageId-id',
    company: exports.fakeEcmCompany,
    jobTitle: 'test job',
    location: 'fake location',
    mobile: '000000000',
    telephone: '11111111',
    statusUpdatedAt: 'fake-date',
    userStatus: 'active',
    enabled: true,
    emailNotificationsEnabled: true
};
exports.fakeEcmEditedUser = {
    id: 'fake-id',
    firstName: 'fake-first-name',
    lastName: 'fake-last-name',
    description: 'i am a fake user for test',
    avatarId: 'fake-avatar-id',
    email: 'fakeEcm@ecmUser.com',
    skypeId: 'fake-skype-id',
    googleId: 'fake-googleId-id',
    instantMessageId: 'fake-instantMessageId-id',
    company: exports.fakeEcmCompany,
    jobTitle: 'test job',
    location: 'fake location',
    mobile: '000000000',
    telephone: '11111111',
    statusUpdatedAt: 'fake-date',
    userStatus: 'active',
    enabled: true,
    emailNotificationsEnabled: true
};
var FakeEcmUserService = (function () {
    function FakeEcmUserService() {
        this.userNeeded = 0;
        this.usersList = [exports.fakeEcmUser, exports.fakeEcmUserNoImage, exports.fakeEcmEditedUser];
    }
    FakeEcmUserService.prototype.getUserInfo = function (userName) {
        return this.lastPromise = Rx_1.Observable.of(this.usersList[this.userNeeded]);
    };
    ;
    FakeEcmUserService.prototype.getCurrentUserInfo = function () {
        return this.getUserInfo('fake-id');
    };
    ;
    FakeEcmUserService.prototype.getUserProfileImage = function (avatarId) {
        if (avatarId) {
            return 'src/assets/images/ecmImg.gif';
        }
    };
    ;
    FakeEcmUserService.prototype.respondWithTheUserWithoutImage = function () {
        this.userNeeded = 1;
    };
    ;
    FakeEcmUserService.prototype.respondWithTheUserWithImage = function () {
        this.userNeeded = 0;
    };
    ;
    FakeEcmUserService.prototype.respondWithEditedUser = function () {
        this.userNeeded = 2;
    };
    ;
    return FakeEcmUserService;
}());
exports.FakeEcmUserService = FakeEcmUserService;
