// ng2-activiti-form/src/components/widgets/hyperlink/hyperlink.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var core_1 = require("@angular/core");
var widget_component_1 = require("./../widget.component");
var HyperlinkWidget = (function (_super) {
    __extends(HyperlinkWidget, _super);
    function HyperlinkWidget() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.linkUrl = widget_component_1.WidgetComponent.DEFAULT_HYPERLINK_URL;
        _this.linkText = null;
        return _this;
    }
    HyperlinkWidget.prototype.ngOnInit = function () {
        if (this.field) {
            this.linkUrl = this.getHyperlinkUrl(this.field);
            this.linkText = this.getHyperlinkText(this.field);
        }
    };
    return HyperlinkWidget;
}(widget_component_1.WidgetComponent));
HyperlinkWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'hyperlink-widget',
        template: "<div class=\"hyperlink-widget\">     <div>         <span>{{field.name}}</span>     </div>     <div>         <a [href]=\"linkUrl\" target=\"_blank\" rel=\"nofollow\">{{linkText}}</a>     </div> </div>",
        styles: [".hyperlink-widget {}"]
    })
], HyperlinkWidget);
exports.HyperlinkWidget = HyperlinkWidget;
