// ng2-activiti-diagrams/src/components/raphael/raphael.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var RaphaelService = (function () {
    function RaphaelService() {
        this.width = 300;
        this.height = 400;
    }
    RaphaelService.prototype.getInstance = function (element) {
        if (!this.paper) {
            this.ctx = element.nativeElement;
            this.refresh();
        }
        return this.paper;
    };
    RaphaelService.prototype.refresh = function () {
        this.ngOnDestroy();
        this.paper = this.getPaperBuilder(this.ctx);
    };
    RaphaelService.prototype.getPaperBuilder = function (ctx) {
        if (typeof Raphael === 'undefined') {
            throw new Error('ng2-charts configuration issue: Embedding Chart.js lib is mandatory');
        }
        var paper = new Raphael(ctx, this.width, this.height);
        return paper;
    };
    RaphaelService.prototype.ngOnDestroy = function () {
        if (this.paper) {
            this.paper.clear();
            this.paper = void 0;
        }
    };
    RaphaelService.prototype.setting = function (width, height) {
        this.width = width;
        this.height = height;
    };
    RaphaelService.prototype.reset = function () {
        this.ngOnDestroy();
    };
    return RaphaelService;
}());
RaphaelService = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [])
], RaphaelService);
exports.RaphaelService = RaphaelService;
