// ng2-activiti-diagrams/src/assets/diagramGateways.mock.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
exports.exclusiveGatway = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    type: 'ExclusiveGateway',
    width: 40,
    height: 40,
    x: 40,
    y: 30,
    properties: [{}]
};
exports.inclusiveGatway = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    type: 'InclusiveGateway',
    width: 40,
    height: 40,
    x: 40,
    y: 30,
    properties: [{}]
};
exports.parallelGatway = {
    id: 'sid-14EE23CE-0731-4E23-80F3-C557DA2A0CFC',
    type: 'ParallelGateway',
    width: 40,
    height: 40,
    x: 40,
    y: 30,
    properties: [{}]
};
exports.eventGatway = {
    id: 'sid-14EE23CE-0731-4E23-80F3-C557DA2A0CFC',
    type: 'EventGateway',
    width: 40,
    height: 40,
    x: 40,
    y: 30,
    properties: [{}]
};
