// ng2-activiti-diagrams/src/assets/diagramEvents.mock.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
exports.startEvent = {
    id: 'startEvent1',
    type: 'StartEvent',
    width: 30,
    height: 30,
    x: 15,
    y: 48.5,
    properties: [{}]
};
exports.startTimeEvent = {
    id: 'startEvent1',
    type: 'StartEvent',
    width: 30,
    height: 30,
    x: 15,
    y: 48.5,
    eventDefinition: { type: 'timer' },
    properties: [{}]
};
exports.startSignalEvent = {
    id: 'startEvent1',
    type: 'StartEvent',
    width: 30,
    height: 30,
    x: 15,
    y: 48.5,
    eventDefinition: { type: 'signal' },
    properties: [{}]
};
exports.startMessageEvent = {
    id: 'startEvent1',
    type: 'StartEvent',
    width: 30,
    height: 30,
    x: 15,
    y: 48.5,
    eventDefinition: { type: 'message' },
    properties: [{}]
};
exports.startErrorEvent = {
    id: 'startEvent1',
    type: 'StartEvent',
    width: 30,
    height: 30,
    x: 15,
    y: 48.5,
    eventDefinition: { type: 'error' },
    properties: [{}]
};
exports.endEvent = {
    id: 'sid-CED2A8DB-47E2-4057-A7B8-3ABBE5CE795E',
    type: 'EndEvent',
    width: 28,
    height: 28,
    x: 15,
    y: 48.5,
    properties: [{}]
};
exports.endErrorEvent = {
    id: 'sid-CED2A8DB-47E2-4057-A7B8-3ABBE5CE795E',
    type: 'EndEvent',
    width: 28,
    height: 28,
    x: 15,
    y: 48.5,
    eventDefinition: { type: 'error' },
    properties: [{}]
};
