// ng2-activiti-diagrams/src/assets/diagramActivities.mock.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
exports.userTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake User task',
    type: 'UserTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}]
};
exports.manualTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Manual task',
    type: 'ManualTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}]
};
exports.serviceTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Service task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}]
};
exports.receiveTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Receive task',
    type: 'ReceiveTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}]
};
exports.scriptTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Script task',
    type: 'ScriptTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}]
};
exports.businessRuleTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake BusinessRule task',
    type: 'BusinessRuleTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}]
};
exports.mailTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Mail task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}],
    taskType: 'mail'
};
exports.camelTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Camel task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}],
    taskType: 'camel'
};
exports.restCallTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Rest Call task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}],
    taskType: 'rest_call'
};
exports.muleTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Mule task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}],
    taskType: 'mule'
};
exports.alfrescoPublishTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Alfresco Publish task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}],
    taskType: 'alfresco_publish'
};
exports.googleDrivePublishTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Google Drive Publish task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}],
    taskType: 'google_drive_publish'
};
exports.boxPublishTask = {
    id: 'sid-C05B7CB7-1CFD-4AE4-9E01-C2C91E35E5A7',
    name: 'Fake Box Publish task',
    type: 'ServiceTask',
    width: 82,
    height: 67,
    x: 75.56890135999998,
    y: 30,
    properties: [{}],
    taskType: 'box_publish'
};
