// ng2-activiti-diagrams/src/assets/diagramSwimlanes.mock.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
exports.pool = {
    id: 'sid-0FF10DA3-E2BD-4E6A-9013-6D66FC8A4716',
    name: 'Activiti',
    width: 600,
    height: 500,
    x: 60,
    y: 45,
    properties: [{}]
};
exports.poolLanes = {
    id: 'sid-0FF10DA3-E2BD-4E6A-9013-6D66FC8A4716',
    name: 'Activiti',
    width: 600,
    height: 500,
    x: 60,
    y: 45,
    lanes: [{
            id: 'sid-332204AB-D0F8-44CD-87B3-BF9DF59FF8AB',
            name: 'Beckend',
            width: 570,
            height: 250,
            x: 90,
            y: 45
        }],
    properties: [{}]
};
