// ng2-activiti-analytics/src/components/widgets/duration/duration.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var number_widget_1 = require("./../number/number.widget");
var report_model_1 = require("./../../../models/report.model");
var forms_1 = require("@angular/forms");
var DurationWidget = (function (_super) {
    __extends(DurationWidget, _super);
    function DurationWidget(elementRef) {
        var _this = _super.call(this, elementRef) || this;
        _this.elementRef = elementRef;
        _this.required = false;
        return _this;
    }
    DurationWidget.prototype.ngOnInit = function () {
        var timeType = new forms_1.FormControl();
        this.formGroup.addControl('timeType', timeType);
        if (this.required) {
            this.formGroup.get(this.controllerName).setValidators(forms_1.Validators.required);
        }
        if (this.field.value === null) {
            this.field.value = 0;
        }
        var paramOptions = [];
        paramOptions.push(new report_model_1.ParameterValueModel({ id: '1', name: 'Seconds' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: '60', name: 'Minutes' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: '3600', name: 'Hours' }));
        paramOptions.push(new report_model_1.ParameterValueModel({ id: '86400', name: 'Days', selected: true }));
        this.duration = new report_model_1.ReportParameterDetailsModel({ id: 'duration', name: 'duration', options: paramOptions });
        this.duration.value = paramOptions[0].id;
    };
    DurationWidget.prototype.calculateDuration = function () {
        if (this.field && this.duration.value) {
            this.currentValue = parseInt(this.field.value, 10) * parseInt(this.duration.value, 10);
            this.formGroup.get(this.controllerName).setValue(this.currentValue);
            this.fieldChanged.emit({ value: this.currentValue });
        }
    };
    return DurationWidget;
}(number_widget_1.NumberWidget));
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], DurationWidget.prototype, "field", void 0);
__decorate([
    core_1.Input('group'),
    __metadata("design:type", forms_1.FormGroup)
], DurationWidget.prototype, "formGroup", void 0);
__decorate([
    core_1.Input('controllerName'),
    __metadata("design:type", String)
], DurationWidget.prototype, "controllerName", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], DurationWidget.prototype, "required", void 0);
DurationWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'duration-widget',
        template: "<div class=\"mdl-grid\">     <div class=\"mdl-cell mdl-cell--6-col\">         <div class=\"mdl-textfield mdl-js-textfield mdl-textfield--floating-label number-widget\">             <input class=\"mdl-textfield__input\"                    type=\"text\"                    pattern=\"-?[0-9]*(\\.[0-9]+)?\"                    [attr.id]=\"field.id\"                    [value]=\"field.value\"                    [(ngModel)]=\"field.value\"                    (ngModelChange)=\"calculateDuration()\">             <label class=\"mdl-textfield__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label>         </div>     </div>     <div class=\"mdl-cell mdl-cell--6-col\">         <div style=\"margin-top: 30px\">             <dropdown-widget [field]=\"duration\" [group]=\"formGroup\" [controllerName]=\"'timeType'\"                              [showDefaultOption]=\"false\"                              (fieldChanged)=\"calculateDuration()\"></dropdown-widget>         </div>     </div> </div>",
        styles: [".number-widget {     width: 100%; }  .number-widget__invalid .mdl-textfield__input {     border-color: #d50000; }  .number-widget__invalid .mdl-textfield__label {     color: #d50000; }  .number-widget__invalid .mdl-textfield__label:after {     background-color: #d50000; }  .number-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }),
    __metadata("design:paramtypes", [core_1.ElementRef])
], DurationWidget);
exports.DurationWidget = DurationWidget;
