// ng2-activiti-analytics/src/components/widgets/number/number.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var widget_component_1 = require("./../widget.component");
var forms_1 = require("@angular/forms");
var NumberWidget = (function (_super) {
    __extends(NumberWidget, _super);
    function NumberWidget(elementRef) {
        var _this = _super.call(this) || this;
        _this.elementRef = elementRef;
        _this.required = false;
        return _this;
    }
    NumberWidget.prototype.ngOnInit = function () {
        if (this.required) {
            this.formGroup.get(this.controllerName).setValidators(forms_1.Validators.required);
        }
    };
    NumberWidget.prototype.setupMaterialComponents = function (handler) {
        if (handler) {
            handler.upgradeAllRegistered();
            if (this.elementRef && this.hasValue()) {
                var container = this.elementRef.nativeElement.querySelector('.mdl-textfield');
                if (container && container.MaterialTextfield) {
                    container.MaterialTextfield.change(this.field.value.toString());
                }
            }
            return true;
        }
        return false;
    };
    return NumberWidget;
}(widget_component_1.WidgetComponent));
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], NumberWidget.prototype, "field", void 0);
__decorate([
    core_1.Input('group'),
    __metadata("design:type", forms_1.FormGroup)
], NumberWidget.prototype, "formGroup", void 0);
__decorate([
    core_1.Input('controllerName'),
    __metadata("design:type", String)
], NumberWidget.prototype, "controllerName", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], NumberWidget.prototype, "required", void 0);
NumberWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'number-widget',
        template: "<div class=\"mdl-textfield mdl-js-textfield mdl-textfield--floating-label number-widget\" [formGroup]=\"formGroup\">     <input formControlName=\"{{controllerName}}\" class=\"mdl-textfield__input\"            type=\"text\"            pattern=\"-?[0-9]*(\\.[0-9]+)?\"            [attr.id]=\"field.id\"            [value]=\"field.value\"            [(ngModel)]=\"field.value\"            (ngModelChange)=\"changeValue(field)\">     <label class=\"mdl-textfield__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label> </div>",
        styles: [".number-widget {     width: 100%; }  .number-widget__invalid .mdl-textfield__input {     border-color: #d50000; }  .number-widget__invalid .mdl-textfield__label {     color: #d50000; }  .number-widget__invalid .mdl-textfield__label:after {     background-color: #d50000; }  .number-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }),
    __metadata("design:paramtypes", [core_1.ElementRef])
], NumberWidget);
exports.NumberWidget = NumberWidget;
