// ng2-activiti-tasklist/src/models/filter.model.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var AppDefinitionRepresentationModel = (function () {
    function AppDefinitionRepresentationModel(obj) {
        this.defaultAppId = obj && obj.defaultAppId || null;
        this.deploymentId = obj && obj.deploymentId || false;
        this.name = obj && obj.name || null;
        this.description = obj && obj.description || null;
        this.theme = obj && obj.theme || null;
        this.icon = obj && obj.icon || null;
        this.id = obj && obj.id;
        this.modelId = obj && obj.modelId;
        this.tenantId = obj && obj.tenantId;
    }
    return AppDefinitionRepresentationModel;
}());
exports.AppDefinitionRepresentationModel = AppDefinitionRepresentationModel;
var FilterRepresentationModel = (function () {
    function FilterRepresentationModel(obj) {
        this.appId = obj && obj.appId || null;
        this.name = obj && obj.name || null;
        this.recent = obj && obj.recent || false;
        this.icon = obj && obj.icon || null;
        this.filter = new FilterParamRepresentationModel(obj.filter);
        this.index = obj && obj.index;
    }
    FilterRepresentationModel.prototype.hasFilter = function () {
        return this.filter ? true : false;
    };
    return FilterRepresentationModel;
}());
exports.FilterRepresentationModel = FilterRepresentationModel;
var FilterParamRepresentationModel = (function () {
    function FilterParamRepresentationModel(obj) {
        this.processDefinitionId = obj && obj.processDefinitionId || null;
        this.processDefinitionKey = obj && obj.processDefinitionKey || null;
        this.name = obj && obj.name || null;
        this.state = obj && obj.state || null;
        this.sort = obj && obj.sort || null;
        this.assignment = obj && obj.assignment || null;
        this.dueAfter = obj && obj.dueAfter || null;
        this.dueBefore = obj && obj.dueBefore || null;
    }
    return FilterParamRepresentationModel;
}());
exports.FilterParamRepresentationModel = FilterParamRepresentationModel;
var TaskQueryRequestRepresentationModel = (function () {
    function TaskQueryRequestRepresentationModel(obj) {
        this.appDefinitionId = obj && obj.appDefinitionId || null;
        this.processInstanceId = obj && obj.processInstanceId || null;
        this.processDefinitionId = obj && obj.processDefinitionId || null;
        this.processDefinitionKey = obj && obj.processDefinitionKey || null;
        this.text = obj && obj.text || null;
        this.assignment = obj && obj.assignment || null;
        this.state = obj && obj.state || null;
        this.sort = obj && obj.sort || null;
        this.page = obj && obj.page || 0;
        this.size = obj && obj.size || 25;
        this.landingTaskId = obj && obj.landingTaskId || '';
    }
    return TaskQueryRequestRepresentationModel;
}());
exports.TaskQueryRequestRepresentationModel = TaskQueryRequestRepresentationModel;
