// ng2-activiti-form/src/components/widgets/dropdown/dropdown.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var form_service_1 = require("../../../services/form.service");
var widget_component_1 = require("./../widget.component");
var widget_visibility_service_1 = require("../../../services/widget-visibility.service");
var DropdownWidget = (function (_super) {
    __extends(DropdownWidget, _super);
    function DropdownWidget(formService, visibilityService, logService) {
        var _this = _super.call(this) || this;
        _this.formService = formService;
        _this.visibilityService = visibilityService;
        _this.logService = logService;
        return _this;
    }
    DropdownWidget.prototype.ngOnInit = function () {
        if (this.field && this.field.restUrl) {
            if (this.field.form.taskId) {
                this.getValuesByTaskId();
            }
            else {
                this.getValuesByProcessDefinitionId();
            }
        }
    };
    DropdownWidget.prototype.getValuesByTaskId = function () {
        var _this = this;
        this.formService
            .getRestFieldValues(this.field.form.taskId, this.field.id)
            .subscribe(function (result) {
            var options = [];
            if (_this.field.emptyOption) {
                options.push(_this.field.emptyOption);
            }
            _this.field.options = options.concat((result || []));
            _this.field.updateForm();
        }, function (err) { return _this.handleError(err); });
    };
    DropdownWidget.prototype.getValuesByProcessDefinitionId = function () {
        var _this = this;
        this.formService
            .getRestFieldValuesByProcessId(this.field.form.processDefinitionId, this.field.id)
            .subscribe(function (result) {
            var options = [];
            if (_this.field.emptyOption) {
                options.push(_this.field.emptyOption);
            }
            _this.field.options = options.concat((result || []));
            _this.field.updateForm();
        }, function (err) { return _this.handleError(err); });
    };
    DropdownWidget.prototype.getOptionValue = function (option, fieldValue) {
        var optionValue = '';
        if (option.id === 'empty' || option.name !== fieldValue) {
            optionValue = option.id;
        }
        else {
            optionValue = option.name;
        }
        return optionValue;
    };
    DropdownWidget.prototype.checkVisibility = function () {
        this.visibilityService.refreshVisibility(this.field.form);
    };
    DropdownWidget.prototype.handleError = function (error) {
        this.logService.error(error);
    };
    return DropdownWidget;
}(widget_component_1.WidgetComponent));
DropdownWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'dropdown-widget',
        template: "<div class=\"dropdown-widget\"      [class.dropdown-widget__invalid]=\"!field.isValid\" *ngIf=\"field?.isVisible\">     <label class=\"dropdown-widget__label\" [attr.for]=\"field.id\">{{field.name}}</label>     <select class=\"dropdown-widget__select\"             [attr.id]=\"field.id\"             [(ngModel)]=\"field.value\"             (ngModelChange)=\"checkVisibility(field)\">         <option *ngFor=\"let opt of field.options\" [value]=\"getOptionValue(opt, field.value)\" [id]=\"opt.id\">{{opt.name}}</option>     </select>     <span *ngIf=\"field.validationSummary\" class=\"mdl-textfield__error\">{{field.validationSummary}}</span> </div>",
        styles: [".dropdown-widget {     width: 100%;     padding-top: 20px;     padding-bottom: 20px; }  .dropdown-widget__select {     width: 100%;     margin-top: 5px;     margin-bottom: 5px; }  .dropdown-widget__invalid .dropdown-widget__select {     border-color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label {     color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label:after {     background-color: #d50000; }  .dropdown-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }),
    __metadata("design:paramtypes", [form_service_1.FormService,
        widget_visibility_service_1.WidgetVisibilityService,
        ng2_alfresco_core_1.LogService])
], DropdownWidget);
exports.DropdownWidget = DropdownWidget;
