// ng2-alfresco-core/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var http_1 = require("@angular/http");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var ng2_translate_1 = require("ng2-translate/ng2-translate");
var material_1 = require("@angular/material");
var index_1 = require("./src/services/index");
var index_2 = require("./src/components/material/index");
var index_3 = require("./src/components/context-menu/index");
__export(require("./src/services/index"));
__export(require("./src/components/index"));
__export(require("./src/utils/index"));
exports.ALFRESCO_CORE_PROVIDERS = [
    index_1.NotificationService,
    index_1.LogService,
    index_1.LogServiceMock,
    index_1.AlfrescoAuthenticationService,
    index_1.AlfrescoContentService,
    index_1.AlfrescoSettingsService,
    index_1.StorageService,
    index_1.AlfrescoApiService,
    index_1.AlfrescoTranslateLoader,
    index_1.AlfrescoTranslationService,
    index_1.RenditionsService,
    index_1.AuthGuard,
    index_1.AuthGuardEcm,
    index_1.AuthGuardBpm
].concat(index_3.CONTEXT_MENU_PROVIDERS);
function createTranslateLoader(http, logService) {
    return new index_1.AlfrescoTranslateLoader(http, logService);
}
exports.createTranslateLoader = createTranslateLoader;
var CoreModule = CoreModule_1 = (function () {
    function CoreModule() {
    }
    CoreModule.forRoot = function () {
        return {
            ngModule: CoreModule_1,
            providers: exports.ALFRESCO_CORE_PROVIDERS.slice()
        };
    };
    return CoreModule;
}());
CoreModule = CoreModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            common_1.CommonModule,
            forms_1.FormsModule,
            forms_1.ReactiveFormsModule,
            http_1.HttpModule,
            material_1.MaterialModule.forRoot(),
            ng2_translate_1.TranslateModule.forRoot({
                provide: ng2_translate_1.TranslateLoader,
                useFactory: (createTranslateLoader),
                deps: [http_1.Http, index_1.LogService]
            })
        ],
        declarations: index_2.MATERIAL_DESIGN_DIRECTIVES.concat(index_3.CONTEXT_MENU_DIRECTIVES),
        providers: exports.ALFRESCO_CORE_PROVIDERS.slice(),
        exports: [
            common_1.CommonModule,
            forms_1.FormsModule,
            material_1.MaterialModule,
            forms_1.ReactiveFormsModule,
            http_1.HttpModule,
            ng2_translate_1.TranslateModule
        ].concat(index_2.MATERIAL_DESIGN_DIRECTIVES, index_3.CONTEXT_MENU_DIRECTIVES)
    })
], CoreModule);
exports.CoreModule = CoreModule;
var CoreModule_1;
