// ng2-alfresco-tag/src/components/tag-list.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var tag_service_1 = require("../services/tag.service");
var TagList = (function () {
    function TagList(tagService) {
        this.tagService = tagService;
        this.resultsEmitter = new core_1.EventEmitter();
    }
    TagList.prototype.ngOnInit = function (changes) {
        return this.refreshTagEcm();
    };
    TagList.prototype.refreshTagEcm = function () {
        var _this = this;
        this.tagService.getAllTheTags().subscribe(function (data) {
            _this.tagsEntries = data.list.entries;
            _this.resultsEmitter.emit(_this.tagsEntries);
        });
    };
    return TagList;
}());
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], TagList.prototype, "resultsEmitter", void 0);
TagList = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-tag-list',
        template: "<div class=\"tag-container\">     <div id=\"table-cards-container\">         <ul class=\"tag-list-control mdl-list\">             <li class=\"mdl-list__item tag-element\" *ngFor=\"let currentEntry of tagsEntries; let idx = index\">                 <span id=\"tag_name_{{idx}}\"  class=\"mdl-list__item-primary-content\">{{currentEntry.entry.tag}}</span>             </li>         </ul>     </div> </div>"
    }),
    __metadata("design:paramtypes", [tag_service_1.TagService])
], TagList);
exports.TagList = TagList;
