// ng2-alfresco-tag/src/components/tag-node-list.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var tag_service_1 = require("../services/tag.service");
var TagNodeList = (function () {
    function TagNodeList(tagService) {
        this.tagService = tagService;
        this.resultsEmitter = new core_1.EventEmitter();
    }
    TagNodeList.prototype.ngOnChanges = function (changes) {
        return this.refreshTag();
    };
    TagNodeList.prototype.refreshTag = function () {
        var _this = this;
        this.tagService.getTagsByNodeId(this.nodeId).subscribe(function (data) {
            _this.tagsEntries = data.list.entries;
            _this.resultsEmitter.emit(_this.tagsEntries);
        });
    };
    TagNodeList.prototype.removeTag = function (tag) {
        var _this = this;
        this.tagService.removeTag(this.nodeId, tag).subscribe(function () {
            _this.refreshTag();
        });
    };
    return TagNodeList;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], TagNodeList.prototype, "nodeId", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], TagNodeList.prototype, "resultsEmitter", void 0);
TagNodeList = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-tag-node-list',
        template: "<div class=\"tag-list-container\">     <ul class=\"tags\">         <li *ngFor=\"let currentEntry of tagsEntries; let idx = index\">             <span class=\"mdl-chip mdl-chip--deletable\">                  <span id=\"tag_name_{{idx}}\" class=\"mdl-chip__text\">{{currentEntry.entry.tag}}</span>                       <button id=\"tag_delete_{{idx}}\" type=\"button\" class=\"mdl-chip__action\"                               (click)=\"removeTag(currentEntry.entry.id)\">                           <i class=\"material-icons\">cancel</i>                       </button>             </span>         </li>     </ul> </div>",
        styles: [".tags {     padding-left: 7px;     padding-right: 7px; }  .tags li {     display: inline; }"]
    }),
    __metadata("design:paramtypes", [tag_service_1.TagService])
], TagNodeList);
exports.TagNodeList = TagNodeList;
