// ng2-alfresco-login/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var login_header_directive_1 = require("./src/directives/login-header.directive");
var login_footer_directive_1 = require("./src/directives/login-footer.directive");
var alfresco_login_component_1 = require("./src/components/alfresco-login.component");
__export(require("./src/directives/login-header.directive"));
__export(require("./src/directives/login-footer.directive"));
__export(require("./src/components/alfresco-login.component"));
exports.ALFRESCO_LOGIN_DIRECTIVES = [
    alfresco_login_component_1.AlfrescoLoginComponent,
    login_footer_directive_1.LoginFooterDirective,
    login_header_directive_1.LoginHeaderDirective
];
var LoginModule = LoginModule_1 = (function () {
    function LoginModule() {
    }
    LoginModule.forRoot = function () {
        return {
            ngModule: LoginModule_1
        };
    };
    return LoginModule;
}());
LoginModule = LoginModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng2_alfresco_core_1.CoreModule
        ],
        declarations: exports.ALFRESCO_LOGIN_DIRECTIVES.slice(),
        providers: [],
        exports: exports.ALFRESCO_LOGIN_DIRECTIVES.slice()
    })
], LoginModule);
exports.LoginModule = LoginModule;
var LoginModule_1;
