// ng2-activiti-form/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var activiti_form_component_1 = require("./src/components/activiti-form.component");
var form_field_component_1 = require("./src/components/form-field/form-field.component");
var activiti_start_form_component_1 = require("./src/components/activiti-start-form.component");
var form_service_1 = require("./src/services/form.service");
var ecm_model_service_1 = require("./src/services/ecm-model.service");
var node_service_1 = require("./src/services/node.service");
var widget_visibility_service_1 = require("./src/services/widget-visibility.service");
var activiti_alfresco_service_1 = require("./src/services/activiti-alfresco.service");
var form_rendering_service_1 = require("./src/services/form-rendering.service");
var http_1 = require("@angular/http");
var index_1 = require("./src/components/widgets/index");
__export(require("./src/components/activiti-form.component"));
__export(require("./src/components/activiti-start-form.component"));
__export(require("./src/services/form.service"));
__export(require("./src/components/widgets/index"));
__export(require("./src/services/ecm-model.service"));
__export(require("./src/services/node.service"));
__export(require("./src/services/form-rendering.service"));
__export(require("./src/events/index"));
exports.ACTIVITI_FORM_DIRECTIVES = [
    activiti_form_component_1.ActivitiForm,
    activiti_start_form_component_1.ActivitiStartForm,
    form_field_component_1.FormFieldComponent
].concat(index_1.WIDGET_DIRECTIVES);
exports.ACTIVITI_FORM_PROVIDERS = [
    form_service_1.FormService,
    ecm_model_service_1.EcmModelService,
    node_service_1.NodeService,
    widget_visibility_service_1.WidgetVisibilityService,
    activiti_alfresco_service_1.ActivitiAlfrescoContentService,
    form_rendering_service_1.FormRenderingService
];
var ActivitiFormModule = ActivitiFormModule_1 = (function () {
    function ActivitiFormModule() {
    }
    ActivitiFormModule.forRoot = function () {
        return {
            ngModule: ActivitiFormModule_1,
            providers: exports.ACTIVITI_FORM_PROVIDERS.slice()
        };
    };
    return ActivitiFormModule;
}());
ActivitiFormModule = ActivitiFormModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng2_alfresco_core_1.CoreModule,
            http_1.HttpModule
        ],
        declarations: exports.ACTIVITI_FORM_DIRECTIVES.slice(),
        entryComponents: index_1.WIDGET_DIRECTIVES.slice(),
        providers: exports.ACTIVITI_FORM_PROVIDERS.slice(),
        exports: exports.ACTIVITI_FORM_DIRECTIVES.slice()
    })
], ActivitiFormModule);
exports.ActivitiFormModule = ActivitiFormModule;
var ActivitiFormModule_1;
