// ng2-alfresco-documentlist/bundles/ng2-alfresco-documentlist.js
!function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in v||(v[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==g.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=v[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(g.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=v[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}n=!n}}function u(e){return y[e]||(y[e]={name:e,dependencies:[],exports:{},importers:[]})}function d(r){if(!r.module){var t=r.module=u(r.name),n=r.module.exports,o=r.declare.call(e,function(e,r){if(t.locked=!0,"object"==typeof e)for(var o in e)n[o]=e[o];else n[e]=r;for(var a=0,u=t.importers.length;u>a;a++){var d=t.importers[a];if(!d.locked)for(var i=0;i<d.dependencies.length;++i)d.dependencies[i]===t&&d.setters[i](n)}return t.locked=!1,r},{id:r.name});t.setters=o.setters,t.execute=o.execute;for(var a=0,i=r.normalizedDeps.length;i>a;a++){var l,s=r.normalizedDeps[a],c=v[s],f=y[s];f?l=f.exports:c&&!c.declarative?l=c.esModule:c?(d(c),f=c.module,l=f.exports):l=p(s),f&&f.importers?(f.importers.push(t),t.dependencies.push(f)):t.dependencies.push(null),t.setters[a]&&t.setters[a](l)}}}function i(e){var r,t=v[e];if(t)t.declarative?f(e,[]):t.evaluated||l(t),r=t.module.exports;else if(r=p(e),!r)throw new Error("Unable to load dependency "+e+".");return(!t||t.declarative)&&r&&r.__useDefault?r["default"]:r}function l(r){if(!r.module){var t={},n=r.module={exports:t,id:r.name};if(!r.executingRequire)for(var o=0,a=r.normalizedDeps.length;a>o;o++){var u=r.normalizedDeps[o],d=v[u];d&&l(d)}r.evaluated=!0;var c=r.execute.call(e,function(e){for(var t=0,n=r.deps.length;n>t;t++)if(r.deps[t]==e)return i(r.normalizedDeps[t]);throw new TypeError("Module "+e+" not declared as a dependency.")},t,n);void 0!==c&&(n.exports=c),t=n.exports,t&&t.__esModule?r.esModule=t:r.esModule=s(t)}}function s(r){var t={};if(("object"==typeof r||"function"==typeof r)&&r!==e)if(m)for(var n in r)"default"!==n&&c(t,r,n);else{var o=r&&r.hasOwnProperty;for(var n in r)"default"===n||o&&!r.hasOwnProperty(n)||(t[n]=r[n])}return t["default"]=r,x(t,"__useDefault",{value:!0}),t}function c(e,r,t){try{var n;(n=Object.getOwnPropertyDescriptor(r,t))&&x(e,t,n)}catch(o){return e[t]=r[t],!1}}function f(r,t){var n=v[r];if(n&&!n.evaluated&&n.declarative){t.push(r);for(var o=0,a=n.normalizedDeps.length;a>o;o++){var u=n.normalizedDeps[o];-1==g.call(t,u)&&(v[u]?f(u,t):p(u))}n.evaluated||(n.evaluated=!0,n.module.execute.call(e))}}function p(e){if(I[e])return I[e];if("@node/"==e.substr(0,6))return I[e]=s(D(e.substr(6)));var r=v[e];if(!r)throw"Module "+e+" not present.";return a(e),f(e,[]),v[e]=void 0,r.declarative&&x(r.module.exports,"__esModule",{value:!0}),I[e]=r.declarative?r.module.exports:r.esModule}var v={},g=Array.prototype.indexOf||function(e){for(var r=0,t=this.length;t>r;r++)if(this[r]===e)return r;return-1},m=!0;try{Object.getOwnPropertyDescriptor({a:0},"a")}catch(h){m=!1}var x;!function(){try{Object.defineProperty({},"a",{})&&(x=Object.defineProperty)}catch(e){x=function(e,r,t){try{e[r]=t.value||t.get.call(e)}catch(n){}}}}();var y={},D="undefined"!=typeof System&&System._nodeRequire||"undefined"!=typeof require&&"undefined"!=typeof require.resolve&&"undefined"!=typeof process&&process.platform&&require,I={"@empty":{}};return function(e,n,o,a){return function(u){u(function(u){for(var d={_nodeRequire:D,register:r,registerDynamic:t,get:p,set:function(e,r){I[e]=r},newModule:function(e){return e}},i=0;i<n.length;i++)(function(e,r){r&&r.__esModule?I[e]=r:I[e]=s(r)})(n[i],arguments[i]);a(d);var l=p(e[0]);if(e.length>1)for(var i=1;i<e.length;i++)p(e[i]);return o?l["default"]:l})}}}("undefined"!=typeof self?self:global)

(["1"], ["13","3","4","10","f"], true, function($__System) {
var require = this.require, exports = this.exports, module = this.module;
$__System.registerDynamic("2", ["3", "4", "5"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("4");
    var document_list_service_1 = $__require("5");
    var ERROR_FOLDER_ALREADY_EXIST = 409;
    var DocumentMenuActionComponent = function () {
        function DocumentMenuActionComponent(documentListService, translateService, logService) {
            this.documentListService = documentListService;
            this.translateService = translateService;
            this.logService = logService;
            this.success = new core_1.EventEmitter();
            this.error = new core_1.EventEmitter();
            this.actions = [];
            this.folderName = '';
            if (translateService) {
                translateService.addTranslationFolder('ng2-alfresco-documentlist', 'node_modules/ng2-alfresco-documentlist/src');
            }
        }
        DocumentMenuActionComponent.prototype.createFolder = function (name) {
            var _this = this;
            this.cancel();
            this.documentListService.createFolder(name, this.folderId).subscribe(function (res) {
                _this.folderName = '';
                _this.logService.info(res.entry);
                _this.success.emit({ node: res.entry });
            }, function (error) {
                var errorMessagePlaceholder = _this.getErrorMessage(error.response);
                if (errorMessagePlaceholder) {
                    _this.message = _this.formatString(errorMessagePlaceholder, [name]);
                    _this.error.emit({ message: _this.message });
                    _this.logService.error(_this.message);
                } else {
                    _this.error.emit(error);
                    _this.logService.error(error);
                }
            });
        };
        DocumentMenuActionComponent.prototype.showDialog = function () {
            if (!this.dialog.nativeElement.showModal) {
                dialogPolyfill.registerDialog(this.dialog.nativeElement);
            }
            this.dialog.nativeElement.showModal();
        };
        DocumentMenuActionComponent.prototype.cancel = function () {
            if (this.dialog) {
                this.dialog.nativeElement.close();
            }
        };
        DocumentMenuActionComponent.prototype.getErrorMessage = function (response) {
            if (response.body && response.body.error.statusCode === ERROR_FOLDER_ALREADY_EXIST) {
                var errorMessage = void 0;
                errorMessage = this.translateService.get('FILE_UPLOAD.MESSAGES.FOLDER_ALREADY_EXIST');
                return errorMessage.value;
            }
        };
        DocumentMenuActionComponent.prototype.formatString = function (message, keys) {
            var i = keys.length;
            while (i--) {
                message = message.replace(new RegExp('\\{' + i + '\\}', 'gm'), keys[i]);
            }
            return message;
        };
        DocumentMenuActionComponent.prototype.isFolderNameEmpty = function () {
            return this.folderName === '' ? true : false;
        };
        return DocumentMenuActionComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String)], DocumentMenuActionComponent.prototype, "folderId", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], DocumentMenuActionComponent.prototype, "success", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], DocumentMenuActionComponent.prototype, "error", void 0);
    __decorate([core_1.ViewChild('dialog'), __metadata("design:type", Object)], DocumentMenuActionComponent.prototype, "dialog", void 0);
    DocumentMenuActionComponent = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-document-menu-action',
        styles: [".container {     display: -webkit-box;     display: -moz-box;     display: -ms-flexbox;     display: -webkit-flex;     display: flex;     -webkit-flex-direction: row;     flex-direction: row;     background-color: #fafafa;     border-bottom: 1px solid transparent;     border-top: 1px solid #e5e5e5;     -webkit-box-shadow: 0 2px 4px rgba(0,0,0,.2);     box-shadow: 0 2px 4px rgba(0,0,0,.2);     height: 53px;     position: relative;     -webkit-transition: height .35s cubic-bezier(0.4,0.0,1,1),border-color .4s;     transition: height .35s cubic-bezier(0.4,0.0,1,1),border-color .4s;     z-index: 5; }  .action {     max-width: 394px;     min-width: 150px; }  .action {     -webkit-align-items: flex-end;     align-items: flex-end;     display: -webkit-box;     display: -moz-box;     display: -ms-flexbox;     display: -webkit-flex;     display: flex;     -webkit-box-sizing: border-box;     box-sizing: border-box;     padding: 13px 0 11px 30px;     -webkit-transition: padding .35s cubic-bezier(0.4, 0.0, 1, 1);     transition: padding .35s cubic-bezier(0.4, 0.0, 1, 1); }  .mdl-menu__item-primary-content {     box-sizing: border-box;     display: -webkit-flex;     display: -ms-flexbox;     display: flex;     -webkit-align-items: center;     -ms-flex-align: center;     align-items: center; }  .mdl-menu__item-primary-content {     -webkit-order: 0;     -ms-flex-order: 0;     order: 0;     -webkit-flex-grow: 2;     -ms-flex-positive: 2;     flex-grow: 2;     text-decoration: none; }  .mdl-menu__item-primary-content {     box-sizing: border-box;     display: -webkit-flex;     display: -ms-flexbox;     display: flex;     -webkit-align-items: center;     -ms-flex-align: center;     align-items: center;  }  .mdl-menu__item-icon {     margin-right: 32px;     margin-top: 10px;     margin-left: 10px; }  .mdl-menu--bottom-left {     width: 200px; }  .mdl-menu__text {     float: right;     margin-right: 22px; }"],
        template: "<div class=\"container\">     <div class=\"action\">         <button id=\"actions\" class=\"mdl-button mdl-js-button mdl-button--raised\">             <i class=\"material-icons\">add</i> {{ 'BUTTON.ACTION_CREATE' | translate }}         </button>         <ul alfresco-mdl-menu class=\"mdl-menu--bottom-left\"             [attr.for]=\"'actions'\">             <li class=\"mdl-menu__item\"                 (click)=\"showDialog()\" >                 <i style=\"float: left;\" class=\"material-icons mdl-menu__item-icon\">folder</i>                 <span class=\"mdl-menu__text\">{{ 'BUTTON.ACTION_NEW_FOLDER' | translate }}</span>             </li>         </ul>     </div> </div>  <dialog class=\"mdl-dialog\" #dialog>     <h4 class=\"mdl-dialog__title\">{{ 'BUTTON.ACTION_NEW_FOLDER' | translate }}</h4>     <div class=\"mdl-dialog__content\">         <div class=\"mdl-textfield mdl-js-textfield mdl-textfield--floating-label\">             <input                     type=\"text\"                     class=\"mdl-textfield__input\"                     id=\"name\"                     required                     [(ngModel)]=\"folderName\"                     placeholder=\"Folder name\"                     data-automation-id=\"name\"                     autocapitalize=\"none\" #name/>         </div>     </div>     <div class=\"mdl-dialog__actions\">         <button type=\"button\" [disabled]=\"isFolderNameEmpty()\" (click)=\"createFolder(folderName)\" class=\"mdl-button\">{{ 'BUTTON.CREATE' | translate }}</button>         <button type=\"button\" (click)=\"cancel()\" class=\"mdl-button close\">{{ 'BUTTON.CANCEL' | translate}}</button>     </div> </dialog>"
    }), __metadata("design:paramtypes", [document_list_service_1.DocumentListService, ng2_alfresco_core_1.AlfrescoTranslationService, ng2_alfresco_core_1.LogService])], DocumentMenuActionComponent);
    exports.DocumentMenuActionComponent = DocumentMenuActionComponent;
});
$__System.registerDynamic("6", ["3", "7"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var content_column_list_component_1 = $__require("7");
    var ContentColumnComponent = function () {
        function ContentColumnComponent(list) {
            this.list = list;
            this.type = 'text';
            this.sortable = false;
            this.title = '';
        }
        ContentColumnComponent.prototype.ngOnInit = function () {
            if (!this.srTitle && this.key === '$thumbnail') {
                this.srTitle = 'Thumbnail';
            }
        };
        ContentColumnComponent.prototype.ngAfterContentInit = function () {
            this.register();
        };
        ContentColumnComponent.prototype.register = function () {
            if (this.list) {
                return this.list.registerColumn(this);
            }
            return false;
        };
        return ContentColumnComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentColumnComponent.prototype, "key", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentColumnComponent.prototype, "type", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentColumnComponent.prototype, "format", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], ContentColumnComponent.prototype, "sortable", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentColumnComponent.prototype, "title", void 0);
    __decorate([core_1.ContentChild(core_1.TemplateRef), __metadata("design:type", Object)], ContentColumnComponent.prototype, "template", void 0);
    __decorate([core_1.Input('sr-title'), __metadata("design:type", String)], ContentColumnComponent.prototype, "srTitle", void 0);
    __decorate([core_1.Input('class'), __metadata("design:type", String)], ContentColumnComponent.prototype, "cssClass", void 0);
    ContentColumnComponent = __decorate([core_1.Component({
        selector: 'content-column',
        template: ''
    }), __metadata("design:paramtypes", [content_column_list_component_1.ContentColumnListComponent])], ContentColumnComponent);
    exports.ContentColumnComponent = ContentColumnComponent;
});
$__System.registerDynamic("7", ["3", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var document_list_component_1 = $__require("8");
    var ContentColumnListComponent = function () {
        function ContentColumnListComponent(documentList) {
            this.documentList = documentList;
        }
        ContentColumnListComponent.prototype.registerColumn = function (column) {
            if (this.documentList && column) {
                var columns = this.documentList.data.getColumns();
                columns.push(column);
                return true;
            }
            return false;
        };
        return ContentColumnListComponent;
    }();
    ContentColumnListComponent = __decorate([core_1.Component({
        selector: 'content-columns',
        template: ''
    }), __metadata("design:paramtypes", [document_list_component_1.DocumentListComponent])], ContentColumnListComponent);
    exports.ContentColumnListComponent = ContentColumnListComponent;
});
$__System.registerDynamic("9", ["3", "a", "b", "c", "d"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var content_action_list_component_1 = $__require("a");
    var content_action_model_1 = $__require("b");
    var document_actions_service_1 = $__require("c");
    var folder_actions_service_1 = $__require("d");
    var ContentActionComponent = function () {
        function ContentActionComponent(list, documentActions, folderActions) {
            this.list = list;
            this.documentActions = documentActions;
            this.folderActions = folderActions;
            this.title = 'Action';
            this.execute = new core_1.EventEmitter();
            this.model = new content_action_model_1.ContentActionModel();
        }
        ContentActionComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.model = new content_action_model_1.ContentActionModel({
                title: this.title,
                icon: this.icon,
                target: this.target
            });
            if (this.handler) {
                this.model.handler = this.getSystemHandler(this.target, this.handler);
            } else if (this.execute) {
                this.model.handler = function (document) {
                    _this.execute.emit({
                        value: document
                    });
                };
            }
            this.register();
        };
        ContentActionComponent.prototype.register = function () {
            if (this.list) {
                return this.list.registerAction(this.model);
            }
            return false;
        };
        ContentActionComponent.prototype.ngOnChanges = function (changes) {
            this.model.title = this.title;
        };
        ContentActionComponent.prototype.getSystemHandler = function (target, name) {
            if (target) {
                var ltarget = target.toLowerCase();
                if (ltarget === 'document') {
                    if (this.documentActions) {
                        return this.documentActions.getHandler(name);
                    }
                    return null;
                }
                if (ltarget === 'folder') {
                    if (this.folderActions) {
                        return this.folderActions.getHandler(name);
                    }
                    return null;
                }
            }
            return null;
        };
        return ContentActionComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentActionComponent.prototype, "title", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentActionComponent.prototype, "icon", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentActionComponent.prototype, "handler", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], ContentActionComponent.prototype, "target", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], ContentActionComponent.prototype, "execute", void 0);
    ContentActionComponent = __decorate([core_1.Component({
        selector: 'content-action',
        template: ''
    }), __metadata("design:paramtypes", [content_action_list_component_1.ContentActionListComponent, document_actions_service_1.DocumentActionsService, folder_actions_service_1.FolderActionsService])], ContentActionComponent);
    exports.ContentActionComponent = ContentActionComponent;
});
$__System.registerDynamic("a", ["3", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var document_list_component_1 = $__require("8");
    var ContentActionListComponent = function () {
        function ContentActionListComponent(documentList) {
            this.documentList = documentList;
        }
        ContentActionListComponent.prototype.registerAction = function (action) {
            if (this.documentList && action) {
                this.documentList.actions.push(action);
                return true;
            }
            return false;
        };
        return ContentActionListComponent;
    }();
    ContentActionListComponent = __decorate([core_1.Component({
        selector: 'content-actions',
        template: ''
    }), __metadata("design:paramtypes", [document_list_component_1.DocumentListComponent])], ContentActionListComponent);
    exports.ContentActionListComponent = ContentActionListComponent;
});
$__System.registerDynamic("e", ["3", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var document_list_component_1 = $__require("8");
    var EmptyFolderContentComponent = function () {
        function EmptyFolderContentComponent(documentList) {
            this.documentList = documentList;
        }
        EmptyFolderContentComponent.prototype.ngAfterContentInit = function () {
            this.documentList.emptyFolderTemplate = this.template;
            this.documentList.dataTable.noContentTemplate = this.template;
        };
        return EmptyFolderContentComponent;
    }();
    __decorate([core_1.ContentChild(core_1.TemplateRef), __metadata("design:type", Object)], EmptyFolderContentComponent.prototype, "template", void 0);
    EmptyFolderContentComponent = __decorate([core_1.Directive({
        selector: 'empty-folder-content'
    }), __metadata("design:paramtypes", [document_list_component_1.DocumentListComponent])], EmptyFolderContentComponent);
    exports.EmptyFolderContentComponent = EmptyFolderContentComponent;
});
$__System.registerDynamic("8", ["3", "f", "4", "10", "5", "11"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("f");
    var ng2_alfresco_core_1 = $__require("4");
    var ng2_alfresco_datatable_1 = $__require("10");
    var document_list_service_1 = $__require("5");
    var share_datatable_adapter_1 = $__require("11");
    var DocumentListComponent = DocumentListComponent_1 = function () {
        function DocumentListComponent(documentListService, ngZone, translateService) {
            this.documentListService = documentListService;
            this.ngZone = ngZone;
            this.translateService = translateService;
            this.baseComponentPath = module.id.replace('/components/document-list.component.js', '');
            this.fallbackThumbnail = this.baseComponentPath + '/assets/images/ft_ic_miscellaneous.svg';
            this.navigate = true;
            this.navigationMode = DocumentListComponent_1.DOUBLE_CLICK_NAVIGATION;
            this.thumbnails = false;
            this.multiselect = false;
            this.enablePagination = true;
            this.contentActions = false;
            this.contextMenuActions = false;
            this.creationMenuActions = true;
            this.pageSize = DocumentListComponent_1.DEFAULT_PAGE_SIZE;
            this.skipCount = 0;
            this.currentFolderId = null;
            this.folderNode = null;
            this.node = null;
            this.nodeClick = new core_1.EventEmitter();
            this.nodeDblClick = new core_1.EventEmitter();
            this.folderChange = new core_1.EventEmitter();
            this.preview = new core_1.EventEmitter();
            this.success = new core_1.EventEmitter();
            this.error = new core_1.EventEmitter();
            this.actions = [];
            this.contextActionHandler = new Rx_1.Subject();
            this.data = new share_datatable_adapter_1.ShareDataTableAdapter(this.documentListService, this.baseComponentPath, []);
            if (translateService) {
                translateService.addTranslationFolder('ng2-alfresco-documentlist', 'node_modules/ng2-alfresco-documentlist/src');
            }
        }
        Object.defineProperty(DocumentListComponent.prototype, "rowFilter", {
            set: function (value) {
                if (this.data && value && this.currentFolderId) {
                    this.data.setFilter(value);
                    this.loadFolderNodesByFolderNodeId(this.currentFolderId, this.pageSize, this.skipCount);
                }
            },
            enumerable: true,
            configurable: true
        });
        ;
        Object.defineProperty(DocumentListComponent.prototype, "imageResolver", {
            set: function (value) {
                if (this.data) {
                    this.data.setImageResolver(value);
                }
            },
            enumerable: true,
            configurable: true
        });
        DocumentListComponent.prototype.getContextActions = function (node) {
            var _this = this;
            if (node && node.entry) {
                var actions = this.getNodeActions(node);
                if (actions && actions.length > 0) {
                    return actions.map(function (a) {
                        return {
                            model: a,
                            node: node,
                            subject: _this.contextActionHandler
                        };
                    });
                }
            }
            return null;
        };
        DocumentListComponent.prototype.contextActionCallback = function (action) {
            if (action) {
                this.executeContentAction(action.node, action.model);
            }
        };
        DocumentListComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.data.thumbnails = this.thumbnails;
            this.contextActionHandler.subscribe(function (val) {
                return _this.contextActionCallback(val);
            });
            this.enforceSingleClickNavigationForMobile();
        };
        DocumentListComponent.prototype.enforceSingleClickNavigationForMobile = function () {
            if (this.isMobile()) {
                this.navigationMode = DocumentListComponent_1.SINGLE_CLICK_NAVIGATION;
            }
        };
        DocumentListComponent.prototype.ngAfterContentInit = function () {
            var columns = this.data.getColumns();
            if (!columns || columns.length === 0) {
                this.setupDefaultColumns();
            }
        };
        DocumentListComponent.prototype.ngOnChanges = function (changes) {
            if (changes['folderNode'] && changes['folderNode'].currentValue) {
                this.loadFolder();
                return;
            }
            if (changes['currentFolderId'] && changes['currentFolderId'].currentValue) {
                this.loadFolderByNodeId(changes['currentFolderId'].currentValue);
                return;
            }
            if (changes['node'] && changes['node'].currentValue) {
                this.data.loadPage(changes['node'].currentValue);
                return;
            }
        };
        DocumentListComponent.prototype.reload = function () {
            var _this = this;
            this.ngZone.run(function () {
                if (_this.folderNode) {
                    _this.loadFolder();
                    return;
                }
                if (_this.currentFolderId) {
                    _this.loadFolderByNodeId(_this.currentFolderId);
                    return;
                }
                if (_this.node) {
                    _this.data.loadPage(_this.node);
                    return;
                }
            });
        };
        DocumentListComponent.prototype.isEmptyTemplateDefined = function () {
            if (this.dataTable) {
                if (this.emptyFolderTemplate) {
                    return true;
                }
            }
            return false;
        };
        DocumentListComponent.prototype.isMobile = function () {
            return !!/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
        };
        DocumentListComponent.prototype.getNodeActions = function (node) {
            var target = null;
            if (node && node.entry) {
                if (node.entry.isFile) {
                    target = 'document';
                }
                if (node.entry.isFolder) {
                    target = 'folder';
                }
                if (target) {
                    var ltarget_1 = target.toLowerCase();
                    return this.actions.filter(function (entry) {
                        return entry.target.toLowerCase() === ltarget_1;
                    });
                }
            }
            return [];
        };
        DocumentListComponent.prototype.onShowContextMenu = function (e) {
            if (e) {
                e.preventDefault();
            }
        };
        DocumentListComponent.prototype.performNavigation = function (node) {
            if (node && node.entry && node.entry.isFolder) {
                this.currentFolderId = node.entry.id;
                this.folderNode = node.entry;
                this.loadFolder();
                this.folderChange.emit({ node: node.entry });
                return true;
            }
            return false;
        };
        DocumentListComponent.prototype.executeContentAction = function (node, action) {
            if (node && node.entry && action) {
                action.handler(node, this);
            }
        };
        DocumentListComponent.prototype.loadFolder = function () {
            var _this = this;
            var nodeId = this.folderNode ? this.folderNode.id : this.currentFolderId;
            if (nodeId) {
                this.loadFolderNodesByFolderNodeId(nodeId, this.pageSize, this.skipCount).catch(function (err) {
                    return _this.error.emit(err);
                });
            }
        };
        DocumentListComponent.prototype.loadFolderByNodeId = function (nodeId) {
            var _this = this;
            this.documentListService.getFolderNode(nodeId).then(function (node) {
                _this.folderNode = node;
                _this.currentFolderId = node.id;
                _this.loadFolderNodesByFolderNodeId(node.id, _this.pageSize, _this.skipCount).catch(function (err) {
                    return _this.error.emit(err);
                });
            }).catch(function (err) {
                return _this.error.emit(err);
            });
        };
        DocumentListComponent.prototype.loadFolderNodesByFolderNodeId = function (id, maxItems, skipCount) {
            var _this = this;
            return new Promise(function (resolve, reject) {
                if (id && _this.documentListService) {
                    _this.documentListService.getFolder(null, {
                        maxItems: maxItems,
                        skipCount: skipCount,
                        rootFolderId: id
                    }).subscribe(function (val) {
                        _this.data.loadPage(val);
                        _this.pagination = val.list.pagination;
                        resolve(true);
                    }, function (error) {
                        reject(error);
                    });
                } else {
                    resolve(false);
                }
            });
        };
        DocumentListComponent.prototype.setupDefaultColumns = function () {
            var colThumbnail = new ng2_alfresco_datatable_1.ObjectDataColumn({
                type: 'image',
                key: '$thumbnail',
                title: '',
                srTitle: 'Thumbnail'
            });
            var colName = new ng2_alfresco_datatable_1.ObjectDataColumn({
                type: 'text',
                key: 'name',
                title: 'Name',
                cssClass: 'full-width',
                sortable: true
            });
            this.data.setColumns([colThumbnail, colName]);
        };
        DocumentListComponent.prototype.onPreviewFile = function (node) {
            if (node) {
                this.preview.emit({
                    value: node
                });
            }
        };
        DocumentListComponent.prototype.onNodeClick = function (node) {
            this.nodeClick.emit({
                value: node
            });
            if (this.navigate && this.navigationMode === DocumentListComponent_1.SINGLE_CLICK_NAVIGATION) {
                if (node && node.entry) {
                    if (node.entry.isFile) {
                        this.onPreviewFile(node);
                    }
                    if (node.entry.isFolder) {
                        this.performNavigation(node);
                    }
                }
            }
        };
        DocumentListComponent.prototype.onRowClick = function (event) {
            var item = event.value.node;
            this.onNodeClick(item);
        };
        DocumentListComponent.prototype.onNodeDblClick = function (node) {
            this.nodeDblClick.emit({
                value: node
            });
            if (this.navigate && this.navigationMode === DocumentListComponent_1.DOUBLE_CLICK_NAVIGATION) {
                if (node && node.entry) {
                    if (node.entry.isFile) {
                        this.onPreviewFile(node);
                    }
                    if (node.entry.isFolder) {
                        this.performNavigation(node);
                    }
                }
            }
        };
        DocumentListComponent.prototype.onRowDblClick = function (event) {
            var item = event.value.node;
            this.onNodeDblClick(item);
        };
        DocumentListComponent.prototype.onShowRowContextMenu = function (event) {
            if (this.contextMenuActions) {
                var args = event.args;
                var node = args.row.node;
                if (node) {
                    args.actions = this.getContextActions(node) || [];
                }
            }
        };
        DocumentListComponent.prototype.onShowRowActionsMenu = function (event) {
            if (this.contentActions) {
                var args = event.args;
                var node = args.row.node;
                if (node) {
                    args.actions = this.getNodeActions(node) || [];
                }
            }
        };
        DocumentListComponent.prototype.onExecuteRowAction = function (event) {
            if (this.contentActions) {
                var args = event.args;
                var node = args.row.node;
                var action = args.action;
                this.executeContentAction(node, action);
            }
        };
        DocumentListComponent.prototype.onActionMenuError = function (event) {
            this.error.emit(event);
        };
        DocumentListComponent.prototype.onActionMenuSuccess = function (event) {
            this.reload();
            this.success.emit(event);
        };
        DocumentListComponent.prototype.onChangePageSize = function (event) {
            this.pageSize = event.maxItems;
            this.reload();
        };
        DocumentListComponent.prototype.onNextPage = function (event) {
            this.skipCount = event.skipCount;
            this.reload();
        };
        DocumentListComponent.prototype.onPrevPage = function (event) {
            this.skipCount = event.skipCount;
            this.reload();
        };
        return DocumentListComponent;
    }();
    DocumentListComponent.SINGLE_CLICK_NAVIGATION = 'click';
    DocumentListComponent.DOUBLE_CLICK_NAVIGATION = 'dblclick';
    DocumentListComponent.DEFAULT_PAGE_SIZE = 20;
    __decorate([core_1.Input(), __metadata("design:type", String)], DocumentListComponent.prototype, "fallbackThumbnail", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DocumentListComponent.prototype, "navigate", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], DocumentListComponent.prototype, "navigationMode", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DocumentListComponent.prototype, "thumbnails", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DocumentListComponent.prototype, "multiselect", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DocumentListComponent.prototype, "enablePagination", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DocumentListComponent.prototype, "contentActions", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DocumentListComponent.prototype, "contextMenuActions", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], DocumentListComponent.prototype, "creationMenuActions", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Number)], DocumentListComponent.prototype, "pageSize", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Function), __metadata("design:paramtypes", [Function])], DocumentListComponent.prototype, "rowFilter", null);
    __decorate([core_1.Input(), __metadata("design:type", Function), __metadata("design:paramtypes", [Function])], DocumentListComponent.prototype, "imageResolver", null);
    __decorate([core_1.Input(), __metadata("design:type", String)], DocumentListComponent.prototype, "currentFolderId", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Object)], DocumentListComponent.prototype, "folderNode", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Object)], DocumentListComponent.prototype, "node", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DocumentListComponent.prototype, "nodeClick", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DocumentListComponent.prototype, "nodeDblClick", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DocumentListComponent.prototype, "folderChange", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DocumentListComponent.prototype, "preview", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DocumentListComponent.prototype, "success", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DocumentListComponent.prototype, "error", void 0);
    __decorate([core_1.ViewChild(ng2_alfresco_datatable_1.DataTableComponent), __metadata("design:type", ng2_alfresco_datatable_1.DataTableComponent)], DocumentListComponent.prototype, "dataTable", void 0);
    __decorate([core_1.HostListener('contextmenu', ['$event']), __metadata("design:type", Function), __metadata("design:paramtypes", [Event]), __metadata("design:returntype", void 0)], DocumentListComponent.prototype, "onShowContextMenu", null);
    DocumentListComponent = DocumentListComponent_1 = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-document-list',
        styles: [".document-list_empty_template {     text-align: center;     margin-top: 20px;     margin-bottom: 20px; }  .document-list__this-space-is-empty {     height: 32px;     opacity: 0.26;     font-family: Muli, Helvetica, Arial, sans-serif;     font-size: 24px;     line-height: 1.33;     letter-spacing: -1px;     color: #000000; }  .document-list__drag-drop {     height: 56px;     opacity: 0.54;     font-family: Muli, Helvetica, Arial, sans-serif;     font-size: 56px;     line-height: 1;     letter-spacing: -2px;     color: #000000;     margin-top: 40px; }  .document-list__any-files-here-to-add {     height: 24px;     opacity: 0.54;     font-family: Muli, Helvetica, Arial, sans-serif;     font-size: 16px;     line-height: 1.5;     letter-spacing: -0.4px;     color: #000000;     margin-top: 17px; }  .document-list__empty_doc_lib {     width: 565px;     height: 161px;     object-fit: contain;     margin-top: 17px; }"],
        template: "<alfresco-document-menu-action     *ngIf=\"creationMenuActions\"     [folderId]=\"currentFolderId\"     (success)=\"onActionMenuSuccess($event)\"     (error)=\"onActionMenuError($event)\"> </alfresco-document-menu-action> <alfresco-datatable     [data]=\"data\"     [actions]=\"contentActions\"     [multiselect]=\"multiselect\"     [fallbackThumbnail]=\"fallbackThumbnail\"     (showRowContextMenu)=\"onShowRowContextMenu($event)\"     (showRowActionsMenu)=\"onShowRowActionsMenu($event)\"     (executeRowAction)=\"onExecuteRowAction($event)\"     (rowClick)=\"onRowClick($event)\"     (rowDblClick)=\"onRowDblClick($event)\">     <div *ngIf=\"!isEmptyTemplateDefined()\">         <no-content-template>             <template>                 <div class=\"document-list_empty_template\">                     <div class=\"document-list__this-space-is-empty\">This folder is empty</div>                     <div class=\"document-list__drag-drop\">Drag and Drop</div>                     <div class=\"document-list__any-files-here-to-add\">any files here to add</div>                     <img [src]=\"baseComponentPath + '/assets/images/empty_doc_lib.svg'\" class=\"document-list__empty_doc_lib\">                 </div>             </template>         </no-content-template>     </div> </alfresco-datatable> <alfresco-pagination *ngIf=\"enablePagination\"     (changePageSize)=\"onChangePageSize($event)\"     (nextPage)=\"onNextPage($event)\"     (prevPage)=\"onPrevPage($event)\"     [pagination]=\"pagination\"     [supportedPageSizes]=\"[5, 10, 15, 20]\"> </alfresco-pagination>"
    }), __metadata("design:paramtypes", [document_list_service_1.DocumentListService, core_1.NgZone, ng2_alfresco_core_1.AlfrescoTranslationService])], DocumentListComponent);
    exports.DocumentListComponent = DocumentListComponent;
    var DocumentListComponent_1;
});
$__System.registerDynamic("12", ["3", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var document_list_component_1 = $__require("8");
    var DocumentListBreadcrumbComponent = function () {
        function DocumentListBreadcrumbComponent() {
            this.route = [];
            this.navigate = new core_1.EventEmitter();
        }
        DocumentListBreadcrumbComponent.prototype.ngOnChanges = function (changes) {
            if (changes['folderNode']) {
                var node = changes['folderNode'].currentValue;
                if (node) {
                    var route = node.path.elements || [];
                    route.push({
                        id: node.id,
                        name: node.name
                    });
                    this.route = route;
                }
            }
        };
        DocumentListBreadcrumbComponent.prototype.onRoutePathClick = function (route, e) {
            if (e) {
                e.preventDefault();
            }
            if (route) {
                this.navigate.emit({
                    value: route
                });
                if (this.target) {
                    this.target.loadFolderByNodeId(route.id);
                }
            }
        };
        return DocumentListBreadcrumbComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", Object)], DocumentListBreadcrumbComponent.prototype, "folderNode", void 0);
    __decorate([core_1.Input(), __metadata("design:type", document_list_component_1.DocumentListComponent)], DocumentListBreadcrumbComponent.prototype, "target", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], DocumentListBreadcrumbComponent.prototype, "navigate", void 0);
    DocumentListBreadcrumbComponent = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-document-list-breadcrumb',
        template: "<div>     <ol *ngIf=\"folderNode\" data-automation-id=\"breadcrumb\" class=\"breadcrumb\">         <li *ngFor=\"let r of route; let last = last\" [class.active]=\"last\" [ngSwitch]=\"last\">             <span *ngSwitchCase=\"true\">{{r.name}}</span>             <a *ngSwitchDefault href=\"#\" [attr.data-automation-id]=\"'breadcrumb_' + r.name\" (click)=\"onRoutePathClick(r, $event)\">                 {{r.name}}             </a>         </li>     </ol> </div>",
        styles: ["/* breadcrumb */  :host .breadcrumb {     text-align: left;     padding: 8px 15px;     list-style: none;     background-color: #fafafa;     margin: 0 0 4px; }  :host .breadcrumb > li {     display: inline-block;     box-sizing: border-box; }  :host .breadcrumb > li+li:before {     content: \">\\00a0\";     padding: 0 0 0 5px;     opacity: 0.54;     color: #000000; }  :host .breadcrumb > li > a {     text-decoration: none;     opacity: 0.54;     font-family: 'Muli', \"Helvetica\", \"Arial\", sans-serif;     font-size: 14px;     font-weight: 600;     line-height: 1.43;     letter-spacing: -0.2px;     color: #000000; }  :host .breadcrumb > li:hover > a, :host .breadcrumb > .active {     opacity: 0.87;     font-size: 14px;     font-weight: 600;     line-height: 1.43;     letter-spacing: -0.2px;     color: #000000; }"]
    })], DocumentListBreadcrumbComponent);
    exports.DocumentListBreadcrumbComponent = DocumentListBreadcrumbComponent;
});
$__System.registerDynamic("d", ["3", "5"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var document_list_service_1 = $__require("5");
    var FolderActionsService = function () {
        function FolderActionsService(documentListService) {
            this.documentListService = documentListService;
            this.handlers = {};
            this.setupActionHandlers();
        }
        FolderActionsService.prototype.getHandler = function (key) {
            if (key) {
                var lkey = key.toLowerCase();
                return this.handlers[lkey] || null;
            }
            return null;
        };
        FolderActionsService.prototype.setHandler = function (key, handler) {
            if (key) {
                var lkey = key.toLowerCase();
                this.handlers[lkey] = handler;
                return true;
            }
            return false;
        };
        FolderActionsService.prototype.canExecuteAction = function (obj) {
            return this.documentListService && obj && obj.entry.isFolder === true;
        };
        FolderActionsService.prototype.setupActionHandlers = function () {
            this.handlers['delete'] = this.deleteNode.bind(this);
            this.handlers['system1'] = this.handleStandardAction1.bind(this);
            this.handlers['system2'] = this.handleStandardAction2.bind(this);
        };
        FolderActionsService.prototype.handleStandardAction1 = function (document) {
            window.alert('standard folder action 1');
        };
        FolderActionsService.prototype.handleStandardAction2 = function (document) {
            window.alert('standard folder action 2');
        };
        FolderActionsService.prototype.deleteNode = function (obj, target) {
            if (this.canExecuteAction(obj) && obj.entry && obj.entry.id) {
                this.documentListService.deleteNode(obj.entry.id).subscribe(function () {
                    if (target && typeof target.reload === 'function') {
                        target.reload();
                    }
                });
            }
        };
        return FolderActionsService;
    }();
    FolderActionsService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [document_list_service_1.DocumentListService])], FolderActionsService);
    exports.FolderActionsService = FolderActionsService;
});
$__System.registerDynamic("c", ["3", "5", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var document_list_service_1 = $__require("5");
    var ng2_alfresco_core_1 = $__require("4");
    var DocumentActionsService = function () {
        function DocumentActionsService(documentListService, contentService) {
            this.documentListService = documentListService;
            this.contentService = contentService;
            this.handlers = {};
            this.setupActionHandlers();
        }
        DocumentActionsService.prototype.getHandler = function (key) {
            if (key) {
                var lkey = key.toLowerCase();
                return this.handlers[lkey] || null;
            }
            return null;
        };
        DocumentActionsService.prototype.setHandler = function (key, handler) {
            if (key) {
                var lkey = key.toLowerCase();
                this.handlers[lkey] = handler;
                return true;
            }
            return false;
        };
        DocumentActionsService.prototype.canExecuteAction = function (obj) {
            return this.documentListService && obj && obj.entry.isFile === true;
        };
        DocumentActionsService.prototype.setupActionHandlers = function () {
            this.handlers['download'] = this.download.bind(this);
            this.handlers['delete'] = this.deleteNode.bind(this);
            this.handlers['system1'] = this.handleStandardAction1.bind(this);
            this.handlers['system2'] = this.handleStandardAction2.bind(this);
        };
        DocumentActionsService.prototype.handleStandardAction1 = function (obj) {
            window.alert('standard document action 1');
        };
        DocumentActionsService.prototype.handleStandardAction2 = function (obj) {
            window.alert('standard document action 2');
        };
        DocumentActionsService.prototype.download = function (obj) {
            if (this.canExecuteAction(obj) && this.contentService) {
                var link = document.createElement('a');
                document.body.appendChild(link);
                link.setAttribute('download', 'download');
                link.href = this.contentService.getContentUrl(obj);
                link.click();
                document.body.removeChild(link);
                return true;
            }
            return false;
        };
        DocumentActionsService.prototype.deleteNode = function (obj, target) {
            if (this.canExecuteAction(obj) && obj.entry && obj.entry.id) {
                this.documentListService.deleteNode(obj.entry.id).subscribe(function () {
                    if (target && typeof target.reload === 'function') {
                        target.reload();
                    }
                });
            }
        };
        return DocumentActionsService;
    }();
    DocumentActionsService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [document_list_service_1.DocumentListService, ng2_alfresco_core_1.AlfrescoContentService])], DocumentActionsService);
    exports.DocumentActionsService = DocumentActionsService;
});
$__System.registerDynamic("5", ["3", "f", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("f");
    var ng2_alfresco_core_1 = $__require("4");
    var DocumentListService = DocumentListService_1 = function () {
        function DocumentListService(authService, contentService, apiService, logService) {
            this.authService = authService;
            this.contentService = contentService;
            this.apiService = apiService;
            this.logService = logService;
            this.mimeTypeIcons = {
                'image/png': 'ft_ic_raster_image.svg',
                'image/jpeg': 'ft_ic_raster_image.svg',
                'image/gif': 'ft_ic_raster_image.svg',
                'application/pdf': 'ft_ic_pdf.svg',
                'application/vnd.ms-excel': 'ft_ic_ms_excel.svg',
                'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet': 'ft_ic_ms_excel.svg',
                'application/vnd.openxmlformats-officedocument.spreadsheetml.template': 'ft_ic_ms_excel.svg',
                'application/msword': 'ft_ic_ms_word.svg',
                'application/vnd.openxmlformats-officedocument.wordprocessingml.document': 'ft_ic_ms_word.svg',
                'application/vnd.openxmlformats-officedocument.wordprocessingml.template': 'ft_ic_ms_word.svg',
                'application/vnd.ms-powerpoint': 'ft_ic_ms_powerpoint.svg',
                'application/vnd.openxmlformats-officedocument.presentationml.presentation': 'ft_ic_ms_powerpoint.svg',
                'application/vnd.openxmlformats-officedocument.presentationml.template': 'ft_ic_ms_powerpoint.svg',
                'application/vnd.openxmlformats-officedocument.presentationml.slideshow': 'ft_ic_ms_powerpoint.svg',
                'video/mp4': 'ft_ic_video.svg',
                'text/plain': 'ft_ic_document.svg',
                'application/x-javascript': 'ft_ic_document.svg',
                'application/json': 'ft_ic_document.svg',
                'image/svg+xml': 'ft_ic_vector_image.svg',
                'text/html': 'ft_ic_website.svg',
                'application/x-compressed': 'ft_ic_archive.svg',
                'application/x-zip-compressed': 'ft_ic_archive.svg',
                'application/zip': 'ft_ic_archive.svg',
                'application/vnd.apple.keynote': 'ft_ic_presentation.svg',
                'application/vnd.apple.pages': 'ft_ic_document.svg',
                'application/vnd.apple.numbers': 'ft_ic_spreadsheet.svg'
            };
        }
        DocumentListService.prototype.getNodesPromise = function (folder, opts) {
            var rootNodeId = DocumentListService_1.ROOT_ID;
            if (opts && opts.rootFolderId) {
                rootNodeId = opts.rootFolderId;
            }
            var params = {
                includeSource: true,
                include: ['path', 'properties']
            };
            if (folder) {
                params.relativePath = folder;
            }
            if (opts) {
                if (opts.maxItems) {
                    params.maxItems = opts.maxItems;
                }
                if (opts.skipCount) {
                    params.skipCount = opts.skipCount;
                }
            }
            return this.apiService.getInstance().nodes.getNodeChildren(rootNodeId, params);
        };
        DocumentListService.prototype.deleteNode = function (nodeId) {
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().nodes.deleteNode(nodeId));
        };
        DocumentListService.prototype.createFolder = function (name, parentId) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().nodes.createFolder(name, '/', parentId)).catch(function (err) {
                return _this.handleError(err);
            });
        };
        DocumentListService.prototype.getFolder = function (folder, opts) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.getNodesPromise(folder, opts)).map(function (res) {
                return res;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        DocumentListService.prototype.getFolderNode = function (nodeId) {
            var opts = {
                includeSource: true,
                include: ['path', 'properties']
            };
            var nodes = this.apiService.getInstance().nodes;
            return nodes.getNodeInfo(nodeId, opts);
        };
        DocumentListService.prototype.getDocumentThumbnailUrl = function (node) {
            if (node && this.contentService) {
                return this.contentService.getDocumentThumbnailUrl(node);
            }
            return null;
        };
        DocumentListService.prototype.getMimeTypeIcon = function (mimeType) {
            var icon = this.mimeTypeIcons[mimeType];
            return icon || DocumentListService_1.DEFAULT_MIME_TYPE_ICON;
        };
        DocumentListService.prototype.handleError = function (error) {
            this.logService.error(error);
            return Rx_1.Observable.throw(error || 'Server error');
        };
        return DocumentListService;
    }();
    DocumentListService.DEFAULT_MIME_TYPE_ICON = 'ft_ic_miscellaneous.svg';
    DocumentListService.ROOT_ID = '-root-';
    DocumentListService = DocumentListService_1 = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoAuthenticationService, ng2_alfresco_core_1.AlfrescoContentService, ng2_alfresco_core_1.AlfrescoApiService, ng2_alfresco_core_1.LogService])], DocumentListService);
    exports.DocumentListService = DocumentListService;
    var DocumentListService_1;
});
$__System.registerDynamic("11", ["13", "4", "10"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var common_1 = $__require("13");
    var ng2_alfresco_core_1 = $__require("4");
    var ng2_alfresco_datatable_1 = $__require("10");
    var ShareDataTableAdapter = function () {
        function ShareDataTableAdapter(documentListService, basePath, schema) {
            this.documentListService = documentListService;
            this.basePath = basePath;
            this.ERR_ROW_NOT_FOUND = 'Row not found';
            this.ERR_COL_NOT_FOUND = 'Column not found';
            this.DEFAULT_DATE_FORMAT = 'medium';
            this.thumbnails = false;
            this.rows = [];
            this.columns = schema || [];
        }
        ShareDataTableAdapter.prototype.getRows = function () {
            return this.rows;
        };
        ShareDataTableAdapter.prototype.setRows = function (rows) {
            this.rows = rows || [];
            this.sort();
        };
        ShareDataTableAdapter.prototype.getColumns = function () {
            return this.columns;
        };
        ShareDataTableAdapter.prototype.setColumns = function (columns) {
            this.columns = columns || [];
        };
        ShareDataTableAdapter.prototype.getValue = function (row, col) {
            if (!row) {
                throw new Error(this.ERR_ROW_NOT_FOUND);
            }
            if (!col) {
                throw new Error(this.ERR_COL_NOT_FOUND);
            }
            var dataRow = row;
            var value = row.getValue(col.key);
            if (dataRow.cache[col.key] !== undefined) {
                return dataRow.cache[col.key];
            }
            if (col.type === 'date') {
                var datePipe = new common_1.DatePipe('en-US');
                var format = col.format || this.DEFAULT_DATE_FORMAT;
                try {
                    var result = datePipe.transform(value, format);
                    return dataRow.cacheValue(col.key, result);
                } catch (err) {
                    console.error("Error parsing date " + value + " to format " + format);
                    return 'Error';
                }
            }
            if (col.type === 'image') {
                if (this.imageResolver) {
                    var resolved = this.imageResolver(row, col);
                    if (resolved) {
                        return resolved;
                    }
                }
                if (col.key === '$thumbnail') {
                    var node = row.node;
                    if (node.entry.isFolder) {
                        return this.getImagePath('ft_ic_folder.svg');
                    }
                    if (node.entry.isFile) {
                        if (this.thumbnails) {
                            if (this.documentListService) {
                                return this.documentListService.getDocumentThumbnailUrl(node);
                            }
                            return null;
                        }
                        if (node.entry.content) {
                            var mimeType = node.entry.content.mimeType;
                            if (mimeType) {
                                var icon = this.documentListService.getMimeTypeIcon(mimeType);
                                if (icon) {
                                    return this.getImagePath(icon);
                                }
                            }
                        }
                    }
                    return this.getImagePath('ft_ic_miscellaneous.svg');
                }
            }
            return dataRow.cacheValue(col.key, value);
        };
        ShareDataTableAdapter.prototype.getSorting = function () {
            return this.sorting;
        };
        ShareDataTableAdapter.prototype.setSorting = function (sorting) {
            this.sorting = sorting;
            this.sortRows(this.rows, this.sorting);
        };
        ShareDataTableAdapter.prototype.sort = function (key, direction) {
            var sorting = this.sorting || new ng2_alfresco_datatable_1.DataSorting();
            if (key) {
                sorting.key = key;
                sorting.direction = direction || 'asc';
            }
            this.setSorting(sorting);
        };
        ShareDataTableAdapter.prototype.setFilter = function (filter) {
            this.filter = filter;
        };
        ShareDataTableAdapter.prototype.setImageResolver = function (resolver) {
            this.imageResolver = resolver;
        };
        ShareDataTableAdapter.prototype.sortRows = function (rows, sorting) {
            if (sorting && sorting.key && rows && rows.length > 0) {
                rows.sort(function (a, b) {
                    if (a.node.entry.isFolder !== b.node.entry.isFolder) {
                        return a.node.entry.isFolder ? -1 : 1;
                    }
                    var left = a.getValue(sorting.key);
                    if (left) {
                        left = left instanceof Date ? left.valueOf().toString() : left.toString();
                    } else {
                        left = '';
                    }
                    var right = b.getValue(sorting.key);
                    if (right) {
                        right = right instanceof Date ? right.valueOf().toString() : right.toString();
                    } else {
                        right = '';
                    }
                    return sorting.direction === 'asc' ? left.localeCompare(right) : right.localeCompare(left);
                });
            }
        };
        ShareDataTableAdapter.prototype.loadPage = function (page) {
            this.page = page;
            var rows = [];
            if (page && page.list) {
                var data = page.list.entries;
                if (data && data.length > 0) {
                    rows = data.map(function (item) {
                        return new ShareDataRow(item);
                    });
                    if (this.filter) {
                        rows = rows.filter(this.filter);
                    }
                    if (this.columns && this.columns.length > 0) {
                        var sorting = this.getSorting();
                        if (sorting) {
                            this.sortRows(rows, sorting);
                        } else {
                            var sortable = this.columns.filter(function (c) {
                                return c.sortable;
                            });
                            if (sortable.length > 0) {
                                this.sort(sortable[0].key, 'asc');
                            } else {
                                this.sort(this.columns[0].key, 'asc');
                            }
                        }
                    }
                }
            }
            this.rows = rows;
        };
        ShareDataTableAdapter.prototype.getImagePath = function (id) {
            return this.basePath + "/assets/images/" + id;
        };
        return ShareDataTableAdapter;
    }();
    exports.ShareDataTableAdapter = ShareDataTableAdapter;
    var ShareDataRow = function () {
        function ShareDataRow(obj) {
            this.obj = obj;
            this.cache = {};
            this.isSelected = false;
            if (!obj) {
                throw new Error(ShareDataRow.ERR_OBJECT_NOT_FOUND);
            }
        }
        Object.defineProperty(ShareDataRow.prototype, "node", {
            get: function () {
                return this.obj;
            },
            enumerable: true,
            configurable: true
        });
        ShareDataRow.prototype.cacheValue = function (key, value) {
            this.cache[key] = value;
            return value;
        };
        ShareDataRow.prototype.getValue = function (key) {
            if (this.cache[key] !== undefined) {
                return this.cache[key];
            }
            return ng2_alfresco_core_1.ObjectUtils.getValue(this.obj.entry, key);
        };
        ShareDataRow.prototype.hasValue = function (key) {
            return this.getValue(key) ? true : false;
        };
        return ShareDataRow;
    }();
    ShareDataRow.ERR_OBJECT_NOT_FOUND = 'Object source not found';
    exports.ShareDataRow = ShareDataRow;
});
$__System.registerDynamic('b', [], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __extends = this && this.__extends || function (d, b) {
        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
    var ContentActionModel = function () {
        function ContentActionModel(obj) {
            if (obj) {
                this.icon = obj.icon;
                this.title = obj.title;
                this.handler = obj.handler;
                this.target = obj.target;
            }
        }
        return ContentActionModel;
    }();
    exports.ContentActionModel = ContentActionModel;
    var DocumentActionModel = function (_super) {
        __extends(DocumentActionModel, _super);
        function DocumentActionModel(json) {
            var _this = _super.call(this, json) || this;
            _this.target = 'document';
            return _this;
        }
        return DocumentActionModel;
    }(ContentActionModel);
    exports.DocumentActionModel = DocumentActionModel;
    var FolderActionModel = function (_super) {
        __extends(FolderActionModel, _super);
        function FolderActionModel(json) {
            var _this = _super.call(this, json) || this;
            _this.target = 'folder';
            return _this;
        }
        return FolderActionModel;
    }(ContentActionModel);
    exports.FolderActionModel = FolderActionModel;
});
$__System.registerDynamic("14", [], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var NodePaging = function () {
        function NodePaging() {}
        return NodePaging;
    }();
    exports.NodePaging = NodePaging;
    var NodePagingList = function () {
        function NodePagingList() {}
        return NodePagingList;
    }();
    exports.NodePagingList = NodePagingList;
    var NodeMinimalEntry = function () {
        function NodeMinimalEntry() {}
        return NodeMinimalEntry;
    }();
    exports.NodeMinimalEntry = NodeMinimalEntry;
    var Pagination = function () {
        function Pagination() {}
        return Pagination;
    }();
    exports.Pagination = Pagination;
    var NodeMinimal = function () {
        function NodeMinimal() {
            this.properties = {};
        }
        return NodeMinimal;
    }();
    exports.NodeMinimal = NodeMinimal;
    var UserInfo = function () {
        function UserInfo() {}
        return UserInfo;
    }();
    exports.UserInfo = UserInfo;
    var ContentInfo = function () {
        function ContentInfo() {}
        return ContentInfo;
    }();
    exports.ContentInfo = ContentInfo;
    var PathInfoEntity = function () {
        function PathInfoEntity() {}
        return PathInfoEntity;
    }();
    exports.PathInfoEntity = PathInfoEntity;
    var PathElementEntity = function () {
        function PathElementEntity() {}
        return PathElementEntity;
    }();
    exports.PathElementEntity = PathElementEntity;
});
$__System.registerDynamic("1", ["3", "4", "10", "8", "2", "6", "7", "9", "a", "e", "12", "d", "c", "5", "11", "b", "14"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("4");
    var ng2_alfresco_datatable_1 = $__require("10");
    var document_list_component_1 = $__require("8");
    var document_menu_action_component_1 = $__require("2");
    var content_column_component_1 = $__require("6");
    var content_column_list_component_1 = $__require("7");
    var content_action_component_1 = $__require("9");
    var content_action_list_component_1 = $__require("a");
    var empty_folder_content_component_1 = $__require("e");
    var breadcrumb_component_1 = $__require("12");
    var folder_actions_service_1 = $__require("d");
    var document_actions_service_1 = $__require("c");
    var document_list_service_1 = $__require("5");
    __export($__require("8"));
    __export($__require("6"));
    __export($__require("7"));
    __export($__require("9"));
    __export($__require("a"));
    __export($__require("e"));
    __export($__require("12"));
    __export($__require("11"));
    __export($__require("d"));
    __export($__require("c"));
    __export($__require("5"));
    __export($__require("b"));
    __export($__require("14"));
    exports.DOCUMENT_LIST_DIRECTIVES = [document_list_component_1.DocumentListComponent, document_menu_action_component_1.DocumentMenuActionComponent, content_column_component_1.ContentColumnComponent, content_column_list_component_1.ContentColumnListComponent, content_action_component_1.ContentActionComponent, content_action_list_component_1.ContentActionListComponent, empty_folder_content_component_1.EmptyFolderContentComponent, breadcrumb_component_1.DocumentListBreadcrumbComponent];
    exports.DOCUMENT_LIST_PROVIDERS = [document_list_service_1.DocumentListService, folder_actions_service_1.FolderActionsService, document_actions_service_1.DocumentActionsService];
    var DocumentListModule = DocumentListModule_1 = function () {
        function DocumentListModule() {}
        DocumentListModule.forRoot = function () {
            return {
                ngModule: DocumentListModule_1,
                providers: exports.DOCUMENT_LIST_PROVIDERS.slice()
            };
        };
        return DocumentListModule;
    }();
    DocumentListModule = DocumentListModule_1 = __decorate([core_1.NgModule({
        imports: [ng2_alfresco_core_1.CoreModule, ng2_alfresco_datatable_1.DataTableModule],
        declarations: exports.DOCUMENT_LIST_DIRECTIVES.slice(),
        providers: exports.DOCUMENT_LIST_PROVIDERS.slice(),
        exports: [ng2_alfresco_datatable_1.DataTableModule].concat(exports.DOCUMENT_LIST_DIRECTIVES)
    })], DocumentListModule);
    exports.DocumentListModule = DocumentListModule;
    var DocumentListModule_1;
    
});
})
(function(factory) {
  if (typeof define == 'function' && define.amd)
    define(["@angular/common","@angular/core","ng2-alfresco-core","ng2-alfresco-datatable","rxjs/Rx"], factory);
  else if (typeof module == 'object' && module.exports && typeof require == 'function')
    module.exports = factory(require("@angular/common"), require("@angular/core"), require("ng2-alfresco-core"), require("ng2-alfresco-datatable"), require("rxjs/Rx"));
  else
    throw new Error("Module must be loaded as AMD or CommonJS");
});
