// ng2-activiti-analytics/src/components/analytics.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var analytics_service_1 = require("../services/analytics.service");
var analytics_generator_component_1 = require("./analytics-generator.component");
var AnalyticsComponent = (function () {
    function AnalyticsComponent(translateService, analyticsService, logService) {
        this.translateService = translateService;
        this.analyticsService = analyticsService;
        this.logService = logService;
        this.debug = false;
        this.editReport = new core_1.EventEmitter();
        logService.info('AnalyticsComponent');
        if (translateService) {
            translateService.addTranslationFolder('ng2-activiti-analytics', 'node_modules/ng2-activiti-analytics/src');
        }
    }
    AnalyticsComponent.prototype.ngOnChanges = function (changes) {
        this.analyticsgenerator.reset();
    };
    AnalyticsComponent.prototype.showReport = function ($event) {
        this.analyticsgenerator.generateReport(this.reportId, $event);
    };
    AnalyticsComponent.prototype.reset = function () {
        this.analyticsgenerator.reset();
    };
    AnalyticsComponent.prototype.onEditReport = function (name) {
        this.editReport.emit(name);
    };
    return AnalyticsComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", Number)
], AnalyticsComponent.prototype, "appId", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Number)
], AnalyticsComponent.prototype, "reportId", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], AnalyticsComponent.prototype, "debug", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], AnalyticsComponent.prototype, "editReport", void 0);
__decorate([
    core_1.ViewChild('analyticsgenerator'),
    __metadata("design:type", analytics_generator_component_1.AnalyticsGeneratorComponent)
], AnalyticsComponent.prototype, "analyticsgenerator", void 0);
AnalyticsComponent = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'activiti-analytics',
        template: "<div class=\"col-md-6\">     <analytics-report-parameters [appId]=\"appId\" [reportId]=\"reportId\"                                  (onFormValueChanged)=\"reset()\"                                  (onSuccess)=\"showReport($event)\"                                  (onEdit)=\"onEditReport($event)\">     </analytics-report-parameters>      <activiti-analytics-generator [reportId]=\"reportId\"                                   [reportParamQuery]=\"reportParamQuery\"                                   #analyticsgenerator>     </activiti-analytics-generator> </div>",
        styles: [".chart {display: block; width: 100%;}"]
    }),
    __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoTranslationService,
        analytics_service_1.AnalyticsService,
        ng2_alfresco_core_1.LogService])
], AnalyticsComponent);
exports.AnalyticsComponent = AnalyticsComponent;
