// ng2-activiti-analytics/src/components/analytics-report-list.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var Rx_1 = require("rxjs/Rx");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var analytics_service_1 = require("../services/analytics.service");
var AnalyticsReportListComponent = (function () {
    function AnalyticsReportListComponent(analyticsService, logService) {
        var _this = this;
        this.analyticsService = analyticsService;
        this.logService = logService;
        this.reportClick = new core_1.EventEmitter();
        this.onSuccess = new core_1.EventEmitter();
        this.onError = new core_1.EventEmitter();
        this.reports = [];
        this.report$ = new Rx_1.Observable(function (observer) { return _this.reportObserver = observer; }).share();
    }
    AnalyticsReportListComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.report$.subscribe(function (report) {
            _this.reports.push(report);
        });
        this.getReportList();
    };
    AnalyticsReportListComponent.prototype.reload = function () {
        this.reset();
        this.getReportList();
    };
    AnalyticsReportListComponent.prototype.getReportList = function () {
        var _this = this;
        this.analyticsService.getReportList().subscribe(function (res) {
            if (res && res.length === 0) {
                _this.createDefaultReports();
            }
            else {
                res.forEach(function (report) {
                    _this.reportObserver.next(report);
                });
                _this.onSuccess.emit(res);
            }
        }, function (err) {
            _this.onError.emit(err);
            _this.logService.error(err);
        });
    };
    AnalyticsReportListComponent.prototype.createDefaultReports = function () {
        var _this = this;
        this.analyticsService.createDefaultReports().subscribe(function () {
            _this.analyticsService.getReportList().subscribe(function (response) {
                response.forEach(function (report) {
                    _this.reportObserver.next(report);
                });
                _this.onSuccess.emit(response);
            });
        });
    };
    AnalyticsReportListComponent.prototype.isReportsEmpty = function () {
        return this.reports === undefined || (this.reports && this.reports.length === 0);
    };
    AnalyticsReportListComponent.prototype.reset = function () {
        if (!this.isReportsEmpty()) {
            this.reports = [];
        }
    };
    AnalyticsReportListComponent.prototype.selectReport = function (report) {
        this.currentReport = report;
        this.reportClick.emit(report);
    };
    return AnalyticsReportListComponent;
}());
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], AnalyticsReportListComponent.prototype, "reportClick", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], AnalyticsReportListComponent.prototype, "onSuccess", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], AnalyticsReportListComponent.prototype, "onError", void 0);
AnalyticsReportListComponent = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'analytics-report-list',
        template: "<div class=\"menu-container\">     <ul class='mdl-list'>         <li class=\"mdl-list__item activiti-filters__entry\" (click)=\"selectReport(report)\" *ngFor=\"let report of reports; let idx = index\"             [class.active]=\"currentReport === report\">             <span [attr.id]=\"'report-list-' + idx\" class=\"mdl-list__item-primary-content\">             <i class=\"material-icons mdl-list__item-icon activiti-filters__entry-icon\" [attr.data-automation-id]=\"report.name + '_filter'\">assignment</i>             <span class=\"text\">{{report.name}}</span>             </span>         </li>     </ul> </div>",
        styles: [".mdl-list__item {     cursor: pointer; }  .activiti-filters__entry {     cursor: pointer; }  .activiti-filters__entry-icon {     margin-right: 12px !important; }  .activiti-filters__entry.active {     color: rgb(68,138,255); }  .activiti-filters__entry.active .activiti-filters__entry-icon {     color: rgb(68,138,255); }"]
    }),
    __metadata("design:paramtypes", [analytics_service_1.AnalyticsService,
        ng2_alfresco_core_1.LogService])
], AnalyticsReportListComponent);
exports.AnalyticsReportListComponent = AnalyticsReportListComponent;
