// ng2-activiti-diagrams/src/components/tooltip/diagram-tooltip.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var DiagramTooltip = (function () {
    function DiagramTooltip() {
    }
    DiagramTooltip.prototype.getTooltipHeader = function (data) {
        var headerValue = data.name || data.id;
        return data.type + ' ' + headerValue;
    };
    DiagramTooltip.prototype.getTooltipMessage = function (data) {
        return (data.value !== undefined && data.value !== null) ? data.value + ' ' + data.dataType : '';
    };
    return DiagramTooltip;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], DiagramTooltip.prototype, "data", void 0);
DiagramTooltip = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'diagram-tooltip',
        template: "<div class=\"mdl-tooltip mdl-tooltip--large mdl-tooltip-diagram\" [attr.for]=\"data.id\">     <div>         <div class=\"mdl-tooltip-header__message\"><span>{{getTooltipHeader(data)}}</span></div>         <span class=\"mdl-tooltip-body__message\">{{getTooltipMessage(data)}}</span>     </div> </div>",
        styles: [".mdl-tooltip {     will-change: unset; }  .mdl-tooltip-diagram {     background: white;     padding: 0px; }  .mdl-tooltip-header__message {     background: black;     color: white;     margin-top: 0px;     margin-bottom: 3px;     width:100%; }  .mdl-tooltip-body__message {     color: black;     text-align: center; }"]
    })
], DiagramTooltip);
exports.DiagramTooltip = DiagramTooltip;
