// ng2-alfresco-core/src/services/alfresco-api.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var alfrescoApi = require("alfresco-js-api");
var alfresco_settings_service_1 = require("./alfresco-settings.service");
var storage_service_1 = require("./storage.service");
var AlfrescoApiService = (function () {
    function AlfrescoApiService(settingsService, storage) {
        var _this = this;
        this.settingsService = settingsService;
        this.storage = storage;
        this.provider = this.settingsService.getProviders();
        this.ticketEcm = this.getTicketEcm();
        this.ticketBpm = this.getTicketBpm();
        this.hostEcm = this.settingsService.ecmHost;
        this.hostBpm = this.settingsService.bpmHost;
        this.contextRoot = 'alfresco';
        this.disableCsrf = false;
        this.init();
        settingsService.bpmHostSubject.subscribe(function (hostBpm) {
            _this.hostBpm = hostBpm;
            _this.init();
        });
        settingsService.ecmHostSubject.subscribe(function (hostEcm) {
            _this.hostEcm = hostEcm;
            _this.init();
        });
        settingsService.csrfSubject.subscribe(function (disableCsrf) {
            _this.disableCsrf = disableCsrf;
            _this.init();
        });
        settingsService.providerSubject.subscribe(function (provider) {
            _this.provider = provider;
            _this.init();
        });
    }
    AlfrescoApiService.prototype.getInstance = function () {
        return this.alfrescoApi;
    };
    AlfrescoApiService.prototype.init = function () {
        this.alfrescoApi = new alfrescoApi({
            provider: this.provider,
            ticketEcm: this.ticketEcm,
            ticketBpm: this.ticketBpm,
            hostEcm: this.hostEcm,
            hostBpm: this.hostBpm,
            contextRoot: this.contextRoot,
            disableCsrf: this.disableCsrf
        });
    };
    AlfrescoApiService.prototype.getTicketEcm = function () {
        return this.storage.getItem('ticket-ECM');
    };
    AlfrescoApiService.prototype.getTicketBpm = function () {
        return this.storage.getItem('ticket-BPM');
    };
    return AlfrescoApiService;
}());
AlfrescoApiService = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [alfresco_settings_service_1.AlfrescoSettingsService,
        storage_service_1.StorageService])
], AlfrescoApiService);
exports.AlfrescoApiService = AlfrescoApiService;
