// ng2-alfresco-core/src/services/alfresco-settings.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var core_1 = require("@angular/core");
var Subject_1 = require("rxjs/Subject");
var AlfrescoSettingsService = AlfrescoSettingsService_1 = (function () {
    function AlfrescoSettingsService() {
        this._ecmHost = AlfrescoSettingsService_1.DEFAULT_ECM_ADDRESS;
        this._bpmHost = AlfrescoSettingsService_1.DEFAULT_BPM_ADDRESS;
        this._csrfDisabled = AlfrescoSettingsService_1.DEFAULT_CSRF_CONFIG;
        this._bpmContextPath = AlfrescoSettingsService_1.DEFAULT_BPM_CONTEXT_PATH;
        this.providers = 'ALL';
        this.bpmHostSubject = new Subject_1.Subject();
        this.ecmHostSubject = new Subject_1.Subject();
        this.csrfSubject = new Subject_1.Subject();
        this.providerSubject = new Subject_1.Subject();
    }
    Object.defineProperty(AlfrescoSettingsService.prototype, "ecmHost", {
        get: function () {
            return this._ecmHost;
        },
        set: function (ecmHostUrl) {
            this.ecmHostSubject.next(ecmHostUrl);
            this._ecmHost = ecmHostUrl;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AlfrescoSettingsService.prototype, "csrfDisabled", {
        set: function (csrfDisabled) {
            this.csrfSubject.next(csrfDisabled);
            this._csrfDisabled = csrfDisabled;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AlfrescoSettingsService.prototype, "bpmHost", {
        get: function () {
            return this._bpmHost;
        },
        set: function (bpmHostUrl) {
            this.bpmHostSubject.next(bpmHostUrl);
            this._bpmHost = bpmHostUrl;
        },
        enumerable: true,
        configurable: true
    });
    AlfrescoSettingsService.prototype.getBPMApiBaseUrl = function () {
        return this._bpmHost + this._bpmContextPath;
    };
    AlfrescoSettingsService.prototype.getProviders = function () {
        return this.providers;
    };
    AlfrescoSettingsService.prototype.setProviders = function (providers) {
        this.providerSubject.next(providers);
        this.providers = providers;
    };
    return AlfrescoSettingsService;
}());
AlfrescoSettingsService.DEFAULT_ECM_ADDRESS = 'http://' + window.location.hostname + ':8080';
AlfrescoSettingsService.DEFAULT_BPM_ADDRESS = 'http://' + window.location.hostname + ':9999';
AlfrescoSettingsService.DEFAULT_CSRF_CONFIG = false;
AlfrescoSettingsService.DEFAULT_BPM_CONTEXT_PATH = '/activiti-app';
AlfrescoSettingsService = AlfrescoSettingsService_1 = __decorate([
    core_1.Injectable()
], AlfrescoSettingsService);
exports.AlfrescoSettingsService = AlfrescoSettingsService;
var AlfrescoSettingsService_1;
