// ng2-alfresco-core/src/services/log.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var core_1 = require("@angular/core");
var LogService = (function () {
    function LogService() {
    }
    Object.defineProperty(LogService.prototype, "assert", {
        get: function () {
            return console.assert.bind(console);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LogService.prototype, "error", {
        get: function () {
            return console.error.bind(console);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LogService.prototype, "group", {
        get: function () {
            return console.group.bind(console);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LogService.prototype, "groupEnd", {
        get: function () {
            return console.groupEnd.bind(console);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LogService.prototype, "info", {
        get: function () {
            return console.info.bind(console);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LogService.prototype, "log", {
        get: function () {
            return console.log.bind(console);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LogService.prototype, "warn", {
        get: function () {
            return console.warn.bind(console);
        },
        enumerable: true,
        configurable: true
    });
    return LogService;
}());
LogService = __decorate([
    core_1.Injectable()
], LogService);
exports.LogService = LogService;
var LogServiceMock = (function () {
    function LogServiceMock() {
    }
    LogServiceMock.prototype.assert = function (message) {
        var optionalParams = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            optionalParams[_i - 1] = arguments[_i];
        }
    };
    LogServiceMock.prototype.error = function (message) {
        var optionalParams = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            optionalParams[_i - 1] = arguments[_i];
        }
    };
    LogServiceMock.prototype.group = function (message) {
        var optionalParams = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            optionalParams[_i - 1] = arguments[_i];
        }
    };
    LogServiceMock.prototype.groupEnd = function (message) {
        var optionalParams = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            optionalParams[_i - 1] = arguments[_i];
        }
    };
    LogServiceMock.prototype.info = function (message) {
        var optionalParams = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            optionalParams[_i - 1] = arguments[_i];
        }
    };
    LogServiceMock.prototype.log = function (message) {
        var optionalParams = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            optionalParams[_i - 1] = arguments[_i];
        }
    };
    LogServiceMock.prototype.warn = function (message) {
        var optionalParams = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            optionalParams[_i - 1] = arguments[_i];
        }
    };
    return LogServiceMock;
}());
exports.LogServiceMock = LogServiceMock;
