// ng2-alfresco-core/src/services/renditions.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var Rx_1 = require("rxjs/Rx");
var alfresco_api_service_1 = require("./alfresco-api.service");
var log_service_1 = require("./log.service");
var RenditionsService = (function () {
    function RenditionsService(apiService, logService) {
        this.apiService = apiService;
        this.logService = logService;
    }
    RenditionsService.prototype.isRenditionAvailable = function (nodeId, encoding) {
        var _this = this;
        return Rx_1.Observable.create(function (observer) {
            _this.getRendition(nodeId, encoding).subscribe(function (res) {
                var isAvailable = true;
                if (res.entry.status === 'NOT_CREATED') {
                    isAvailable = false;
                }
                observer.next(isAvailable);
                observer.complete();
            }, function () {
                observer.next(false);
                observer.complete();
            });
        });
    };
    RenditionsService.prototype.isConversionPossible = function (nodeId, encoding) {
        var _this = this;
        return Rx_1.Observable.create(function (observer) {
            _this.getRendition(nodeId, encoding).subscribe(function () {
                observer.next(true);
                observer.complete();
            }, function () {
                observer.next(false);
                observer.complete();
            });
        });
    };
    RenditionsService.prototype.getRendition = function (nodeId, encoding) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.renditionsApi.getRendition(nodeId, encoding))
            .catch(function (err) { return _this.handleError(err); });
    };
    RenditionsService.prototype.getRenditionsListByNodeId = function (nodeId) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.renditionsApi.getRenditions(nodeId))
            .catch(function (err) { return _this.handleError(err); });
    };
    RenditionsService.prototype.createRendition = function (nodeId, encoding) {
        var _this = this;
        return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.renditionsApi.createRendition(nodeId, { id: encoding }))
            .catch(function (err) { return _this.handleError(err); });
    };
    RenditionsService.prototype.handleError = function (error) {
        this.logService.error(error);
        return Rx_1.Observable.throw(error || 'Server error');
    };
    return RenditionsService;
}());
RenditionsService = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [alfresco_api_service_1.AlfrescoApiService,
        log_service_1.LogService])
], RenditionsService);
exports.RenditionsService = RenditionsService;
