// ng2-activiti-analytics/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var ng2_activiti_diagrams_1 = require("ng2-activiti-diagrams");
var analytics_report_list_component_1 = require("./src/components/analytics-report-list.component");
var analytics_report_parameters_component_1 = require("./src/components/analytics-report-parameters.component");
var analytics_component_1 = require("./src/components/analytics.component");
var analytics_generator_component_1 = require("./src/components/analytics-generator.component");
var analytics_report_heat_map_component_1 = require("./src/components/analytics-report-heat-map.component");
var analytics_service_1 = require("./src/services/analytics.service");
var ng2_charts_1 = require("ng2-charts/ng2-charts");
var index_1 = require("./src/components/widgets/index");
__export(require("./src/components/analytics.component"));
__export(require("./src/components/analytics-generator.component"));
__export(require("./src/components/analytics-report-list.component"));
__export(require("./src/components/analytics-report-parameters.component"));
__export(require("./src/services/analytics.service"));
__export(require("./src/components/widgets/index"));
exports.ANALYTICS_DIRECTIVES = [
    analytics_component_1.AnalyticsComponent,
    analytics_report_list_component_1.AnalyticsReportListComponent,
    analytics_report_parameters_component_1.AnalyticsReportParametersComponent,
    analytics_generator_component_1.AnalyticsGeneratorComponent,
    analytics_report_heat_map_component_1.AnalyticsReportHeatMapComponent,
    index_1.WIDGET_DIRECTIVES
];
exports.ANALYTICS_PROVIDERS = [
    analytics_service_1.AnalyticsService
];
var AnalyticsModule = AnalyticsModule_1 = (function () {
    function AnalyticsModule() {
    }
    AnalyticsModule.forRoot = function () {
        return {
            ngModule: AnalyticsModule_1,
            providers: exports.ANALYTICS_PROVIDERS.slice()
        };
    };
    return AnalyticsModule;
}());
AnalyticsModule = AnalyticsModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng2_alfresco_core_1.CoreModule,
            ng2_activiti_diagrams_1.DiagramsModule
        ],
        declarations: exports.ANALYTICS_DIRECTIVES.concat(ng2_charts_1.CHART_DIRECTIVES),
        providers: exports.ANALYTICS_PROVIDERS.slice(),
        exports: exports.ANALYTICS_DIRECTIVES.slice()
    })
], AnalyticsModule);
exports.AnalyticsModule = AnalyticsModule;
var AnalyticsModule_1;
