// ng2-activiti-form/src/components/widgets/core/group-user.model.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var GroupUserModel = (function () {
    function GroupUserModel(json) {
        if (json) {
            this.company = json.company;
            this.email = json.email;
            this.firstName = json.firstName;
            this.id = json.id;
            this.lastName = json.lastName;
        }
    }
    return GroupUserModel;
}());
exports.GroupUserModel = GroupUserModel;
