// ng2-activiti-form/src/components/widgets/dynamic-table/editors/row.editor.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var dynamic_table_widget_model_1 = require("./../dynamic-table.widget.model");
var RowEditorComponent = (function () {
    function RowEditorComponent() {
        this.save = new core_1.EventEmitter();
        this.cancel = new core_1.EventEmitter();
        this.validationSummary = { isValid: true, text: null };
    }
    RowEditorComponent.prototype.onCancelChanges = function () {
        this.cancel.emit({
            table: this.table,
            row: this.row,
            column: this.column
        });
    };
    RowEditorComponent.prototype.onSaveChanges = function () {
        this.validate();
        if (this.isValid()) {
            this.save.emit({
                table: this.table,
                row: this.row,
                column: this.column
            });
        }
    };
    RowEditorComponent.prototype.isValid = function () {
        return this.validationSummary && this.validationSummary.isValid;
    };
    RowEditorComponent.prototype.validate = function () {
        this.validationSummary = this.table.validateRow(this.row);
    };
    return RowEditorComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", dynamic_table_widget_model_1.DynamicTableModel)
], RowEditorComponent.prototype, "table", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], RowEditorComponent.prototype, "row", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], RowEditorComponent.prototype, "column", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], RowEditorComponent.prototype, "save", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], RowEditorComponent.prototype, "cancel", void 0);
RowEditorComponent = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'row-editor',
        template: "<div class=\"row-editor mdl-shadow--2dp\"     [class.row-editor__invalid]=\"!validationSummary.isValid\">     <div class=\"mdl-grid\" *ngFor=\"let column of table.columns\">         <div class=\"mdl-cell mdl-cell--6-col\" [ngSwitch]=\"column.type\">             <div *ngSwitchCase=\"'Dropdown'\">                 <alf-dropdown-editor                     [table]=\"table\"                     [row]=\"row\"                     [column]=\"column\">                 </alf-dropdown-editor>                 </div>                 <div *ngSwitchCase=\"'Date'\">                     <alf-date-editor                     [table]=\"table\"                     [row]=\"row\"                     [column]=\"column\">                 </alf-date-editor>                 </div>                  <div *ngSwitchCase=\"'Boolean'\">                 <alf-boolean-editor                     [table]=\"table\"                     [row]=\"row\"                     [column]=\"column\">                 </alf-boolean-editor>                 </div>                 <div *ngSwitchDefault>                 <alf-text-editor                     [table]=\"table\"                     [row]=\"row\"                     [column]=\"column\">                 </alf-text-editor>             </div>         </div>     </div>     <div class=\"row-editor__validation-summary\" *ngIf=\"!validationSummary.isValid\">{{validationSummary.text}}</div>     <div>         <button             class=\"mdl-button mdl-js-button mdl-js-ripple-effect\"             (click)=\"onCancelChanges()\">Cancel</button>         <button             class=\"mdl-button mdl-js-button mdl-js-ripple-effect\"             (click)=\"onSaveChanges()\">Save</button>     </div> </div>",
        styles: [".row-editor {     padding: 8px; }  .row-editor__validation-summary {     visibility: hidden; }  .row-editor__invalid .row-editor__validation-summary {     padding-left: 16px;     padding-right: 16px;     padding-top: 8px;     padding-bottom: 8px;     color: #d50000;     visibility: visible; }"]
    })
], RowEditorComponent);
exports.RowEditorComponent = RowEditorComponent;
