// ng2-alfresco-documentlist/src/components/content-action/content-action.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var content_action_list_component_1 = require("./content-action-list.component");
var content_action_model_1 = require("./../../models/content-action.model");
var document_actions_service_1 = require("../../services/document-actions.service");
var folder_actions_service_1 = require("../../services/folder-actions.service");
var ContentActionComponent = (function () {
    function ContentActionComponent(list, documentActions, folderActions) {
        this.list = list;
        this.documentActions = documentActions;
        this.folderActions = folderActions;
        this.title = 'Action';
        this.execute = new core_1.EventEmitter();
        this.model = new content_action_model_1.ContentActionModel();
    }
    ContentActionComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.model = new content_action_model_1.ContentActionModel({
            title: this.title,
            icon: this.icon,
            target: this.target
        });
        if (this.handler) {
            this.model.handler = this.getSystemHandler(this.target, this.handler);
        }
        else if (this.execute) {
            this.model.handler = function (document) {
                _this.execute.emit({
                    value: document
                });
            };
        }
        this.register();
    };
    ContentActionComponent.prototype.register = function () {
        if (this.list) {
            return this.list.registerAction(this.model);
        }
        return false;
    };
    ContentActionComponent.prototype.ngOnChanges = function (changes) {
        this.model.title = this.title;
    };
    ContentActionComponent.prototype.getSystemHandler = function (target, name) {
        if (target) {
            var ltarget = target.toLowerCase();
            if (ltarget === 'document') {
                if (this.documentActions) {
                    return this.documentActions.getHandler(name);
                }
                return null;
            }
            if (ltarget === 'folder') {
                if (this.folderActions) {
                    return this.folderActions.getHandler(name);
                }
                return null;
            }
        }
        return null;
    };
    return ContentActionComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentActionComponent.prototype, "title", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentActionComponent.prototype, "icon", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentActionComponent.prototype, "handler", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ContentActionComponent.prototype, "target", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], ContentActionComponent.prototype, "execute", void 0);
ContentActionComponent = __decorate([
    core_1.Component({
        selector: 'content-action',
        template: ''
    }),
    __metadata("design:paramtypes", [content_action_list_component_1.ContentActionListComponent,
        document_actions_service_1.DocumentActionsService,
        folder_actions_service_1.FolderActionsService])
], ContentActionComponent);
exports.ContentActionComponent = ContentActionComponent;
