// ng2-activiti-tasklist/src/services/activiti-people.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var Rx_1 = require("rxjs/Rx");
var ActivitiPeopleService = (function () {
    function ActivitiPeopleService(alfrescoJsApi, logService) {
        this.alfrescoJsApi = alfrescoJsApi;
        this.logService = logService;
    }
    ActivitiPeopleService.prototype.getWorkflowUsers = function (taskId, searchWord) {
        var _this = this;
        var option = { excludeTaskId: taskId, filter: searchWord };
        return Rx_1.Observable.fromPromise(this.getWorkflowUserApi(option))
            .map(function (response) { return response.data || []; })
            .catch(function (err) { return _this.handleError(err); });
    };
    ActivitiPeopleService.prototype.involveUserWithTask = function (taskId, idToInvolve) {
        var _this = this;
        var node = { userId: idToInvolve };
        return Rx_1.Observable.fromPromise(this.involveUserToTaskApi(taskId, node))
            .catch(function (err) { return _this.handleError(err); });
    };
    ActivitiPeopleService.prototype.removeInvolvedUser = function (taskId, idToRemove) {
        var _this = this;
        var node = { userId: idToRemove };
        return Rx_1.Observable.fromPromise(this.removeInvolvedUserFromTaskApi(taskId, node))
            .catch(function (err) { return _this.handleError(err); });
    };
    ActivitiPeopleService.prototype.getWorkflowUserApi = function (options) {
        return this.alfrescoJsApi.getInstance().activiti.usersWorkflowApi.getUsers(options);
    };
    ActivitiPeopleService.prototype.involveUserToTaskApi = function (taskId, node) {
        return this.alfrescoJsApi.getInstance().activiti.taskActionsApi.involveUser(taskId, node);
    };
    ActivitiPeopleService.prototype.removeInvolvedUserFromTaskApi = function (taskId, node) {
        return this.alfrescoJsApi.getInstance().activiti.taskActionsApi.removeInvolvedUser(taskId, node);
    };
    ActivitiPeopleService.prototype.handleError = function (error) {
        this.logService.error(error);
        return Rx_1.Observable.throw(error || 'Server error');
    };
    return ActivitiPeopleService;
}());
ActivitiPeopleService = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoApiService,
        ng2_alfresco_core_1.LogService])
], ActivitiPeopleService);
exports.ActivitiPeopleService = ActivitiPeopleService;
