// ng2-activiti-analytics/src/components/widgets/dropdown/dropdown.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var widget_component_1 = require("./../widget.component");
var DropdownWidget = (function (_super) {
    __extends(DropdownWidget, _super);
    function DropdownWidget() {
        var _this = _super.call(this) || this;
        _this.fieldChanged = new core_1.EventEmitter();
        _this.showDefaultOption = true;
        _this.required = false;
        _this.defaultOptionText = 'Choose One';
        return _this;
    }
    DropdownWidget.prototype.ngOnInit = function () {
        if (this.required) {
            this.formGroup.get(this.controllerName).setValidators(forms_1.Validators.required);
        }
    };
    return DropdownWidget;
}(widget_component_1.WidgetComponent));
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], DropdownWidget.prototype, "field", void 0);
__decorate([
    core_1.Input('group'),
    __metadata("design:type", forms_1.FormGroup)
], DropdownWidget.prototype, "formGroup", void 0);
__decorate([
    core_1.Input('controllerName'),
    __metadata("design:type", String)
], DropdownWidget.prototype, "controllerName", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], DropdownWidget.prototype, "fieldChanged", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], DropdownWidget.prototype, "showDefaultOption", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], DropdownWidget.prototype, "required", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], DropdownWidget.prototype, "defaultOptionText", void 0);
DropdownWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'dropdown-widget',
        template: "<div class=\"dropdown-widget\" [formGroup]=\"formGroup\">     <label class=\"dropdown-widget__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label>     <select [formControlName]=\"controllerName\" [attr.id]=\"'select-' + field.id\" class=\"dropdown-widget__select\"             [(ngModel)]=\"field.value\" (ngModelChange)=\"changeValue($event)\">         <option *ngIf=\"showDefaultOption\" value=\"null\">{{defaultOptionText}}</option>         <option *ngFor=\"let opt of field.options\" [value]=\"opt.id\">{{opt.label}}</option>     </select> </div>",
        styles: [".dropdown-widget {     width: 100%; }  .dropdown-widget__select {     width: 100%; }  .dropdown-widget__invalid .dropdown-widget__select {     border-color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label {     color: #d50000; }  .dropdown-widget__invalid .dropdown-widget__label:after {     background-color: #d50000; }  .dropdown-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }),
    __metadata("design:paramtypes", [])
], DropdownWidget);
exports.DropdownWidget = DropdownWidget;
