// ng2-alfresco-tag/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var tag_actions_component_1 = require("./src/components/tag-actions.component");
var tag_list_component_1 = require("./src/components/tag-list.component");
var tag_node_list_component_1 = require("./src/components/tag-node-list.component");
var tag_service_1 = require("./src/services/tag.service");
__export(require("./src/components/tag-actions.component"));
__export(require("./src/components/tag-list.component"));
__export(require("./src/components/tag-node-list.component"));
__export(require("./src/services/tag.service"));
exports.TAG_DIRECTIVES = [
    tag_actions_component_1.TagActionsComponent,
    tag_list_component_1.TagList,
    tag_node_list_component_1.TagNodeList
];
exports.TAG_PROVIDERS = [
    tag_service_1.TagService
];
var TagModule = TagModule_1 = (function () {
    function TagModule() {
    }
    TagModule.forRoot = function () {
        return {
            ngModule: TagModule_1,
            providers: exports.TAG_DIRECTIVES.slice()
        };
    };
    return TagModule;
}());
TagModule = TagModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng2_alfresco_core_1.CoreModule
        ],
        declarations: exports.TAG_DIRECTIVES.slice(),
        providers: exports.TAG_PROVIDERS.slice(),
        exports: exports.TAG_DIRECTIVES.slice()
    })
], TagModule);
exports.TagModule = TagModule;
var TagModule_1;
