// ng2-activiti-form/src/components/widgets/upload/upload.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var widget_component_1 = require("./../widget.component");
var form_service_1 = require("../../../services/form.service");
var UploadWidget = (function (_super) {
    __extends(UploadWidget, _super);
    function UploadWidget(formService, logService) {
        var _this = _super.call(this) || this;
        _this.formService = formService;
        _this.logService = logService;
        return _this;
    }
    UploadWidget.prototype.ngOnInit = function () {
        if (this.field &&
            this.field.value &&
            this.field.value.length > 0) {
            this.hasFile = true;
            var file = this.field.value[0];
            this.fileName = file.name;
            this.displayText = decodeURI(file.name);
        }
    };
    UploadWidget.prototype.reset = function () {
        this.hasFile = false;
        this.fileName = null;
        this.displayText = null;
        if (this.field) {
            this.field.value = null;
            this.field.json.value = null;
        }
    };
    UploadWidget.prototype.onFileChanged = function (event) {
        var _this = this;
        var files = event.target.files;
        if (files && files.length > 0) {
            var file = files[0];
            this.formService.createTemporaryRawRelatedContent(file)
                .subscribe(function (response) {
                _this.logService.info(response);
                _this.field.value = [response];
                _this.field.json.value = [response];
            }, function (error) {
                _this.logService.error(error);
                window.alert('Error uploading file. See console output for more details.');
            });
        }
    };
    return UploadWidget;
}(widget_component_1.WidgetComponent));
UploadWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'upload-widget',
        template: "<div class=\"upload-widget\"      [class.upload-widget__invalid]=\"!field.isValid\">     <label class=\"upload-widget__label\" [attr.for]=\"field.id\">{{field.name}}</label>     <div>         <i *ngIf=\"hasFile\" class=\"material-icons upload-widget__icon\">attachment</i>         <span *ngIf=\"hasFile\" class=\"upload-widget__file\">{{displayText}}</span>         <input *ngIf=\"!hasFile\"                #file                type=\"file\"                [attr.id]=\"field.id\"                class=\"upload-widget__file\"                (change)=\"onFileChanged($event)\">         <button *ngIf=\"hasFile\" (click)=\"reset(file);\"                 class=\"mdl-button mdl-js-button mdl-button--icon upload-widget__reset\">             <i class=\"material-icons\">highlight_off</i>         </button>     </div>     <span *ngIf=\"field.validationSummary\" class=\"mdl-textfield__error\">{{field.validationSummary}}</span> </div>",
        styles: [".upload-widget {     width:100%;     word-break: break-all; }  .upload-widget__icon {     float: left; }  .upload-widget__file {     float: left;     margin-top: 4px; }  .upload-widget__reset {     float: left;     margin-top: -2px; }  .upload-widget__invalid .upload-widget__label {     color: #d50000; }  .upload-widget__invalid .upload-widget__label:after {     background-color: #d50000; }  .upload-widget__invalid .upload-widget__file {     color: #d50000; }  .upload-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    }),
    __metadata("design:paramtypes", [form_service_1.FormService,
        ng2_alfresco_core_1.LogService])
], UploadWidget);
exports.UploadWidget = UploadWidget;
