// ng2-alfresco-viewer/src/componets/mediaPlayer.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var MediaPlayerComponent = (function () {
    function MediaPlayerComponent() {
    }
    MediaPlayerComponent.prototype.ngOnChanges = function (changes) {
        if (!this.urlFile) {
            throw new Error('Attribute urlFile is required');
        }
    };
    return MediaPlayerComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], MediaPlayerComponent.prototype, "urlFile", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], MediaPlayerComponent.prototype, "mimeType", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], MediaPlayerComponent.prototype, "nameFile", void 0);
MediaPlayerComponent = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'media-player',
        template: "<div class=\"viewer-video-content\">     <div class=\"viewer-video-row\">         <div class=\"viewer-video-cell\">             <video controls  >                 <source [src]=\"urlFile\" [type]=\"mimeType\" />             </video>         </div>     </div> </div>",
        styles: [".viewer-video-row {     display: -webkit-box;     display: -moz-box;     display: -ms-flexbox;     display: -webkit-flex;     display: flex;      -webkit-box-orient: horizontal;     -moz-box-orient: horizontal;     box-orient: horizontal;     flex-direction: row;      -webkit-box-pack: center;     -moz-box-pack: center;     box-pack: center;     justify-content: center;      -webkit-box-align: center;     -moz-box-align: center;     box-align: center;     align-items: center; }  .viewer-video-cell {     -webkit-box-flex: 1;     -moz-box-flex: 1;     box-flex: 1;     -webkit-flex: 1 1 auto;     flex: 1 1 auto;      padding: 10px;     margin: 10px;      text-align: center; }  video {     max-height: 80vh;     display: flex;     flex-direction: row;     flex-wrap: nowrap;     justify-content: center;     align-items: center;     align-content: center;     max-width: 100%;     margin-left: auto;     margin-right: auto; }"]
    })
], MediaPlayerComponent);
exports.MediaPlayerComponent = MediaPlayerComponent;
