// ng2-alfresco-viewer/src/componets/notSupportedFormat.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var NotSupportedFormat = (function () {
    function NotSupportedFormat() {
    }
    NotSupportedFormat.prototype.download = function () {
        window.open(this.urlFile);
    };
    return NotSupportedFormat;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], NotSupportedFormat.prototype, "nameFile", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], NotSupportedFormat.prototype, "urlFile", void 0);
NotSupportedFormat = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'not-supported-format',
        template: "<section class=\"section--center mdl-grid mdl-grid--no-spacing\">     <div class=\"viewer-margin mdl-card mdl-cell mdl-cell--9-col-desktop mdl-cell--6-col-tablet mdl-cell--4-col-phone mdl-shadow--2dp\">         <div class=\"viewer-download-text mdl-card__supporting-text viewer-margin\">             <h4>File '<span>{{nameFile}}</span>' is of an unsupported format</h4>         </div>         <div class=\"center-element mdl-card__actions\">             <button id=\"viewer-download-button\" class=\"mdl-button mdl-js-button mdl-button--raised mdl-js-ripple-effect mdl-button--accent\" (click)=\"download()\">                 <i class=\"viewer-margin-cloud-download material-icons\">cloud_download</i>   Download             </button>         </div>     </div> </section>",
        styles: [" .viewer-download-text  {     text-align: center;     word-wrap: break-word; }  .viewer-margin-cloud-download{     margin-right: 20px; }  .viewer-margin {     margin: auto !important; }  .center-element {     display: flex;     align-items: center;     justify-content: center; }  .full_width{     width :95% !important; }"]
    })
], NotSupportedFormat);
exports.NotSupportedFormat = NotSupportedFormat;
