// ng2-alfresco-viewer/src/componets/pdfViewer.component.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var rendering_queue_services_1 = require("../services/rendering-queue.services");
var PdfViewerComponent = (function () {
    function PdfViewerComponent(renderingQueueServices, logService) {
        this.renderingQueueServices = renderingQueueServices;
        this.logService = logService;
        this.showToolbar = true;
        this.currentScaleMode = 'auto';
        this.MAX_AUTO_SCALE = 1.25;
        this.DEFAULT_SCALE_DELTA = 1.1;
        this.MIN_SCALE = 0.25;
        this.MAX_SCALE = 10.0;
    }
    PdfViewerComponent.prototype.ngOnChanges = function (changes) {
        var _this = this;
        if (!this.urlFile) {
            throw new Error('Attribute urlFile is required');
        }
        if (this.urlFile) {
            return new Promise(function (resolve, reject) {
                var loadingTask = _this.getPDFJS().getDocument(_this.urlFile);
                loadingTask.onProgress = function (progressData) {
                    var level = progressData.loaded / progressData.total;
                    _this.laodingPercent = Math.round(level * 100);
                };
                loadingTask.then(function (pdfDocument) {
                    _this.currentPdfDocument = pdfDocument;
                    _this.totalPages = pdfDocument.numPages;
                    _this.page = 1;
                    _this.displayPage = 1;
                    _this.initPDFViewer(_this.currentPdfDocument);
                    _this.currentPdfDocument.getPage(1).then(function () {
                        _this.scalePage('auto');
                        resolve();
                    }, function (error) {
                        reject(error);
                    });
                }, function (error) {
                    reject(error);
                });
            });
        }
    };
    PdfViewerComponent.prototype.getPDFJS = function () {
        return PDFJS;
    };
    PdfViewerComponent.prototype.initPDFViewer = function (pdfDocument) {
        var _this = this;
        PDFJS.verbosity = 1;
        PDFJS.disableWorker = false;
        var documentContainer = document.getElementById('viewer-pdf-container');
        var viewer = document.getElementById('viewer-viewerPdf');
        window.document.addEventListener('scroll', function (event) {
            _this.watchScroll(event.target);
        }, true);
        this.pdfViewer = new PDFJS.PDFViewer({
            container: documentContainer,
            viewer: viewer,
            renderingQueue: this.renderingQueueServices
        });
        this.renderingQueueServices.setViewer(this.pdfViewer);
        this.pdfViewer.setDocument(pdfDocument);
    };
    PdfViewerComponent.prototype.scalePage = function (scaleMode) {
        this.currentScaleMode = scaleMode;
        if (this.pdfViewer) {
            var viewerContainer = document.getElementById('viewer-main-container');
            var documentContainer = document.getElementById('viewer-pdf-container');
            var widthContainer = void 0;
            var heigthContainer = void 0;
            if (viewerContainer && viewerContainer.clientWidth <= documentContainer.clientWidth) {
                widthContainer = viewerContainer.clientWidth;
                heigthContainer = viewerContainer.clientHeight;
            }
            else {
                widthContainer = documentContainer.clientWidth;
                heigthContainer = documentContainer.clientHeight;
            }
            var currentPage = this.pdfViewer._pages[this.pdfViewer._currentPageNumber];
            var padding = 20;
            var pageWidthScale = (widthContainer - padding) / currentPage.width * currentPage.scale;
            var pageHeightScale = (heigthContainer - padding) / currentPage.width * currentPage.scale;
            var scale = void 0;
            switch (this.currentScaleMode) {
                case 'page-actual':
                    scale = 1;
                    break;
                case 'page-width':
                    scale = pageWidthScale;
                    break;
                case 'page-height':
                    scale = pageHeightScale;
                    break;
                case 'page-fit':
                    scale = Math.min(pageWidthScale, pageHeightScale);
                    break;
                case 'auto':
                    var horizontalScale = void 0;
                    if (this.isLandscape) {
                        horizontalScale = Math.min(pageHeightScale, pageWidthScale);
                    }
                    else {
                        horizontalScale = pageWidthScale;
                    }
                    scale = Math.min(this.MAX_AUTO_SCALE, horizontalScale);
                    break;
                default:
                    this.logService.error('pdfViewSetScale: \'' + scaleMode + '\' is an unknown zoom value.');
                    return;
            }
            this.setScaleUpdatePages(scale);
        }
    };
    PdfViewerComponent.prototype.setScaleUpdatePages = function (newScale) {
        if (!this.isSameScale(this.currentScale, newScale)) {
            this.currentScale = newScale;
            this.pdfViewer._pages.forEach(function (currentPage) {
                currentPage.update(newScale);
            });
            this.pdfViewer.update();
        }
    };
    PdfViewerComponent.prototype.isSameScale = function (oldScale, newScale) {
        return (newScale === oldScale);
    };
    PdfViewerComponent.prototype.isLandscape = function (width, height) {
        return (width > height);
    };
    PdfViewerComponent.prototype.onResize = function () {
        this.scalePage(this.currentScaleMode);
    };
    PdfViewerComponent.prototype.pageFit = function () {
        if (this.currentScaleMode !== 'page-fit') {
            this.scalePage('page-fit');
        }
        else {
            this.scalePage('auto');
        }
    };
    PdfViewerComponent.prototype.zoomIn = function (ticks) {
        var newScale = this.currentScale;
        do {
            newScale = (newScale * this.DEFAULT_SCALE_DELTA).toFixed(2);
            newScale = Math.ceil(newScale * 10) / 10;
            newScale = Math.min(this.MAX_SCALE, newScale);
        } while (--ticks > 0 && newScale < this.MAX_SCALE);
        this.currentScaleMode = 'auto';
        this.setScaleUpdatePages(newScale);
    };
    PdfViewerComponent.prototype.zoomOut = function (ticks) {
        var newScale = this.currentScale;
        do {
            newScale = (newScale / this.DEFAULT_SCALE_DELTA).toFixed(2);
            newScale = Math.floor(newScale * 10) / 10;
            newScale = Math.max(this.MIN_SCALE, newScale);
        } while (--ticks > 0 && newScale > this.MIN_SCALE);
        this.currentScaleMode = 'auto';
        this.setScaleUpdatePages(newScale);
    };
    PdfViewerComponent.prototype.previousPage = function () {
        if (this.pdfViewer && this.page > 1) {
            this.page--;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
    };
    PdfViewerComponent.prototype.nextPage = function () {
        if (this.pdfViewer && this.page < this.totalPages) {
            this.page++;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
    };
    PdfViewerComponent.prototype.inputPage = function (page) {
        var pageInput = parseInt(page, 10);
        if (!isNaN(pageInput) && pageInput > 0 && pageInput <= this.totalPages) {
            this.page = pageInput;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
        else {
            this.displayPage = this.page;
        }
    };
    PdfViewerComponent.prototype.watchScroll = function (target) {
        var outputPage = this.getVisibleElement(target);
        if (outputPage) {
            this.page = outputPage.id;
            this.displayPage = this.page;
        }
    };
    PdfViewerComponent.prototype.getVisibleElement = function (target) {
        var _this = this;
        return this.pdfViewer._pages.find(function (page) {
            return _this.isOnScreen(page, target);
        });
    };
    PdfViewerComponent.prototype.isOnScreen = function (page, target) {
        var viewport = {};
        viewport.top = target.scrollTop;
        viewport.bottom = viewport.top + target.scrollHeight;
        var bounds = {};
        bounds.top = page.div.offsetTop;
        bounds.bottom = bounds.top + page.viewport.height;
        return ((bounds.top <= viewport.bottom) && (bounds.bottom >= viewport.top));
    };
    ;
    PdfViewerComponent.prototype.handleKeyboardEvent = function (event) {
        var key = event.keyCode;
        if (key === 39) {
            this.nextPage();
        }
        else if (key === 37) {
            this.previousPage();
        }
    };
    return PdfViewerComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], PdfViewerComponent.prototype, "urlFile", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], PdfViewerComponent.prototype, "nameFile", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], PdfViewerComponent.prototype, "showToolbar", void 0);
__decorate([
    core_1.HostListener('document:keydown', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [KeyboardEvent]),
    __metadata("design:returntype", void 0)
], PdfViewerComponent.prototype, "handleKeyboardEvent", null);
PdfViewerComponent = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'pdf-viewer',
        template: "<!-- Start Pdf Canvas --> <div id=\"viewer-pdf-container\" class=\"viewer-pdf-container\" (keypress)=eventHandler() (window:resize)=\"onResize($event)\">     <div id=\"viewer-viewerPdf\" class=\"pdfViewer\">         <div id=\"loader-container\" class=\"loader-container\">             <div class=\"loader-item\">                 <div id=\"loader-spin\" class=\"mdl-progress mdl-js-progress mdl-progress__indeterminate\"></div>                 <div id=\"loader-text\" class=\"loader-text\">Loading <span>{{nameFile}}</span> {{laodingPercent}}%</div>             </div >         </div>     </div> </div> <!-- End Pdf Canvas -->   <!-- Pagination toolbar start --> <div *ngIf=\"showToolbar\" id=\"viewer-toolbar-pagination\" class=\"viewer-toolbar-pagination mdl-cell--hide-tablet mdl-cell--hide-phone\">     <div id=\"viewer-previous-page-button\" class=\"button-page left\" (click)=\"previousPage()\">         <i class=\"icon material-icons\">keyboard_arrow_left</i>     </div>      <div class=\"viewer-page-counter left\" >         <input id=\"viewer-pagenumber-input\"  #page                (keyup.enter)=\"inputPage(page.value)\" class=\"viewer-pagenumber-input left\" type=\"text\" pattern=\"-?[0-9]*(\\.[0-9]+)?\" value=\"{{displayPage}}\">         <div id=\"viewer-total-pages\"  class=\"left viewer-total-pages\">/ {{totalPages}}</div>     </div>      <div id=\"viewer-next-page-button\" class=\"button-page left\"  (click)=\"nextPage()\" >         <i class=\"icon material-icons\" >keyboard_arrow_right</i>     </div> </div> <!-- Pagination toolbar end -->  <!-- Command toolbar start --> <div *ngIf=\"showToolbar\"  id=\"viewer-toolbar-command\" class=\"viewer-toolbar-command\">     <div id=\"viewer-scale-page-button\" class=\"button-page left\" (click)=\"pageFit()\">         <i class=\"icon material-icons\">zoom_out_map</i>     </div>     <div id=\"viewer-zoom-in-button\" class=\"button-page left\" (click)=\"zoomIn()\">         <i class=\"icon material-icons\">zoom_in</i>     </div>     <div id=\"viewer-zoom-out-button\" class=\"button-page left\" (click)=\"zoomOut()\">         <i class=\"icon material-icons\">zoom_out</i>     </div> </div> <!-- Command toolbar end -->",
        styles: [".loader-container {     display: -webkit-box;      /* OLD - iOS 6-, Safari 3.1-6 */     display: -moz-box;         /* OLD - Firefox 19- (buggy but mostly works) */     display: -webkit-flex;     /* NEW - Chrome */     display: flex;             /* NEW, Spec - Opera 12.1, Firefox 20+ */     -webkit-box-flex-direction: row;     -moz-box-flex-direction: row;     -webkit-flex-direction: row;     flex-direction: row;     height:100%; }  .loader-item {     margin: auto;     max-height:100px;     max-width:300px; }  .loader-text{     white-space: nowrap;     text-align: center;     position: relative;     color : #fff; }  .left {     float: left; }  .viewer-toolbar-pagination{     padding-top: 4px;     top: 80px;     right:35px;     width:auto;     position:absolute;     border-radius: 10px;     background: #3E3E3E;     color: white; }  .viewer-toolbar-command{     height: 30px;     padding-top: 4px;     top: 80px;     left:35px;     width:auto;     position:absolute;     border-radius: 10px;     background: #3E3E3E;     color: white; }  .viewer-pagenumber-input {     border: none;     display: block;     font-size: 16px;     padding: 4px 0;     background: 0 0;     text-align: right;     color: inherit;     width: 33px;     margin-right: 4px;     height: 20px; }  .viewer-total-pages {     border: medium none;     display: flex;     font-size: 16px;     padding: 4px 0px;     background: transparent none repeat scroll 0px 0px;     text-align: right;     color: inherit;     margin-right: 4px;     height: 20px;     align-items: center;     justify-content: center; }  .viewer-page-counter {     margin-right: 20px; }  .button-page {     margin-right: 4px;     height: 24px;     width: 24px;     margin-left: 4px;     cursor: pointer; }  .button-page:hover {     cursor: pointer;     background: grey;     border-radius: 24px; } :host >>> .textLayer {     position: absolute;     left: 0;     top: 0;     right: 0;     bottom: 0;     overflow: hidden;     opacity: 0.2;     line-height: 1.0; }  :host >>> .textLayer > div {     color: transparent;     position: absolute;     white-space: pre;     cursor: text;     -webkit-transform-origin: 0% 0%;     -moz-transform-origin: 0% 0%;     -o-transform-origin: 0% 0%;     -ms-transform-origin: 0% 0%;     transform-origin: 0% 0%; }  :host >>> .textLayer .highlight {     margin: -1px;     padding: 1px;      background-color: rgb(180, 0, 170);     border-radius: 4px; }  :host >>> .textLayer .highlight.begin {     border-radius: 4px 0px 0px 4px; }  :host >>> .textLayer .highlight.end {     border-radius: 0px 4px 4px 0px; }  :host >>> .textLayer .highlight.middle {     border-radius: 0px; }  :host >>> .textLayer .highlight.selected {     background-color: rgb(0, 100, 0); }  :host >>> .textLayer ::selection { background: rgb(0,0,255); } :host >>> .textLayer ::-moz-selection { background: rgb(0,0,255); }  :host >>> .textLayer .endOfContent {     display: block;     position: absolute;     left: 0px;     top: 100%;     right: 0px;     bottom: 0px;     z-index: -1;     cursor: default;     -webkit-user-select: none;     -ms-user-select: none;     -moz-user-select: none; }  :host >>> .textLayer .endOfContent.active {     top: 0px; }   :host >>> .annotationLayer section {     position: absolute; }  :host >>> .annotationLayer .linkAnnotation > a {     position: absolute;     font-size: 1em;     top: 0;     left: 0;     width: 100%;     height: 100%; }  :host >>> .annotationLayer .linkAnnotation > a /* -ms-a */  {     background: url(\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\") 0 0 repeat; }  :host >>> .annotationLayer .linkAnnotation > a:hover {     opacity: 0.2;     background: #ff0;     box-shadow: 0px 2px 10px #ff0; }  :host >>> .annotationLayer .textAnnotation img {     position: absolute;     cursor: pointer; }  :host >>> .annotationLayer .popupWrapper {     position: absolute;     width: 20em; }  :host >>> .annotationLayer .popup {     position: absolute;     z-index: 200;     max-width: 20em;     background-color: #FFFF99;     box-shadow: 0px 2px 5px #333;     border-radius: 2px;     padding: 0.6em;     margin-left: 5px;     cursor: pointer;     word-wrap: break-word; }  :host >>> .annotationLayer .popup h1 {     font-size: 1em;     border-bottom: 1px solid #000000;     padding-bottom: 0.2em; }  :host >>> .annotationLayer .popup p {     padding-top: 0.2em; }  :host >>> .annotationLayer .highlightAnnotation, .annotationLayer .underlineAnnotation, .annotationLayer .squigglyAnnotation, .annotationLayer .strikeoutAnnotation, .annotationLayer .fileAttachmentAnnotation {     cursor: pointer; }  :host >>> .pdfViewer .canvasWrapper {     overflow: hidden; }  :host >>> .pdfViewer .page {     direction: ltr;     width: 816px;     height: 1056px;     margin: 1px auto -8px auto;     position: relative;     overflow: visible;     border: 9px solid transparent;     background-clip: content-box;     background-color: white; }  :host >>> .pdfViewer.removePageBorders .page {     margin: 0px auto 10px auto;     border: none; }  :host >>> .pdfViewer .page canvas {     margin: 0;     display: block; }  :host >>> .pdfViewer .page .loadingIcon {     position: absolute;     display: block;     left: 0;     top: 0;     right: 0;     bottom: 0; }  :host >>> .loadingIcon {     width: 100px;     height: 100px;     left: 50% !important;     top: 50% !important;      margin-top: -50px;     margin-left: -50px;      font-size: 5px;     text-indent: -9999em;     border-top: 1.1em solid rgba(3,0,2, 0.2);     border-right: 1.1em solid rgba(3,0,2, 0.2);     border-bottom: 1.1em solid rgba(3,0,2, 0.2);     border-left: 1.1em solid #030002;     -webkit-transform: translateZ(0);     -ms-transform: translateZ(0);     transform: translateZ(0);     -webkit-animation: load8 1.1s infinite linear;     animation: load8 1.1s infinite linear; } :host >>> .loadingIcon, :host >>> .loadingIcon:after {     border-radius: 50%; } @-webkit-keyframes load8 {     0% {         -webkit-transform: rotate(0deg);         transform: rotate(0deg);     }     100% {         -webkit-transform: rotate(360deg);         transform: rotate(360deg);     } } @keyframes load8 {     0% {         -webkit-transform: rotate(0deg);         transform: rotate(0deg);     }     100% {         -webkit-transform: rotate(360deg);         transform: rotate(360deg);     } }  :host >>> * {     padding: 0;     margin: 0; }  :host >>> html {     height: 100%;     width: 100%;     /* Font size is needed to make the activity bar the correct size. */     font-size: 10px; }  :host >>> body {     height: 100%;     width: 100%;     background-color: #404040;     background-image: url(images/texture.png); }  :host >>> body, input, button, select {     font: message-box;     outline: none; }  :host >>> .hidden {     display: none !important; } :host >>> [hidden] {     display: none !important; }   #viewer-pdf-container {     overflow: auto;     -webkit-overflow-scrolling: touch;     position: absolute;     top: 32px;     right: 0;     bottom: 0;     left: 0;     outline: none; } html[dir='ltr'] #viewer-pdf-container {     box-shadow: inset 1px 0 0 hsla(0,0%,100%,.05); } html[dir='rtl'] #viewer-pdf-container {     box-shadow: inset -1px 0 0 hsla(0,0%,100%,.05); }"],
        providers: [rendering_queue_services_1.RenderingQueueServices]
    }),
    __metadata("design:paramtypes", [rendering_queue_services_1.RenderingQueueServices,
        ng2_alfresco_core_1.LogService])
], PdfViewerComponent);
exports.PdfViewerComponent = PdfViewerComponent;
