// ng2-activiti-diagrams/src/services/diagrams.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var Rx_1 = require("rxjs/Rx");
var http_1 = require("@angular/http");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
var DiagramsService = (function () {
    function DiagramsService(authService, http, settingsService, logService) {
        this.authService = authService;
        this.http = http;
        this.settingsService = settingsService;
        this.logService = logService;
    }
    DiagramsService.prototype.getProcessDefinitionModel = function (processDefinitionId) {
        var _this = this;
        var url = this.settingsService.getBPMApiBaseUrl() + "/app/rest/process-definitions/" + processDefinitionId + "/model-json";
        var options = this.getRequestOptions();
        return this.http
            .get(url, options)
            .map(function (res) {
            var body = res.json();
            return body;
        }).catch(function (err) { return _this.handleError(err); });
    };
    DiagramsService.prototype.getHeaders = function () {
        return new http_1.Headers({
            'Accept': 'application/json',
            'Content-Type': 'application/json',
            'Authorization': this.authService.getTicketBpm()
        });
    };
    DiagramsService.prototype.getRequestOptions = function (param) {
        var headers = this.getHeaders();
        return new http_1.RequestOptions({ headers: headers, withCredentials: true, search: param });
    };
    DiagramsService.prototype.handleError = function (error) {
        this.logService.error(error);
        return Rx_1.Observable.throw(error.json().error || 'Server error');
    };
    return DiagramsService;
}());
DiagramsService = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoAuthenticationService,
        http_1.Http,
        ng2_alfresco_core_1.AlfrescoSettingsService,
        ng2_alfresco_core_1.LogService])
], DiagramsService);
exports.DiagramsService = DiagramsService;
