// ng2-activiti-diagrams/src/services/diagram-color.service.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var DiagramColorService = DiagramColorService_1 = (function () {
    function DiagramColorService() {
    }
    DiagramColorService.prototype.setTotalColors = function (totalColors) {
        this.totalColors = totalColors;
    };
    DiagramColorService.prototype.getFillOpacity = function () {
        return '0.6';
    };
    DiagramColorService.prototype.getFillColour = function (key) {
        if (this.totalColors && this.totalColors.hasOwnProperty(key)) {
            var colorPercentage = this.totalColors[key];
            return this.convertColorToHsb(colorPercentage);
        }
        else {
            return DiagramColorService_1.ACTIVITY_FILL_COLOR;
        }
    };
    DiagramColorService.prototype.getBpmnColor = function (data, defaultColor) {
        if (data.current) {
            return DiagramColorService_1.CURRENT_COLOR;
        }
        else if (data.completed) {
            return DiagramColorService_1.COMPLETED_COLOR;
        }
        else {
            return defaultColor;
        }
    };
    DiagramColorService.prototype.getBpmnStrokeWidth = function (data) {
        if (data.current || data.completed) {
            return DiagramColorService_1.TASK_HIGHLIGHT_STROKE;
        }
        else {
            return DiagramColorService_1.TASK_STROKE;
        }
    };
    DiagramColorService.prototype.convertColorToHsb = function (colorPercentage) {
        var hue = (120.0 - (colorPercentage * 1.2)) / 360.0;
        return 'hsb(' + hue + ', 1, 1)';
    };
    return DiagramColorService;
}());
DiagramColorService.CURRENT_COLOR = '#017501';
DiagramColorService.COMPLETED_COLOR = '#2632aa';
DiagramColorService.ACTIVITY_STROKE_COLOR = '#bbbbbb';
DiagramColorService.MAIN_STROKE_COLOR = '#585858';
DiagramColorService.ACTIVITY_FILL_COLOR = '#f9f9f9';
DiagramColorService.TASK_STROKE = 1;
DiagramColorService.TASK_HIGHLIGHT_STROKE = 2;
DiagramColorService.CALL_ACTIVITY_STROKE = 2;
DiagramColorService = DiagramColorService_1 = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [])
], DiagramColorService);
exports.DiagramColorService = DiagramColorService;
var DiagramColorService_1;
