// ng2-alfresco-datatable/index.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var ng2_alfresco_core_1 = require("ng2-alfresco-core");
__export(require("./src/data/index"));
__export(require("./src/components/index"));
__export(require("./src/components/pagination/index"));
var datatable_component_1 = require("./src/components/datatable/datatable.component");
var no_content_template_component_1 = require("./src/components/datatable/no-content-template.component");
var pagination_component_1 = require("./src/components/pagination/pagination.component");
exports.ALFRESCO_DATATABLE_DIRECTIVES = [
    datatable_component_1.DataTableComponent,
    no_content_template_component_1.NoContentTemplateComponent,
    pagination_component_1.PaginationComponent
];
var DataTableModule = DataTableModule_1 = (function () {
    function DataTableModule() {
    }
    DataTableModule.forRoot = function () {
        return {
            ngModule: DataTableModule_1
        };
    };
    return DataTableModule;
}());
DataTableModule = DataTableModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            ng2_alfresco_core_1.CoreModule
        ],
        declarations: exports.ALFRESCO_DATATABLE_DIRECTIVES.slice(),
        exports: exports.ALFRESCO_DATATABLE_DIRECTIVES.slice()
    })
], DataTableModule);
exports.DataTableModule = DataTableModule;
var DataTableModule_1;
