// ng2-alfresco-userinfo/bundles/ng2-alfresco-userinfo.js
!function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in v||(v[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==g.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=v[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(g.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=v[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}n=!n}}function u(e){return y[e]||(y[e]={name:e,dependencies:[],exports:{},importers:[]})}function d(r){if(!r.module){var t=r.module=u(r.name),n=r.module.exports,o=r.declare.call(e,function(e,r){if(t.locked=!0,"object"==typeof e)for(var o in e)n[o]=e[o];else n[e]=r;for(var a=0,u=t.importers.length;u>a;a++){var d=t.importers[a];if(!d.locked)for(var i=0;i<d.dependencies.length;++i)d.dependencies[i]===t&&d.setters[i](n)}return t.locked=!1,r},{id:r.name});t.setters=o.setters,t.execute=o.execute;for(var a=0,i=r.normalizedDeps.length;i>a;a++){var l,s=r.normalizedDeps[a],c=v[s],f=y[s];f?l=f.exports:c&&!c.declarative?l=c.esModule:c?(d(c),f=c.module,l=f.exports):l=p(s),f&&f.importers?(f.importers.push(t),t.dependencies.push(f)):t.dependencies.push(null),t.setters[a]&&t.setters[a](l)}}}function i(e){var r,t=v[e];if(t)t.declarative?f(e,[]):t.evaluated||l(t),r=t.module.exports;else if(r=p(e),!r)throw new Error("Unable to load dependency "+e+".");return(!t||t.declarative)&&r&&r.__useDefault?r["default"]:r}function l(r){if(!r.module){var t={},n=r.module={exports:t,id:r.name};if(!r.executingRequire)for(var o=0,a=r.normalizedDeps.length;a>o;o++){var u=r.normalizedDeps[o],d=v[u];d&&l(d)}r.evaluated=!0;var c=r.execute.call(e,function(e){for(var t=0,n=r.deps.length;n>t;t++)if(r.deps[t]==e)return i(r.normalizedDeps[t]);throw new TypeError("Module "+e+" not declared as a dependency.")},t,n);void 0!==c&&(n.exports=c),t=n.exports,t&&t.__esModule?r.esModule=t:r.esModule=s(t)}}function s(r){var t={};if(("object"==typeof r||"function"==typeof r)&&r!==e)if(m)for(var n in r)"default"!==n&&c(t,r,n);else{var o=r&&r.hasOwnProperty;for(var n in r)"default"===n||o&&!r.hasOwnProperty(n)||(t[n]=r[n])}return t["default"]=r,x(t,"__useDefault",{value:!0}),t}function c(e,r,t){try{var n;(n=Object.getOwnPropertyDescriptor(r,t))&&x(e,t,n)}catch(o){return e[t]=r[t],!1}}function f(r,t){var n=v[r];if(n&&!n.evaluated&&n.declarative){t.push(r);for(var o=0,a=n.normalizedDeps.length;a>o;o++){var u=n.normalizedDeps[o];-1==g.call(t,u)&&(v[u]?f(u,t):p(u))}n.evaluated||(n.evaluated=!0,n.module.execute.call(e))}}function p(e){if(I[e])return I[e];if("@node/"==e.substr(0,6))return I[e]=s(D(e.substr(6)));var r=v[e];if(!r)throw"Module "+e+" not present.";return a(e),f(e,[]),v[e]=void 0,r.declarative&&x(r.module.exports,"__esModule",{value:!0}),I[e]=r.declarative?r.module.exports:r.esModule}var v={},g=Array.prototype.indexOf||function(e){for(var r=0,t=this.length;t>r;r++)if(this[r]===e)return r;return-1},m=!0;try{Object.getOwnPropertyDescriptor({a:0},"a")}catch(h){m=!1}var x;!function(){try{Object.defineProperty({},"a",{})&&(x=Object.defineProperty)}catch(e){x=function(e,r,t){try{e[r]=t.value||t.get.call(e)}catch(n){}}}}();var y={},D="undefined"!=typeof System&&System._nodeRequire||"undefined"!=typeof require&&"undefined"!=typeof require.resolve&&"undefined"!=typeof process&&process.platform&&require,I={"@empty":{}};return function(e,n,o,a){return function(u){u(function(u){for(var d={_nodeRequire:D,register:r,registerDynamic:t,get:p,set:function(e,r){I[e]=r},newModule:function(e){return e}},i=0;i<n.length;i++)(function(e,r){r&&r.__esModule?I[e]=r:I[e]=s(r)})(n[i],arguments[i]);a(d);var l=p(e[0]);if(e.length>1)for(var i=1;i<e.length;i++)p(e[i]);return o?l["default"]:l})}}}("undefined"!=typeof self?self:global)

(["1"], ["3","4","7"], true, function($__System) {
var require = this.require, exports = this.exports, module = this.module;
$__System.registerDynamic("2", ["3", "4", "5", "6"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("4");
    var ecm_user_service_1 = $__require("5");
    var bpm_user_service_1 = $__require("6");
    var UserInfoComponent = function () {
        function UserInfoComponent(ecmUserService, bpmUserService, authService, translateService) {
            var _this = this;
            this.ecmUserService = ecmUserService;
            this.bpmUserService = bpmUserService;
            this.authService = authService;
            this.translateService = translateService;
            this.menuOpenType = 'right';
            this.baseComponentPath = module.id.replace('components/user-info.component.js', '');
            this.anonymousImageUrl = this.baseComponentPath + '/assets/images/anonymous.gif';
            if (translateService) {
                translateService.addTranslationFolder('ng2-alfresco-userinfo', 'node_modules/ng2-alfresco-userinfo/src');
            }
            authService.loginSubject.subscribe(function (response) {
                _this.getUserInfo();
            });
        }
        UserInfoComponent.prototype.ngOnInit = function () {
            this.getUserInfo();
        };
        UserInfoComponent.prototype.getUserInfo = function () {
            this.getEcmUserInfo();
            this.getBpmUserInfo();
        };
        UserInfoComponent.prototype.isLoggedIn = function () {
            return this.authService.isLoggedIn();
        };
        UserInfoComponent.prototype.getEcmUserInfo = function () {
            var _this = this;
            if (this.authService.isEcmLoggedIn()) {
                this.ecmUserService.getCurrentUserInfo().subscribe(function (res) {
                    _this.ecmUser = res;
                    _this.getEcmAvatar();
                });
            } else {
                this.ecmUser = null;
                this.ecmUserImage = null;
            }
        };
        UserInfoComponent.prototype.getBpmUserInfo = function () {
            var _this = this;
            if (this.authService.isBpmLoggedIn()) {
                this.bpmUserService.getCurrentUserInfo().subscribe(function (res) {
                    _this.bpmUser = res;
                });
                this.bpmUserImage = this.bpmUserService.getCurrentUserProfileImage();
            } else {
                this.bpmUser = null;
                this.bpmUserImage = null;
            }
        };
        UserInfoComponent.prototype.onImageLoadingError = function (event) {
            if (event) {
                var element = event.target;
                element.src = this.fallBackThumbnailImage || this.anonymousImageUrl;
            }
        };
        UserInfoComponent.prototype.stopClosing = function (event) {
            event.stopPropagation();
        };
        UserInfoComponent.prototype.getUserNameHeaderFor = function (env) {
            if (this.ecmUser && env === 'ECM') {
                return this.ecmUser.firstName || this.ecmUser.lastName;
            }
            if (this.bpmUser && env === 'BPM') {
                return this.formatValue(this.bpmUser.firstName) || this.formatValue(this.bpmUser.lastName) || this.formatValue(this.bpmUser.fullname);
            }
        };
        UserInfoComponent.prototype.getEcmAvatar = function () {
            this.ecmUserImage = this.ecmUserService.getUserProfileImage(this.ecmUser.avatarId);
        };
        UserInfoComponent.prototype.getUserAvatar = function () {
            return this.ecmUserImage || this.bpmUserImage;
        };
        UserInfoComponent.prototype.getBpmUserAvatar = function () {
            return this.bpmUserImage;
        };
        UserInfoComponent.prototype.getEcmUserAvatar = function () {
            return this.ecmUserImage;
        };
        UserInfoComponent.prototype.formatValue = function (value) {
            return value === 'null' ? null : value;
        };
        return UserInfoComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String)], UserInfoComponent.prototype, "ecmBackgroundImage", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], UserInfoComponent.prototype, "bpmBackgroundImage", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], UserInfoComponent.prototype, "menuOpenType", void 0);
    __decorate([core_1.Input(), __metadata("design:type", String)], UserInfoComponent.prototype, "fallBackThumbnailImage", void 0);
    UserInfoComponent = __decorate([core_1.Component({
        selector: 'ng2-alfresco-userinfo',
        moduleId: module.id,
        styles: [".profile-image {   text-align: center;   border-radius: 90%;   width: 40px;   height: 40px;   margin-right: 0%;   cursor: pointer;   border: 1px solid #999999;   vertical-align: middle;   background-color:white;; }  .button-profile {   display: inline-block;   border: 0px;   vertical-align: middle; }  .detail-user-profile-list-mdl{     padding: 0px 0; }  @media only screen and (min-device-width: 480px) {     .user-profile-list-mdl{         max-height: 450px;         min-width: 450px;         overflow: auto;         padding:0px;     } }  .header-profile{     color: rgb(255,152,0);     margin-left: 10px; }  hr.title-start {     border: 0;     height: 1px;     background: #333;     background-image: linear-gradient(to right, #ccc, #333, #ccc); }  span.role-label-user{     font-weight: 400;     line-height: 1;     letter-spacing: 0;     color: rgba(0,0,0,.87); }  .custom-role-style{     font-size: 14px;     color: #9e9e9e; }  .truncate-long-names{     text-overflow: ellipsis; } .demo-card-wide.mdl-card {     width: inherit; } .demo-card-wide > .mdl-card__title {     color: #fff;     height: 176px; } .demo-card-wide > .mdl-card__menu {     color: #fff; }  .card-title__option{     height: 100px!important; }  img.profile-picture {     background-color: white;     background-size: cover;     border-radius: 50%;     box-shadow: inset 1px 1px 3px rgba(0,0,0,0.2), 1px 1px 4px rgba(0,0,0,0.3);     height: 80px;     width: 80px;     z-index: 3;     margin-left: 0px;     margin-right: 8px; }"],
        template: "<div id=\"userinfo_container\" *ngIf=\"isLoggedIn()\">     <div *ngIf=\"ecmUser || bpmUser\">         <div class=\"button-profile\" id=\"user-profile\" data-automation-id=\"user-profile\">             <img id=\"logged-user-img\"                  [src]=\"getUserAvatar()\"                  alt=\"user-info-profile-button\"                  (error)=\"onImageLoadingError($event)\"                  class=\"profile-image\"/>         </div>         <div mdl id=\"user-profile-lists\" class=\"user-profile-list-mdl mdl-menu mdl-js-menu mdl-js-ripple-effect\"              [class.mdl-menu--bottom-left]=\"menuOpenType === 'left'? true : false\"              [class.mdl-menu--bottom-right]=\"menuOpenType === 'right'? true : false\"              for=\"user-profile\" (click)=\"stopClosing($event)\">             <div class=\"mdl-tabs mdl-js-tabs mdl-js-ripple-effect menu-container__items\">                 <div id=\"tab-bar-env\" class=\"mdl-tabs__tab-bar\" [hidden]=\"!(ecmUser && bpmUser)\">                     <a href=\"#ecm-panel\" id=\"ecm-tab\" class=\"mdl-tabs__tab is-active\">ECM</a>                     <a href=\"#bpm-panel\" id=\"bpm-tab\" class=\"mdl-tabs__tab\">BPM</a>                 </div>                 <div class=\"mdl-tabs__panel\" [class.is-active]=\"ecmUser?true:false\" id=\"ecm-panel\">                     <div class=\"detail-user-profile-list-mdl mdl-list\" *ngIf=\"ecmUser\">                         <div class=\"demo-card-wide mdl-card\">                             <div class=\"card-title__option mdl-card__title\"                                  id=\"ecm-background-image\"                                  [style.background-image]=\"'url(' + ( ecmBackgroundImage || baseComponentPath + '/assets/images/ecm-background.png')+')'\">                                 <img class=\"profile-picture\"                                      id=\"ecm-user-detail-image\"                                      alt=\"ecm-profile-image\"                                      (error)=\"onImageLoadingError($event)\"                                      [src]=\"getEcmUserAvatar()\"/>                                 <h2 class=\"mdl-card__title-text\" id=\"ecm-username\">{{getUserNameHeaderFor('ECM')}}</h2>                             </div>                             <div class=\"mdl-card__supporting-text\">                                 <li class=\"mdl-list__item mdl-list__item--two-line\">                                     <span class=\"mdl-list__item-primary-content\">                                     <span id=\"ecm-full-name\" class=\"truncate-long-names\">                                         {{ formatValue(ecmUser.firstName) }} {{ formatValue(ecmUser.lastName) }}                                     </span>                                     <span id=\"ecm-email\" class=\"mdl-list__item-sub-title\">{{ecmUser.email}}</span>                                     </span>                                     <span id=\"ecm-job-title\" class=\"mdl-list__item-secondary-content custom-role-style\">                                         <span                                             class=\"role-label-user\">{{ 'USER_PROFILE.LABELS.ECM.JOB_TITLE' | translate }}</span>                                         {{ecmUser.jobTitle?ecmUser.jobTitle:'N/A'}}                                     </span>                                 </li>                             </div>                         </div>                     </div>                 </div>                 <div class=\"mdl-tabs__panel\" [class.is-active]=\"bpmUser && !ecmUser?true:false\" id=\"bpm-panel\">                     <div class=\"detail-user-profile-list-mdl mdl-list\" *ngIf=\"bpmUser\">                         <div class=\"demo-card-wide mdl-card\">                             <div class=\"card-title__option mdl-card__title\"                                  id=\"bpm-background-image\"                                  [style.background-image]=\"'url(' + (bpmBackgroundImage || baseComponentPath + '/assets/images/bpm-background.png')+')'\">                                 <img class=\"profile-picture\"                                      id=\"bpm-user-detail-image\"                                      alt=\"bpm-profile-image\"                                      (error)=\"onImageLoadingError($event)\"                                      [src]=\"getBpmUserAvatar()\"/>                                 <h2 class=\"mdl-card__title-text\" id=\"bpm-username\">{{getUserNameHeaderFor('BPM')}}</h2>                             </div>                             <div class=\"mdl-card__supporting-text\">                                 <li class=\"mdl-list__item mdl-list__item--two-line\">                             <span class=\"mdl-list__item-primary-content\">                             <span id=\"bpm-full-name\" class=\"truncate-long-names\">                                 {{ formatValue(bpmUser.firstName) }} {{ formatValue(bpmUser.lastName) }}                             </span>                             <span id=\"bpm-email\" class=\"mdl-list__item-sub-title\">{{bpmUser.email}}</span>                             </span>                                     <span id=\"bpm-tenant\" class=\"mdl-list__item-secondary-content custom-role-style\">                                 <span class=\"role-label-user\">{{ 'USER_PROFILE.LABELS.BPM.TENANT' | translate }}</span>                                 {{bpmUser.tenantName}}                             </span>                                 </li>                             </div>                         </div>                     </div>                 </div>             </div>         </div>      </div> </div>"
    }), __metadata("design:paramtypes", [ecm_user_service_1.EcmUserService, bpm_user_service_1.BpmUserService, ng2_alfresco_core_1.AlfrescoAuthenticationService, ng2_alfresco_core_1.AlfrescoTranslationService])], UserInfoComponent);
    exports.UserInfoComponent = UserInfoComponent;
});
$__System.registerDynamic("5", ["3", "7", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("7");
    var ng2_alfresco_core_1 = $__require("4");
    var EcmUserService = function () {
        function EcmUserService(apiService, contentService, logService) {
            this.apiService = apiService;
            this.contentService = contentService;
            this.logService = logService;
        }
        EcmUserService.prototype.getUserInfo = function (userName) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.callApiGetPersonInfo(userName)).map(function (data) {
                return data['entry'];
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        EcmUserService.prototype.getCurrentUserInfo = function () {
            return this.getUserInfo('-me-');
        };
        EcmUserService.prototype.callApiGetPersonInfo = function (userName, opts) {
            return this.apiService.getInstance().core.peopleApi.getPerson(userName, opts);
        };
        EcmUserService.prototype.getUserProfileImage = function (avatarId) {
            if (avatarId) {
                var nodeObj = { entry: { id: avatarId } };
                return this.contentService.getContentUrl(nodeObj);
            }
        };
        EcmUserService.prototype.handleError = function (error) {
            this.logService.error(error);
            return Rx_1.Observable.throw(error || 'Server error');
        };
        return EcmUserService;
    }();
    EcmUserService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoApiService, ng2_alfresco_core_1.AlfrescoContentService, ng2_alfresco_core_1.LogService])], EcmUserService);
    exports.EcmUserService = EcmUserService;
});
$__System.registerDynamic("6", ["3", "7", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("7");
    var ng2_alfresco_core_1 = $__require("4");
    var BpmUserService = function () {
        function BpmUserService(apiService, logService) {
            this.apiService = apiService;
            this.logService = logService;
        }
        BpmUserService.prototype.getCurrentUserInfo = function () {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().activiti.profileApi.getProfile()).map(function (data) {
                return data;
            }).catch(function (err) {
                return _this.handleError(err);
            });
        };
        BpmUserService.prototype.getCurrentUserProfileImage = function () {
            return this.apiService.getInstance().activiti.profileApi.getProfilePictureUrl();
        };
        BpmUserService.prototype.handleError = function (error) {
            this.logService.error(error);
            return Rx_1.Observable.throw(error || 'Server error');
        };
        return BpmUserService;
    }();
    BpmUserService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoApiService, ng2_alfresco_core_1.LogService])], BpmUserService);
    exports.BpmUserService = BpmUserService;
});
$__System.registerDynamic("1", ["3", "4", "2", "5", "6"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("4");
    var user_info_component_1 = $__require("2");
    var ecm_user_service_1 = $__require("5");
    var bpm_user_service_1 = $__require("6");
    __export($__require("2"));
    __export($__require("6"));
    __export($__require("5"));
    exports.USER_INFO_DIRECTIVES = [user_info_component_1.UserInfoComponent];
    exports.USER_INFO_PROVIDERS = [ecm_user_service_1.EcmUserService, bpm_user_service_1.BpmUserService];
    var UserInfoComponentModule = UserInfoComponentModule_1 = function () {
        function UserInfoComponentModule() {}
        UserInfoComponentModule.forRoot = function () {
            return {
                ngModule: UserInfoComponentModule_1,
                providers: exports.USER_INFO_PROVIDERS.slice()
            };
        };
        return UserInfoComponentModule;
    }();
    UserInfoComponentModule = UserInfoComponentModule_1 = __decorate([core_1.NgModule({
        imports: [ng2_alfresco_core_1.CoreModule],
        declarations: exports.USER_INFO_DIRECTIVES.slice(),
        providers: exports.USER_INFO_PROVIDERS.slice(),
        exports: exports.USER_INFO_DIRECTIVES.slice()
    })], UserInfoComponentModule);
    exports.UserInfoComponentModule = UserInfoComponentModule;
    var UserInfoComponentModule_1;
    
});
})
(function(factory) {
  if (typeof define == 'function' && define.amd)
    define(["@angular/core","ng2-alfresco-core","rxjs/Rx"], factory);
  else if (typeof module == 'object' && module.exports && typeof require == 'function')
    module.exports = factory(require("@angular/core"), require("ng2-alfresco-core"), require("rxjs/Rx"));
  else
    throw new Error("Module must be loaded as AMD or CommonJS");
});
