// ng2-activiti-form/src/components/widgets/amount/amount.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var core_1 = require("@angular/core");
var widget_component_1 = require("./../widget.component");
var AmountWidget = AmountWidget_1 = (function (_super) {
    __extends(AmountWidget, _super);
    function AmountWidget() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.currency = AmountWidget_1.DEFAULT_CURRENCY;
        return _this;
    }
    AmountWidget.prototype.ngOnInit = function () {
        if (this.field && this.field.currency) {
            this.currency = this.field.currency;
        }
    };
    return AmountWidget;
}(widget_component_1.WidgetComponent));
AmountWidget.DEFAULT_CURRENCY = '$';
AmountWidget = AmountWidget_1 = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'amount-widget',
        template: "<div class=\"mdl-textfield mdl-js-textfield amount-widget\"      [class.amount-widget__invalid]=\"!field.isValid\">     <label [attr.for]=\"field.id\">{{field.name}}, {{currency}}</label>     <input class=\"mdl-textfield__input\"            type=\"text\"            [attr.id]=\"field.id\"            [attr.required]=\"isRequired()\"            [(ngModel)]=\"field.value\"            (ngModelChange)=\"checkVisibility(field)\"            [disabled]=\"field.readOnly\"            placeholder=\"{{field.placeholder}}\">     <span *ngIf=\"field.validationSummary\" class=\"mdl-textfield__error\">{{field.validationSummary}}</span> </div>",
        styles: [".amount-widget {     width: 100%; }   .amount-widget__invalid .mdl-textfield__input {     border-color: #d50000; }  .amount-widget__invalid .mdl-textfield__label {     color: #d50000; }  .amount-widget__invalid .mdl-textfield__label:after {     background-color: #d50000; }  .amount-widget__invalid .mdl-textfield__error {     visibility: visible !important; }"]
    })
], AmountWidget);
exports.AmountWidget = AmountWidget;
var AmountWidget_1;
