// ng2-activiti-form/src/components/widgets/container/container.widget.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var core_1 = require("@angular/core");
var container_widget_model_1 = require("./container.widget.model");
var widget_component_1 = require("./../widget.component");
var ContainerWidget = (function (_super) {
    __extends(ContainerWidget, _super);
    function ContainerWidget() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ContainerWidget.prototype.onExpanderClicked = function () {
        if (this.content && this.content.isCollapsible()) {
            this.content.isExpanded = !this.content.isExpanded;
        }
    };
    ContainerWidget.prototype.ngOnInit = function () {
        if (this.field) {
            this.content = new container_widget_model_1.ContainerWidgetModel(this.field);
        }
    };
    ContainerWidget.prototype.ngAfterViewInit = function () {
        this.setupMaterialComponents();
    };
    ContainerWidget.prototype.setupMaterialComponents = function () {
        if (componentHandler) {
            componentHandler.upgradeAllRegistered();
            return true;
        }
        return false;
    };
    return ContainerWidget;
}(widget_component_1.WidgetComponent));
ContainerWidget = __decorate([
    core_1.Component({
        moduleId: module.id,
        selector: 'container-widget',
        template: "<div class=\"container-widget\">     <div [ngClass]=\"{'hidden':!(content?.isGroup() && content?.isVisible)}\" class=\"container-widget__header\">         <h4 class=\"container-widget__header-text\" id=\"container-header\"             [class.collapsible]=\"content?.isCollapsible()\">             <button *ngIf=\"content?.isCollapsible()\"                     alfresco-mdl-button                     class=\"mdl-button--icon\"                     (click)=\"onExpanderClicked()\">                 <i class=\"material-icons\">{{ content?.isExpanded ? 'expand_less' : 'expand_more' }}</i>             </button>             <span (click)=\"onExpanderClicked()\" id=\"container-header-label\">{{content.name}}</span>         </h4>     </div>     <div class=\"mdl-grid container-widget__mdl-grid\" [ngClass]=\"{'hidden':!(content?.isVisible && content?.isExpanded)}\">         <div *ngFor=\"let col of content.columns\" class=\"mdl-cell mdl-cell--{{col.size}}-col\">             <div class=\"mdl-grid container-widget__mdl-grid\" *ngIf=\"col.hasFields()\">                 <div *ngFor=\"let field of col.fields\" class=\"mdl-cell mdl-cell--12-col\">                     <form-field [field]=\"field\"></form-field>                 </div>             </div>         </div>     </div> </div>",
        styles: [".container-widget {}  .container-widget__header {}  .container-widget__header-text {     border-bottom: 1px solid rgba(0,0,0,.87);     margin-left: 10px;     margin-right: 10px;     cursor: default;      user-select: none;     -webkit-user-select: none;   /* Chrome/Safari/Opera */     -moz-user-select: none;      /* Firefox */     -ms-user-select: none;       /* IE/Edge */     -webkit-touch-callout: none; /* iOS Safari */ }  .container-widget__header-text.collapsible {     cursor: pointer; }  .container-widget__mdl-grid {     align-items: center; }"]
    })
], ContainerWidget);
exports.ContainerWidget = ContainerWidget;
