// ng2-activiti-processlist/src/assets/activiti-process.model.mock.js
/*!
 * @license
 * Copyright 2016 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var process_instance_model_1 = require("../models/process-instance.model");
var ProcessList = (function () {
    function ProcessList(data) {
        this.data = data || [];
    }
    return ProcessList;
}());
exports.ProcessList = ProcessList;
var SingleProcessList = (function (_super) {
    __extends(SingleProcessList, _super);
    function SingleProcessList(name) {
        var _this = this;
        var instance = new process_instance_model_1.ProcessInstance({
            id: '123',
            name: name
        });
        _this = _super.call(this, [instance]) || this;
        return _this;
    }
    return SingleProcessList;
}(ProcessList));
exports.SingleProcessList = SingleProcessList;
exports.exampleProcess = new process_instance_model_1.ProcessInstance({
    id: '123',
    name: 'Process 123',
    started: '2016-11-10T03:37:30.010+0000',
    startedBy: {
        id: 1001,
        firstName: 'Bob',
        lastName: 'Jones',
        email: 'bob@app.activiti.com'
    }
});
exports.exampleProcessNoName = new process_instance_model_1.ProcessInstance({
    id: '123',
    name: null,
    started: '2016-11-10T03:37:30.010+0000',
    startedBy: {
        id: 1001,
        firstName: 'Bob',
        lastName: 'Jones',
        email: 'bob@app.activiti.com'
    },
    processDefinitionName: 'My Process'
});
