// ng2-alfresco-tag/bundles/ng2-alfresco-tag.js
!function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in v||(v[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==g.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=v[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(g.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=v[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}n=!n}}function u(e){return y[e]||(y[e]={name:e,dependencies:[],exports:{},importers:[]})}function d(r){if(!r.module){var t=r.module=u(r.name),n=r.module.exports,o=r.declare.call(e,function(e,r){if(t.locked=!0,"object"==typeof e)for(var o in e)n[o]=e[o];else n[e]=r;for(var a=0,u=t.importers.length;u>a;a++){var d=t.importers[a];if(!d.locked)for(var i=0;i<d.dependencies.length;++i)d.dependencies[i]===t&&d.setters[i](n)}return t.locked=!1,r},{id:r.name});t.setters=o.setters,t.execute=o.execute;for(var a=0,i=r.normalizedDeps.length;i>a;a++){var l,s=r.normalizedDeps[a],c=v[s],f=y[s];f?l=f.exports:c&&!c.declarative?l=c.esModule:c?(d(c),f=c.module,l=f.exports):l=p(s),f&&f.importers?(f.importers.push(t),t.dependencies.push(f)):t.dependencies.push(null),t.setters[a]&&t.setters[a](l)}}}function i(e){var r,t=v[e];if(t)t.declarative?f(e,[]):t.evaluated||l(t),r=t.module.exports;else if(r=p(e),!r)throw new Error("Unable to load dependency "+e+".");return(!t||t.declarative)&&r&&r.__useDefault?r["default"]:r}function l(r){if(!r.module){var t={},n=r.module={exports:t,id:r.name};if(!r.executingRequire)for(var o=0,a=r.normalizedDeps.length;a>o;o++){var u=r.normalizedDeps[o],d=v[u];d&&l(d)}r.evaluated=!0;var c=r.execute.call(e,function(e){for(var t=0,n=r.deps.length;n>t;t++)if(r.deps[t]==e)return i(r.normalizedDeps[t]);throw new TypeError("Module "+e+" not declared as a dependency.")},t,n);void 0!==c&&(n.exports=c),t=n.exports,t&&t.__esModule?r.esModule=t:r.esModule=s(t)}}function s(r){var t={};if(("object"==typeof r||"function"==typeof r)&&r!==e)if(m)for(var n in r)"default"!==n&&c(t,r,n);else{var o=r&&r.hasOwnProperty;for(var n in r)"default"===n||o&&!r.hasOwnProperty(n)||(t[n]=r[n])}return t["default"]=r,x(t,"__useDefault",{value:!0}),t}function c(e,r,t){try{var n;(n=Object.getOwnPropertyDescriptor(r,t))&&x(e,t,n)}catch(o){return e[t]=r[t],!1}}function f(r,t){var n=v[r];if(n&&!n.evaluated&&n.declarative){t.push(r);for(var o=0,a=n.normalizedDeps.length;a>o;o++){var u=n.normalizedDeps[o];-1==g.call(t,u)&&(v[u]?f(u,t):p(u))}n.evaluated||(n.evaluated=!0,n.module.execute.call(e))}}function p(e){if(I[e])return I[e];if("@node/"==e.substr(0,6))return I[e]=s(D(e.substr(6)));var r=v[e];if(!r)throw"Module "+e+" not present.";return a(e),f(e,[]),v[e]=void 0,r.declarative&&x(r.module.exports,"__esModule",{value:!0}),I[e]=r.declarative?r.module.exports:r.esModule}var v={},g=Array.prototype.indexOf||function(e){for(var r=0,t=this.length;t>r;r++)if(this[r]===e)return r;return-1},m=!0;try{Object.getOwnPropertyDescriptor({a:0},"a")}catch(h){m=!1}var x;!function(){try{Object.defineProperty({},"a",{})&&(x=Object.defineProperty)}catch(e){x=function(e,r,t){try{e[r]=t.value||t.get.call(e)}catch(n){}}}}();var y={},D="undefined"!=typeof System&&System._nodeRequire||"undefined"!=typeof require&&"undefined"!=typeof require.resolve&&"undefined"!=typeof process&&process.platform&&require,I={"@empty":{}};return function(e,n,o,a){return function(u){u(function(u){for(var d={_nodeRequire:D,register:r,registerDynamic:t,get:p,set:function(e,r){I[e]=r},newModule:function(e){return e}},i=0;i<n.length;i++)(function(e,r){r&&r.__esModule?I[e]=r:I[e]=s(r)})(n[i],arguments[i]);a(d);var l=p(e[0]);if(e.length>1)for(var i=1;i<e.length;i++)p(e[i]);return o?l["default"]:l})}}}("undefined"!=typeof self?self:global)

(["1"], ["3","8","7"], true, function($__System) {
var require = this.require, exports = this.exports, module = this.module;
$__System.registerDynamic("2", ["3", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var tag_service_1 = $__require("4");
    var TagActionsComponent = function () {
        function TagActionsComponent(tagService) {
            this.tagService = tagService;
            this.isContextMenu = false;
            this.addEmitter = new core_1.EventEmitter();
            this.resultsEmitter = new core_1.EventEmitter();
        }
        TagActionsComponent.prototype.ngOnChanges = function (changes) {
            return this.refreshTag();
        };
        TagActionsComponent.prototype.refreshTag = function () {
            var _this = this;
            this.tagService.getTagsByNodeId(this.nodeId).subscribe(function (data) {
                _this.tagsEntries = data.list.entries;
                _this.resultsEmitter.emit(_this.tagsEntries);
            });
        };
        TagActionsComponent.prototype.addTag = function () {
            var _this = this;
            this.tagService.addTag(this.nodeId, this.newTagName).subscribe(function (res) {
                _this.refreshTag();
                _this.addEmitter.emit(_this.nodeId);
            });
        };
        TagActionsComponent.prototype.removeTag = function (tag) {
            var _this = this;
            this.tagService.removeTag(this.nodeId, tag).subscribe(function () {
                _this.refreshTag();
            });
        };
        return TagActionsComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String)], TagActionsComponent.prototype, "nodeId", void 0);
    __decorate([core_1.Input(), __metadata("design:type", Boolean)], TagActionsComponent.prototype, "isContextMenu", void 0);
    __decorate([core_1.Output(), __metadata("design:type", core_1.EventEmitter)], TagActionsComponent.prototype, "addEmitter", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], TagActionsComponent.prototype, "resultsEmitter", void 0);
    TagActionsComponent = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-tag-node-actions-list',
        template: "<!--isContextMenu--> <div *ngIf=\"isContextMenu\">     <li class=\"mdl-menu__item\" *ngFor=\"let currentEntry of tagsEntries; let idx = index\">             <span id=\"tag_name_{{idx}}\" class=\"mdl-list__item-primary-content\">                 {{currentEntry.entry.tag}}             </span>         <div id=\"tag_delete_{{idx}}\" class=\"mdl-list__item-secondary-action\" (click)=\"removeTag(currentEntry.entry.id)\">             <i class=\"material-icons\">delete</i>         </div>     </li>     <li>         <div class=\"mdl-textfield mdl-js-textfield tag-container-input\">             <input class=\"mdl-textfield__input tag-input\" type=\"text\" id=\"new-tag-text\" [(ngModel)]=\"newTagName\"/>             <label class=\"mdl-textfield__label tag-label\" for=\"new-tag-text\">New Tag...</label>         </div>         <button id=\"add-tag\" class=\"mdl-button mdl-js-button mdl-button--raised button\" (click)=\"addTag()\">             Add tag         </button>     </li> </div>  <!--not ContextMenu--> <ul *ngIf=\"!isContextMenu\" class=\"tag-list-control mdl-list\">     <li class=\"mdl-list__item tag-element\" *ngFor=\"let currentEntry of tagsEntries; let idx = index\">         <span id=\"tag_name_{{idx}}\" class=\"mdl-list__item-primary-content\">{{currentEntry.entry.tag}}</span>         <div id=\"tag_delete_{{idx}}\" class=\"mdl-list__item-secondary-action\" (click)=\"removeTag(currentEntry.entry.id)\">             <i class=\"material-icons\">delete</i>         </div>     </li>     <li>         <div class=\"mdl-textfield mdl-js-textfield tag-container-input\">             <input class=\"mdl-textfield__input tag-input\" type=\"text\" id=\"new-tag-text\" [(ngModel)]=\"newTagName\"/>             <label class=\"mdl-textfield__label tag-label\" for=\"new-tag-text\">New Tag...</label>         </div>         <button id=\"add-tag\" class=\"mdl-button mdl-js-button mdl-button--raised button\" (click)=\"addTag()\">             Add tag         </button>     </li> </ul>",
        styles: [".tag-container {     width: 300px; }  .tag-element {     padding: 0px 0px 0px 9px!important; }  .tag-input {     float: left;     margin-left: 11px!important; }  .tag-label {     left: 10px!important;     top: 12px!important; }  .tag-container-input{     padding-top: 11px;     width: 190px;     float: left;     margin-right: 15px; }  .tag-button{     float:left }"]
    }), __metadata("design:paramtypes", [tag_service_1.TagService])], TagActionsComponent);
    exports.TagActionsComponent = TagActionsComponent;
});
$__System.registerDynamic("5", ["3", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var tag_service_1 = $__require("4");
    var TagList = function () {
        function TagList(tagService) {
            this.tagService = tagService;
            this.resultsEmitter = new core_1.EventEmitter();
        }
        TagList.prototype.ngOnInit = function (changes) {
            return this.refreshTagEcm();
        };
        TagList.prototype.refreshTagEcm = function () {
            var _this = this;
            this.tagService.getAllTheTags().subscribe(function (data) {
                _this.tagsEntries = data.list.entries;
                _this.resultsEmitter.emit(_this.tagsEntries);
            });
        };
        return TagList;
    }();
    __decorate([core_1.Output(), __metadata("design:type", Object)], TagList.prototype, "resultsEmitter", void 0);
    TagList = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-tag-list',
        template: "<div class=\"tag-container\">     <div id=\"table-cards-container\">         <ul class=\"tag-list-control mdl-list\">             <li class=\"mdl-list__item tag-element\" *ngFor=\"let currentEntry of tagsEntries; let idx = index\">                 <span id=\"tag_name_{{idx}}\"  class=\"mdl-list__item-primary-content\">{{currentEntry.entry.tag}}</span>             </li>         </ul>     </div> </div>"
    }), __metadata("design:paramtypes", [tag_service_1.TagService])], TagList);
    exports.TagList = TagList;
});
$__System.registerDynamic("6", ["3", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var tag_service_1 = $__require("4");
    var TagNodeList = function () {
        function TagNodeList(tagService) {
            this.tagService = tagService;
            this.resultsEmitter = new core_1.EventEmitter();
        }
        TagNodeList.prototype.ngOnChanges = function (changes) {
            return this.refreshTag();
        };
        TagNodeList.prototype.refreshTag = function () {
            var _this = this;
            this.tagService.getTagsByNodeId(this.nodeId).subscribe(function (data) {
                _this.tagsEntries = data.list.entries;
                _this.resultsEmitter.emit(_this.tagsEntries);
            });
        };
        TagNodeList.prototype.removeTag = function (tag) {
            var _this = this;
            this.tagService.removeTag(this.nodeId, tag).subscribe(function () {
                _this.refreshTag();
            });
        };
        return TagNodeList;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String)], TagNodeList.prototype, "nodeId", void 0);
    __decorate([core_1.Output(), __metadata("design:type", Object)], TagNodeList.prototype, "resultsEmitter", void 0);
    TagNodeList = __decorate([core_1.Component({
        moduleId: module.id,
        selector: 'alfresco-tag-node-list',
        template: "<div class=\"tag-list-container\">     <ul class=\"tags\">         <li *ngFor=\"let currentEntry of tagsEntries; let idx = index\">             <span class=\"mdl-chip mdl-chip--deletable\">                  <span id=\"tag_name_{{idx}}\" class=\"mdl-chip__text\">{{currentEntry.entry.tag}}</span>                       <button id=\"tag_delete_{{idx}}\" type=\"button\" class=\"mdl-chip__action\"                               (click)=\"removeTag(currentEntry.entry.id)\">                           <i class=\"material-icons\">cancel</i>                       </button>             </span>         </li>     </ul> </div>",
        styles: [".tags {     padding-left: 7px;     padding-right: 7px; }  .tags li {     display: inline; }"]
    }), __metadata("design:paramtypes", [tag_service_1.TagService])], TagNodeList);
    exports.TagNodeList = TagNodeList;
});
$__System.registerDynamic("4", ["3", "7", "8"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1 = $__require("3");
    var Rx_1 = $__require("7");
    var ng2_alfresco_core_1 = $__require("8");
    var TagService = function () {
        function TagService(apiService, logService) {
            this.apiService = apiService;
            this.logService = logService;
        }
        TagService.prototype.getTagsByNodeId = function (nodeId) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.tagsApi.getNodeTags(nodeId)).catch(function (err) {
                return _this.handleError(err);
            });
        };
        TagService.prototype.getAllTheTags = function () {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.tagsApi.getTags()).catch(function (err) {
                return _this.handleError(err);
            });
        };
        TagService.prototype.addTag = function (nodeId, tagName) {
            var _this = this;
            var alfrescoApi = this.apiService.getInstance();
            var tagBody = new alfrescoApi.core.TagBody();
            tagBody.tag = tagName;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.tagsApi.addTag(nodeId, tagBody)).catch(function (err) {
                return _this.handleError(err);
            });
        };
        TagService.prototype.removeTag = function (nodeId, tag) {
            var _this = this;
            return Rx_1.Observable.fromPromise(this.apiService.getInstance().core.tagsApi.removeTag(nodeId, tag)).catch(function (err) {
                return _this.handleError(err);
            });
        };
        TagService.prototype.handleError = function (error) {
            this.logService.error(error);
            return Rx_1.Observable.throw(error || 'Server error');
        };
        return TagService;
    }();
    TagService = __decorate([core_1.Injectable(), __metadata("design:paramtypes", [ng2_alfresco_core_1.AlfrescoApiService, ng2_alfresco_core_1.LogService])], TagService);
    exports.TagService = TagService;
});
$__System.registerDynamic("1", ["3", "8", "2", "5", "6", "4"], true, function ($__require, exports, module) {
    /*!
     * @license
     * Copyright 2016 Alfresco Software, Ltd.
     *
     * Licensed under the Apache License, Version 2.0 (the "License");
     * you may not use this file except in compliance with the License.
     * You may obtain a copy of the License at
     *
     *     http://www.apache.org/licenses/LICENSE-2.0
     *
     * Unless required by applicable law or agreed to in writing, software
     * distributed under the License is distributed on an "AS IS" BASIS,
     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     * See the License for the specific language governing permissions and
     * limitations under the License.
     */
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    function __export(m) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }
    var core_1 = $__require("3");
    var ng2_alfresco_core_1 = $__require("8");
    var tag_actions_component_1 = $__require("2");
    var tag_list_component_1 = $__require("5");
    var tag_node_list_component_1 = $__require("6");
    var tag_service_1 = $__require("4");
    __export($__require("2"));
    __export($__require("5"));
    __export($__require("6"));
    __export($__require("4"));
    exports.TAG_DIRECTIVES = [tag_actions_component_1.TagActionsComponent, tag_list_component_1.TagList, tag_node_list_component_1.TagNodeList];
    exports.TAG_PROVIDERS = [tag_service_1.TagService];
    var TagModule = TagModule_1 = function () {
        function TagModule() {}
        TagModule.forRoot = function () {
            return {
                ngModule: TagModule_1,
                providers: exports.TAG_DIRECTIVES.slice()
            };
        };
        return TagModule;
    }();
    TagModule = TagModule_1 = __decorate([core_1.NgModule({
        imports: [ng2_alfresco_core_1.CoreModule],
        declarations: exports.TAG_DIRECTIVES.slice(),
        providers: exports.TAG_PROVIDERS.slice(),
        exports: exports.TAG_DIRECTIVES.slice()
    })], TagModule);
    exports.TagModule = TagModule;
    var TagModule_1;
    
});
})
(function(factory) {
  if (typeof define == 'function' && define.amd)
    define(["@angular/core","ng2-alfresco-core","rxjs/Rx"], factory);
  else if (typeof module == 'object' && module.exports && typeof require == 'function')
    module.exports = factory(require("@angular/core"), require("ng2-alfresco-core"), require("rxjs/Rx"));
  else
    throw new Error("Module must be loaded as AMD or CommonJS");
});
